<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Translational Omics in Alzheimers Disease: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css">
<script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/incubator/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicons/incubator/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicons/incubator/favicon-16x16.png">
<link rel="manifest" href="favicons/incubator/site.webmanifest">
<link rel="mask-icon" href="favicons/incubator/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/aio.html';">Instructor View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Translational Omics in Alzheimers Disease
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Translational Omics in Alzheimers Disease
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Translational Omics in Alzheimers Disease
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/aio.html">Instructor View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction to Omics Translation in AD</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="exploring-data-in-synapse.html">2. Synapse and AD Knowledge Portal</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="differential-expression-analysis.html">3. Differential expression analysis</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="mouse-human-alignment.html">4. Cross Species Functional Alignment</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="choosing_the_right_model.html">5. Choosing the right animal models</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-introduction"><p>Content from <a href="introduction.html">Introduction to Omics Translation in AD</a></p>
<hr>
<p>Last updated on 2026-02-12 |

        <a href="https://github.com/asliuyar/translational-omics-AD/edit/main/episodes/introduction.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>This workshop will deliver hands-on training using the most recent
computational techniques and omics data resources to improve awareness
and utility of integrating multi-scale data from model systems and
humans to uncover the mechanistic drivers of Alzheimer’s Disease
(AD).</p>
<p>Despite considerable effort in drug development for AD, high rate of
failure in clinical trials emphasize the need for more effective
preclinical strategies, including targeted use of mouse models to study
heterogenous disease phenotypes. In this era of omics-driven research,
interspecies alignment of disease relevant molecular signatures relies
heavily on bioinformatics tools and field-specific data
infrastructures.</p>
<p>The goal of this workshop is to accelerate translational research in
AD by training interested scientists on how to develop robust
computational workflows for both hypothesis-driven and data-driven
research strategies. The AD-specific, bespoke lessons will use resources
from the <a href="https://adknowledgeportal.synapse.org" class="external-link">AD Knowledge
Portal</a>, a NIA designated FAIR data repository that shares data from
human and non-human studies generated by multiple collaborative research
programs focused on aging, dementia, and AD.</p>
</section><section><h2 class="section-heading" id="lesson-material">Lesson material<a class="anchor" aria-label="anchor" href="#lesson-material"></a>
</h2>
<hr class="half-width">
<p>Here are the <a href="https://thejacksonlaboratory.box.com/s/j5a9orof8vrecpgrzx0d6nf6xyepluyi" class="external-link">R
scripts and data files</a> used in this workshop. <!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 --></p>
</section></section><section id="aio-exploring-data-in-synapse"><p>Content from <a href="exploring-data-in-synapse.html">Synapse and AD Knowledge Portal</a></p>
<hr>
<p>Last updated on 2026-02-12 |

        <a href="https://github.com/asliuyar/translational-omics-AD/edit/main/episodes/exploring-data-in-synapse.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can we work with the Synapse R client?</li>
<li>How can we work with data in AD Knowledge Portal?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Explain how to use <code>Synapser</code> Package.</li>
<li>Demonstrate how to locate data and metadata in the AD Knowledge
Portal.</li>
<li>Demonstrate how to download data from the AD Knowledge Portal
programmatically.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="working-with-ad-portal-metadata">Working with AD Portal metadata<a class="anchor" aria-label="anchor" href="#working-with-ad-portal-metadata"></a>
</h2>
<hr class="half-width">
<p>We have now downloaded several metadata files and an RNAseq counts
file from the portal. For our next exercises, we want to read those
files in as R data so we can work with them.</p>
<p>We can see from the <code>download_table</code> we got during the
bulk download step that we have five metadata files. Two of these should
be the individual and biospecimen files, and three of them are assay
metadata files.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">download_table</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">metadataType</span>, <span class="va">assay</span><span class="op">)</span></span></code></pre>
</div>
<p>We are only interested in RNAseq data, so we will only read in the
individual, biospecimen, and RNAseq assay metadata files.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># counts matrix</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="st">"data/htseqcounts_5XFAD.txt"</span>, </span>
<span>                   show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># individual metadata</span></span>
<span><span class="va">ind_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_individual_metadata.csv"</span>, </span>
<span>                     show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biospecimen metadata</span></span>
<span><span class="va">bio_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_biospecimen_metadata.csv"</span>, </span>
<span>                     show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># assay metadata</span></span>
<span><span class="va">rna_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_assay_RNAseq_metadata.csv"</span>, </span>
<span>                     show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s examine the data and metadata files a bit before we begin our
analyses. We start by exploring the <code>counts</code> data that we
read in using the tidyverse <code>read_tsv()</code>
(<em>read</em>_<em>t</em>ab-<em>s</em>eparated <em>v</em>alues)
function. This function reads data in as a <em>tibble</em>, a kind of
data table with some nice features that avoid some bad habits of the
base R <code>read.csv()</code> function. Calling a <code>tibble</code>
object will print the first ten rows in a nice tidy output. Doing the
same for a base R dataframe read in with <code>read.csv()</code> will
print the whole thing until it runs out of memory. If you want to
inspect a large dataframe, use <code>head(df)</code> to view the first
several rows only.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 55,489 × 73
   gene_id `32043rh` `32044rh` `32046rh` `32047rh` `32048rh` `32049rh` `32050rh`
   &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
 1 ENSG00…     22554         0         0         0     16700         0         0
 2 ENSG00…    344489         4         0         1    260935         6         8
 3 ENSMUS…      5061      3483      3941      3088      2756      3067      2711
 4 ENSMUS…         0         0         0         0         0         0         0
 5 ENSMUS…       208       162       138       127        95       154       165
 6 ENSMUS…        44        17        14        28        23        24        14
 7 ENSMUS…       143        88       121       117       115       109        75
 8 ENSMUS…        22         6        10        11        11        19        24
 9 ENSMUS…      7165      5013      5581      4011      4104      5254      4345
10 ENSMUS…      3728      2316      2238      1965      1822      1999      1809
# ℹ 55,479 more rows
# ℹ 65 more variables: `32052rh` &lt;dbl&gt;, `32053rh` &lt;dbl&gt;, `32057rh` &lt;dbl&gt;,
#   `32059rh` &lt;dbl&gt;, `32061rh` &lt;dbl&gt;, `32062rh` &lt;dbl&gt;, `32065rh` &lt;dbl&gt;,
#   `32067rh` &lt;dbl&gt;, `32068rh` &lt;dbl&gt;, `32070rh` &lt;dbl&gt;, `32073rh` &lt;dbl&gt;,
#   `32074rh` &lt;dbl&gt;, `32075rh` &lt;dbl&gt;, `32078rh` &lt;dbl&gt;, `32081rh` &lt;dbl&gt;,
#   `32088rh` &lt;dbl&gt;, `32640rh` &lt;dbl&gt;, `46105rh` &lt;dbl&gt;, `46106rh` &lt;dbl&gt;,
#   `46107rh` &lt;dbl&gt;, `46108rh` &lt;dbl&gt;, `46109rh` &lt;dbl&gt;, `46110rh` &lt;dbl&gt;, …</code></pre>
</div>
<p>The data file has a column of ENSEMBL <code>gene_id</code>s and then
a bunch of columns with count data, where the column headers correspond
to the <code>specimenID</code>s. These <code>specimenID</code>s should
all be in the RNAseq assay metadata file, so let’s check.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># what does the RNAseq assay metadata look like?</span></span>
<span><span class="va">rna_meta</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 72 × 12
   specimenID platform   RIN   rnaBatch libraryBatch sequencingBatch libraryPrep
   &lt;chr&gt;      &lt;chr&gt;      &lt;lgl&gt;    &lt;dbl&gt;        &lt;dbl&gt;           &lt;dbl&gt; &lt;chr&gt;
 1 32043rh    IlluminaN… NA           1            1               1 polyAselec…
 2 32044rh    IlluminaN… NA           1            1               1 polyAselec…
 3 32046rh    IlluminaN… NA           1            1               1 polyAselec…
 4 32047rh    IlluminaN… NA           1            1               1 polyAselec…
 5 32049rh    IlluminaN… NA           1            1               1 polyAselec…
 6 32057rh    IlluminaN… NA           1            1               1 polyAselec…
 7 32061rh    IlluminaN… NA           1            1               1 polyAselec…
 8 32065rh    IlluminaN… NA           1            1               1 polyAselec…
 9 32067rh    IlluminaN… NA           1            1               1 polyAselec…
10 32070rh    IlluminaN… NA           1            1               1 polyAselec…
# ℹ 62 more rows
# ℹ 5 more variables: libraryPreparationMethod &lt;lgl&gt;, isStranded &lt;lgl&gt;,
#   readStrandOrigin &lt;lgl&gt;, runType &lt;chr&gt;, readLength &lt;dbl&gt;</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># are all the column headers from the counts matrix </span></span>
<span><span class="co"># (except the first "gene_id" column) in the assay metadata?</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">rna_meta</span><span class="op">$</span><span class="va">`specimenID`</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] TRUE</code></pre>
</div>
<p><strong>Assay metadata</strong></p>
<p>The assay metadata contains information about how data was generated
on each sample in the assay. Each <code>specimenID</code> represents a
unique sample. We can use some tools from <code>dplyr</code> to explore
the metadata.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># how many unique specimens were sequenced?</span></span>
<span><span class="fu">n_distinct</span><span class="op">(</span><span class="va">rna_meta</span><span class="op">$</span><span class="va">`specimenID`</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 72</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># were the samples all sequenced on the same platform?</span></span>
<span><span class="fu">distinct</span><span class="op">(</span><span class="va">rna_meta</span>, <span class="va">platform</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 1 × 1
  platform
  &lt;chr&gt;
1 IlluminaNovaseq6000</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># were there multiple sequencing batches reported?</span></span>
<span><span class="fu">distinct</span><span class="op">(</span><span class="va">rna_meta</span>, <span class="va">sequencingBatch</span><span class="op">)</span> </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 1 × 1
  sequencingBatch
            &lt;dbl&gt;
1               1</code></pre>
</div>
<p><strong>Biospecimen metadata</strong></p>
<p>The biospecimen metadata contains specimen-level information,
including organ and tissue the specimen was taken from, how it was
prepared, etc. Each <code>specimenID</code> is mapped to an
<code>individualID</code>.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># all specimens from the RNAseq assay metadata file should be in the biospecimen file</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="va">rna_meta</span><span class="op">$</span><span class="va">`specimenID`</span> <span class="op">%in%</span> <span class="va">bio_meta</span><span class="op">$</span><span class="va">`specimenID`</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] TRUE</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># but the biospecimen file also contains specimens from different assays</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="va">bio_meta</span><span class="op">$</span><span class="va">`specimenID`</span> <span class="op">%in%</span> <span class="va">rna_meta</span><span class="op">$</span><span class="va">`specimenID`</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] FALSE</code></pre>
</div>
<p><strong>Individual metadata</strong></p>
<p>The individual metadata contains information about all the
individuals in the study, represented by unique
<code>individualID</code>s. For humans, this includes information on
age, sex, race, diagnosis, etc. For MODEL-AD mouse models, the
individual metadata has information on model genotypes, stock numbers,
diet, and more.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># all `individualID`s in the biospecimen file should be in the individual file</span></span>
<span><span class="fu">all</span><span class="op">(</span><span class="va">bio_meta</span><span class="op">$</span><span class="va">`individualID`</span> <span class="op">%in%</span> <span class="va">ind_meta</span><span class="op">$</span><span class="va">`individualID`</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] TRUE</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># which model genotypes are in this study?</span></span>
<span><span class="fu">distinct</span><span class="op">(</span><span class="va">ind_meta</span>, <span class="va">genotype</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 2 × 1
  genotype
  &lt;chr&gt;
1 5XFAD_carrier
2 5XFAD_noncarrier</code></pre>
</div>
<p><strong>Joining metadata</strong></p>
<p>We use the three-file structure for our metadata because it allows us
to store metadata for each study in a tidy format. Every line in the
assay and biospecimen files represents a unique specimen, and every line
in the individual file represents a unique individual. This means the
files can be easily joined by <code>specimenID</code> and
<code>individualID</code> to get all levels of metadata that apply to a
particular data file. We will use the <code>left_join()</code> function
from the <code>dplyr</code> package, and the <code>%\&gt;%</code>
operator from the <code>magrittr</code> package. If you are unfamiliar
with the pipe, think of it as a shorthand for “take this (the preceding
object) and do that (the subsequent command)”. See <a href="https://magrittr.tidyverse.org/" class="external-link">magrittr</a> for more info on
piping in R.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># join all the rows in the assay metadata</span></span>
<span><span class="co"># that have a match in the biospecimen metadata</span></span>
<span><span class="va">joined_meta</span> <span class="op">&lt;-</span> <span class="va">rna_meta</span> <span class="op">%&gt;%</span>               <span class="co"># start with the rnaseq assay metadata</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">bio_meta</span>, by <span class="op">=</span> <span class="st">"specimenID"</span><span class="op">)</span> <span class="op">%&gt;%</span>  <span class="co"># join rows from biospecimen </span></span>
<span>                                                <span class="co"># that match specimenID </span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">ind_meta</span>, by <span class="op">=</span> <span class="st">"individualID"</span><span class="op">)</span> <span class="co"># join rows from individual </span></span>
<span>                                              <span class="co"># that match individualID</span></span>
<span></span>
<span><span class="va">joined_meta</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 72 × 53
   specimenID platform   RIN   rnaBatch libraryBatch sequencingBatch libraryPrep
   &lt;chr&gt;      &lt;chr&gt;      &lt;lgl&gt;    &lt;dbl&gt;        &lt;dbl&gt;           &lt;dbl&gt; &lt;chr&gt;
 1 32043rh    IlluminaN… NA           1            1               1 polyAselec…
 2 32044rh    IlluminaN… NA           1            1               1 polyAselec…
 3 32046rh    IlluminaN… NA           1            1               1 polyAselec…
 4 32047rh    IlluminaN… NA           1            1               1 polyAselec…
 5 32049rh    IlluminaN… NA           1            1               1 polyAselec…
 6 32057rh    IlluminaN… NA           1            1               1 polyAselec…
 7 32061rh    IlluminaN… NA           1            1               1 polyAselec…
 8 32065rh    IlluminaN… NA           1            1               1 polyAselec…
 9 32067rh    IlluminaN… NA           1            1               1 polyAselec…
10 32070rh    IlluminaN… NA           1            1               1 polyAselec…
# ℹ 62 more rows
# ℹ 46 more variables: libraryPreparationMethod &lt;lgl&gt;, isStranded &lt;lgl&gt;,
#   readStrandOrigin &lt;lgl&gt;, runType &lt;chr&gt;, readLength &lt;dbl&gt;,
#   individualID &lt;dbl&gt;, specimenIdSource &lt;chr&gt;, organ &lt;chr&gt;, tissue &lt;chr&gt;,
#   BrodmannArea &lt;lgl&gt;, sampleStatus &lt;chr&gt;, tissueWeight &lt;lgl&gt;,
#   tissueVolume &lt;lgl&gt;, nucleicAcidSource &lt;lgl&gt;, cellType &lt;lgl&gt;,
#   fastingState &lt;lgl&gt;, isPostMortem &lt;lgl&gt;, samplingAge &lt;lgl&gt;, …</code></pre>
</div>
<p>We now have a very wide dataframe that contains all the available
metadata on each specimen in the RNAseq data from this study. This
procedure can be used to join the three types of metadata files for
every study in the AD Knowledge Portal, allowing you to filter
individuals and specimens as needed based on your analysis criteria!</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># convert columns of strings to month-date-year format</span></span>
<span><span class="va">joined_meta_time</span> <span class="op">&lt;-</span> <span class="va">joined_meta</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>dateBirth <span class="op">=</span> <span class="fu">mdy</span><span class="op">(</span><span class="va">dateBirth</span><span class="op">)</span>, dateDeath <span class="op">=</span> <span class="fu">mdy</span><span class="op">(</span><span class="va">dateDeath</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  </span>
<span>  <span class="co"># create a new column that subtracts dateBirth from dateDeath in days, </span></span>
<span>  <span class="co"># then divide by 30 to get months</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>timepoint <span class="op">=</span> <span class="fu">as.numeric</span><span class="op">(</span><span class="fu">difftime</span><span class="op">(</span><span class="va">dateDeath</span>, <span class="va">dateBirth</span>, units <span class="op">=</span><span class="st">"days"</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fl">30</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  </span>
<span>  <span class="co"># convert numeric ages to timepoint categories</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>timepoint <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span><span class="va">timepoint</span> <span class="op">&gt;</span> <span class="fl">10</span> <span class="op">~</span> <span class="st">"12 mo"</span>,</span>
<span>                               <span class="va">timepoint</span> <span class="op">&lt;</span> <span class="fl">10</span> <span class="op">&amp;</span> <span class="va">timepoint</span> <span class="op">&gt;</span> <span class="fl">5</span> <span class="op">~</span> <span class="st">"6 mo"</span>,</span>
<span>                               <span class="va">timepoint</span> <span class="op">&lt;</span> <span class="fl">5</span> <span class="op">~</span> <span class="st">"4 mo"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">covars_5XFAD</span> <span class="op">&lt;-</span> <span class="va">joined_meta_time</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">individualID</span>, <span class="va">specimenID</span>, <span class="va">sex</span>, <span class="va">genotype</span>, <span class="va">timepoint</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">covars_5XFAD</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">covars_5XFAD</span><span class="op">$</span><span class="va">specimenID</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">covars_5XFAD</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>        individualID specimenID    sex         genotype timepoint
32043rh        32043    32043rh female    5XFAD_carrier     12 mo
32044rh        32044    32044rh   male 5XFAD_noncarrier     12 mo
32046rh        32046    32046rh   male 5XFAD_noncarrier     12 mo
32047rh        32047    32047rh   male 5XFAD_noncarrier     12 mo
32049rh        32049    32049rh female 5XFAD_noncarrier     12 mo
32057rh        32057    32057rh female 5XFAD_noncarrier     12 mo</code></pre>
</div>
<p>We will save <code>joined_meta</code> for the next lesson.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">covars_5XFAD</span>, file <span class="op">=</span> <span class="st">"data/covars_5XFAD.rds"</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="single-specimen-files">Single Specimen files<a class="anchor" aria-label="anchor" href="#single-specimen-files"></a>
</h2>
<hr class="half-width">
<p>For files that contain data from a single specimen (e.g. raw
sequencing files, raw mass spectra, etc.), we can use the Synapse
annotations to associate these files with the appropriate metadata.</p>
<div id="exercise" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="exercise" class="callout-inner">
<h3 class="callout-title">Exercise</h3>
<div class="callout-content">
<p>Use <em>Explore Data</em> to find all RNAseq files from the
<code>Jax.IU.Pitt_5XFAD</code> study. If we filter for data where
<code>Study = "Jax.IU.Pitt_5XFAD"</code> and
<code>Assay = "rnaSeq"</code> we will get a list of 148 files, including
raw fastqs and processed counts data.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<p>We can use the function <code>synGetAnnotations</code> to view the
annotations associated with any file without downloading the file.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the synID of a random fastq file from this list</span></span>
<span><span class="va">random_fastq</span> <span class="op">&lt;-</span> <span class="st">"syn22108503"</span></span>
<span></span>
<span><span class="co"># extract the annotations as a nested list</span></span>
<span><span class="va">fastq_annotations</span> <span class="op">&lt;-</span> <span class="fu">synGetAnnotations</span><span class="op">(</span><span class="va">random_fastq</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fastq_annotations</span></span></code></pre>
</div>
<p>The file annotations let us see which study the file is associated
with (<code>Jax.IU.Pitt.5XFAD</code>), which species it is from (mouse),
which assay generated the file (rnaSeq), and a whole bunch of other
properties. Most importantly, single-specimen files are annotated with
the <code>specimenID</code> of the specimen in the file, and the
<code>individualID</code> of the individual that specimen was taken
from. We can use these annotations to link files to the rest of the
metadata, including metadata that is not in annotations. This is
especially helpful for human studies, as potentially identifying
information like age, race, and diagnosis is not included in file
annotations.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># find records belonging to the individual this file maps</span></span>
<span><span class="co"># to in our joined metadata</span></span>
<span><span class="va">joined_meta</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">`individualID`</span> <span class="op">==</span> <span class="va">fastq_annotations</span><span class="op">$</span><span class="va">individualID</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="annotations-during-bulk-download">Annotations during bulk download<a class="anchor" aria-label="anchor" href="#annotations-during-bulk-download"></a>
</h2>
<hr class="half-width">
<p>When bulk downloading many files, the best practice is to preserve
the download manifest that is generated which lists all the files, their
<code>synIDs</code>, and all their annotations. If using the Synapse R
client, follow the instructions in the Bulk download files section
above.</p>
<p>If we use the “Programmatic Options” tab in the AD Portal download
menu to download all 148 rnaSeq files from the 5XFAD study, we would get
a table query that looks like this:</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu">synTableQuery</span><span class="op">(</span><span class="st">"SELECT * FROM syn11346063.37 </span></span>
<span><span class="st">                       WHERE ( ( \"study\" HAS ( 'Jax.IU.Pitt_5XFAD' ) ) </span></span>
<span><span class="st">                       AND ( \"assay\" HAS ( 'rnaSeq' ) ) )"</span><span class="op">)</span></span></code></pre>
</div>
<p>As we saw previously, this downloads a csv file with the results of
our AD Portal query. Opening that file lets us see which specimens are
associated with which files:</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotations_table</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">query</span><span class="op">$</span><span class="va">filepath</span>, </span>
<span>                              show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">annotations_table</span></span></code></pre>
</div>
<p>You could then use
<code>purrr::walk(download_table$id, ~synGet(.x, downloadLocation = ))</code>
to walk through the column of <code>synIDs</code> and download all 148
files. However, because these are large files, it might be preferable to
use the Python client or command line client for increased speed.</p>
<p>Once you’ve downloaded all the files in the <code>id</code> column,
you can link those files to their annotations by the <code>name</code>
column.</p>
<p>We’ll use the “random fastq” that we got annotations for earlier to
avoid downloading the whole 3GB file, we’ll use synGet with
<code>downloadFile = FALSE</code> to get only the Synapse entity
information, rather than the file. If we downloaded the actual file, we
could find it in the directory and search using the filename. Since
we’re just downloading the Synapse entity wrapper object, we’ll use the
file name listed in the object properties.</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fastq</span> <span class="op">&lt;-</span> <span class="fu">synGet</span><span class="op">(</span><span class="va">random_fastq</span>, downloadFile <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter the annotations table to rows that match the fastq filename</span></span>
<span><span class="va">annotations_table</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="va">fastq</span><span class="op">$</span><span class="va">properties</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="multispecimen-files">Multispecimen files<a class="anchor" aria-label="anchor" href="#multispecimen-files"></a>
</h2>
<hr class="half-width">
<p>Multispecimen files in the AD Knowledge Portal are files that contain
data or information from more than one specimen. They are not annotated
with <code>individualID</code>s or <code>specimenID</code>s, since these
files may contain numbers of specimens that exceed the annotation
limits. These files are usually processed or summary data (gene counts,
peptide quantifications, etc), and are always annotated with
<code>isMultiSpecimen = TRUE</code>.</p>
<p>If we look at the processed data files in the table of 5XFAD RNAseq
file annotations we just downloaded, we will see that it
<code>isMultiSpecimen = TRUE</code>, but <code>individualID</code> and
<code>specimenID</code> are blank:</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotations_table</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">fileFormat</span> <span class="op">==</span> <span class="st">"txt"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">individualID</span>, <span class="va">specimenID</span>, <span class="va">isMultiSpecimen</span><span class="op">)</span></span></code></pre>
</div>
<p>The multispecimen file should contain a row or column of
<code>specimenID</code>s that correspond to the <code>specimenID</code>s
used in a study’s metadata, as we have seen with the 5XFAD counts
file.</p>
<p>In this example, we take a slice of the counts data to reduce
computation, transpose it so that each row represents a single specimen,
and then join it to the joined metadata by the
<code>specimenID</code>.</p>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">slice_head</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">t</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span>rownames <span class="op">=</span> <span class="st">"specimenID"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">joined_meta</span>, by <span class="op">=</span> <span class="st">"specimenID"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 73 × 58
   specimenID V1    V2    V3    V4    V5    platform RIN   rnaBatch libraryBatch
   &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;lgl&gt;    &lt;dbl&gt;        &lt;dbl&gt;
 1 gene_id    "ENS… "ENS… "ENS… "ENS… "ENS… &lt;NA&gt;     NA          NA           NA
 2 32043rh    " 22… "344… "  5… "   … "   … Illumin… NA           1            1
 3 32044rh    "   … "   … "348… "   … " 16… Illumin… NA           1            1
 4 32046rh    "   … "   … "394… "   … " 13… Illumin… NA           1            1
 5 32047rh    "   … "   … "308… "   … " 12… Illumin… NA           1            1
 6 32048rh    " 16… "260… "  2… "   … "   … Illumin… NA           1            1
 7 32049rh    "   … "   … "306… "   … " 15… Illumin… NA           1            1
 8 32050rh    "   … "   … "271… "   … " 16… Illumin… NA           1            1
 9 32052rh    " 19… "337… "  3… "   … "   … Illumin… NA           1            1
10 32053rh    " 14… "206… "  3… "   … "   … Illumin… NA           1            1
# ℹ 63 more rows
# ℹ 48 more variables: sequencingBatch &lt;dbl&gt;, libraryPrep &lt;chr&gt;,
#   libraryPreparationMethod &lt;lgl&gt;, isStranded &lt;lgl&gt;, readStrandOrigin &lt;lgl&gt;,
#   runType &lt;chr&gt;, readLength &lt;dbl&gt;, individualID &lt;dbl&gt;,
#   specimenIdSource &lt;chr&gt;, organ &lt;chr&gt;, tissue &lt;chr&gt;, BrodmannArea &lt;lgl&gt;,
#   sampleStatus &lt;chr&gt;, tissueWeight &lt;lgl&gt;, tissueVolume &lt;lgl&gt;,
#   nucleicAcidSource &lt;lgl&gt;, cellType &lt;lgl&gt;, fastingState &lt;lgl&gt;, …</code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul><li>
</li></ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-differential-expression-analysis"><p>Content from <a href="differential-expression-analysis.html">Differential expression analysis</a></p>
<hr>
<p>Last updated on 2026-02-12 |

        <a href="https://github.com/asliuyar/translational-omics-AD/edit/main/episodes/differential-expression-analysis.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What transcriptomic changes do we observe in mouse models carrying
AD-related mutations?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Read in a count matrix and metadata.</li>
<li>Understand the data from AD mouse models</li>
<li>Format the data for differential analysis</li>
<li>Perform differential analysis using DESeq2.</li>
<li>Pathway enrichment of differentially expressed genes</li>
<li>Save data for next lessons</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="differential-expression-analysis">Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h1>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: replacing previous import 'S4Arrays::makeNindexFromArrayViewport' by
'DelayedArray::makeNindexFromArrayViewport' when loading 'SummarizedExperiment'</code></pre>
</div>
<p><strong>Reading Gene Expression Count matrix from previous
lesson</strong></p>
<p>In this lesson, we will use the raw counts matrix and metadata
downloaded in the previous lesson and will perform differential
expression analysis.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">read.delim</span><span class="op">(</span><span class="st">"data/htseqcounts_5XFAD.txt"</span>, </span>
<span>                      check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Reading Sample Metadata from Previous Lesson</strong></p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covars</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">"data/covars_5XFAD.rds"</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s explore the data:</p>
<p>Let’s look at the top of the metadata.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">covars</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>        individualID specimenID    sex         genotype timepoint
32043rh        32043    32043rh female    5XFAD_carrier     12 mo
32044rh        32044    32044rh   male 5XFAD_noncarrier     12 mo
32046rh        32046    32046rh   male 5XFAD_noncarrier     12 mo
32047rh        32047    32047rh   male 5XFAD_noncarrier     12 mo
32049rh        32049    32049rh female 5XFAD_noncarrier     12 mo
32057rh        32057    32057rh female 5XFAD_noncarrier     12 mo</code></pre>
</div>
<p>identify distinct groups using sample metadata</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">distinct</span><span class="op">(</span><span class="va">covars</span>, <span class="va">sex</span>, <span class="va">genotype</span>, <span class="va">timepoint</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>           sex         genotype timepoint
32043rh female    5XFAD_carrier     12 mo
32044rh   male 5XFAD_noncarrier     12 mo
32049rh female 5XFAD_noncarrier     12 mo
46105rh female 5XFAD_noncarrier      6 mo
46108rh   male 5XFAD_noncarrier      6 mo
46131rh female 5XFAD_noncarrier      4 mo
46877rh   male 5XFAD_noncarrier      4 mo
46887rh female    5XFAD_carrier      4 mo
32053rh   male    5XFAD_carrier     12 mo
46111rh female    5XFAD_carrier      6 mo
46865rh   male    5XFAD_carrier      6 mo
46866rh   male    5XFAD_carrier      4 mo</code></pre>
</div>
<p>How many mice were used to produce this data?</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covars</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sex</span>, <span class="va">genotype</span>, <span class="va">timepoint</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 12 × 4
# Groups:   sex, genotype, timepoint [12]
   sex    genotype         timepoint     n
   &lt;chr&gt;  &lt;chr&gt;            &lt;chr&gt;     &lt;int&gt;
 1 female 5XFAD_carrier    12 mo         6
 2 female 5XFAD_carrier    4 mo          6
 3 female 5XFAD_carrier    6 mo          6
 4 female 5XFAD_noncarrier 12 mo         6
 5 female 5XFAD_noncarrier 4 mo          6
 6 female 5XFAD_noncarrier 6 mo          6
 7 male   5XFAD_carrier    12 mo         6
 8 male   5XFAD_carrier    4 mo          6
 9 male   5XFAD_carrier    6 mo          6
10 male   5XFAD_noncarrier 12 mo         6
11 male   5XFAD_noncarrier 4 mo          6
12 male   5XFAD_noncarrier 6 mo          6</code></pre>
</div>
<p>How many rows and columns are there in counts?</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 55489    73</code></pre>
</div>
<p>In the counts matrix, genes are in rows and samples are in columns.
Let’s look at the first few rows.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">counts</span>, n<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             gene_id 32043rh 32044rh 32046rh 32047rh 32048rh 32049rh 32050rh
1    ENSG00000080815   22554       0       0       0   16700       0       0
2    ENSG00000142192  344489       4       0       1  260935       6       8
3 ENSMUSG00000000001    5061    3483    3941    3088    2756    3067    2711
4 ENSMUSG00000000003       0       0       0       0       0       0       0
5 ENSMUSG00000000028     208     162     138     127      95     154     165
  32052rh 32053rh 32057rh 32059rh 32061rh 32062rh 32065rh 32067rh 32068rh
1   19748   14023       0   17062       0   15986      10       0   18584
2  337456  206851       1  264748       0  252248     172       4  300398
3    3334    3841    4068    3306    4076    3732    3940    4238    3257
4       0       0       0       0       0       0       0       0       0
5     124     103     164     116     108     134     204     239     148
  32070rh 32073rh 32074rh 32075rh 32078rh 32081rh 32088rh 32640rh 46105rh
1       1       0       0   22783   17029   16626   15573   12721       4
2       4       2       9  342655  280968  258597  243373  188818      19
3    3351    3449    4654    4844    3132    3334    3639    3355    4191
4       0       0       0       0       0       0       0       0       0
5     159     167     157     211     162     149     160     103     158
  46106rh 46107rh 46108rh 46109rh 46110rh 46111rh 46112rh 46113rh 46115rh
1       0       0       0       0       0   17931       0   19087       0
2       0       0       1       5       1  293409       8  273704       1
3    3058    4265    3248    3638    3747    3971    3192    3805    3753
4       0       0       0       0       0       0       0       0       0
5     167     199     113     168     175     203     158     108     110
  46121rh 46131rh 46132rh 46134rh 46138rh 46141rh 46142rh 46862rh 46863rh
1       0       0   12703   18833       0   18702   17666       0   14834
2       0       1  187975  285048       0  284499  250600       0  218494
3    4134    3059    3116    3853    3682    2844    3466    3442    3300
4       0       0       0       0       0       0       0       0       0
5     179     137     145     183     171     138      88     154     157
  46865rh 46866rh 46867rh 46868rh 46871rh 46872rh 46873rh 46874rh 46875rh
1   10546   10830   10316   10638   15248       0       0   11608   11561
2  169516  152769  151732  190150  229063       6       1  165941  171303
3    3242    3872    3656    3739    3473    3154    5510    3657    4121
4       0       0       0       0       0       0       0       0       0
5     131     152     152     155     140      80     240     148     112
  46876rh 46877rh 46878rh 46879rh 46881rh 46882rh 46883rh 46884rh 46885rh
1       0       0   12683   15613       0   14084   20753       0       0
2       0       2  183058  216122       0  199448  306081       0       5
3    3422    3829    3996    4324    2592    2606    4600    2913    3614
4       0       0       0       0       0       0       0       0       0
5     147     166     169     215     115     101     174     127     151
  46886rh 46887rh 46888rh 46889rh 46890rh 46891rh 46892rh 46893rh 46895rh
1   16639   16072       0   16680   13367       0   25119      92       0
2  242543  258061       0  235530  196721       0  371037    1116       0
3    3294    3719    3899    4173    4008    3037    5967    3459    4262
4       0       0       0       0       0       0       0       0       0
5     139     128     210     127     156     116     260     161     189
  46896rh 46897rh
1   15934       0
2  235343       6
3    3923    3486
4       0       0
5     179     117</code></pre>
</div>
<p>As you can see, the gene ids are ENSEMBL IDs. There is some risk that
these may not be unique. Let’s check whether any of the gene symbols are
duplicated. We will sum the number of duplicated gene symbols.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="fu">duplicated</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 0</code></pre>
</div>
<p>The sum equals zero, so there are no duplicated gene symbols, which
is good. Similarly, samples should be unique. Once again, let’s verify
this:</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="fu">duplicated</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 0</code></pre>
</div>
<p><strong>Formatting the count matrix</strong></p>
<p>Now, as we see that <code>gene_id</code> is in first column of count
matrix, but we will need only count data in matrix, so we will change
the <code>gene_id</code> column to <code>rownames</code>.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Converting the `gene_id` as `rownames` of `counts` matrix</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span> <span class="op">%&gt;%</span></span>
<span>          <span class="fu">column_to_rownames</span><span class="op">(</span><span class="va">.</span>, var <span class="op">=</span> <span class="st">"gene_id"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>          <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s confirm if change is done correctly.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">counts</span>, n<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                   32043rh 32044rh 32046rh 32047rh 32048rh 32049rh 32050rh
ENSG00000080815      22554       0       0       0   16700       0       0
ENSG00000142192     344489       4       0       1  260935       6       8
ENSMUSG00000000001    5061    3483    3941    3088    2756    3067    2711
ENSMUSG00000000003       0       0       0       0       0       0       0
ENSMUSG00000000028     208     162     138     127      95     154     165
                   32052rh 32053rh 32057rh 32059rh 32061rh 32062rh 32065rh
ENSG00000080815      19748   14023       0   17062       0   15986      10
ENSG00000142192     337456  206851       1  264748       0  252248     172
ENSMUSG00000000001    3334    3841    4068    3306    4076    3732    3940
ENSMUSG00000000003       0       0       0       0       0       0       0
ENSMUSG00000000028     124     103     164     116     108     134     204
                   32067rh 32068rh 32070rh 32073rh 32074rh 32075rh 32078rh
ENSG00000080815          0   18584       1       0       0   22783   17029
ENSG00000142192          4  300398       4       2       9  342655  280968
ENSMUSG00000000001    4238    3257    3351    3449    4654    4844    3132
ENSMUSG00000000003       0       0       0       0       0       0       0
ENSMUSG00000000028     239     148     159     167     157     211     162
                   32081rh 32088rh 32640rh 46105rh 46106rh 46107rh 46108rh
ENSG00000080815      16626   15573   12721       4       0       0       0
ENSG00000142192     258597  243373  188818      19       0       0       1
ENSMUSG00000000001    3334    3639    3355    4191    3058    4265    3248
ENSMUSG00000000003       0       0       0       0       0       0       0
ENSMUSG00000000028     149     160     103     158     167     199     113
                   46109rh 46110rh 46111rh 46112rh 46113rh 46115rh 46121rh
ENSG00000080815          0       0   17931       0   19087       0       0
ENSG00000142192          5       1  293409       8  273704       1       0
ENSMUSG00000000001    3638    3747    3971    3192    3805    3753    4134
ENSMUSG00000000003       0       0       0       0       0       0       0
ENSMUSG00000000028     168     175     203     158     108     110     179
                   46131rh 46132rh 46134rh 46138rh 46141rh 46142rh 46862rh
ENSG00000080815          0   12703   18833       0   18702   17666       0
ENSG00000142192          1  187975  285048       0  284499  250600       0
ENSMUSG00000000001    3059    3116    3853    3682    2844    3466    3442
ENSMUSG00000000003       0       0       0       0       0       0       0
ENSMUSG00000000028     137     145     183     171     138      88     154
                   46863rh 46865rh 46866rh 46867rh 46868rh 46871rh 46872rh
ENSG00000080815      14834   10546   10830   10316   10638   15248       0
ENSG00000142192     218494  169516  152769  151732  190150  229063       6
ENSMUSG00000000001    3300    3242    3872    3656    3739    3473    3154
ENSMUSG00000000003       0       0       0       0       0       0       0
ENSMUSG00000000028     157     131     152     152     155     140      80
                   46873rh 46874rh 46875rh 46876rh 46877rh 46878rh 46879rh
ENSG00000080815          0   11608   11561       0       0   12683   15613
ENSG00000142192          1  165941  171303       0       2  183058  216122
ENSMUSG00000000001    5510    3657    4121    3422    3829    3996    4324
ENSMUSG00000000003       0       0       0       0       0       0       0
ENSMUSG00000000028     240     148     112     147     166     169     215
                   46881rh 46882rh 46883rh 46884rh 46885rh 46886rh 46887rh
ENSG00000080815          0   14084   20753       0       0   16639   16072
ENSG00000142192          0  199448  306081       0       5  242543  258061
ENSMUSG00000000001    2592    2606    4600    2913    3614    3294    3719
ENSMUSG00000000003       0       0       0       0       0       0       0
ENSMUSG00000000028     115     101     174     127     151     139     128
                   46888rh 46889rh 46890rh 46891rh 46892rh 46893rh 46895rh
ENSG00000080815          0   16680   13367       0   25119      92       0
ENSG00000142192          0  235530  196721       0  371037    1116       0
ENSMUSG00000000001    3899    4173    4008    3037    5967    3459    4262
ENSMUSG00000000003       0       0       0       0       0       0       0
ENSMUSG00000000028     210     127     156     116     260     161     189
                   46896rh 46897rh
ENSG00000080815      15934       0
ENSG00000142192     235343       6
ENSMUSG00000000001    3923    3486
ENSMUSG00000000003       0       0
ENSMUSG00000000028     179     117</code></pre>
</div>
<p>As you can see from count table there are some genes that start with
<code>ENSG</code> and others start with <code>ENSMUSG</code>.
<code>ENSG</code> refers to human gene ENSEMBL id and
<code>ENSMUSG</code> refer to mouse ENSEMBL id. Let’s check how many
<code>gene_ids</code> are NOT from the mouse genome by searching for the
string “MUS” (as in <em>Mus musculus</em>) in the <code>rownames</code>
of the <code>counts</code> matrix.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">str_detect</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, <span class="st">"MUS"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                32043rh 32044rh 32046rh 32047rh 32048rh 32049rh
ENSG00000080815   22554       0       0       0   16700       0
ENSG00000142192  344489       4       0       1  260935       6</code></pre>
</div>
<p>Ok, so we see there are two human genes in out count matrix. Why?
What genes are they?</p>
<p>Briefly, the 5xFAD mouse strain harbors two human transgenes APP
(<code>ENSG00000142192</code>) and PSEN1 (<code>ENSG00000080815</code>)
and inserted into exon 2 of the mouse Thy1 gene. To validate 5XFAD
strain and capture expression of human transgene APP and PS1, a custom
mouse genomic sequence was created and we quantified expression of human
as well as mouse App (<code>ENSMUSG00000022892</code>) and Psen1
(<code>ENSMUSG00000019969</code>) genes by our MODEL-AD RNA-Seq
pipeline.</p>
<p><strong>Validation of 5xFAD mouse strain</strong></p>
<p>First we convert the dataframe to longer format and join our
covariates by <code>MouseID</code>.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count_tpose</span> <span class="op">&lt;-</span>  <span class="va">counts</span>  <span class="op">%&gt;%</span></span>
<span>                <span class="fu">rownames_to_column</span><span class="op">(</span><span class="va">.</span>, var <span class="op">=</span> <span class="st">"gene_id"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                <span class="fu">filter</span><span class="op">(</span><span class="va">gene_id</span> <span class="op">%in%</span></span>
<span>                      <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000080815"</span>,</span>
<span>                        <span class="st">"ENSMUSG00000019969"</span>,</span>
<span>                        <span class="st">"ENSG00000142192"</span>,</span>
<span>                        <span class="st">"ENSMUSG00000022892"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                <span class="fu">pivot_longer</span><span class="op">(</span><span class="va">.</span>, cols    <span class="op">=</span> <span class="op">-</span><span class="st">"gene_id"</span>,</span>
<span>                              names_to  <span class="op">=</span> <span class="st">"specimenID"</span>,</span>
<span>                              values_to <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                <span class="fu">left_join</span><span class="op">(</span><span class="va">covars</span>, by<span class="op">=</span><span class="st">"specimenID"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">count_tpose</span><span class="op">)</span> </span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          gene_id specimenID counts individualID    sex         genotype
1 ENSG00000080815    32043rh  22554        32043 female    5XFAD_carrier
2 ENSG00000080815    32044rh      0        32044   male 5XFAD_noncarrier
3 ENSG00000080815    32046rh      0        32046   male 5XFAD_noncarrier
4 ENSG00000080815    32047rh      0        32047   male 5XFAD_noncarrier
5 ENSG00000080815    32048rh  16700        32048 female    5XFAD_carrier
6 ENSG00000080815    32049rh      0        32049 female 5XFAD_noncarrier
  timepoint
1     12 mo
2     12 mo
3     12 mo
4     12 mo
5     12 mo
6     12 mo</code></pre>
</div>
<p>Rename the APP and PSEN1 genes to specify whether mouse or human.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># make the age column a factor and re-order the levels</span></span>
<span><span class="va">count_tpose</span><span class="op">$</span><span class="va">timepoint</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">count_tpose</span><span class="op">$</span><span class="va">timepoint</span>,</span>
<span>                                levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"4 mo"</span>, <span class="st">"6 mo"</span>, <span class="st">"12 mo"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># rename the gene id to gene symbol</span></span>
<span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="st">"ENSG00000142192"</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="st">"Human APP"</span></span>
<span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="st">"ENSG00000080815"</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="st">"Human PSEN1"</span></span>
<span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000022892"</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="st">"Mouse App"</span></span>
<span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="va">count_tpose</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000019969"</span><span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="st">"Mouse Psen1"</span></span></code></pre>
</div>
<p>Visualize orthologous genes.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create simple box plots showing normalized counts </span></span>
<span><span class="co"># by genotype and time point faceted by sex.</span></span>
<span><span class="va">count_tpose</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">timepoint</span>, y <span class="op">=</span> <span class="va">counts</span>, color <span class="op">=</span> <span class="va">genotype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu">geom_point</span><span class="op">(</span>position <span class="op">=</span> <span class="fu">position_jitterdodge</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">gene_id</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/differential-expression-analysis-rendered-unnamed-chunk-16-1.png" alt="" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>You will notice expression of Human APP is higher in 5XFAD carriers
but lower in non-carriers. However mouse App expressed in both 5XFAD
carrier and non-carrier.</p>
<p>We are going to sum the counts from both orthologous genes (human APP
and mouse App; human PSEN1 and mouse Psen1) and save the summed
expression as expression of mouse genes, respectively to match with gene
names in control mice.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># merge mouse and human APP gene raw count</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>   <span class="op">%in%</span> <span class="st">"ENSMUSG00000022892"</span>, <span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000022892"</span>, <span class="op">]</span> <span class="op">+</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSG00000142192"</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000142192"</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># merge mouse and human PS1 gene raw count</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>   <span class="op">%in%</span> <span class="st">"ENSMUSG00000019969"</span>, <span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000019969"</span>, <span class="op">]</span> <span class="op">+</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSG00000080815"</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000080815"</span><span class="op">)</span>, <span class="op">]</span></span></code></pre>
</div>
<p>Let’s verify if expression of both human genes have been merged or
not:</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">str_detect</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, <span class="st">"MUS"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 32043rh 32044rh 32046rh 32047rh 32048rh 32049rh
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<p>What proportion of genes have zero counts in all samples?</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sums</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>gene_id <span class="op">=</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>,</span>
<span>                        sums    <span class="op">=</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu">rowSums</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">sum</span><span class="op">(</span><span class="va">gene_sums</span><span class="op">$</span><span class="va">sums</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 9691</code></pre>
</div>
<p>We can see that 9,691 (17%) genes have no reads at all associated
with them. In the next lesson, we will remove genes that have no counts
in any samples.</p>
<div class="section level2">
<h2 id="differential-analysis-using-deseq2">Differential Analysis using DESeq2<a class="anchor" aria-label="anchor" href="#differential-analysis-using-deseq2"></a>
</h2>
<p>Now, after exploring and formatting the data, We will look for
differential expression between the control and 5xFAD mice at different
ages for both sexes. The differentially expressed genes (DEGs) can
inform our understanding of how the 5XFAD mutation affects biological
processes.</p>
<p>DESeq2 analysis consist of multiple steps. We are going to briefly
understand some of the important steps using a subset of data and then
we will perform differential analysis on the whole dataset.</p>
<p>First, order the data (so counts and metadata <code>specimenID</code>
orders match) and save as another variable name.</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rawdata</span>  <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span>, <span class="fu">sort</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="va">covars</span><span class="op">[</span><span class="fu">sort</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">covars</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span></code></pre>
</div>
<p>Subset the counts matrix and sample metadata to include only 12-month
old male mice. You can amend the code to compare wild type and 5XFAD
mice from either sex, at any time point.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta.12M.Male</span> <span class="op">&lt;-</span> <span class="va">metadata</span><span class="op">[</span><span class="op">(</span><span class="va">metadata</span><span class="op">$</span><span class="va">sex</span>           <span class="op">==</span> <span class="st">"male"</span> <span class="op">&amp;</span> </span>
<span>                           <span class="va">metadata</span><span class="op">$</span><span class="va">timepoint</span>     <span class="op">==</span> <span class="st">"12 mo"</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">meta.12M.Male</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>        individualID specimenID  sex         genotype timepoint
32044rh        32044    32044rh male 5XFAD_noncarrier     12 mo
32046rh        32046    32046rh male 5XFAD_noncarrier     12 mo
32047rh        32047    32047rh male 5XFAD_noncarrier     12 mo
32053rh        32053    32053rh male    5XFAD_carrier     12 mo
32059rh        32059    32059rh male    5XFAD_carrier     12 mo
32061rh        32061    32061rh male 5XFAD_noncarrier     12 mo
32062rh        32062    32062rh male    5XFAD_carrier     12 mo
32073rh        32073    32073rh male 5XFAD_noncarrier     12 mo
32074rh        32074    32074rh male 5XFAD_noncarrier     12 mo
32075rh        32075    32075rh male    5XFAD_carrier     12 mo
32088rh        32088    32088rh male    5XFAD_carrier     12 mo
32640rh        32640    32640rh male    5XFAD_carrier     12 mo</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">as.matrix</span><span class="op">(</span><span class="va">rawdata</span><span class="op">[</span> , <span class="fu">colnames</span><span class="op">(</span><span class="va">rawdata</span><span class="op">)</span> <span class="op">%in%</span></span>
<span>                            <span class="fu">rownames</span><span class="op">(</span><span class="va">meta.12M.Male</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu">colnames</span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "32044rh" "32046rh" "32047rh" "32053rh" "32059rh" "32061rh" "32062rh"
 [8] "32073rh" "32074rh" "32075rh" "32088rh" "32640rh"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rownames</span><span class="op">(</span><span class="va">meta.12M.Male</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "32044rh" "32046rh" "32047rh" "32053rh" "32059rh" "32061rh" "32062rh"
 [8] "32073rh" "32074rh" "32075rh" "32088rh" "32640rh"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">match</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">dat</span><span class="op">)</span>, <span class="fu">rownames</span><span class="op">(</span><span class="va">meta.12M.Male</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1]  1  2  3  4  5  6  7  8  9 10 11 12</code></pre>
</div>
<p>Next, we build the <code>DESeqDataSet</code> using the following
function:</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData <span class="op">=</span> <span class="va">dat</span>, </span>
<span>                                   colData   <span class="op">=</span> <span class="va">meta.12M.Male</span>, </span>
<span>                                   design    <span class="op">=</span> <span class="op">~</span> <span class="va">genotype</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>class: DESeqDataSet
dim: 55487 12
metadata(1): version
assays(1): counts
rownames(55487): ENSMUSG00000000001 ENSMUSG00000000003 ...
  ENSMUSG00000118487 ENSMUSG00000118488
rowData names(0):
colnames(12): 32044rh 32046rh ... 32088rh 32640rh
colData names(5): individualID specimenID sex genotype timepoint</code></pre>
</div>
<p><strong>Pre-filtering</strong></p>
<p>While it is not necessary to pre-filter low count genes before
running the DESeq2 functions, there are two reasons which make
pre-filtering useful: by removing rows in which there are very few
reads, we reduce the memory size of the dds data object, and we increase
the speed of the transformation and testing functions within DESeq2. It
can also improve visualizations, as features with no information for
differential expression are not plotted.</p>
<p>Here we perform a minimal pre-filtering to keep only rows that have
at least 10 reads total.</p>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="va">ddsHTSeq</span><span class="op">[</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">ddsHTSeq</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>class: DESeqDataSet
dim: 33059 12
metadata(1): version
assays(1): counts
rownames(33059): ENSMUSG00000000001 ENSMUSG00000000028 ...
  ENSMUSG00000118486 ENSMUSG00000118487
rowData names(0):
colnames(12): 32044rh 32046rh ... 32088rh 32640rh
colData names(5): individualID specimenID sex genotype timepoint</code></pre>
</div>
<p><strong>Reference level</strong></p>
<p>By default, R will choose a reference level for factors based on
alphabetical order. Then, if you never tell the <code>DESeq2</code>
functions which level you want to compare against (<em>e.g.</em> which
level represents the control group), the comparisons will be based on
the alphabetical order of the levels.</p>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># specifying the reference-level to `5XFAD_noncarrier`</span></span>
<span><span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">genotype</span> <span class="op">&lt;-</span> <span class="fu">relevel</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">genotype</span>, ref <span class="op">=</span> <span class="st">"5XFAD_noncarrier"</span><span class="op">)</span>  </span></code></pre>
</div>
<p>Run the standard differential expression analysis steps that is
wrapped into a single function, <code>DESeq</code>.</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">ddsHTSeq</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating size factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>gene-wise dispersion estimates: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>mean-dispersion relationship</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>final dispersion estimates, fitting model and testing: 2 workers</code></pre>
</div>
<p>Results tables are generated using the function results, which
extracts a results table with log2 fold changes, p-values and adjusted
p-values. By default the argument <code>alpha</code> is set to 0.1. If
the adjusted p-value cutoff will be a value other than 0.1, alpha should
be set to that value:</p>
<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">results</span><span class="op">(</span><span class="va">dds</span>, alpha<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span>  <span class="co"># setting 0.05 as significant threshold</span></span>
<span><span class="va">res</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>log2 fold change (MLE): genotype 5XFAD carrier vs 5XFAD noncarrier
Wald test p-value: genotype 5XFAD carrier vs 5XFAD noncarrier
DataFrame with 33059 rows and 6 columns
                    baseMean log2FoldChange     lfcSE      stat    pvalue
                   &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
ENSMUSG00000000001 3737.9009      0.0148125 0.0466948  0.317219 0.7510777
ENSMUSG00000000028  138.5635     -0.0712500 0.1550131 -0.459639 0.6457754
ENSMUSG00000000031   29.2983      0.6705922 0.3563442  1.881866 0.0598541
ENSMUSG00000000037  123.6482     -0.2184054 0.1554362 -1.405113 0.1599876
ENSMUSG00000000049   15.1733      0.3657555 0.3924376  0.932010 0.3513316
...                      ...            ...       ...       ...       ...
ENSMUSG00000118473   1.18647      -0.377971  1.531586 -0.246784  0.805075
ENSMUSG00000118477  59.10359      -0.144081  0.226690 -0.635586  0.525046
ENSMUSG00000118479  24.64566      -0.181992  0.341445 -0.533006  0.594029
ENSMUSG00000118486   1.92048       0.199838  1.253875  0.159376  0.873372
ENSMUSG00000118487  65.78311      -0.191362  0.218593 -0.875427  0.381342
                        padj
                   &lt;numeric&gt;
ENSMUSG00000000001  0.943421
ENSMUSG00000000028  0.913991
ENSMUSG00000000031  0.352346
ENSMUSG00000000037  0.566360
ENSMUSG00000000049  0.765640
...                      ...
ENSMUSG00000118473        NA
ENSMUSG00000118477  0.863565
ENSMUSG00000118479  0.893356
ENSMUSG00000118486        NA
ENSMUSG00000118487  0.785846</code></pre>
</div>
<p>We can order our results table by the smallest p-value:</p>
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">resOrdered</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">[</span><span class="fu">order</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">resOrdered</span>, n<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>log2 fold change (MLE): genotype 5XFAD carrier vs 5XFAD noncarrier
Wald test p-value: genotype 5XFAD carrier vs 5XFAD noncarrier
DataFrame with 10 rows and 6 columns
                     baseMean log2FoldChange     lfcSE      stat       pvalue
                    &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;
ENSMUSG00000019969  13860.942        1.90740 0.0432685   44.0828  0.00000e+00
ENSMUSG00000030579   2367.096        2.61215 0.0749326   34.8600 2.99982e-266
ENSMUSG00000046805   7073.296        2.12247 0.0635035   33.4229 6.38461e-245
ENSMUSG00000032011  80423.476        1.36195 0.0424007   32.1210 2.24203e-226
ENSMUSG00000022892 271265.838        1.36140 0.0434167   31.3567 7.88742e-216
ENSMUSG00000038642  10323.969        1.69717 0.0549488   30.8864 1.81875e-209
ENSMUSG00000023992   2333.227        2.62290 0.0882819   29.7105 5.61838e-194
ENSMUSG00000079293    761.313        5.12514 0.1738382   29.4822 4.86644e-191
ENSMUSG00000040552    617.149        2.22726 0.0781799   28.4889 1.60609e-178
ENSMUSG00000069516   2604.926        2.34471 0.0847390   27.6697 1.61630e-168
                           padj
                      &lt;numeric&gt;
ENSMUSG00000019969  0.00000e+00
ENSMUSG00000030579 3.60954e-262
ENSMUSG00000046805 5.12152e-241
ENSMUSG00000032011 1.34886e-222
ENSMUSG00000022892 3.79622e-212
ENSMUSG00000038642 7.29469e-206
ENSMUSG00000023992 1.93152e-190
ENSMUSG00000079293 1.46389e-187
ENSMUSG00000040552 4.29450e-175
ENSMUSG00000069516 3.88961e-165</code></pre>
</div>
<p>We can summarize some basic tallies using the summary function.</p>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
out of 33059 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 1098, 3.3%
LFC &lt; 0 (down)     : 505, 1.5%
outliers [1]       : 33, 0.1%
low counts [2]     : 8961, 27%
(mean count &lt; 8)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<p>How many adjusted p-values were less than 0.05?</p>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 1603</code></pre>
</div>
<p>How many adjusted p-values were less than 0.1?</p>
<div class="codewrapper sourceCode" id="cb62">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sum</span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.1</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 2001</code></pre>
</div>
<p><strong>Function to convert ensembleIDs to common gene
names</strong></p>
<p>We’ll use a package to translate mouse ENSEMBL IDS to gene names. Run
this function and they will be called up when assembling results from
the differential expression analysis.</p>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">map_function.df</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">inputtype</span>, <span class="va">outputtype</span><span class="op">)</span> <span class="op">{</span></span>
<span>                            <span class="fu">mapIds</span><span class="op">(</span> <span class="va">org.Mm.eg.db</span>,</span>
<span>                                    keys      <span class="op">=</span> <span class="fu">row.names</span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,</span>
<span>                                    column    <span class="op">=</span> <span class="va">outputtype</span>,</span>
<span>                                    keytype   <span class="op">=</span> <span class="va">inputtype</span>,</span>
<span>                                    multiVals <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span></span>
<span>                            <span class="op">}</span></span></code></pre>
</div>
<p><strong>Generating Results table</strong></p>
<p>Here we will call the function to get the <code>symbol</code> names
of the genes incorporated into the results table, along with the columns
we are most interested in.</p>
<div class="codewrapper sourceCode" id="cb65">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">All_res</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>           </span>
<span>           <span class="co"># run map_function to add symbol of gene corresponding to ENSEMBL ID</span></span>
<span>           <span class="fu">mutate</span><span class="op">(</span>symbol <span class="op">=</span> <span class="fu">map_function.df</span><span class="op">(</span><span class="va">res</span>, <span class="st">"ENSEMBL"</span>, <span class="st">"SYMBOL"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>    </span>
<span>           </span>
<span>           <span class="co"># run map_function to add Entrez ID of gene corresponding to ENSEMBL ID</span></span>
<span>           <span class="fu">mutate</span><span class="op">(</span>EntrezGene <span class="op">=</span> <span class="fu">map_function.df</span><span class="op">(</span><span class="va">res</span>, <span class="st">"ENSEMBL"</span>, <span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>  </span>
<span>           <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="st">"symbol"</span>,</span>
<span>                         <span class="st">"EntrezGene"</span>,</span>
<span>                         <span class="st">"baseMean"</span>, </span>
<span>                         <span class="st">"log2FoldChange"</span>, </span>
<span>                         <span class="st">"lfcSE"</span>, </span>
<span>                         <span class="st">"stat"</span>, </span>
<span>                         <span class="st">"pvalue"</span>, </span>
<span>                         <span class="st">"padj"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb67">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">All_res</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                   symbol EntrezGene   baseMean log2FoldChange      lfcSE
ENSMUSG00000000001  Gnai3      14679 3737.90089     0.01481247 0.04669481
ENSMUSG00000000028  Cdc45      12544  138.56354    -0.07125004 0.15501305
ENSMUSG00000000031    H19      14955   29.29832     0.67059217 0.35634418
ENSMUSG00000000037  Scml2     107815  123.64823    -0.21840544 0.15543617
ENSMUSG00000000049   Apoh      11818   15.17325     0.36575555 0.39243756
ENSMUSG00000000056   Narf      67608 5017.30216    -0.06713961 0.04466809
                         stat     pvalue      padj
ENSMUSG00000000001  0.3172187 0.75107766 0.9434210
ENSMUSG00000000028 -0.4596390 0.64577537 0.9139907
ENSMUSG00000000031  1.8818665 0.05985415 0.3523459
ENSMUSG00000000037 -1.4051134 0.15998756 0.5663602
ENSMUSG00000000049  0.9320095 0.35133160 0.7656400
ENSMUSG00000000056 -1.5030778 0.13281898 0.5203335</code></pre>
</div>
<p><strong>Extracting genes that are significantly
expressed</strong></p>
<p>Let’s subset all the genes with a p-value &lt; 0.05.</p>
<div class="codewrapper sourceCode" id="cb69">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dseq_res</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">All_res</span><span class="op">[</span><span class="fu">order</span><span class="op">(</span><span class="va">All_res</span><span class="op">$</span><span class="va">padj</span><span class="op">)</span>, <span class="op">]</span>, <span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre>
</div>
<p>Wow! We have a lot of genes with apparently very strong statistically
significant differences between the control and 5xFAD carrier.</p>
<div class="codewrapper sourceCode" id="cb70">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">dseq_res</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 1603    8</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb72">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">dseq_res</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                   symbol EntrezGene   baseMean log2FoldChange      lfcSE
ENSMUSG00000019969  Psen1      19164  13860.942       1.907397 0.04326854
ENSMUSG00000030579 Tyrobp      22177   2367.096       2.612152 0.07493257
ENSMUSG00000046805  Mpeg1      17476   7073.296       2.122467 0.06350348
ENSMUSG00000032011   Thy1      21838  80423.476       1.361953 0.04240065
ENSMUSG00000022892    App      11820 271265.838       1.361405 0.04341673
ENSMUSG00000038642   Ctss      13040  10323.969       1.697172 0.05494883
                       stat        pvalue          padj
ENSMUSG00000019969 44.08277  0.000000e+00  0.000000e+00
ENSMUSG00000030579 34.86004 2.999825e-266 3.609539e-262
ENSMUSG00000046805 33.42285 6.384608e-245 5.121519e-241
ENSMUSG00000032011 32.12104 2.242032e-226 1.348863e-222
ENSMUSG00000022892 31.35669 7.887422e-216 3.796216e-212
ENSMUSG00000038642 30.88640 1.818747e-209 7.294691e-206</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="exploring-and-exporting-results">Exploring and exporting results<a class="anchor" aria-label="anchor" href="#exploring-and-exporting-results"></a>
</h2>
<p><strong>Exporting results to CSV files</strong></p>
<p>We can save results file into a csv file like this:</p>
<div class="codewrapper sourceCode" id="cb74">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">write.csv</span><span class="op">(</span><span class="va">All_res</span>, file<span class="op">=</span><span class="st">"results/All_5xFAD_12months_male.csv"</span><span class="op">)</span></span>
<span><span class="fu">write.csv</span><span class="op">(</span><span class="va">dseq_res</span>, file<span class="op">=</span><span class="st">"results/DEG_5xFAD_12months_male.csv"</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Volcano plot</strong></p>
<p>We can visualize the differential expression results using the
volcano plot function from <code>EnhancedVolcano</code> package. For the
most basic volcano plot, only a single data frame, data matrix, or
tibble of test results is required, containing point labels, log2FC, and
adjusted or unadjusted p-values. The default cut-off for log2FC is
&gt;|2|; the default cut-off for p-value is 10e-6.</p>
<div class="codewrapper sourceCode" id="cb75">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">EnhancedVolcano</span><span class="op">(</span><span class="va">All_res</span>,</span>
<span>                lab            <span class="op">=</span> <span class="op">(</span><span class="va">All_res</span><span class="op">$</span><span class="va">symbol</span><span class="op">)</span>,</span>
<span>                x              <span class="op">=</span> <span class="st">'log2FoldChange'</span>,</span>
<span>                y              <span class="op">=</span> <span class="st">'padj'</span>,</span>
<span>                legendPosition <span class="op">=</span> <span class="st">'none'</span>,</span>
<span>                title          <span class="op">=</span> <span class="st">'Volcano plot:Differential Expression Results'</span>,</span>
<span>                subtitle       <span class="op">=</span> <span class="st">''</span>,</span>
<span>                FCcutoff       <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                pCutoff        <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                xlim           <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">6</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: One or more p-values is 0. Converting to 10^-1 * current lowest
non-zero p-value...</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
ℹ The deprecated feature was likely used in the EnhancedVolcano package.
  Please report the issue to the authors.
This warning is displayed once per session.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.
ℹ The deprecated feature was likely used in the EnhancedVolcano package.
  Please report the issue to the authors.
This warning is displayed once per session.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
</div>
<figure><img src="fig/differential-expression-analysis-rendered-unnamed-chunk-41-1.png" alt="" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>You can see that some top significantly expressed are
immune/inflammation-related genes such as Ctsd, C4b, Csf1 etc. These
genes are upregulated in the 5XFAD strain.</p>
<p><strong>Principal component plot of the samples</strong></p>
<p>Principal component analysis is a dimension reduction technique that
reduces the dimensionality of these large matrixes into a linear
coordinate system, so that we can more easily visualize what factors are
contributing the most to variation in the dataset by graphing the
principal components.</p>
<div class="codewrapper sourceCode" id="cb79">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData <span class="op">=</span> <span class="fu">as.matrix</span><span class="op">(</span><span class="va">rawdata</span><span class="op">)</span>, </span>
<span>                                   colData   <span class="op">=</span> <span class="va">metadata</span>, </span>
<span>                                   design    <span class="op">=</span> <span class="op">~</span> <span class="va">genotype</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb81">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="va">ddsHTSeq</span><span class="op">[</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span>, <span class="op">]</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">ddsHTSeq</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating size factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>gene-wise dispersion estimates: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>mean-dispersion relationship</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>final dispersion estimates, fitting model and testing: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>-- replacing outliers and refitting for 42 genes
-- DESeq argument 'minReplicatesForReplace' = 7
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>fitting model and testing</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb90">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vsd</span> <span class="op">&lt;-</span> <span class="fu">varianceStabilizingTransformation</span><span class="op">(</span><span class="va">dds</span>, blind <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">plotPCA</span><span class="op">(</span><span class="va">vsd</span>, intgroup <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"genotype"</span>, <span class="st">"sex"</span>, <span class="st">"timepoint"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>using ntop=500 top features by variance</code></pre>
</div>
<figure><img src="fig/differential-expression-analysis-rendered-unnamed-chunk-44-1.png" alt="" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We can see that clustering is occurring, though it’s kind of hard to
see exactly how they are clustering in this visualization.</p>
<p>It is also possible to customize the PCA plot using the
<code>ggplot</code> function.</p>
<div class="codewrapper sourceCode" id="cb92">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pcaData</span> <span class="op">&lt;-</span> <span class="fu">plotPCA</span><span class="op">(</span><span class="va">vsd</span>, </span>
<span>                   intgroup   <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"genotype"</span>, <span class="st">"sex"</span>,<span class="st">"timepoint"</span><span class="op">)</span>,</span>
<span>                   returnData <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb94">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">percentVar</span> <span class="op">&lt;-</span> <span class="fu">round</span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu">attr</span><span class="op">(</span><span class="va">pcaData</span>, <span class="st">"percentVar"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">pcaData</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">PC1</span>, <span class="va">PC2</span>,color <span class="op">=</span> <span class="va">genotype</span>, shape <span class="op">=</span> <span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>       <span class="fu">geom_point</span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">geom_text</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">timepoint</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0.5</span>, vjust<span class="op">=</span><span class="fl">2</span>, size <span class="op">=</span><span class="fl">3.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="fu">paste0</span><span class="op">(</span><span class="st">"PC1: "</span>, <span class="va">percentVar</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="st">"% variance"</span><span class="op">)</span>, </span>
<span>            y <span class="op">=</span> <span class="fu">paste0</span><span class="op">(</span><span class="st">"PC2: "</span>, <span class="va">percentVar</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="st">"% variance"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/differential-expression-analysis-rendered-unnamed-chunk-45-1.png" alt="" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>PCA identified genotype and sex being a major source of variation in
between 5XFAD and WT mice. Female and male samples from the 5XFAD
carriers clustered distinctly at all ages, suggesting the presence of
sex-biased molecular changes in animals.</p>
<p><strong>Function for Differential analysis using DESeq2</strong></p>
<p>Finally, we can build a function for differential analysis that
consists of all above discussed steps. It will require to input sorted
raw count matrix, sample metadata and define the reference group.</p>
<div class="codewrapper sourceCode" id="cb95">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DEG</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">rawdata</span>, <span class="va">meta</span>, <span class="va">include.batch</span> <span class="op">=</span> <span class="cn">FALSE</span>, <span class="va">ref</span> <span class="op">=</span> <span class="va">ref</span><span class="op">)</span> <span class="op">{</span></span>
<span>                <span class="va">dseq_res</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span><span class="op">)</span></span>
<span>                <span class="va">All_res</span>  <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span><span class="op">)</span></span>
<span>      </span>
<span>                <span class="kw">if</span> <span class="op">(</span><span class="va">include.batch</span><span class="op">)</span> <span class="op">{</span></span>
<span>                    <span class="fu">cat</span><span class="op">(</span><span class="st">"Including batch as covariate\n"</span><span class="op">)</span></span>
<span>                    <span class="va">design_formula</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">Batch</span> <span class="op">+</span> <span class="va">genotype</span></span>
<span>                    <span class="op">}</span></span>
<span>                <span class="kw">else</span> <span class="op">{</span></span>
<span>                    <span class="va">design_formula</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">genotype</span></span>
<span>                    <span class="op">}</span></span>
<span>  </span>
<span>                <span class="va">dat2</span> <span class="op">&lt;-</span> <span class="fu">as.matrix</span><span class="op">(</span><span class="va">rawdata</span><span class="op">[</span>, <span class="fu">colnames</span><span class="op">(</span><span class="va">rawdata</span><span class="op">)</span> <span class="op">%in%</span> </span>
<span>                                            <span class="fu">rownames</span><span class="op">(</span><span class="va">meta</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span>  </span>
<span>                <span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData <span class="op">=</span> <span class="va">dat2</span>,</span>
<span>                            colData <span class="op">=</span> <span class="va">meta</span>,</span>
<span>                            design <span class="op">=</span> <span class="va">design_formula</span><span class="op">)</span></span>
<span></span>
<span>                <span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="va">ddsHTSeq</span><span class="op">[</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span>, <span class="op">]</span></span>
<span>  </span>
<span>                <span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">genotype</span> <span class="op">&lt;-</span> <span class="fu">relevel</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">genotype</span>, ref <span class="op">=</span> <span class="va">ref</span><span class="op">)</span></span>
<span> </span>
<span>                <span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">ddsHTSeq</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  </span>
<span>                <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">results</span><span class="op">(</span><span class="va">dds</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span>                <span class="co">#summary(res)</span></span>
<span>  </span>
<span>                <span class="va">res</span><span class="op">$</span><span class="va">symbol</span> <span class="op">&lt;-</span> <span class="fu">map_function.df</span><span class="op">(</span><span class="va">res</span>,<span class="st">"ENSEMBL"</span>,<span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span>  </span>
<span>                <span class="va">res</span><span class="op">$</span><span class="va">EntrezGene</span> <span class="op">&lt;-</span> <span class="fu">map_function.df</span><span class="op">(</span><span class="va">res</span>,<span class="st">"ENSEMBL"</span>,<span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span>  </span>
<span>                <span class="va">All_res</span> <span class="op">&lt;&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">res</span><span class="op">[</span>, <span class="fu">c</span><span class="op">(</span><span class="st">"symbol"</span>, </span>
<span>                                                  <span class="st">"EntrezGene"</span>,</span>
<span>                                                  <span class="st">"baseMean"</span>, </span>
<span>                                                  <span class="st">"log2FoldChange"</span>, </span>
<span>                                                  <span class="st">"lfcSE"</span>, </span>
<span>                                                  <span class="st">"stat"</span>, </span>
<span>                                                  <span class="st">"pvalue"</span>, </span>
<span>                                                  <span class="st">"padj"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span>              <span class="op">}</span></span></code></pre>
</div>
<p>Let’s use this function to analyze all groups present in our
data.</p>
<p><strong>Differential Analysis of all groups</strong></p>
<p>First, we add a <code>Group</code> column to our metadata table that
will combine all variables of interest (<code>genotype</code>,
<code>sex</code>, and <code>timepoint</code>) for each sample.</p>
<div class="codewrapper sourceCode" id="cb96">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span><span class="op">$</span><span class="va">Group</span> <span class="op">&lt;-</span> <span class="fu">paste0</span><span class="op">(</span><span class="va">metadata</span><span class="op">$</span><span class="va">genotype</span>,</span>
<span>                         <span class="st">"-"</span>,</span>
<span>                         <span class="va">metadata</span><span class="op">$</span><span class="va">sex</span>,</span>
<span>                         <span class="st">"-"</span>,</span>
<span>                         <span class="va">metadata</span><span class="op">$</span><span class="va">timepoint</span><span class="op">)</span></span>
<span><span class="fu">unique</span><span class="op">(</span><span class="va">metadata</span><span class="op">$</span><span class="va">Group</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "5XFAD_carrier-female-12 mo"    "5XFAD_noncarrier-male-12 mo"
 [3] "5XFAD_noncarrier-female-12 mo" "5XFAD_carrier-male-12 mo"
 [5] "5XFAD_noncarrier-female-6 mo"  "5XFAD_noncarrier-male-6 mo"
 [7] "5XFAD_carrier-female-6 mo"     "5XFAD_noncarrier-female-4 mo"
 [9] "5XFAD_carrier-female-4 mo"     "5XFAD_carrier-male-6 mo"
[11] "5XFAD_carrier-male-4 mo"       "5XFAD_noncarrier-male-4 mo"   </code></pre>
</div>
<p>Next, we create a comparison table that has all cases and controls
that we would like to compare with each other. Here I have made
comparison groups for age and sex-matched 5xFAD carriers vs
5xFAD_noncarriers, with carriers as the cases and noncarriers as the
controls:</p>
<div class="codewrapper sourceCode" id="cb98">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comparisons</span> <span class="op">&lt;-</span>  <span class="fu">data.frame</span><span class="op">(</span>control <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"5XFAD_noncarrier-male-4 mo"</span>, </span>
<span>                                       <span class="st">"5XFAD_noncarrier-female-4 mo"</span>, </span>
<span>                                       <span class="st">"5XFAD_noncarrier-male-6 mo"</span>, </span>
<span>                                       <span class="st">"5XFAD_noncarrier-female-6 mo"</span>,</span>
<span>                                       <span class="st">"5XFAD_noncarrier-male-12 mo"</span>, </span>
<span>                                       <span class="st">"5XFAD_noncarrier-female-12 mo"</span><span class="op">)</span>, </span>
<span>                           case    <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"5XFAD_carrier-male-4 mo"</span>,</span>
<span>                                       <span class="st">"5XFAD_carrier-female-4 mo"</span>,</span>
<span>                                       <span class="st">"5XFAD_carrier-male-6 mo"</span>, </span>
<span>                                       <span class="st">"5XFAD_carrier-female-6 mo"</span>,</span>
<span>                                       <span class="st">"5XFAD_carrier-male-12 mo"</span>,</span>
<span>                                       <span class="st">"5XFAD_carrier-female-12 mo"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb99">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comparisons</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                        control                       case
1    5XFAD_noncarrier-male-4 mo    5XFAD_carrier-male-4 mo
2  5XFAD_noncarrier-female-4 mo  5XFAD_carrier-female-4 mo
3    5XFAD_noncarrier-male-6 mo    5XFAD_carrier-male-6 mo
4  5XFAD_noncarrier-female-6 mo  5XFAD_carrier-female-6 mo
5   5XFAD_noncarrier-male-12 mo   5XFAD_carrier-male-12 mo
6 5XFAD_noncarrier-female-12 mo 5XFAD_carrier-female-12 mo</code></pre>
</div>
<p>Finally, we implement our <code>DEG</code> function on each
case/control comparison of interest and store the result table in a list
and data frame:</p>
<div class="codewrapper sourceCode" id="cb101">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># initiate an empty list and data frame to save results</span></span>
<span><span class="va">DE_5xFAD.list</span> <span class="op">&lt;-</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">DE_5xFAD.df</span>   <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu">nrow</span><span class="op">(</span><span class="va">comparisons</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">{</span></span>
<span>    <span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">metadata</span><span class="op">[</span><span class="va">metadata</span><span class="op">$</span><span class="va">Group</span> <span class="op">%in%</span> <span class="va">comparisons</span><span class="op">[</span><span class="va">i</span>,<span class="op">]</span>,<span class="op">]</span></span>
<span>    <span class="fu">DEG</span><span class="op">(</span><span class="va">rawdata</span>, <span class="va">meta</span>, ref <span class="op">=</span> <span class="st">"5XFAD_noncarrier"</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="co"># append results in data frame</span></span>
<span>    <span class="va">DE_5xFAD.df</span> <span class="op">&lt;-</span> <span class="fu">rbind</span><span class="op">(</span><span class="va">DE_5xFAD.df</span>,</span>
<span>                         <span class="va">All_res</span> <span class="op">%&gt;%</span> </span>
<span>                         <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="st">"5xFAD"</span>,</span>
<span>                                sex   <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span>,</span>
<span>                                age   <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">timepoint</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="co"># append results in list</span></span>
<span>    <span class="va">DE_5xFAD.list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">All_res</span></span>
<span>    <span class="fu">names</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">paste0</span><span class="op">(</span><span class="va">comparisons</span><span class="op">[</span><span class="va">i</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating size factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>gene-wise dispersion estimates: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>mean-dispersion relationship</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>final dispersion estimates, fitting model and testing: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating size factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>gene-wise dispersion estimates: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>mean-dispersion relationship</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>final dispersion estimates, fitting model and testing: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating size factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>gene-wise dispersion estimates: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>mean-dispersion relationship</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>final dispersion estimates, fitting model and testing: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating size factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>gene-wise dispersion estimates: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>mean-dispersion relationship</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>final dispersion estimates, fitting model and testing: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating size factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>gene-wise dispersion estimates: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>mean-dispersion relationship</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>final dispersion estimates, fitting model and testing: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating size factors</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>estimating dispersions</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>gene-wise dispersion estimates: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>mean-dispersion relationship</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>final dispersion estimates, fitting model and testing: 2 workers</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<p>Let’s explore the result stored in our list:</p>
<div class="codewrapper sourceCode" id="cb144">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "5XFAD_carrier-male-4 mo"    "5XFAD_carrier-female-4 mo"
[3] "5XFAD_carrier-male-6 mo"    "5XFAD_carrier-female-6 mo"
[5] "5XFAD_carrier-male-12 mo"   "5XFAD_carrier-female-12 mo"</code></pre>
</div>
<p>We can easily extract the result table for any group of interest by
using $ and name of group. Let’s check top few rows from
<code>5XFAD_carrier-male-4 mo</code> group:</p>
<div class="codewrapper sourceCode" id="cb146">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">$</span><span class="va">`5XFAD_carrier-male-4 mo`</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                   symbol EntrezGene   baseMean log2FoldChange      lfcSE
ENSMUSG00000000001  Gnai3      14679 3707.53159   -0.023085867 0.03816431
ENSMUSG00000000028  Cdc45      12544  159.76225   -0.009444942 0.13226126
ENSMUSG00000000031    H19      14955   35.96987    0.453401511 0.27852555
ENSMUSG00000000037  Scml2     107815  126.82414    0.089394568 0.13774048
ENSMUSG00000000049   Apoh      11818   19.99721    0.115325773 0.31548606
ENSMUSG00000000056   Narf      67608 5344.21741   -0.100413295 0.03811800
                          stat      pvalue      padj
ENSMUSG00000000001 -0.60490724 0.545240632 0.9999514
ENSMUSG00000000028 -0.07141125 0.943070457 0.9999514
ENSMUSG00000000031  1.62786329 0.103553876 0.9999514
ENSMUSG00000000037  0.64900725 0.516333691 0.9999514
ENSMUSG00000000049  0.36554951 0.714701258 0.9999514
ENSMUSG00000000056 -2.63427474 0.008431723 0.5696567</code></pre>
</div>
<p>Let’s check the result stored as dataframe:</p>
<div class="codewrapper sourceCode" id="cb148">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">DE_5xFAD.df</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                   symbol EntrezGene   baseMean log2FoldChange      lfcSE
ENSMUSG00000000001  Gnai3      14679 3707.53159   -0.023085867 0.03816431
ENSMUSG00000000028  Cdc45      12544  159.76225   -0.009444942 0.13226126
ENSMUSG00000000031    H19      14955   35.96987    0.453401511 0.27852555
ENSMUSG00000000037  Scml2     107815  126.82414    0.089394568 0.13774048
ENSMUSG00000000049   Apoh      11818   19.99721    0.115325773 0.31548606
ENSMUSG00000000056   Narf      67608 5344.21741   -0.100413295 0.03811800
                          stat      pvalue      padj model  sex  age
ENSMUSG00000000001 -0.60490724 0.545240632 0.9999514 5xFAD male 4 mo
ENSMUSG00000000028 -0.07141125 0.943070457 0.9999514 5xFAD male 4 mo
ENSMUSG00000000031  1.62786329 0.103553876 0.9999514 5xFAD male 4 mo
ENSMUSG00000000037  0.64900725 0.516333691 0.9999514 5xFAD male 4 mo
ENSMUSG00000000049  0.36554951 0.714701258 0.9999514 5xFAD male 4 mo
ENSMUSG00000000056 -2.63427474 0.008431723 0.5696567 5xFAD male 4 mo</code></pre>
</div>
<p>Check if result is present for all ages:</p>
<div class="codewrapper sourceCode" id="cb150">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">unique</span><span class="op">(</span><span class="op">(</span><span class="va">DE_5xFAD.df</span><span class="op">$</span><span class="va">age</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "4 mo"  "6 mo"  "12 mo"</code></pre>
</div>
<p>Check if result is present for both sexes:</p>
<div class="codewrapper sourceCode" id="cb152">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">unique</span><span class="op">(</span><span class="op">(</span><span class="va">DE_5xFAD.df</span><span class="op">$</span><span class="va">sex</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "male"   "female"</code></pre>
</div>
<p>Check number of genes in each group:</p>
<div class="codewrapper sourceCode" id="cb154">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">DE_5xFAD.df</span>, <span class="va">model</span>, <span class="va">sex</span>, <span class="va">age</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  model    sex   age     n
1 5xFAD female 12 mo 33120
2 5xFAD female  4 mo 32930
3 5xFAD female  6 mo 33249
4 5xFAD   male 12 mo 33059
5 5xFAD   male  4 mo 33119
6 5xFAD   male  6 mo 33375</code></pre>
</div>
<p>Check number of genes significantly differentially expressed in all
cases compared to age and sex-matched controls:</p>
<div class="codewrapper sourceCode" id="cb156">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">degs.up</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">DE_5xFAD.list</span>,</span>
<span>               <span class="op">~</span> <span class="fu">length</span><span class="op">(</span><span class="fu">which</span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">padj</span>          <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span></span>
<span>                             <span class="va">.x</span><span class="op">$</span><span class="va">log2FoldChange</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">degs.down</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">DE_5xFAD.list</span>,</span>
<span>                 <span class="op">~</span> <span class="fu">length</span><span class="op">(</span><span class="fu">which</span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">padj</span>           <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> </span>
<span>                                <span class="va">.x</span><span class="op">$</span><span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">deg</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>Case              <span class="op">=</span> <span class="fu">names</span><span class="op">(</span><span class="va">degs.up</span><span class="op">)</span>,</span>
<span>                  Up_DEGs.pval.05   <span class="op">=</span> <span class="fu">as.vector</span><span class="op">(</span><span class="fu">unlist</span><span class="op">(</span><span class="va">degs.up</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  Down_DEGs.pval.05 <span class="op">=</span> <span class="fu">as.vector</span><span class="op">(</span><span class="fu">unlist</span><span class="op">(</span><span class="va">degs.down</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu">kable</span><span class="op">(</span><span class="va">deg</span><span class="op">)</span></span></code></pre>
</div>
<table class="table">
<thead><tr>
<th align="left">Case</th>
<th align="right">Up_DEGs.pval.05</th>
<th align="right">Down_DEGs.pval.05</th>
</tr></thead>
<tbody>
<tr>
<td align="left">5XFAD_carrier-male-4 mo</td>
<td align="right">86</td>
<td align="right">11</td>
</tr>
<tr>
<td align="left">5XFAD_carrier-female-4 mo</td>
<td align="right">522</td>
<td align="right">90</td>
</tr>
<tr>
<td align="left">5XFAD_carrier-male-6 mo</td>
<td align="right">714</td>
<td align="right">488</td>
</tr>
<tr>
<td align="left">5XFAD_carrier-female-6 mo</td>
<td align="right">1081</td>
<td align="right">409</td>
</tr>
<tr>
<td align="left">5XFAD_carrier-male-12 mo</td>
<td align="right">1098</td>
<td align="right">505</td>
</tr>
<tr>
<td align="left">5XFAD_carrier-female-12 mo</td>
<td align="right">1494</td>
<td align="right">1023</td>
</tr>
</tbody>
</table>
<p>Interestingly, in females more genes are differentially expressed at
all age groups, and more genes are differentially expressed the older
the mice get in both sexes.</p>
</div>
<div class="section level2">
<h2 id="pathway-enrichment">Pathway Enrichment<a class="anchor" aria-label="anchor" href="#pathway-enrichment"></a>
</h2>
<p>We may wish to look for enrichment of biological pathways in a list
of differentially expressed genes. Here we will test for enrichment of
KEGG pathways using using the <code>enrichKEGG</code> function in the
<code>clusterProfiler</code> package.</p>
<div class="codewrapper sourceCode" id="cb157">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">list</span><span class="op">(</span>FAD_M_4 <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">$</span><span class="va">`5XFAD_carrier-male-4 mo`</span><span class="op">[</span><span class="fu">order</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">$</span><span class="va">`5XFAD_carrier-male-4 mo`</span><span class="op">$</span><span class="va">padj</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>                             <span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>            <span class="fu">pull</span><span class="op">(</span><span class="va">EntrezGene</span><span class="op">)</span>,</span>
<span>                 FAD_F_4 <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">$</span><span class="va">`5XFAD_carrier-female-4 mo`</span><span class="op">[</span><span class="fu">order</span><span class="op">(</span><span class="va">DE_5xFAD.list</span><span class="op">$</span><span class="va">`5XFAD_carrier-female-4 mo`</span><span class="op">$</span><span class="va">padj</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>                                  <span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>            <span class="fu">pull</span><span class="op">(</span><span class="va">EntrezGene</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># perform enrichment analysis</span></span>
<span><span class="va">enrich_pathway</span> <span class="op">&lt;-</span> <span class="fu">compareCluster</span><span class="op">(</span><span class="va">dat</span>,</span>
<span>                                 fun          <span class="op">=</span> <span class="st">"enrichKEGG"</span>,</span>
<span>                                 pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                                 organism     <span class="op">=</span> <span class="st">"mmu"</span></span>
<span>                                 <span class="op">)</span></span>
<span></span>
<span><span class="va">enrich_pathway</span><span class="op">@</span><span class="va">compareClusterResult</span><span class="op">$</span><span class="va">Description</span> <span class="op">&lt;-</span> </span>
<span>                          <span class="fu">gsub</span><span class="op">(</span><span class="st">" - Mus musculus \\(house mouse)"</span>,</span>
<span>                               <span class="st">""</span>,</span>
<span>                               <span class="va">enrich_pathway</span><span class="op">@</span><span class="va">compareClusterResult</span><span class="op">$</span><span class="va">Description</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s plot top enriched functions using the <code>dotplot</code>
function of the <code>clusterProfiler</code> package.</p>
<div class="codewrapper sourceCode" id="cb158">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu">dotplot</span><span class="op">(</span><span class="va">enrich_pathway</span>,</span>
<span>                         showCategory <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                         font.size    <span class="op">=</span> <span class="fl">14</span>,</span>
<span>                         title        <span class="op">=</span> <span class="st">"Enriched KEGG Pathways"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/differential-expression-analysis-rendered-unnamed-chunk-59-1.png" alt="" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>What does this plot infer?</p>
</div>
<div class="section level2">
<h2 id="save-data-for-next-lesson">Save Data for Next Lesson<a class="anchor" aria-label="anchor" href="#save-data-for-next-lesson"></a>
</h2>
<p>We will use the results data in the next lesson. Save it now and we
will load it at the beginning of the next lesson. We will use R’s
<code>save</code> command to save the objects in compressed, binary
format. The <code>save</code> command is useful when you want to save
multiple objects in one file.</p>
<div class="codewrapper sourceCode" id="cb159">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">save</span><span class="op">(</span><span class="va">DE_5xFAD.df</span>, <span class="va">DE_5xFAD.list</span>, file <span class="op">=</span> <span class="st">"results/DEAnalysis_5XFAD.Rdata"</span><span class="op">)</span></span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul><li>
</li></ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</div></section><section id="aio-mouse-human-alignment"><p>Content from <a href="mouse-human-alignment.html">Cross Species Functional Alignment</a></p>
<hr>
<p>Last updated on 2026-02-12 |

        <a href="https://github.com/asliuyar/translational-omics-AD/edit/main/episodes/mouse-human-alignment.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do we perfrom a cross-species comparison?</li>
<li>What transcriptomic changes do we observe in mouse models?</li>
<li>Which aspects of disease does a model capture?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Approaches to align mouse data to human data</li>
<li>Understand the human AD co-expression modules</li>
<li>Understand the data from AD mouse models</li>
<li>Perform differential analysis using DESeq2</li>
<li>Perform correlation analysis between mouse models and human
modules</li>
<li>Understand the biological domains and subdomains of AD</li>
<li>Use domain annotations to compare between species</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Author: Gregory Cary, Jackson Laboratory</p>
<section><h2 class="section-heading" id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<hr class="half-width">
<p>First, let’s install the necessary packages into the workspace,
starting with the annotation packages:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Mouse and Human annotation databases</span></span>
<span><span class="co"># BiocManager::install(c('org.Mm.eg.db','org.Hs.eg.db', 'GO.db'))</span></span></code></pre>
</div>
<p>Next we’ll install the <code>fgsea</code> and
<code>clusterProfiler</code> packages, which will be used for GO term
enrichment.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># fgsea package</span></span>
<span><span class="co"># BiocManager::install(c('fgsea', 'clusterProfiler'))</span></span></code></pre>
</div>
<p>Finally we can instally the <code>synapseclient</code> python package
using the <code>reticulate</code> R package, this will enable us
comand-line access to objects in the Synapse data repository.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># synapseclient python package</span></span>
<span><span class="co"># reticulate::py_install('synapseclient')</span></span></code></pre>
</div>
<p>Finally we’ll need to generate a personal access token on Synapse.
Login to Synapse, go to
<code>Your Account &gt; Account Settings &gt; Personal Access Tokens &gt; Manage Personal Access Tokens</code>.
Click on <code>Create New Token</code> and make sure to enable both View
and Download permissions. Paste the resulting PAT in the quotes below
and save it to your workspace.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># synToken &lt;- </span></span></code></pre>
</div>
<p><strong>NOTE</strong>: This is for learning purposes only. Storing
your PAT in an environmental variable in your workspace is not secure
and you could easily loose access to the PAT. The preferred method is to
store it in a separate file called <code>.synapseConfig</code> under
your home directory.</p>
<p>Now we should be able to login to Synapse through our R session using
the following commands:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># import the synapseclient python package</span></span>
<span><span class="va">syn.client</span> <span class="op">&lt;-</span> <span class="fu">reticulate</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'synapseclient'</span><span class="op">)</span></span>
<span><span class="va">syn</span>        <span class="op">&lt;-</span> <span class="va">syn.client</span><span class="op">$</span><span class="fu">Synapse</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># log in to Synapse</span></span>
<span><span class="co"># syn$login('', authToken = synToken)</span></span></code></pre>
</div>
<p>Finally, let’s load the R packages we’ll need for today’s lesson:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load necessary libraries for the analysis</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">DESeq2</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">clusterProfiler</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">fgsea</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># set ggplot plotting theme (personal preference)</span></span>
<span><span class="fu">theme_set</span><span class="op">(</span> <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="aligning-human-and-mouse-phenotypes">[1] Aligning Human and Mouse Phenotypes<a class="anchor" aria-label="anchor" href="#aligning-human-and-mouse-phenotypes"></a>
</h2>
<hr class="half-width">
<p>Alzheimer’s Disease (AD) is complex, and we can not expect any single
animal model to fully recapitulate all aspects of late onset AD (LOAD)
pathology. To study AD with animal models we must find dimensions
through which we can align phenotypes between the models and human
cohorts. In MODEL-AD we use the following data modalities to identify
commonalities between mouse models and human cohorts:</p>
<ol style="list-style-type: decimal">
<li>Imaging (i.e. MRI and PET) to correspond with human imaging studies
(e.g. <a href="https://adni.loni.usc.edu/" class="external-link">ADNI</a>)<br>
</li>
<li>Neuropathological and biomarker phenotypes<br>
</li>
<li>
<em>Lots</em> of ’omics — genomics, proteomics, and
metabolomics</li>
</ol>
<p>The ’omics comparisons allow for very rich comparisons because a
significant proportion of genes are shared between these two species.
Furthermore, homology at the anatomical and neuropathological levels is
less clear.</p>
<figure><img src="fig/dimensions_of_model_alignment.png" alt="image showing humans and mice along with neuroimaging, neuropathology, and heatmap images representing the different modalities to align." class="figure mx-auto d-block"><div class="figcaption">Aligning Human and Mouse Phenotypes</div>
</figure><p>In this session we will explore several ways to compare ’omics
signatures between human AD patients and mouse models. We’ll focus on
transcriptomic alignment for this session, but we’ll consider other
modalities in later sessions. We’ll consider several different
approaches to compare gene expression between human cohorts and model
systems:</p>
<ol style="list-style-type: decimal">
<li>Correlation of genes within human co-expression modules<br>
</li>
</ol>
<ul>
<li>correlations will be generally weak for all expression, but animal
models may recapitulate specific aspects of the disease<br>
</li>
<li>we can use subsets of genes from co-expression modules, which
represent genes expressed in similar patterns in AD, and look for
correlations within these subsets<br>
</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Correlation of functional enrichment results<br>
</li>
</ol>
<ul>
<li>another approach is to consider the functional annotation enriched
among differentially expressed genes in human and mouse.</li>
<li>we can similarly sub-divide these groups of co-functional genes into
biological domains to aid our interpretation</li>
</ul>
<p>Let’s start by briefly reviewing how to assess differential
expression in our mouse RNA-seq datasets using the <code>DESeq2</code>
package. Then we’ll move on to discussing the background of the human
cohorts and co-expression modules.</p>
</section><section><h2 class="section-heading" id="differential-expression-analysis-in-ad-mouse-models">[2] Differential Expression Analysis in AD Mouse Models<a class="anchor" aria-label="anchor" href="#differential-expression-analysis-in-ad-mouse-models"></a>
</h2>
<hr class="half-width">
<p>Let’s analyze the 5xFAD RNA-seq expression data we explored
yesterday. Specifically, we want to know which genes are differentially
expressed at each age as a result of the transgenes that constitute the
5xFAD model.</p>
<p>The 5xFAD mouse model is a 5x transgenic model consisting of mutatnt
human transgenes of the amyloid precursor protein (APP) and presenilin 1
(PSEN1) genes. The specific variants are all causal variants for
Familial Alzheimer’s Disease (FAD) and include three variants in the APP
gene - Swedish (K670N, M671L), Florida (I716V), and London (V717I) - and
two in the PSEN1 gene - M146L and L286V. The expression of both
transgenes is under control of the neural-specific elements of the mouse
Thy1 promoter, which drives overexpression of the transgenes in the
brain. More information about this generation and maintenance of this
strain can be obtained from the <a href="https://www.jax.org/strain/008730" class="external-link">JAX strain catalog</a>.</p>
<p>This model has been extensively characterized by the MODEL-AD
consortium, and others, including the study that we explored yesterday
(i.e. <a href="https://www.synapse.org/Synapse:syn21983020" class="external-link">syn21983020</a>). For
more information about this specific MODEL-AD study, see the publication
by <a href="https://pubmed.ncbi.nlm.nih.gov/34366832/" class="external-link">Oblak et al
(2021)</a>. The primary patho-physiological phenotypes are summarized in
the figure below from <a href="https://www.alzforum.org/research-models/5xfad-b6sjl" class="external-link">AlzForum</a>,
and include (1) early deposition of amyloid plaques, (2) gliosis and
neuroinflammation, (3) synaptic changes and cognitive impairment, and
(4) neuronal loss, specifically in cortical layer V and the subiculum.
Importantly, Tau tangles are absent from this model.</p>
<figure><img src="fig/5xFAD_phenotypes.png" alt="list of AD phenotypes associated with 5xFAD mouse model" class="figure mx-auto d-block"><div class="figcaption">5xFAD Mouse Phenotypes</div>
</figure><p>But what <strong>else</strong> can the RNA-seq data tell us about the
transcriptomic response to the 5xFAD model in the brain? To know more,
we need to assess the differentially expressed transcripts. We’ll use
the <code>DESeq2</code> package to perform differential expression
analysis of the 5xFAD RNA-seq data.</p>
<div class="section level3">
<h3 id="read-5xfad-rna-seq-count-data">read 5xFAD RNA-seq count data<a class="anchor" aria-label="anchor" href="#read-5xfad-rna-seq-count-data"></a>
</h3>
<p>First, let’s re-access the RNA-seq data and metadata from Synapse</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RNA-seq counts</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">'syn22108847'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">$</span><span class="va">path</span> <span class="op">%&gt;%</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biospecimen  metadata</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">'syn22103213'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">$</span><span class="va">path</span> <span class="op">%&gt;%</span> <span class="fu">read_csv</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">individualID</span>, <span class="va">specimenID</span><span class="op">)</span> </span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="st">"data/htseqcounts_5XFAD.txt"</span>, </span>
<span>                   show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># individual metadata</span></span>
<span><span class="va">ind_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_individual_metadata.csv"</span>, </span>
<span>                     show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biospecimen metadata</span></span>
<span><span class="va">bio_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_biospecimen_metadata.csv"</span>, </span>
<span>                     show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># assay metadata</span></span>
<span><span class="va">rna_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_assay_RNAseq_metadata.csv"</span>, </span>
<span>                     show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># individual metadata, joined to the above</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">'syn22103212'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">$</span><span class="va">path</span> <span class="op">%&gt;%</span> <span class="fu">read_csv</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">meta</span>, by <span class="op">=</span> <span class="st">'individualID'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">specimenID</span><span class="op">)</span><span class="op">)</span> </span></code></pre>
</div>
<p>We can modify the metadata to only include covariates we’ll need for
this analysis</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># order rows that have corresponding IDs in the counts table</span></span>
<span><span class="va">covars</span> <span class="op">&lt;-</span> <span class="va">meta</span> <span class="op">%&gt;%</span> <span class="fu">slice</span><span class="op">(</span> <span class="fu">match</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="va">specimenID</span><span class="op">)</span> <span class="op">)</span> </span>
<span></span>
<span><span class="co"># compute the age of animals in months</span></span>
<span><span class="va">covars</span> <span class="op">&lt;-</span> <span class="va">covars</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    dateBirth <span class="op">=</span> <span class="fu">mdy</span><span class="op">(</span><span class="va">dateBirth</span><span class="op">)</span>,</span>
<span>    dateDeath <span class="op">=</span> <span class="fu">mdy</span><span class="op">(</span><span class="va">dateDeath</span><span class="op">)</span>,</span>
<span>    age <span class="op">=</span> <span class="fu">interval</span><span class="op">(</span><span class="va">dateBirth</span>,<span class="va">dateDeath</span><span class="op">)</span> <span class="op">%/%</span> <span class="fu">months</span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># change the group variable based on the animal genotype</span></span>
<span><span class="va">covars</span> <span class="op">&lt;-</span> <span class="va">covars</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span> group <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">genotype</span> <span class="op">==</span> <span class="st">'5XFAD_carrier'</span>, <span class="st">'5xFAD'</span>, <span class="st">'WT'</span><span class="op">)</span> <span class="op">)</span></span>
<span>    </span>
<span><span class="co"># finally, only keep the columns we'll need</span></span>
<span><span class="va">covars</span> <span class="op">&lt;-</span> <span class="va">covars</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">specimenID</span>, <span class="va">group</span>, <span class="va">sex</span>, <span class="va">age</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">covars</span><span class="op">)</span></span></code></pre>
</div>
<p>First, let’s make sure we have all relevant metadata</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">all</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">==</span><span class="va">covars</span><span class="op">$</span><span class="va">specimenID</span><span class="op">)</span></span></code></pre>
</div>
<p>How many animals do we have in each group?</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covars</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">group</span>, <span class="va">age</span>, <span class="va">sex</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarise</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Looks like we have 6 samples each from two genotypes (5xFAD or WT),
three ages (4 months, 6 months, and 10 months), and both sexes, for a
total of 72 samples.</p>
</div>
<div class="section level3">
<h3 id="accounting-for-transgenes">accounting for transgenes<a class="anchor" aria-label="anchor" href="#accounting-for-transgenes"></a>
</h3>
<p>The 5xFAD model has two copies each of the APP and PSEN1 genes - one
endogenous mouse gene, and the orthologous human transgene. The RNA-seq
data was assessed using a custom transcriptome definition that included
the sequences of both the mouse and human versions of each gene.</p>
<p>Ultimately we are going to sum the counts from both ortholgous genes
(human APP and mouse App; human PSEN1 and mouse Psen1). But first, let’s
look at the expression of each of these genes in the different groups.
To start we’ll filter the counts down to just those four relevant gene
IDs and join the counts up with the covariates to explore the expression
of these genes.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tg.counts</span> <span class="op">&lt;-</span> <span class="va">counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000080815"</span>,<span class="st">"ENSMUSG00000019969"</span>,</span>
<span>                        <span class="st">"ENSG00000142192"</span>,<span class="st">"ENSMUSG00000022892"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="va">.</span>,cols <span class="op">=</span> <span class="op">-</span><span class="st">"gene_id"</span>,names_to <span class="op">=</span> <span class="st">"specimenID"</span>,values_to<span class="op">=</span><span class="st">"counts"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">covars</span> ,by<span class="op">=</span><span class="st">"specimenID"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">tg.counts</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s do a little data housekeeping:</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># make an age column that is a factor and re-order the levels</span></span>
<span><span class="va">tg.counts</span> <span class="op">&lt;-</span> <span class="va">tg.counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    age.m <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">age</span>, <span class="st">'m'</span><span class="op">)</span>,</span>
<span>    age.m <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">age.m</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4m'</span>,<span class="st">'6m'</span>,<span class="st">'10m'</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># add gene symbols</span></span>
<span><span class="va">tg.counts</span> <span class="op">&lt;-</span> <span class="va">tg.counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    symbol <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span></span>
<span>      <span class="va">gene_id</span> <span class="op">==</span> <span class="st">"ENSG00000142192"</span> <span class="op">~</span> <span class="st">"Human APP"</span>,</span>
<span>      <span class="va">gene_id</span> <span class="op">==</span> <span class="st">"ENSG00000080815"</span> <span class="op">~</span> <span class="st">"Human PSEN1"</span>,</span>
<span>      <span class="va">gene_id</span> <span class="op">==</span> <span class="st">"ENSMUSG00000022892"</span> <span class="op">~</span> <span class="st">"Mouse App"</span>,</span>
<span>      <span class="va">gene_id</span> <span class="op">==</span> <span class="st">"ENSMUSG00000019969"</span> <span class="op">~</span> <span class="st">"Mouse Psen1"</span></span>
<span>      <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Okay, now let’s plot the counts for each gene across the samples:</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">tg.counts</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">age.m</span>, y<span class="op">=</span><span class="va">counts</span>, color<span class="op">=</span><span class="va">group</span>, shape <span class="op">=</span> <span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>position<span class="op">=</span><span class="fu">position_jitterdodge</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">symbol</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>The human transgenes all have a counts of zero in the WT animals
(where the transgenes are absent), while the endogenous mouse genes are
expressed relatively consistently across both groups.</p>
<p>Let’s combine the expression of corresponding human and mouse genes
by summing the expression and saving the summed expression as expression
of mouse genes, respectively to match with gene names in control
mice.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># move the gene_id column to rownames, to enable summing across rows</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span> <span class="op">%&gt;%</span> <span class="fu">column_to_rownames</span><span class="op">(</span><span class="st">"gene_id"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">#merge mouse and human APP gene raw count</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000022892"</span>,<span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000022892"</span>,<span class="op">]</span> <span class="op">+</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSG00000142192"</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000142192"</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">#merge mouse and human PS1 gene raw count</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000019969"</span>,<span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000019969"</span>,<span class="op">]</span> <span class="op">+</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSG00000080815"</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000080815"</span><span class="op">)</span>,<span class="op">]</span></span></code></pre>
</div>
<p>We can confirm that the human genes are now absent from the counts
table:</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">str_detect</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, <span class="st">"MUS"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="prepare-data-and-run-deseq-analysis">prepare data and run DESeq analysis<a class="anchor" aria-label="anchor" href="#prepare-data-and-run-deseq-analysis"></a>
</h3>
<p>Next we’ll prepare the data for differential expression analysis.
We’ll use DESeq2 today, though there are other approaches. Another
disclaimer: there are multiple steps to a DESeq2 analysis and we’re not
going to get into nitty-gritty details here. We’ll briefly cover some of
the basics, but for more information, please refer to the <a href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html" class="external-link">DESeq2
vignette</a>.</p>
<p>Let’s perform this analysis stratified by age group while controlling
for the sex of the animals as a covariate. We can start with the
youngest animals (4 months old). Let’s sub-set the data and covariates
to these data:</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covars.4m</span> <span class="op">&lt;-</span> <span class="va">covars</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">age</span> <span class="op">==</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">counts.4m</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span>,<span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">covars.4m</span><span class="op">$</span><span class="va">specimenID</span><span class="op">]</span></span></code></pre>
</div>
<p>Next we’ll build the DESeq object</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData<span class="op">=</span><span class="va">counts.4m</span>, </span>
<span>                                   colData<span class="op">=</span><span class="va">covars.4m</span>, </span>
<span>                                   design <span class="op">=</span> <span class="op">~</span><span class="va">group</span><span class="op">+</span><span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span></span></code></pre>
</div>
<p>Now we have a <code>DESeqDataSet</code> object covering counts data
for 55k genes across 24 mice.</p>
<p>Let’s take a closer look at the counts that go into this object</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts.4m</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre>
</div>
<p>You can see that ENSMUSG00000000003 has <code>0</code> reads across
the samples listed here. Let’s find out how many genes are
<code>0 counts</code> across all samples.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sums</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>gene_id <span class="op">=</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>,</span>
<span>                        sums    <span class="op">=</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu">rowSums</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">sum</span><span class="op">(</span><span class="va">gene_sums</span><span class="op">$</span><span class="va">sums</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span></code></pre>
</div>
<p>We can see that 9691 (17.5%) genes have no reads at all. Let’s filter
these out. While it is not necessary to pre-filter low count genes
before running the DESeq2 functions, there are two reasons which make
pre-filtering useful: by removing rows in which there are very few
reads, we reduce the memory size of the dds data object, and we increase
the speed of the transformation and testing functions within DESeq2. It
can also improve visualizations, as features with no information for
differential expression are not plotted.</p>
<p>Here we perform a minimal pre-filtering to keep only rows that have
at least 10 reads in at least 6 separate samples.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="va">ddsHTSeq</span><span class="op">[</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">6</span>,<span class="op">]</span></span></code></pre>
</div>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>What proportion of the 55k genes we started with remain after this
filter?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span></span></code></pre>
</div>
<p>There are 24765 genes, or 44.6% (24765/55487).</p>
</div>
</div>
</div>
</div>
<p>Let’s also make sure DESeq knows which group is our control or
“reference” group. By default this is arbitrarily assigned to the first
group in the factor. We can use the <code>relevel</code> function to set
the reference group to “WT” (wild type).</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu">relevel</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">group</span>,ref<span class="op">=</span><span class="st">"WT"</span><span class="op">)</span>  </span></code></pre>
</div>
<p>Now we’re ready to run the differential expression analysis; it’ll
take a few seconds to process this step:</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">ddsHTSeq</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>What are the results that have been computed:</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">resultsNames</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></code></pre>
</div>
<p>Because we specified <code>design = ~ group + sex</code> when setting
up the DESeq object, we now have the results from these two contrasts.
We can pull the results specifically for the 5xFAD vs WT comparison
using the <code>results</code> function. The results table contains the
log2 fold change, p-value, and adjusted p-value for each gene in the
analysis.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">results</span><span class="op">(</span><span class="va">dds</span>, contrast <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'group'</span>,<span class="st">'5xFAD'</span>,<span class="st">'WT'</span><span class="op">)</span>, alpha<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span>  </span>
<span><span class="va">fad.res.4m</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre>
</div>
<p>We can get a summary of the DE results using the <code>summary</code>
function:</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre>
</div>
<p>Ok, there are a total of 425 significantly differentially expressed
genes at 4 months of age when comparing 5xFAD to WT brain tissue. The
vast majority of these (392) are expressed at higher levels in 5xFAD
mice relative to WT. Let’s take a look at the most significantly DE
genes by ordering the results table by the adjusted p-value:</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fad.res.4m</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">head</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>This is a little difficult to interpret given all genes are
identified by their ENSEMBL IDs. Let’s map in the gene symbols using the
<code>org.Mm.eg.db</code> package. This package contains a mapping of
ENSEMBL IDs to gene symbols, and we can use the
<code>AnnotationDbi::mapIds</code> function to get the gene symbols for
our results table.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fad.res.4m</span><span class="op">$</span><span class="va">symbol</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">mapIds</span><span class="op">(</span></span>
<span>  <span class="fu">org.Mm.eg.db</span><span class="fu">::</span><span class="va">org.Mm.eg.db</span>,</span>
<span>  keys <span class="op">=</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">fad.res.4m</span><span class="op">)</span>,</span>
<span>  column <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span>  keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  multiVals <span class="op">=</span> <span class="st">"first"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">fad.res.4m</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">head</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Ok! Now we can see that among the most significantly up-regulated
genes in 5xFAD mice brains at 4 months of age are Psen1 and App (which
we saw previously), along with Thy1, Cst7, Ccl6, and Clec7a. Let’s plot
all DE genes:</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">fad.res.4m</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">log2FoldChange</span>, <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.3</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="op">(</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="fu">abs</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fu">log2</span><span class="op">(</span><span class="fl">1.2</span><span class="op">)</span><span class="op">)</span> <span class="op">)</span>, show.legend <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'grey'</span>,<span class="st">'red'</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu">geom_text_repel</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">fad.res.4m</span>, <span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="fu">abs</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fu">log2</span><span class="op">(</span><span class="fl">1.2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span>, min.segment.length <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s put it all together and compute results for each age group.
This code does all of the steps outlined above for each age cohort. The
input data, <code>DESeq</code> objects, and result tables are stored in
columns of a data frame. Feel free to copy and paste this code, but
investigate it and be sure you understand what each part is doing. It
may take about a minute or two to complete.</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">st</span> <span class="op">=</span> <span class="fu">Sys.time</span><span class="op">(</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">fad.deg</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  age <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">4</span>,<span class="fl">6</span>,<span class="fl">10</span><span class="op">)</span>,</span>
<span>  meta <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">age</span>, <span class="op">~</span> <span class="va">covars</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">age</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  counts <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">meta</span>, <span class="op">~</span> <span class="va">counts</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">specimenID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    dds <span class="op">=</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span></span>
<span>      countData <span class="op">=</span> <span class="va">counts</span>,</span>
<span>      colData <span class="op">=</span> <span class="va">meta</span>,</span>
<span>      design <span class="op">=</span> <span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>    </span>
<span>    dds <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">dds</span>, <span class="op">~</span> <span class="va">.x</span><span class="op">[</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">6</span>,<span class="op">]</span><span class="op">)</span>,</span>
<span>    dds <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">dds</span>, <span class="op">~</span> <span class="op">{</span><span class="va">.x</span><span class="op">$</span><span class="va">group</span> <span class="op">=</span> <span class="fu">relevel</span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">group</span>, ref <span class="op">=</span> <span class="st">'WT'</span><span class="op">)</span>; <span class="va">.x</span><span class="op">}</span><span class="op">)</span>,</span>
<span>    dds <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">dds</span>, <span class="op">~</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu">DESeq</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    res <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">dds</span>, <span class="op">~</span> <span class="fu">results</span><span class="op">(</span><span class="va">.x</span>, contrast <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'group'</span>,<span class="st">'5xFAD'</span>,<span class="st">'WT'</span><span class="op">)</span>, alpha<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    res.t <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">res</span>, <span class="op">~</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                  <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">'Ensembl_gene_id'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                  <span class="fu">mutate</span><span class="op">(</span>symbol <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">mapIds</span><span class="op">(</span></span>
<span>                    <span class="fu">org.Mm.eg.db</span><span class="fu">::</span><span class="va">org.Mm.eg.db</span>,</span>
<span>                    keys <span class="op">=</span> <span class="va">Ensembl_gene_id</span>,</span>
<span>                    column <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span>                    keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>                    multiVals <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span></span>
<span><span class="va">ed</span> <span class="op">=</span> <span class="fu">Sys.time</span><span class="op">(</span><span class="op">)</span> <span class="op">-</span> <span class="va">st</span></span>
<span></span>
<span><span class="fu">print</span><span class="op">(</span><span class="va">ed</span><span class="op">)</span></span></code></pre>
</div>
<p>What information does the <code>fad.r</code> tibble contain? Let’s
take a look:</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">glimpse</span><span class="op">(</span><span class="va">fad.deg</span><span class="op">)</span></span></code></pre>
</div>
<p>The first column is the age brackets in integers. The second and
third columns are the metadata and count data for each age cohort (24
samples per age bracket). The fourth column is the DESeq object and the
fifth and sixth columns are the results from the DE analysis.</p>
<p>So by running:</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">fad.deg</span><span class="op">$</span><span class="va">res</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<p>We can see that there are more significantly DE genes at 6 months
than we saw at four months.</p>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p>How many up- and down-regulated genes are found for each age?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">map_dbl</span><span class="op">(</span><span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span>, </span>
<span>  <span class="op">~</span><span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">length</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">map_dbl</span><span class="op">(</span><span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span>, </span>
<span>  <span class="op">~</span><span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">length</span><span class="op">)</span></span></code></pre>
</div>
<p>There are between 392 and 1855 up-regulated genes, and between 33 and
1827 down-regulated genes.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-3" class="callout-inner">
<h3 class="callout-title">Challenge 3</h3>
<div class="callout-content">
<p>How many down-regulated genes overlap between each timepoint?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>The comparisons to check are 4m+6m, 4m+10m, and 6m+10m; these are in
rows 1+2, 1+3, and 2+3, respectively.</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">length</span><span class="op">(</span><span class="fu">intersect</span><span class="op">(</span></span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span>,</span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">length</span><span class="op">(</span><span class="fu">intersect</span><span class="op">(</span></span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span>,</span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">length</span><span class="op">(</span><span class="fu">intersect</span><span class="op">(</span></span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span>,</span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>It looks like there are 22, 22, and 424 overlapping down-regulated
genes, respectively.</p>
</div>
</div>
</div>
</div>
<p>This is a good point to save these data before we move on.</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">fad.deg</span>, <span class="fu">here</span><span class="fu">::</span><span class="fu">here</span><span class="op">(</span><span class="st">"5xFAD_DESeq_analysis.rds"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
</section><section><h2 class="section-heading" id="overview-of-human-cohort-data">[3] Overview of Human cohort data<a class="anchor" aria-label="anchor" href="#overview-of-human-cohort-data"></a>
</h2>
<hr class="half-width">
<p>Now that we have our mouse transcriptomes analyzed, let’s switch
gears and think about the human datasets. The Accelerating Medicines
Partnership-Alzheimer’s Disease Consortium (<a href="https://adknowledgeportal.synapse.org/Explore/Programs/DetailsPage?Program=AMP-AD" class="external-link">AMP-AD</a>)
has generated extensive sets of ’omics data from a variety of human
Alzheimer’s Disease cohorts. AMP-AD researchers are applying systems
biology approaches toward the goal of elucidating AD mechanisms and
highlighting potential therapeutic targets.</p>
<p>There are three large, independent human cohorts that are part of
AMP-AD:</p>
<ol style="list-style-type: decimal">
<li>The Religious Orders Study and the Memory and Aging Project (<a href="https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn3219045" class="external-link">ROSMAP,
syn3219045</a>)</li>
<li>Mount Sinai Brain Bank (<a href="https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn3159438" class="external-link">MSBB,
syn3159438</a>)<br>
</li>
<li>Mayo Clinic (<a href="https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn5550404" class="external-link">Mayo,
syn5550404</a>)</li>
</ol>
<p>These studies have collected postmortem RNA-seq profiles from over
1,200 individuals spanning seven distinct brain regions:<br>
- dorsolateral prefrontal cortex (DLPFC)<br>
- temporal cortex (TCX)<br>
- inferior frontal gyrus (IFG)<br>
- superior temporal gyrus (STG)<br>
- frontal pole (FP)<br>
- parahippocampal gyrus (PHG)<br>
- cerebellum (CBE)</p>
<p>These samples are generally balanced for AD, MCI, and non-affected
controls. The data provide a broad assessment on how AD affects multiple
brain regions in 3 different populations in the US.</p>
<figure><img src="fig/ampad_studies_regions.png" alt="table showing AMP-AD studies, the number of samples, and brain regions sampled" class="figure mx-auto d-block"><div class="figcaption">Transcriptomic studies of Alzheimer’s
Disease</div>
</figure><div class="section level3">
<h3 id="overview-of-human-consensus-rna-seq-coexpression-modules">Overview of Human Consensus RNA-Seq Coexpression Modules<a class="anchor" aria-label="anchor" href="#overview-of-human-consensus-rna-seq-coexpression-modules"></a>
</h3>
<p><a href="https://doi.org/10.1016/j.celrep.2020.107908" class="external-link">Wan, et
al. (2020)</a> performed meta analysis including all available AMP-AD
RNA-seq datasets and systematically defined correspondences between gene
expression changes associated with AD in human brains. Briefly, the
RNA-seq read libraries were normalized and covariates were adjusted for
each human study separately before testing for differential expression
using fixed/mixed effects modeling to account for batch effects. The
expression data from each brain region was used to perform co-expression
analysis using a variety of different algorithms, generating in total
2,978 co-expression modules across all tissues. Of these, 660 modules
showed enrichment for at least one AD-specific differentially expressed
gene from the meta-analysis of all cases compared to controls.</p>
<p>Wan <em>et al</em> clustered these modules together using network
analyses and found 30 co-expression modules related to LOAD pathology.
Among the 30 aggregate co-expression modules, five consensus clusters
were described that span brain region and study cohorts. These consensus
clusters consist of subsets of modules which are associated with similar
AD related changes across brain regions and cohorts. Wan <em>et al</em>
looked for enrichment of cell -type signatures within the modules using
expression-weighted cell type enrichment analysis (<a href="https://doi.org/10.3389/fnins.2016.00016" class="external-link">Skene and Grant
(2016)</a>) and examined enrichment of functional annotations within the
modules.</p>
<figure><img src="fig/ampad_consensus_clusters.png" alt="Heatmap showing the overlap between co-expression clusters and the consensus clusters." class="figure mx-auto d-block"><div class="figcaption">AMP-AD GENE Modules</div>
</figure><p>This figure shows a matrix view of gene content overlap between the
30 co-expression modules. You’ll note a few strongly overlapping group
of modules, implicating similar expression profiles in different studies
and brain regions, which are the consensus clusters (A-E).</p>
<p>The first module block (consensus cluster A) is enriched for
signatures of astrocytes, while the next block (consensus cluster B) is
enriched for signatures of other cell types, including endothelial and
microglial expressed genes, suggesting an inflammation component. The
third module block (consensus cluster C) is strongly enriched for
signatures of neuronal gene expression, linking the modules within this
cluster to neurodegenerative processes. The fourth module block
(consensus cluster D) is enriched for oligodendrocyte and other glial
genes, indicating myelination and other neuronal support functions
associated with these modules. Finally, consensus cluster E contains
mixed modules that don’t have clear cell type enrichments, but do have
enrichments for processes involved in response to stress or unfolded
proteins. Stress response is not cell specific, so the expression
represented by these modules may be throughout many cells in the
brain.</p>
</div>
<div class="section level3">
<h3 id="accessing-amp-ad-module-data">Accessing AMP-AD module data<a class="anchor" aria-label="anchor" href="#accessing-amp-ad-module-data"></a>
</h3>
<p>These AMP-AD co-expression modules are very useful for making
comparisons between animal models and the human cohorts. In order to use
these modules to make the comparisons, we’ll need to download data
pertaining to the 30 co-expression modules. These data are available
from the Synapse data repository (<a href="https://www.synapse.org/Synapse:syn11932957" class="external-link">syn11932957</a>);
let’s download and take a closer look at the data.</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="va">syn</span><span class="op">$</span><span class="fu">tableQuery</span><span class="op">(</span><span class="st">"SELECT * FROM syn11932957"</span><span class="op">)</span></span>
<span><span class="va">module_table</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">query</span><span class="op">$</span><span class="va">filepath</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">module_table</span><span class="op">)</span></span></code></pre>
</div>
<p>Here you see 9 columns in this table. Columns we’re interested in
are:</p>
<ul>
<li>column 3: <code>GeneID</code> contains Ensembl gene IDs<br>
</li>
<li>column 4: <code>Module</code> is the module name in which gene is
clustered<br>
</li>
<li>column 7: <code>brainRegion</code> is the tissue of the
corresponding module<br>
</li>
<li>column 9: <code>external_gene_name</code> are gene symbols</li>
</ul>
<p>How many unique modules are in the table?</p>
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">Module</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>What are the names of the modules?</p>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">unique</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">Module</span><span class="op">)</span></span></code></pre>
</div>
<p>How many genes are in each module?</p>
<div class="codewrapper sourceCode" id="cb40">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">table</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">Module</span><span class="op">)</span></span></code></pre>
</div>
<p>We can visualize this as bar plot using ggplot2 package.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">module_table</span>,<span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">Module</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_bar</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> </span></code></pre>
</div>
<div id="challenge-4" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-4" class="callout-inner">
<h3 class="callout-title">Challenge 4</h3>
<div class="callout-content">
<p>What are other ways to count genes in each module?</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>You could also try:</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">module_table</span> ,<span class="va">Module</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>We can also check the total number of unique genes in the table</p>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">GeneID</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="mouse-orthologs-of-human-module-genes">Mouse orthologs of Human module genes<a class="anchor" aria-label="anchor" href="#mouse-orthologs-of-human-module-genes"></a>
</h4>
<p>In the module table we’ve downloaded we have human ENSEMBL ids and
human gene symbols. In order to compare between human and mouse models,
we will need to identify the corresponding (i.e. orthologous) mouse gene
IDs. We are going to add the gene IDs of orthologous genes in mouse to
the corresponding human genes in the module table.</p>
<p>Orthology mapping can be tricky, but thankfully Wan <em>et al</em>
have already identified mouse orthologs for each of the human genes
using the HGNC Comparison of Orthology Predictions (<a href="https://www.genenames.org/tools/hcop/" class="external-link">HCOP</a>) tool. While there
are a variety of different ways to get data about gene orthology, for
the sake of simplicity we are going to read that table from Synapse (<a href="https://doi.org/10.7303/syn17010253.1" class="external-link">syn17010253</a>).</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouse.human.ortho</span> <span class="op">&lt;-</span> <span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">"syn17010253"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span> <span class="op">%&gt;%</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">mouse.human.ortho</span><span class="op">)</span></span></code></pre>
</div>
<p>There are 15 columns with various gene identifiers for each species.
We’ll add mouse gene symbols from the ortholog table to the module table
by matching the human ENSEMBL IDs from both tables
(i.e. <code>GeneID</code> from the module table and
<code>human_ensembl_gene</code> from the orthology table).</p>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">module_table</span><span class="op">$</span><span class="va">Mouse_gene_symbol</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">mouse.human.ortho</span><span class="op">$</span><span class="va">mouse_symbol</span><span class="op">[</span></span>
<span>    <span class="fu">match</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">GeneID</span>,<span class="va">mouse.human.ortho</span><span class="op">$</span><span class="va">human_ensembl_gene</span><span class="op">)</span></span>
<span>    <span class="op">]</span></span></code></pre>
</div>
<p>Taking a look at the module table, we can see the new column of mouse
orthologs we just added.</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">module_table</span><span class="op">)</span></span></code></pre>
</div>
<p>Some genes don’t have identified orthologs. Also there’s some
redundant information. Let’s only keep the columns of interest and rows
that contain a mouse ortholog mapping:</p>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules</span> <span class="op">&lt;-</span> <span class="va">module_table</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">distinct</span><span class="op">(</span>tissue <span class="op">=</span> <span class="va">brainRegion</span>, module <span class="op">=</span> <span class="va">Module</span>, gene <span class="op">=</span> <span class="va">GeneID</span>, <span class="va">Mouse_gene_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Mouse_gene_symbol</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span></span></code></pre>
</div>
<p>Take a look at this new data table:</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-5" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-5" class="callout-inner">
<h3 class="callout-title">Challenge 5</h3>
<div class="callout-content">
<p>How many human genes are we removing that don’t have identified
orthologs?</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">module_table</span>, <span class="fu">is.na</span><span class="op">(</span><span class="va">Mouse_gene_symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">external_gene_name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nrow</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>2998 genes</p>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="reading-differential-expression-result-of-human-data-from-meta-analysis">Reading differential expression result of human data from
meta-analysis<a class="anchor" aria-label="anchor" href="#reading-differential-expression-result-of-human-data-from-meta-analysis"></a>
</h3>
<p>Now we know the genes that are in each AMP-AD co-expression cluster,
along with the ID of the corresponding orthologous genes in mouse. We’ll
also need to know how the expression of these genes change in AD.</p>
<p>We’ll download the results from differential expression analysis of
reprocessed AMP-AD RNA-seq data (all 7 brain regions). Log fold change
values for human transcripts can be obtained from Synapse (<a href="https://www.synapse.org/#!Synapse:syn14237651" class="external-link">syn14237651</a>).</p>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules_raw</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">"syn14237651"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules_raw</span><span class="op">)</span></span></code></pre>
</div>
<p>Data from which tissues are in this table?</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">unique</span><span class="op">(</span><span class="va">ampad_modules_raw</span><span class="op">$</span><span class="va">Tissue</span><span class="op">)</span></span></code></pre>
</div>
<p>All 7 brain regions are represented.</p>
<p>The AMP-AD data has been processed many ways and using different
models and comparisons. Let’s take a look at how many ways the data have
been analyzed:</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Model</span>, <span class="va">Comparison</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>For our analyses we’ll use data from the “Diagnosis” model and
comparisons between AD cases vs controls. We’ll filter the table for
these conditions and only keep the three columns we’ll need:
<code>Tissue</code>, <code>Gene</code> and <code>logFC</code>.</p>
<div class="codewrapper sourceCode" id="cb53">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_fc</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>, <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span>tissue <span class="op">=</span> <span class="va">Tissue</span>, gene <span class="op">=</span> <span class="va">ensembl_gene_id</span>, ampad_fc <span class="op">=</span> <span class="va">logFC</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="combine-with-modules-so-correlation-can-be-done-per-module">Combine with modules so correlation can be done per module<a class="anchor" aria-label="anchor" href="#combine-with-modules-so-correlation-can-be-done-per-module"></a>
</h4>
<p>Next, we will combine the fold change table we just downloaded
(<code>ampad_fc</code>) and module table from before
(<code>ampad_modules</code>). First, let’s look at both tables to check
how can we merge them together?</p>
<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_fc</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules</span><span class="op">)</span></span></code></pre>
</div>
<p>The columns common to both tables are <code>gene</code> (the human
Ensembl gene IDs) and <code>tissue</code> (the brain region
corresponding to the module/measurement). So we will merge the data sets
using these two columns.</p>
<p><strong>Reminder</strong>: Each gene can be present in multiple brain
regions, but should only be in one module per brain region. Let’s double
check using the first gene in the table:</p>
<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules</span><span class="op">[</span><span class="va">ampad_modules</span><span class="op">$</span><span class="va">gene</span> <span class="op">%in%</span> <span class="st">"ENSG00000168439"</span>,<span class="op">]</span></span></code></pre>
</div>
<p>This gene is present in six different co-expression modules all from
different brain regions. You can do this for any other gene as well.</p>
<p>We’ll merge the two tables using the <code>dplyr::inner_join</code>
function:</p>
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules_fc</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">ampad_modules</span>, <span class="va">ampad_fc</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"tissue"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Mouse_gene_symbol</span>, <span class="va">module</span>, <span class="va">ampad_fc</span><span class="op">)</span> </span></code></pre>
</div>
<p>Take a look at the new table we just made:</p>
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules_fc</span><span class="op">)</span></span></code></pre>
</div>
<p>We will use the data in <code>ampad_modules_fc</code> to compare with
log fold change data measured in mouse models.</p>
</div>
<div class="section level4">
<h4 id="preparing-module-information-for-correlation-plots">Preparing module information for correlation plots<a class="anchor" aria-label="anchor" href="#preparing-module-information-for-correlation-plots"></a>
</h4>
<p>Let’s package up these data and save this progress so far. This is
some manual book-keeping to arrange the modules into consensus clusters
for plotting later. You can just copy-paste this code into your
<code>Rstudio</code> session.</p>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cluster_a</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"TCXblue"</span>, <span class="st">"PHGyellow"</span>, <span class="st">"IFGyellow"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster A (ECM organization)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster A\n(ECM organization)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_b</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"DLPFCblue"</span>, <span class="st">"CBEturquoise"</span>, <span class="st">"STGblue"</span>, <span class="st">"PHGturquoise"</span>, <span class="st">"IFGturquoise"</span>, <span class="st">"TCXturquoise"</span>, <span class="st">"FPturquoise"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster B (Immune system)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster B\n(Immune system)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_c</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"IFGbrown"</span>, <span class="st">"STGbrown"</span>, <span class="st">"DLPFCyellow"</span>, <span class="st">"TCXgreen"</span>, <span class="st">"FPyellow"</span>, <span class="st">"CBEyellow"</span>, <span class="st">"PHGbrown"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster C (Neuronal system)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster C\n(Neuronal system)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_d</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"DLPFCbrown"</span>, <span class="st">"STGyellow"</span>, <span class="st">"PHGgreen"</span>, <span class="st">"CBEbrown"</span>, <span class="st">"TCXyellow"</span>, <span class="st">"IFGblue"</span>, <span class="st">"FPblue"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster D (Cell Cycle, NMD)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster D\n(Cell Cycle, NMD)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_e</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"FPbrown"</span>, <span class="st">"CBEblue"</span>, <span class="st">"DLPFCturquoise"</span>, <span class="st">"TCXbrown"</span>, <span class="st">"STGturquoise"</span>, <span class="st">"PHGblue"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster E (Organelle Biogensis, Cellular stress response)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster E\n(Organelle Biogenesis,\nCellular stress response)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">module_clusters</span> <span class="op">&lt;-</span> <span class="va">cluster_a</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_b</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_c</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_d</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_e</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cluster_label <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">cluster_label</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">module_clusters</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="va">module_clusters</span><span class="op">$</span><span class="va">module</span></span>
<span></span>
<span><span class="fu">save</span><span class="op">(</span><span class="va">ampad_modules_fc</span>,<span class="va">module_clusters</span>,<span class="va">mod</span>, file<span class="op">=</span> <span class="fu">here</span><span class="fu">::</span><span class="fu">here</span><span class="op">(</span><span class="st">"AMPAD_Module_Data.RData"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="correlation-between-mouse-models-and-human-ad-modules">[4] Correlation between mouse models and human AD modules<a class="anchor" aria-label="anchor" href="#correlation-between-mouse-models-and-human-ad-modules"></a>
</h2>
<hr class="half-width">
<p>Now we’ll get to the cross-species alignment. Our goal, as
demonstrated in the plots below, is to identify modules where the gene
expression is correlated between human and mouse orthologs (left) as
well as modules where there is no correlation (right).</p>
<figure><img src="fig/hs_mm_logfc_correlation.png" alt="Example correlation plots for AMP-AD co-expression module genes and mouse model expression." class="figure mx-auto d-block"><div class="figcaption">Human-Mouse logFC correlation</div>
</figure><p>There are two approaches that we commonly use to compute correlation
between mouse data and human AD data:</p>
<ol style="list-style-type: decimal">
<li>Compare change in expression in Human AD cases vs controls with
change in expression in mouse models for each gene in a given
module:<br>
</li>
</ol>
<ul>
<li>LogFC(h) = log fold change in expression of human AD patients
compared to control patients.</li>
<li>LogFC(m) = log fold change in expression of mouse AD models compared
to control mouse models.</li>
</ul>
<p><span class="math display">\[cor.test(LogFC(h),
LogFC(m))\]</span></p>
<ol start="2" style="list-style-type: decimal">
<li>Compare Human AD expression changes to mouse genetic effects for
each gene in a given module:<br>
</li>
</ol>
<ul>
<li>h = human gene expression (Log2 RNA-seq Fold Change AD/control)</li>
<li>β = mouse gene expression effect from linear regression model (Log2
RNA-seq TPM)</li>
</ul>
<p><span class="math display">\[cor.test(LogFC(h), β)\]</span></p>
<p>Both approaches allow us to assess directional coherence between gene
expression for genes in AMP-AD modules and the effects of genetic
manipulations in mice. For this session we are going to use the first
approach; we’ll return to approach #2 later in the week.</p>
<div id="challenge-6" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-6" class="callout-inner">
<h3 class="callout-title">Challenge 6</h3>
<div class="callout-content">
<p>What are the relative advantages of each approach?</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<p><span class="math display">\[cor.test(LogFC(h),
LogFC(m))\]</span></p>
<p>• direct comparison of effect sizes and direction • intuitive
interpretation</p>
<p><span class="math display">\[cor.test(LogFC(h), β)\]</span></p>
<p>• identify the genetic contributions • human AD data often has age,
sex, and other covariates regressed out, to derive the AD specific
effect • controlling for analogous variables by computing the genetic
effect (β) is often advantageous</p>
<p>Others?</p>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="step-0-reading-gene-expression-count-matrix-from-previous-lesson">Step 0: Reading Gene Expression Count matrix from Previous
Lesson<a class="anchor" aria-label="anchor" href="#step-0-reading-gene-expression-count-matrix-from-previous-lesson"></a>
</h4>
<p>First we’ll read the results saved after differential expression
analysis (above). We’ll only keep the information about which age cohort
and the differential expression results.</p>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fad.deg</span> <span class="op">&lt;-</span> <span class="fu">load</span><span class="op">(</span><span class="st">"results/DEAnalysis_5XFAD.Rdata"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'results/DEAnalysis_5XFAD.Rdata', probable reason 'No such file or directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in `readChar()`:
! cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb62">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">fad.deg</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error:
! object 'fad.deg' not found</code></pre>
</div>
<p>Let’s also load the AMP-AD module data.</p>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"data/AMPAD_Module_Data.RData"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="step-1-measure-the-correlation-between-mouse-models-for-each-sex-at-each-age-and-amp-ad-modules-using-common-genes-from-both-datasets">Step 1: Measure the correlation between mouse models for each sex at
each age and AMP-AD modules using common genes from both datasets<a class="anchor" aria-label="anchor" href="#step-1-measure-the-correlation-between-mouse-models-for-each-sex-at-each-age-and-amp-ad-modules-using-common-genes-from-both-datasets"></a>
</h4>
<p>We compute a Pearson correlation between changes in expression for
each gene within a given module (log fold change for cases vs controls)
with each mouse model (log fold change of the 5xFAD mice vs sex- and
age-matched B6 mice).</p>
<p>First, we’ll combine both mouse <code>fad.deg</code> and human
<code>ampad_modules_fc</code> log fold change data sets for all
genes.</p>
<div class="codewrapper sourceCode" id="cb65">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_vs_ampad</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">fad.deg</span>, </span>
<span>                             <span class="va">ampad_modules_fc</span>, </span>
<span>                             by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"symbol"</span><span class="op">)</span>, </span>
<span>                             multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> </span></code></pre>
</div>
<p><strong>Note:</strong> for this join we specify
<code>multiple = "all"</code> so that the same gene can be matched
across multiple human tissues and multiple mouse ages.</p>
<div class="codewrapper sourceCode" id="cb66">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">model_vs_ampad</span><span class="op">)</span></span></code></pre>
</div>
<p>Now we’ll create a list column containing data frames using the <a href="https://tidyr.tidyverse.org/reference/nest.html" class="external-link">tidyr::nest</a>
function. Nesting is implicitly a summarizing operation: you get one row
for each group defined by the non-nested columns.</p>
<div class="codewrapper sourceCode" id="cb67">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="va">model_vs_ampad</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">module</span>, <span class="va">age</span>, <span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">ampad_fc</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">module</span>, <span class="va">age</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<p>And we can also look at some of the nested data:</p>
<div class="codewrapper sourceCode" id="cb68">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">data</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<p>These are the mouse and human log fold-change values for all genes in
the TCXblue module; the mouse data corresponds to 4 month old 5xFAD
mice.</p>
<p>The total number of groups in the data table:</p>
<div class="codewrapper sourceCode" id="cb69">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we’ll compute correlation coefficients using the
<code>cor.test</code> function:</p>
<div class="codewrapper sourceCode" id="cb70">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">df</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>      cor_test <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">"log2FoldChange"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                                      <span class="va">.x</span><span class="op">[[</span><span class="st">"ampad_fc"</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>      estimate <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"estimate"</span><span class="op">)</span>,</span>
<span>      p_value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"p.value"</span><span class="op">)</span></span>
<span>      <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">cor_test</span><span class="op">)</span></span></code></pre>
</div>
<p>Here we’re using <code>purrr::map</code> based functions to apply the
correlation test to every entry in the <code>data</code> column. We can
pull out specific features from the <code>cor_test</code> list column,
including the computed correlation coefficient (<code>estimate</code>)
and the significance (<code>p.value</code>).</p>
<p>In the end we should have correlation coefficients and significance
values for every comparison in our data table:</p>
<div class="codewrapper sourceCode" id="cb71">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cor.df</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="step-2-annotate-correlation-table-to-prepare-for-visualization">Step 2: Annotate correlation table to prepare for visualization<a class="anchor" aria-label="anchor" href="#step-2-annotate-correlation-table-to-prepare-for-visualization"></a>
</h4>
<p>These steps will make it easier to make a nice looking plot during
the next step. We’ll add a column that flags whether the correlation is
significant or not, and we’ll add in the information about which
consensus cluster each module belongs to:</p>
<div class="codewrapper sourceCode" id="cb72">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_module</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>significant <span class="op">=</span> <span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">module_clusters</span>, by <span class="op">=</span> <span class="st">"module"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">cluster</span>, <span class="va">cluster_label</span>, <span class="va">module</span>, <span class="va">age</span>,</span>
<span>                correlation <span class="op">=</span> <span class="va">estimate</span>, <span class="va">p_value</span>, <span class="va">significant</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">model_module</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="step-3-create-a-dataframe-to-use-as-input-for-plotting-the-results">Step 3: Create a dataframe to use as input for plotting the
results<a class="anchor" aria-label="anchor" href="#step-3-create-a-dataframe-to-use-as-input-for-plotting-the-results"></a>
</h4>
<p>More preparations for plotting, here we’ll get all of the values in
the right order so that they are grouped together nicely on the
plot.</p>
<div class="codewrapper sourceCode" id="cb73">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">correlation_for_plot</span> <span class="op">&lt;-</span> <span class="va">model_module</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">arrange</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span> </span>
<span>      module <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">module</span>,levels<span class="op">=</span><span class="va">mod</span><span class="op">)</span>,</span>
<span>      age <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">age</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'10m'</span>,<span class="st">'6m'</span>,<span class="st">'4m'</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="op">)</span> </span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">correlation_for_plot</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="visualizing-the-correlation-plot">Visualizing the Correlation plot<a class="anchor" aria-label="anchor" href="#visualizing-the-correlation-plot"></a>
</h3>
<p>Now, we will use the above matrix and visualize the correlation
results using <code>ggplot2</code> package functions.</p>
<div class="codewrapper sourceCode" id="cb74">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">correlation_for_plot</span></span>
<span><span class="fu">range</span><span class="op">(</span><span class="va">correlation_for_plot</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># the AMP-AD modules will be along the x-axis, the mouse models will be along the y-axis</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">age</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># each tile of the grid will be filled with a circle where the fill and size correspond to the correlation coefficient</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">age</span>, </span>
<span>                              colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># we'll draw a box arround significant correlations</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span>, <span class="va">.data</span><span class="op">$</span><span class="va">significant</span><span class="op">)</span>, </span>
<span>             <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">age</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span>,</span>
<span>             color<span class="op">=</span><span class="st">"black"</span>,shape<span class="op">=</span><span class="fl">0</span>,size<span class="op">=</span><span class="fl">9</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># plot the x-axis on the top of the plot, set the parameters of the scales</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span>guide <span class="op">=</span> <span class="st">"none"</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">0.5</span>, <span class="fl">0.5</span><span class="op">)</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                                 name <span class="op">=</span> <span class="st">"Correlation"</span>, guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># remove axis labels</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># facet the plot based on age range for the mice (rows) and consensus cluster (columns)</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span> rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="st">'5xFAD'</span><span class="op">)</span>, cols <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">cluster_label</span><span class="op">)</span>, </span>
<span>                       scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free"</span>,switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># specify how different aspects of the plot will look</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    strip.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>,colour <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"right"</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>In the above plot, categories along the x-axis are the 30 AMP-AD
co-expression modules grouped into 5 consensus clusters, while the
categories along the y-axis show the different groupings of mouse models
tested (split by age). Positive correlations are shown in blue and
negative correlations in red. Color intensity and size of the circles
are proportional to the correlation coefficient. Black squares around
dots represent significant correlation at p-value=0.05 and
non-significant correlations are left blank.</p>
<p>5xFAD mice display gene expression alterations that are correlated
with human disease across all five consensus clusters, with the
strongest correlations observed among modules/genes in Consensus Cluster
B, which generally consists of immune system pathways and functions.</p>
</div>
<div class="section level3">
<h3 id="examining-individual-correlation-results">Examining individual correlation results<a class="anchor" aria-label="anchor" href="#examining-individual-correlation-results"></a>
</h3>
<p>Let’s say we want to know more about a single comparison in the plot
above and which genes are contributing to the correlation result. Maybe
we’re really interested in the correlations to the FPbrown module
between 5xFAD mice at 4 months and 6 months. We can plot the individual
correlations for each comparison in the plot above with the data we
have. We’ll label genes with large fold change in the mouse using the
<code>ggrepel::geom_label_repel</code> function.</p>
<div class="codewrapper sourceCode" id="cb75">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># specify which comparisons to consider</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="st">'FPbrown'</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4m'</span>,<span class="st">'6m'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter the correlation data frame to these comparisons</span></span>
<span><span class="va">indiv.corr</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">module</span> <span class="op">==</span> <span class="va">m</span>, <span class="va">age</span> <span class="op">%in%</span> <span class="va">a</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span> facet <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">age</span>, <span class="st">'\n'</span>, <span class="st">'r = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">estimate</span>,<span class="fl">3</span><span class="op">)</span>,<span class="st">' ; p = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">p_value</span>,<span class="fl">3</span><span class="op">)</span> <span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">aes</span><span class="op">(</span> <span class="va">log2FoldChange</span> , <span class="va">ampad_fc</span> <span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span> size <span class="op">=</span> <span class="fl">.5</span>, color <span class="op">=</span> <span class="st">'darkred'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span>method <span class="op">=</span> <span class="st">'lm'</span>, lwd <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu">geom_text_repel</span><span class="op">(</span> data <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">desc</span><span class="op">(</span><span class="fu">abs</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                             <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">3</span>, min.segment.length <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">'5xFAD logFC'</span>, y <span class="op">=</span> <span class="st">'AMP-AD logFC'</span>,</span>
<span>       title <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">indiv.corr</span><span class="op">$</span><span class="va">module</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">facet</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Here we can see that the 5xFAD logFC are pretty small in general. The
correlations are relatively weak and driven by individual genes that
have relatively large changes (e.g. Heatr4, Nirp1a, Rpl39l). If we
compare a different module, say STGblue, we can see a stronger
relationship between mouse and human expression changes.</p>
<div class="codewrapper sourceCode" id="cb76">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">&lt;-</span> <span class="st">'STGblue'</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4m'</span>,<span class="st">'6m'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter the correlation data frame to these comparisons</span></span>
<span><span class="va">indiv.corr</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">module</span> <span class="op">==</span> <span class="va">m</span>, <span class="va">age</span> <span class="op">%in%</span> <span class="va">a</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span> facet <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">age</span>, <span class="st">'\n'</span>, <span class="st">'r = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">estimate</span>,<span class="fl">3</span><span class="op">)</span>,<span class="st">' ; p = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">p_value</span>,<span class="fl">3</span><span class="op">)</span> <span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">aes</span><span class="op">(</span> <span class="va">log2FoldChange</span> , <span class="va">ampad_fc</span> <span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span> size <span class="op">=</span> <span class="fl">.5</span>, color <span class="op">=</span> <span class="st">'darkred'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span>method <span class="op">=</span> <span class="st">'lm'</span>, lwd <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu">geom_text_repel</span><span class="op">(</span> data <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">desc</span><span class="op">(</span><span class="fu">abs</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                             <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">3</span>, min.segment.length <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">'5xFAD logFC'</span>, y <span class="op">=</span> <span class="st">'AMP-AD logFC'</span>,</span>
<span>       title <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">indiv.corr</span><span class="op">$</span><span class="va">module</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">facet</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>These correlations are much stronger (r is approximately 0.3 vs 0.1
for the previous module), and there is a consistent pattern between
young mice and old mice, with similar genes being expressed in similar
ways (e.g. Itgax and Clec7a are up-regulated at both ages). These
significant positive correlations suggest that the 5xFAD model captures
inflammatory changes observed in human AD patients.</p>
</div>
</section><section><h2 class="section-heading" id="detecting-functional-coherence-of-gene-sets-from-omics-data">[5] Detecting functional coherence of gene sets from omics data<a class="anchor" aria-label="anchor" href="#detecting-functional-coherence-of-gene-sets-from-omics-data"></a>
</h2>
<hr class="half-width">
<p>Most omics analyses produce data on many thousands of genomic
features (i.e. transcripts, proteins, etc.) for each condition tested.
Simply looking at these lists of genes and associated statistics can be
daunting and uninformative. We need approaches to identify which
biological functions are being impacted by a given experiment from these
systems-level measurements.</p>
<p>Gene functional enrichment analysis describes a variety of
statistical methods that identify groups of genes that share a
particular biological function or process and show differential
association with experimental conditions. Most approaches compare some
statistically valid set of differentially expressed features to sets of
functional annotations for those features. There are many different
functional annotation sets available, some of the more commonly used
include:</p>
<ul>
<li>gene function resources, such as the <a href="amigo.geneontology.org/amigo">Gene Ontology</a> (i.e. GO)<br>
</li>
<li>pathway databases, such as <a href="reactome.org">Reactome</a> or <a href="www.genome.jp/kegg/">KEGG</a><br>
</li>
<li>disease and phenotype ontologies, such as the <a href="hpo.jax.org/app">Human Phenotype Ontology</a>, the <a href="www.informatics.jax.org/vocab/mp_ontology">Mammalian Phenotype
Ontology</a>, and the <a href="https://disease-ontology.org/" class="external-link">Disease
Ontology</a>
</li>
</ul>
<p>These are the resources that are the foundation for many genomics
knowledge bases, such as <a href="www.informatics.jax.org">MGI</a>, <a href="monarchinitiative.org">monarch initiative</a>, and the <a href="alliancegenome.org">Alliance of Genome Resources</a>. The precise
nature of each of these resources varies, but the core information
contained within each is the relationship of sets of genes to
biologically meaningful annotations. These annotations are typically
expertly curated from the published literature.</p>
<p>There are a variety of statistical approaches that can be employed to
test for functional enrichment among genes identified from an omics
dataset. Two of the most common broad classes of tests are
<em>over-representation analysis (ORA)</em> and <em>gene set enrichment
analysis (GSEA)</em>. For example, consider the figure below from <a href="https://doi.org/10.1016/j.tig.2023.01.003" class="external-link">Zhao &amp; Rhee, Trends
in Genetics (2023)</a>. Let’s consider each in a bit more detail.</p>
<figure><img src="fig/pathway_enrichment_Zhao_Rhee_TrendsGenetics.jpg" alt="Figure showing different methods to test functional enrichment." class="figure mx-auto d-block"><div class="figcaption">Functional enrichment approaches</div>
</figure><div class="section level3">
<h3 id="over-representation-analysis">Over-representation analysis<a class="anchor" aria-label="anchor" href="#over-representation-analysis"></a>
</h3>
<p><code>ORA</code> involves statistical tests of overlap between two
lists of genes: one derived from the experiment and one from the
functional annotations. For example, one might ask what is the overlap
between the genes in an annotation class, such as “Lysosome”, and the
genes that are significantly up-regulated in a given experimental
condition. These tests usually rely on some form of Fisher’s exact test
(e.g. <code>fisher.test()</code>) or hypergeometric test
(e.g. <code>phyper()</code>). If the gene lists consist of a larger
number of overlapping genes than would be expected at random - given the
sample sizes involved - then there is said to be a statistical
over-representation of the annotation class within the experimental
condition.</p>
<p>Of course, these overlap tests are generally considered for all
annotation classes, which can number in the hundreds to thousands.
Performing this many statistical tests ensures that many will be
identified as significant by chance. Therefore, there is typically a
requirement to correct for multiple testing errors
(e.g. <code>p.adjust()</code>).</p>
<p>There are many R packages available to handle the statistical tests
and corrections involved in ORA. Today we’ll use
<code>clusterProfiler::enrichGO()</code>, which wraps statistical
testing for overlap with GO terms and multiple test correction in one
function.</p>
<p>Let’s start by considering the enrichments from the mouse data
analyzed previously. We’ll start by considering the genes that are
significantly DE in 6 month old animals</p>
<div class="codewrapper sourceCode" id="cb77">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene.list.up</span> <span class="op">&lt;-</span> <span class="va">fad.deg</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">age</span> <span class="op">==</span> <span class="st">'6m'</span>,</span>
<span>         <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, </span>
<span>         <span class="va">log2FoldChange</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">duplicated</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unique</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gene.list.dn</span> <span class="op">&lt;-</span> <span class="va">fad.deg</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">age</span> <span class="op">==</span> <span class="st">'6m'</span>,</span>
<span>         <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, </span>
<span>         <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">duplicated</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unique</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">length</span><span class="op">(</span><span class="va">gene.list.up</span><span class="op">)</span></span>
<span><span class="fu">length</span><span class="op">(</span><span class="va">gene.list.dn</span><span class="op">)</span></span></code></pre>
</div>
<p>There are a total of # r length(gene.list.up) significantly
up-regulated genes and r length(gene.list.dn) significantly
down-regulated genes in this cohort. Now test for over representation of
GO terms among the DEGs. First, we need to identify the universe of all
possible genes which includes the genes that were both measured by the
experiment and contained within the annotation set.</p>
<div class="codewrapper sourceCode" id="cb78">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">univ</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">org.Mm.egGO</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">bitr</span><span class="op">(</span><span class="va">.</span>, fromType <span class="op">=</span> <span class="st">"ENTREZID"</span>, toType <span class="op">=</span> <span class="st">'SYMBOL'</span>, OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, drop <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="st">'SYMBOL'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">intersect</span><span class="op">(</span><span class="va">.</span>, <span class="va">fad.deg</span><span class="op">$</span><span class="va">symbol</span><span class="op">)</span></span></code></pre>
</div>
<p>Now let’s test for enriched GO terms (this can take 3-4 minutes)</p>
<div class="codewrapper sourceCode" id="cb79">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.up</span> <span class="op">&lt;-</span> <span class="fu">enrichGO</span><span class="op">(</span><span class="va">gene.list.up</span>, </span>
<span>                   ont <span class="op">=</span> <span class="st">'all'</span>, </span>
<span>                   OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, </span>
<span>                   keyType <span class="op">=</span> <span class="st">'SYMBOL'</span>,</span>
<span>                   universe <span class="op">=</span> <span class="va">univ</span></span>
<span>                   <span class="op">)</span></span>
<span><span class="va">enr.dn</span> <span class="op">&lt;-</span> <span class="fu">enrichGO</span><span class="op">(</span><span class="va">gene.list.dn</span>, </span>
<span>                   ont <span class="op">=</span> <span class="st">'all'</span>, </span>
<span>                   OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, </span>
<span>                   keyType <span class="op">=</span> <span class="st">'SYMBOL'</span>,</span>
<span>                   universe <span class="op">=</span> <span class="va">univ</span></span>
<span>                   <span class="op">)</span></span></code></pre>
</div>
<p>How many significant terms are identified:</p>
<div class="codewrapper sourceCode" id="cb80">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.up</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">enr.dn</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-7" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-7" class="callout-inner">
<h3 class="callout-title">Challenge 7</h3>
<div class="callout-content">
<p>How many significant terms are identified from the up-regulated gene
list if you do not specify the “universe”?</p>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7"> Show me the solution </h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" aria-labelledby="headingSolution7" data-bs-parent="#accordionSolution7">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb81">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">enrichGO</span><span class="op">(</span><span class="va">gene.list.up</span>, ont <span class="op">=</span> <span class="st">'all'</span>, OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, keyType <span class="op">=</span> <span class="st">'SYMBOL'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="va">.</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span> </span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Plot the top 10 significant terms:</p>
<div class="codewrapper sourceCode" id="cb82">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu">plot_grid</span><span class="op">(</span></span>
<span>  <span class="fu">dotplot</span><span class="op">(</span><span class="va">enr.dn</span>, showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'down'</span><span class="op">)</span>,</span>
<span>  <span class="fu">dotplot</span><span class="op">(</span><span class="va">enr.up</span>, showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'up'</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<p>From this we can see that nervous system related terms
(e.g. “dendrite development” and “protein localization to synapse”) are
down in 5xFAD mice at 6 months, while immune functions (e.g. “regulation
of innate immune response” and “leukocyte mediated immunity”) are up in
5xFAD mice at 6 months.</p>
</div>
<div class="section level3">
<h3 id="gene-set-enrichment-analysis">Gene set enrichment analysis<a class="anchor" aria-label="anchor" href="#gene-set-enrichment-analysis"></a>
</h3>
<p><code>GSEA</code> is an alternative approach that uses a statistical
measure from the omics data (e.g. the log fold change or significance)
to rank the genes. An “enrichment score” is calculated for each
annotation set based on where the genes annotated to the term sit in the
overall distribution.</p>
<p>Let’s analyze those same data with the
<code>fgsea::fgseaMultilevel()</code> function. First we’ll specify the
gene list and use the <code>log2FoldChange</code> value to rank the
genes in the list.</p>
<div class="codewrapper sourceCode" id="cb83">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene.list</span> <span class="op">&lt;-</span> <span class="va">fad.deg</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">age</span> <span class="op">==</span> <span class="st">'6m'</span>, <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>,<span class="op">!</span><span class="fu">duplicated</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">log2FoldChange</span>, name <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span></span></code></pre>
</div>
<p>We’ll also need a list of GO terms and the genes annotated. We can
get this from the <code>org.Mm.eg.db</code> annotation package using the
<code>AnnotationDbi::select</code> function</p>
<div class="codewrapper sourceCode" id="cb84">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get GO annotations</span></span>
<span><span class="va">go.terms</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span></span>
<span>    <span class="va">org.Mm.eg.db</span>, </span>
<span>    keys <span class="op">=</span> <span class="fu">keys</span><span class="op">(</span> <span class="va">org.Mm.eg.db</span> , keytype <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span>,</span>
<span>    columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"GO"</span>, <span class="st">"SYMBOL"</span><span class="op">)</span>,</span>
<span>    keytype <span class="op">=</span> <span class="st">"SYMBOL"</span> </span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the list of gene sets</span></span>
<span><span class="va">go.gene.sets</span> <span class="op">&lt;-</span> <span class="fu">split</span><span class="op">(</span><span class="va">go.terms</span><span class="op">$</span><span class="va">SYMBOL</span>, <span class="va">go.terms</span><span class="op">$</span><span class="va">GO</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter gene sets by size (minSize and maxSize are arguments in fgsea)</span></span>
<span><span class="va">min_size</span> <span class="op">&lt;-</span> <span class="fl">15</span></span>
<span><span class="va">max_size</span> <span class="op">&lt;-</span> <span class="fl">500</span></span>
<span><span class="va">go.sets.filtered</span> <span class="op">&lt;-</span> <span class="va">go.gene.sets</span><span class="op">[</span><span class="fu">sapply</span><span class="op">(</span><span class="va">go.gene.sets</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">length</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">min_size</span> <span class="op">&amp;&amp;</span> <span class="fu">length</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="va">max_size</span><span class="op">)</span><span class="op">]</span></span></code></pre>
</div>
<p>Now we’ll test for enrichment:</p>
<div class="codewrapper sourceCode" id="cb85">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr</span> <span class="op">&lt;-</span> <span class="fu">fgsea</span><span class="fu">::</span><span class="fu">fgseaMultilevel</span><span class="op">(</span> </span>
<span>    pathways <span class="op">=</span> <span class="va">go.sets.filtered</span>,</span>
<span>    stats <span class="op">=</span> <span class="va">gene.list</span>, </span>
<span>    minSize <span class="op">=</span> <span class="va">min_size</span>,</span>
<span>    maxSize <span class="op">=</span> <span class="va">max_size</span>,</span>
<span>    nproc <span class="op">=</span> <span class="fl">1</span> <span class="op">)</span></span></code></pre>
</div>
<p>Let’s take a look</p>
<div class="codewrapper sourceCode" id="cb86">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">head</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<p>This isn’t especially informative without the GO term names. Let’s
pull those in</p>
<div class="codewrapper sourceCode" id="cb87">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get term names from the GO.db package</span></span>
<span><span class="va">go_term_map</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span></span>
<span>    <span class="fu">GO.db</span><span class="fu">::</span><span class="va">GO.db</span>,</span>
<span>    keys <span class="op">=</span> <span class="va">gse.enr</span><span class="op">$</span><span class="va">pathway</span>,</span>
<span>    columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"GOID"</span>, <span class="st">"TERM"</span><span class="op">)</span>,</span>
<span>    keytype <span class="op">=</span> <span class="st">"GOID"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># join with results table</span></span>
<span><span class="va">gse.enr</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">gse.enr</span>, <span class="va">go_term_map</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>pathway <span class="op">=</span> <span class="va">GOID</span>, <span class="va">TERM</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">relocate</span><span class="op">(</span><span class="va">TERM</span>, .after <span class="op">=</span> <span class="va">pathway</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># now take another look</span></span>
<span><span class="va">gse.enr</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">head</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<p>This is more helpful. We can see that the GO terms associated with
up-regulated genes include “inflammatory response” and “chemotaxis”,
while there’s one term among the top 10 that is associated with
down-regulated genes, “chemical synaptic transmission”.</p>
<p>How many significant terms are identified:</p>
<div class="codewrapper sourceCode" id="cb88">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-8" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-8" class="callout-inner">
<h3 class="callout-title">Challenge 8</h3>
<div class="callout-content">
<p>The tally above represents all genes, both up- and down-regulatd. To
compare between GSEA and ORA, can you identify how many GSEA enriched
terms are associated with up-regulated genes and how many are associated
with down-regulated genes? (Hint: the <code>NES</code> value within the
results relates to the directionality of enrichment).</p>
</div>
</div>
</div>
<div id="accordionSolution8" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution8" aria-expanded="false" aria-controls="collapseSolution8">
  <h4 class="accordion-header" id="headingSolution8"> Show me the solution </h4>
</button>
<div id="collapseSolution8" class="accordion-collapse collapse" aria-labelledby="headingSolution8" data-bs-parent="#accordionSolution8">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb89">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span>  </span>
<span><span class="va">gse.enr</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>40 terms are up-regulated, while only 2 are associated with
down-regulated genes</p>
</div>
</div>
</div>
</div>
<div id="challenge-9" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-9" class="callout-inner">
<h3 class="callout-title">Challenge 9</h3>
<div class="callout-content">
<p>You may notice that your numbers of significantly enriched terms are
slightly different. Why would this be the case?</p>
</div>
</div>
</div>
<div id="accordionSolution9" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution9" aria-expanded="false" aria-controls="collapseSolution9">
  <h4 class="accordion-header" id="headingSolution9"> Show me the solution </h4>
</button>
<div id="collapseSolution9" class="accordion-collapse collapse" aria-labelledby="headingSolution9" data-bs-parent="#accordionSolution9">
<div class="accordion-body">
<p><code>fgsea</code> uses a permutation-based approach to estimate the
significance of the gene set enrichments. Because this involves random
sampling, running the analysis multiple times with the same settings can
result in subtle, minor differences in the output, particularly in the
p-values.</p>
<p>Increasing the <code>nPermSimple</code> parameter increases the
number of permutations performed. This leads to a more thorough sampling
of the null distribution, thereby improving the precision of the
estimated p-values, especially for highly significant pathways. However,
it’s a trade-off, as a higher <code>nPermSimple</code> value will also
increase the computational time required to run the analysis.</p>
</div>
</div>
</div>
</div>
<p>Let’s compute the enriched terms for all age groups:</p>
<div class="codewrapper sourceCode" id="cb90">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fad.enr</span> <span class="op">&lt;-</span> <span class="va">fad.deg</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span></span>
<span>          , <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span></span>
<span>          <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">age</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>gl <span class="op">=</span> <span class="va">log2FoldChange</span> <span class="op">%&gt;%</span> <span class="fu">setNames</span><span class="op">(</span><span class="va">.</span>, <span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">sort</span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>  gse <span class="op">=</span> <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="va">gl</span>,</span>
<span>    <span class="op">~</span> <span class="fu">fgsea</span><span class="fu">::</span><span class="fu">fgseaMultilevel</span><span class="op">(</span></span>
<span>      pathways <span class="op">=</span>  <span class="va">go.sets.filtered</span>,</span>
<span>      stats <span class="op">=</span> <span class="va">.x</span>,</span>
<span>      minSize <span class="op">=</span> <span class="va">min_size</span>,</span>
<span>      maxSize <span class="op">=</span> <span class="va">max_size</span>,</span>
<span>      nproc <span class="op">=</span> <span class="fl">1</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span>,</span>
<span>  res <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">gse</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>    <span class="va">go_term_map</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span></span>
<span>      <span class="fu">GO.db</span><span class="fu">::</span><span class="va">GO.db</span>,</span>
<span>      keys <span class="op">=</span> <span class="va">.x</span><span class="op">$</span><span class="va">pathway</span>,</span>
<span>      columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"GOID"</span>, <span class="st">"TERM"</span><span class="op">)</span>,</span>
<span>      keytype <span class="op">=</span> <span class="st">"GOID"</span></span>
<span>    <span class="op">)</span></span>
<span>    </span>
<span>    <span class="fu">inner_join</span><span class="op">(</span><span class="va">.x</span>, <span class="va">go_term_map</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>pathway <span class="op">=</span> <span class="va">GOID</span>, <span class="va">TERM</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">relocate</span><span class="op">(</span><span class="va">TERM</span>, .after <span class="op">=</span> <span class="va">pathway</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">fad.enr</span>, <span class="fu">here</span><span class="fu">::</span><span class="fu">here</span><span class="op">(</span><span class="st">'results'</span>, <span class="st">'5xFAD_fgsea_results.rds'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="common-pitfalls-best-practices">Common pitfalls &amp; best practices<a class="anchor" aria-label="anchor" href="#common-pitfalls-best-practices"></a>
</h3>
<p>These kinds of functional enrichment analyses are very common, but
not all results reported are equal! A recent paper describes an <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009935" class="external-link">“Urgent
need for consistent standards in functional enrichment analysis”</a>.
They examine how the results of functional enrichment analyses are
reported in the literature, and identify several common shortcomings.
Watch out for these common mistakes when performing and reporting your
own analyses! We’ll have more opportunities to discuss issues with
reproducibility in computational biology research in future
sessions.</p>
<figure><img src="fig/common_pitfalls_Wijesooriya_PLOS_CompBio.png" alt="Figure from the paper showing the number of papers examined that failed to meet certain standards for reporting the results of functional enrichment testing." class="figure mx-auto d-block"><div class="figcaption">Common pitfalls</div>
</figure>
</div>
</section><section><h2 class="section-heading" id="alzheimerss-disease-biological-domains">[6] Alzheimers’s Disease Biological Domains<a class="anchor" aria-label="anchor" href="#alzheimerss-disease-biological-domains"></a>
</h2>
<hr class="half-width">
<p>While these results are informative and help to provide essential
biological context to the results of the omics experiment, it is
difficult to understand all of the enriched terms and what that means
for the biology of disease. It would be useful to group each of the
enriched terms within broader areas of disease biology. There are
several common molecular endophenotypes that are detected within human
Alzheimer’s cohorts across omics data types (transcriptomics,
proteomics, and genetics). Several of these endophenotypic areas are
shown in the figure below (source: Jesse Wiley).</p>
<figure><img src="fig/AD_endophenotypes.png" alt="An image of a brain surrounded by different endophenotype categories that are associated with AD." class="figure mx-auto d-block"><div class="figcaption">Endophenotypes of AD</div>
</figure><p>These endophenotypes describe molecular processes that are
dysregulated by or during the disease process. Very similar sets of
endophenotypes have been identified among the targets of therapeutics in
clinical trials for AD (see below, <a href="https://doi.org/10.1002/trc2.12465" class="external-link">Cummings et al, Alzheimer’s
disease drug development pipeline: 2024, Alz Dem TRCI</a>.</p>
<figure><img src="fig/CADRO_pipeline_2024.png" alt="Circle plot showing therapeutics in various stages of clinical trials coded by the CADRO target class." class="figure mx-auto d-block"><div class="figcaption">Common AD Research Ontology</div>
</figure><figure><img src="fig/CADRO_agent_distribution_2024.png" alt="Bar plot showing the numbers of agents in clinical trials across each class." class="figure mx-auto d-block"><div class="figcaption">Drugs in Pipeline by CADRO</div>
</figure><p>In order to formalize and operationalize these endophenotypic areas,
the <a href="treatad.org/emory-sage-sgc/">Emory-Sage-SGC-JAX (ESSJ)
TREAT-AD center</a> has developed the <a href="https://alz-journals.onlinelibrary.wiley.com/doi/pdfdirect/10.1002/trc2.12461" class="external-link">Biological
Domains of AD</a>. In the enumeration of these domains, several criteria
were established; the defined biological domains should be:</p>
<ul>
<li>
<em>objective:</em> leverage existing well-annotated resources<br>
</li>
<li>
<em>automatable:</em> in CADRO, therapeutics and targets are
manually assigned<br>
</li>
<li>
<em>intelligible:</em> groupings should be easy to understand<br>
</li>
<li>
<em>modifiable:</em> definitions should be (and are!) continuously
be updated.</li>
</ul>
<p>In all, 19 distinct biological domains (aka biodomains) have been
identified. These biological domains are defined using sets of GO terms
that align with the endophenotype. For example, terms like “cytokine
receptor binding” and “leukocyte migration” are annotated to the
<code>Immune Response</code> biodomain, while terms like “postsynaptic
density” and “synaptic vesicle cycle” are annotated to the
<code>Synapse</code> biodomain. Of all terms in the GO, 6,837 terms
(15.7%) are annotated to one of the biological domains. Because the GO
terms have genes annotated, genes can be associated with specific
endophenotypes via the biological domain groupings of terms. In all,
16,275 genes are annotated to at least 1 biodomain term. While the
biodomains exhibit very few overlapping GO terms (panel A), due to gene
pleiotropy (etc) the number of overlapping genes between biological
domains is quite a bit higher (panel B). The development of the
biological domains is described in more detail in the published paper <a href="doi.org/10.1002/trc2.12461">Genetic and Multi-omic Risk Assessment
of Alzheimer’s Disease Implicates Core Associated Biological
Domains</a>.</p>
<figure><img src="fig/AD_biodomains.png" alt="Heatmaps showing the fraction of overlapping GO terms and genes between the various biological domains." class="figure mx-auto d-block"><div class="figcaption">AD Biodomain Demographics</div>
</figure><p>The ESSJ TREAT-AD center has developed approaches to group terms
within each biodomain into functionally coherent sub-domains. The
sub-domains are driven by gene co-annotation and disease risk score
enrichment.</p>
<figure><img src="fig/AD_subdomains.png" alt="Plot showing the number of subdomains associated with each domain and how many GO terms are assigned to each domain." class="figure mx-auto d-block"><div class="figcaption">AD Subdomains</div>
</figure><div class="section level3">
<h3 id="download-and-explore-the-biological-domain-annotations">Download and explore the biological domain annotations<a class="anchor" aria-label="anchor" href="#download-and-explore-the-biological-domain-annotations"></a>
</h3>
<p>First let’s download the biodomain definition file from synapse.</p>
<div class="codewrapper sourceCode" id="cb91">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># biodomain definitions</span></span>
<span><span class="va">biodom</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">'syn25428992'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biodomain labels and colors</span></span>
<span><span class="va">dom.lab</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">'syn26856828'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span></code></pre>
</div>
<p>What is in the <code>dom.lab</code> file?</p>
<div class="codewrapper sourceCode" id="cb92">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">glimpse</span><span class="op">(</span><span class="va">dom.lab</span><span class="op">)</span></span></code></pre>
</div>
<p>This file contains some useful standardized abbreviations and colors
that will be useful as we work with and plot domain information
later.</p>
<p>What is in the <code>biodom</code> file?</p>
<div class="codewrapper sourceCode" id="cb93">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">glimpse</span><span class="op">(</span><span class="va">biodom</span><span class="op">)</span></span></code></pre>
</div>
<p>You can see the file is a list of GO term accessions
(<code>GO_ID</code>) and names (<code>GOterm_Name</code>) as well as the
corresponding endophenotype areas (e.g. <code>Biodomain</code> and
<code>Subdomain</code>) to which the term is annotated. There are also
gene annotations for each term. These annotation sets come from two
sources: (1) the <code>symbol</code> and <code>uniprot</code>
annotations are drawn directly from the Gene Ontology via the provided
API, (2) <code>ensembl_id</code>, <code>entrez_id</code>, and
<code>hgnc_symbol</code> are from BioMart annotations
(e.g. <code>biomaRt::getLDS()</code>).</p>
<p>We can see how many GO terms are annotated to each biodomain:</p>
<div class="codewrapper sourceCode" id="cb94">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biodom</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarise</span><span class="op">(</span>n_term <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">GO_ID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">n_term</span><span class="op">)</span> </span></code></pre>
</div>
<p>The biodomains range in size from <code>Tau Homeostasis</code> (10
terms) up to <code>Synapse</code> (1,379 terms).</p>
<p>What about the size of the subdomains? For simplicity let’s focus on
the Subdomains within the <code>Synapse</code> domain.</p>
<div class="codewrapper sourceCode" id="cb95">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biodom</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="st">'Synapse'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarise</span><span class="op">(</span>n_term <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">GO_ID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">n_term</span><span class="op">)</span> </span></code></pre>
</div>
<p>There are 8 subdomains within the <code>Synapse</code> biodomain,
plus a set of terms that aren’t assigned to any subdomain (i.e. where
<code>Subdomain</code> is <code>NA</code>). The subdomains range in size
from <code>axon regeneration</code> (20 terms) up to
<code>trans-synaptic signaling</code> (162 terms). There are 822 terms
within the <code>Synapse</code> domain that aren’t assigned to any
sub-domains.</p>
<p>We can also investigate the individual genes annotated to each
biodomain GO term.</p>
<div class="codewrapper sourceCode" id="cb96">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">GOterm_Name</span> <span class="op">==</span> <span class="st">'amyloid-beta formation'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unlist</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>So the genes associated with <code>amyloid-beta formation</code>
within the <code>APP Metabolism</code> biodomain include PSEN1, PSEN2,
BACE1, ABCA7, NCSTN, and others.</p>
</div>
<div class="section level3">
<h3 id="annotate-enriched-terms-with-biological-domain">Annotate enriched terms with biological domain<a class="anchor" aria-label="anchor" href="#annotate-enriched-terms-with-biological-domain"></a>
</h3>
<p>Let’s re-characterize the 5xFAD functional enrichments we computed
previously with the biological domain annotations and see if we can get
more context about what is changing in that model. We’ll focus on the
GSEA results and start by annotating the results with biodomain
groupings.</p>
<div class="codewrapper sourceCode" id="cb97">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr.bd</span> <span class="op">=</span> <span class="va">gse.enr</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, pathway <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'pathway'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">relocate</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">gse.enr.bd</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">pathway</span>, <span class="va">TERM</span>, <span class="va">Biodomain</span>, <span class="va">padj</span>, <span class="va">NES</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<p>Not all of the enriched terms are annotated to a biological domain.
Some terms are too broad and not specific (e.g. ‘cell mophogenesis’ or
‘MAPK cascade’), while others may not have been captured by a biological
domain annotation yet (e.g. ‘transcription cis-regulatory region
binding’). <em>Remember</em> that the conception of the biodomains
involved a requirement that they be modifiable, and these terms may be
added to the biodomain definintions in the future. Let’s modify the
<code>Biodomain</code> value for terms that aren’t annotated to a domain
to ‘none’.</p>
<div class="codewrapper sourceCode" id="cb98">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr.bd</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">[</span><span class="fu">is.na</span><span class="op">(</span><span class="va">gse.enr.bd</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'no domain'</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">gse.enr.bd</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">pathway</span>, <span class="va">TERM</span>, <span class="va">Biodomain</span>, <span class="va">padj</span>, <span class="va">NES</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<p>How many terms are enriched from each domain?</p>
<div class="codewrapper sourceCode" id="cb99">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bd.tally</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span>domain <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'no domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    n_term <span class="op">=</span> <span class="va">biodom</span><span class="op">$</span><span class="va">GO_ID</span><span class="op">[</span> <span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span> <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    n_sig_term <span class="op">=</span> <span class="va">gse.enr.bd</span><span class="op">$</span><span class="va">pathway</span><span class="op">[</span> <span class="va">gse.enr.bd</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span> <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu">arrange</span><span class="op">(</span><span class="va">bd.tally</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">n_sig_term</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Many enriched terms don’t map to a domain r bd.tally %&gt;%
filter(domain == ‘no domain’) %&gt;% pull(n_sig_term) %&gt;% sum()), but
nearly half do (r bd.tally %&gt;% filter(domain != ‘no domain’) %&gt;%
pull(n_sig_term) %&gt;% sum()). Of those that do, the vast majority map
into the <code>Immune Response</code> biodomain.</p>
<p>We can plot the enrichment results, stratified by biodomain:</p>
<div class="codewrapper sourceCode" id="cb100">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">gse.enr.bd</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">Biodomain</span>, levels <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">bd.tally</span>, <span class="va">n_sig_term</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">domain</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">padj</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, </span>
<span>              color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>This makes it really clear that in the 6 month old 5xFAD mice, the
enriched <code>Immune Response</code> domain terms are all associated
with up-regulated genes, while the enriched <code>Synapse</code> domain
terms are associated with a mix of up- and down-regulated genes. It also
highlights the several other domains with significantly enriched terms
(e.g. <code>Proteostasis</code>, <code>Structural Stabilization</code>,
<code>Lipid Metabolism</code>, <code>Endolysosome</code>, etc; even
<code>APP Metabolism</code>).</p>
<p>Now let’s look across age groups to see how domain enrichments
change:</p>
<div class="codewrapper sourceCode" id="cb101">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span> <span class="va">fad.enr</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">age</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>pathway <span class="op">=</span> <span class="va">GO_ID</span>, <span class="va">Biodomain</span>, <span class="va">Subdomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Biodomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Biodomain</span><span class="op">)</span>,</span>
<span>    Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    age <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">age</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4m'</span>,<span class="st">'6m'</span>,<span class="st">'10m'</span><span class="op">)</span><span class="op">)</span></span>
<span>         <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_sig <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">fct_reorder</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">n_sig</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">padj</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">age</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, </span>
<span>              color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>The earliest transcriptomic changes in the 5xFAD animals are
associated with <code>Immune Response</code> and
<code>Lipid Metabolism</code> domain terms. By 6 months, there are
enrichments for several other domains (as above), and by 10 months the
term enrichments are even more exaggerated (i.e. <em>more</em>
significance, larger NES values, etc).</p>
<p>We can also break down these results by sub-domain to get a more
clear idea of the processes affected in each case</p>
<div class="codewrapper sourceCode" id="cb102">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span> <span class="va">fad.enr</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">age</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>pathway <span class="op">=</span> <span class="va">GO_ID</span>, <span class="va">Biodomain</span>, <span class="va">Subdomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Biodomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Biodomain</span><span class="op">)</span>,</span>
<span>    Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    sd_lab <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">abbr</span>,<span class="st">'_'</span>,<span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    age <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">age</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4m'</span>,<span class="st">'6m'</span>,<span class="st">'10m'</span><span class="op">)</span><span class="op">)</span></span>
<span>         <span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">sd_lab</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">age</span><span class="op">)</span>, rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">abbr</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">'free'</span>, space <span class="op">=</span> <span class="st">'free_y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>label <span class="op">=</span> <span class="op">~</span> <span class="fu">str_remove_all</span><span class="op">(</span><span class="va">.x</span>, <span class="st">'^[A-Za-z]{2}_'</span><span class="op">)</span>, drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>What trends do you notice from these results? Which sub-domain
processes are affected at the earliest ages? Are there any sub-domains
that change across the age groups?</p>
<div id="challenge-10" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-10" class="callout-inner">
<h3 class="callout-title">Challenge 10</h3>
<div class="callout-content">
<p>How could you plot the results from the ORA to show biodomain
enrichements?</p>
</div>
</div>
</div>
<div id="accordionSolution10" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution10" aria-expanded="false" aria-controls="collapseSolution10">
  <h4 class="accordion-header" id="headingSolution10"> Show me the solution </h4>
</button>
<div id="collapseSolution10" class="accordion-collapse collapse" aria-labelledby="headingSolution10" data-bs-parent="#accordionSolution10">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb103">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.ora</span> <span class="op">=</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">enr.up</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>dir <span class="op">=</span> <span class="st">'up'</span><span class="op">)</span>,</span>
<span>                    <span class="va">enr.dn</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>dir <span class="op">=</span> <span class="st">'dn'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, ID <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'ID'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">relocate</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span></span>
<span></span>
<span><span class="va">enr.ora</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">[</span><span class="fu">is.na</span><span class="op">(</span><span class="va">enr.ora</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'none'</span></span>
<span></span>
<span><span class="va">bd.tally</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span>domain <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    n_term <span class="op">=</span> <span class="va">biodom</span><span class="op">$</span><span class="va">GO_ID</span><span class="op">[</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    n_sig_term <span class="op">=</span> <span class="va">enr.ora</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">enr.ora</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">enr.ora</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">enr.ora</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">Biodomain</span>, levels <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">bd.tally</span>, <span class="va">n_sig_term</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">domain</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">p.adjust</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    signed_logP <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">p.adjust</span><span class="op">)</span>,</span>
<span>    signed_logP <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">dir</span> <span class="op">==</span> <span class="st">'dn'</span>, <span class="op">-</span><span class="fl">1</span> <span class="op">*</span> <span class="va">signed_logP</span>, <span class="va">signed_logP</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.ora</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">signed_logP</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.ora</span>, <span class="va">dir</span> <span class="op">==</span> <span class="st">'up'</span><span class="op">)</span>,</span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>,</span>
<span>              scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.ora</span>, <span class="va">dir</span> <span class="op">==</span> <span class="st">'dn'</span><span class="op">)</span>,</span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>,</span>
<span>              scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span></span>
<span>    <span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="va">Count</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>,</span>
<span>    color <span class="op">=</span> <span class="st">'grey20'</span>,</span>
<span>    shape <span class="op">=</span> <span class="fl">21</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">.3</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Based on the gene list (up or down) we can add a sign to the
significance. When we plot this we can see there are many more
significantly enriched terms from the ORA. The dominant signal is still
the up-regulation of terms from the <code>Immune Response</code>
biodomain. There is also nearly exclusive up-regulation of terms from
the <code>Autophagy</code>, <code>Oxidative Stress</code>, and
<code>APP Metabolism</code> domains. The most down-regulated terms are
from the <code>Synapse</code> biodomain.</p>
</div>
</div>
</div>
</div>
<div id="challenge-11" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-11" class="callout-inner">
<h3 class="callout-title">Challenge 11</h3>
<div class="callout-content">
<p>Which biodomain terms are over-represented among the gene
co-expression modules? Choose one module and test for over-represented
terms at a <code>p.adjust</code> value less than <code>0.05</code>.</p>
<p>What if you look at the <em>most</em> significantly enriched terms,
with <code>p.adjust</code> values less than <code>1e-5</code>?</p>
</div>
</div>
</div>
<div id="accordionSolution11" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution11" aria-expanded="false" aria-controls="collapseSolution11">
  <h4 class="accordion-header" id="headingSolution11"> Show me the solution </h4>
</button>
<div id="collapseSolution11" class="accordion-collapse collapse" aria-labelledby="headingSolution11" data-bs-parent="#accordionSolution11">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb104">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stg_blue.ora</span> <span class="op">&lt;-</span> <span class="va">ampad_modules</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">module</span> <span class="op">==</span> <span class="st">'STGblue'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">Mouse_gene_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">enrichGO</span><span class="op">(</span><span class="va">.</span>, <span class="va">org.Mm.eg.db</span>, keyType <span class="op">=</span> <span class="st">'SYMBOL'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">inner_join</span><span class="op">(</span><span class="va">stg_blue.ora</span><span class="op">@</span><span class="va">result</span>,</span>
<span>           <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>ID <span class="op">=</span> <span class="va">GO_ID</span>, <span class="va">Biodomain</span><span class="op">:</span><span class="va">Subdomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">inner_join</span><span class="op">(</span><span class="va">stg_blue.ora</span><span class="op">@</span><span class="va">result</span>,</span>
<span>           <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>ID <span class="op">=</span> <span class="va">GO_ID</span>, <span class="va">Biodomain</span><span class="op">:</span><span class="va">Subdomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">1e-5</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>At the default p-value the domains with the most terms enriched
include <code>Synapse</code>, <code>Lipid Metabolism</code>, and
<code>Immune Response</code>. If we filter to the <em>most</em>
significantly enriched terms, terms from the
<code>Immune Response</code> and <code>Structural Stabilization</code>
domains are represented.</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="comparing-enrichments">Comparing enrichments<a class="anchor" aria-label="anchor" href="#comparing-enrichments"></a>
</h3>
<p>Now we know which domain processes are affected in the mouse model,
let’s consider how we can compare the functional enrichments in these
models with functional enrichments from the AMP-AD cohorts. We’ll need
to compute and examine the functional enrichments from the human cohort
data.</p>
<p>First let’s get the human GO set gene annotations</p>
<div class="codewrapper sourceCode" id="cb105">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get GO annotations</span></span>
<span><span class="va">hs.go.terms</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span></span>
<span>    <span class="va">org.Hs.eg.db</span>, </span>
<span>    keys <span class="op">=</span> <span class="fu">keys</span><span class="op">(</span> <span class="va">org.Hs.eg.db</span> , keytype <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span>,</span>
<span>    columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"GO"</span>, <span class="st">"SYMBOL"</span><span class="op">)</span>,</span>
<span>    keytype <span class="op">=</span> <span class="st">"SYMBOL"</span> </span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the list of gene sets</span></span>
<span><span class="va">hs.go.gene.sets</span> <span class="op">&lt;-</span> <span class="fu">split</span><span class="op">(</span><span class="va">hs.go.terms</span><span class="op">$</span><span class="va">SYMBOL</span>, <span class="va">hs.go.terms</span><span class="op">$</span><span class="va">GO</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter gene sets by size (minSize and maxSize are arguments in fgsea)</span></span>
<span><span class="va">min_size</span> <span class="op">&lt;-</span> <span class="fl">15</span></span>
<span><span class="va">max_size</span> <span class="op">&lt;-</span> <span class="fl">500</span></span>
<span><span class="va">hs.go.sets.filtered</span> <span class="op">&lt;-</span> <span class="va">hs.go.gene.sets</span><span class="op">[</span><span class="fu">sapply</span><span class="op">(</span><span class="va">hs.go.gene.sets</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">length</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">min_size</span> <span class="op">&amp;&amp;</span> <span class="fu">length</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="va">max_size</span><span class="op">)</span><span class="op">]</span></span></code></pre>
</div>
<p>Now we can run the enrichments</p>
<div class="codewrapper sourceCode" id="cb106">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs.gsea</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>, </span>
<span>    <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span>,</span>
<span>    <span class="va">Tissue</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">'DLPFC'</span>, <span class="st">'PHG'</span>,<span class="st">'TCX'</span><span class="op">)</span>,</span>
<span>    <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">hgnc_symbol</span><span class="op">)</span></span>
<span>    , <span class="va">adj.P.Val</span> <span class="op">&lt;=</span> <span class="fl">0.05</span></span>
<span>    <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Study</span>, <span class="va">Tissue</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarise</span><span class="op">(</span>gl <span class="op">=</span> <span class="va">logFC</span> <span class="op">%&gt;%</span> <span class="fu">setNames</span><span class="op">(</span><span class="va">.</span>, <span class="va">hgnc_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">sort</span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    gse <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">gl</span>, </span>
<span>        <span class="op">~</span><span class="fu">fgsea</span><span class="fu">::</span><span class="fu">fgseaMultilevel</span><span class="op">(</span></span>
<span>              pathways <span class="op">=</span>  <span class="va">hs.go.sets.filtered</span>,</span>
<span>              stats <span class="op">=</span> <span class="va">.x</span>,</span>
<span>              minSize <span class="op">=</span> <span class="va">min_size</span>,</span>
<span>              maxSize <span class="op">=</span> <span class="va">max_size</span>,</span>
<span>              nproc <span class="op">=</span> <span class="fl">1</span></span>
<span>            <span class="op">)</span><span class="op">)</span>,</span>
<span>    res <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">gse</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>            <span class="va">go_term_map</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span></span>
<span>              <span class="fu">GO.db</span><span class="fu">::</span><span class="va">GO.db</span>,</span>
<span>              keys <span class="op">=</span> <span class="va">.x</span><span class="op">$</span><span class="va">pathway</span>,</span>
<span>              columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"GOID"</span>, <span class="st">"TERM"</span><span class="op">)</span>,</span>
<span>              keytype <span class="op">=</span> <span class="st">"GOID"</span></span>
<span>            <span class="op">)</span></span>
<span>            </span>
<span>            <span class="fu">inner_join</span><span class="op">(</span><span class="va">.x</span>, <span class="va">go_term_map</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>pathway <span class="op">=</span> <span class="va">GOID</span>, <span class="va">TERM</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu">relocate</span><span class="op">(</span><span class="va">TERM</span>, .after <span class="op">=</span> <span class="va">pathway</span><span class="op">)</span></span>
<span>          <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Let’s map the biodomains onto the enriched GO terms, and plot the
results</p>
<div class="codewrapper sourceCode" id="cb107">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span> <span class="va">hs.gsea</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Study</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">Subdomain</span>, pathway <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'pathway'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Biodomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Biodomain</span><span class="op">)</span>,</span>
<span>    Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span></span>
<span>         <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_sig <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">fct_reorder</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">n_sig</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">padj</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">Study</span><span class="op">)</span>, space <span class="op">=</span> <span class="st">'free'</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, </span>
<span>              color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>We can see fairly similar domain term enrichments from the
transcriptomic data across AMP-AD cohorts. There is strong evidence for
up-regulation among transcripts annotated to GO terms in the
<code>Immune Response</code>, <code>Structural Stabilization</code>, and
<code>Vasculature</code> domains, along with similarly strong evidence
of down-regulation among transcripts annotated to GO terms in the
<code>Synapse</code> domain.</p>
<p>Now lets compare enrichments between species. First we’ll combine the
enrichment results for both species and plot them side-by-side:</p>
<div class="codewrapper sourceCode" id="cb108">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span>  </span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span></span>
<span>    <span class="va">fad.enr</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'5xFAD, '</span>,<span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">model</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span>,</span>
<span>    <span class="va">hs.gsea</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">Study</span>,<span class="st">', '</span>, <span class="va">Tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">model</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>  </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">Subdomain</span>, pathway <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'pathway'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">model</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Biodomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Biodomain</span><span class="op">)</span>,</span>
<span>    Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span></span>
<span>         <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_sig <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">fct_reorder</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">n_sig</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">padj</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">model</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, </span>
<span>              color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Ok, there’s definitely similarity in terms of the direction of
enrichment for domain terms between the human and mouse data. But how
can we assess if the same processes are affected in both mouse and
human? Well, let’s take a look at <code>Immune Response</code> terms
more specifically.</p>
<div class="codewrapper sourceCode" id="cb109">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comb.gsea</span> <span class="op">&lt;-</span> <span class="va">fad.enr</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'5xFAD, '</span>,<span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">model</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">hs.gsea</span><span class="op">$</span><span class="va">res</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">pathway</span>, hs.NES <span class="op">=</span> <span class="va">NES</span>, hs.padj <span class="op">=</span> <span class="va">padj</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">Subdomain</span>, pathway <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'pathway'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">contains</span><span class="op">(</span><span class="st">'NES'</span><span class="op">)</span>, <span class="op">~</span><span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>, <span class="fl">0</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">comb.gsea</span>, <span class="va">Biodomain</span> <span class="op">==</span> <span class="st">"Immune Response"</span><span class="op">)</span>, </span>
<span>  <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">hs.NES</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">model</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>In this case, there are several <code>Immune Response</code> terms
that are significantly enriched in both datasets and all overlapping
terms are enriched with a positive NES value, indicating they are
up-regulated processes in both AD vs Control and the 5xFAD vs WT. We can
more systematically compare enriched terms within each Biodomain. First,
lets generate a list of GO terms that is specific to each Biodomain and
each Subdomain.</p>
<div class="codewrapper sourceCode" id="cb110">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># curate biodomain and sub-domain gene lists</span></span>
<span><span class="va">bd.terms</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span> set <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span> terms <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">set</span>, <span class="op">~</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">GO_ID</span><span class="op">)</span><span class="op">)</span> <span class="op">)</span></span>
<span></span>
<span><span class="va">sd.terms</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span> <span class="va">biodom</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span><span class="op">=</span><span class="st">'domain'</span><span class="op">)</span> <span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span> Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">abbr</span>, <span class="va">Subdomain</span>, <span class="va">subdomain_idx</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    set <span class="op">=</span> <span class="fu">paste0</span><span class="op">(</span><span class="va">abbr</span>,<span class="st">'_'</span>,<span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    bd <span class="op">=</span> <span class="va">Biodomain</span>, sdi <span class="op">=</span> <span class="va">subdomain_idx</span>,</span>
<span>    terms <span class="op">=</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">bd</span>, <span class="va">subdomain_idx</span> <span class="op">==</span> <span class="va">sdi</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">GO_ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co"># combine</span></span>
<span><span class="va">bd.term.list</span> <span class="op">=</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">bd.terms</span>, <span class="va">sd.terms</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">set</span>, <span class="va">terms</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">!=</span> <span class="st">'NA_none'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">bd.terms</span>, <span class="va">sd.terms</span><span class="op">)</span></span></code></pre>
</div>
<p>Next let’s combine all of the human GSEA results with each of the
5xFAD age-stratified GSEA results. Any term that is enriched with an
adjusted p-value &gt; 0 we will set the NES value to 0 and the p-value
to 1.</p>
<div class="codewrapper sourceCode" id="cb111">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fad.vs.amp</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span>fad <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'5xFAD, '</span>, <span class="va">fad.enr</span><span class="op">$</span><span class="va">age</span><span class="op">)</span>, </span>
<span>                       amp <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">hs.gsea</span><span class="op">$</span><span class="va">Study</span>,<span class="st">', '</span>, <span class="va">hs.gsea</span><span class="op">$</span><span class="va">Tissue</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">fad</span>, <span class="va">amp</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>    <span class="va">mm</span> <span class="op">=</span> <span class="va">fad.enr</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'5xFAD, '</span>, <span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">model</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>          <span class="fu">select</span><span class="op">(</span>GOID <span class="op">=</span> <span class="va">pathway</span>, term <span class="op">=</span> <span class="va">TERM</span>, mm.nes <span class="op">=</span> <span class="va">NES</span>, mm.padj <span class="op">=</span> <span class="va">padj</span><span class="op">)</span></span>
<span>    <span class="va">hs</span> <span class="op">=</span> <span class="va">hs.gsea</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">Study</span>,<span class="st">', '</span>, <span class="va">Tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">model</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>          <span class="fu">select</span><span class="op">(</span>GOID <span class="op">=</span> <span class="va">pathway</span>, term <span class="op">=</span> <span class="va">TERM</span>, hs.nes <span class="op">=</span> <span class="va">NES</span>, hs.padj <span class="op">=</span> <span class="va">padj</span><span class="op">)</span></span>
<span>    <span class="fu">full_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'GOID'</span>,<span class="st">'term'</span><span class="op">)</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span></span>
<span>      <span class="fu">across</span><span class="op">(</span><span class="fu">contains</span><span class="op">(</span><span class="st">'padj'</span><span class="op">)</span>, <span class="op">~</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>      <span class="fu">across</span><span class="op">(</span><span class="fu">contains</span><span class="op">(</span><span class="st">'nes'</span><span class="op">)</span>, <span class="op">~</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>, <span class="fl">0</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>      hs.nes <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">hs.padj</span> <span class="op">&gt;</span> <span class="fl">0.05</span>, <span class="fl">0</span>, <span class="va">hs.nes</span><span class="op">)</span>,</span>
<span>      mm.nes <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">mm.padj</span> <span class="op">&gt;</span> <span class="fl">0.05</span>, <span class="fl">0</span>, <span class="va">mm.nes</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span>    <span class="op">}</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Then we can go through each of the GO term lists and perform the
correlation analysis. First we’ll make a list combining all mouse
comparisons and each GO term list to test using the
<code>crossing</code> function:</p>
<div class="codewrapper sourceCode" id="cb112">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">fad.vs.amp</span>, set <span class="op">=</span> <span class="va">bd.term.list</span><span class="op">$</span><span class="va">set</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">comp</span><span class="op">)</span></span></code></pre>
</div>
<p>Now we can add a column for the term sub-set to consider for each
correlation:</p>
<div class="codewrapper sourceCode" id="cb113">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span> data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">data</span>, <span class="va">set</span>, <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">GOID</span> <span class="op">%in%</span> <span class="va">bd.term.list</span><span class="op">$</span><span class="va">terms</span><span class="op">[[</span><span class="fu">which</span><span class="op">(</span><span class="va">bd.term.list</span><span class="op">$</span><span class="va">set</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            n.terms <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="fu">nrow</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            n.sig.both <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">mm.padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">hs.padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nrow</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            jaccard <span class="op">=</span> <span class="va">n.sig.both</span><span class="op">/</span><span class="va">n.terms</span> <span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">comp</span><span class="op">)</span></span></code></pre>
</div>
<p>For several term sets there aren’t many terms enriched. We won’t be
able to compute correlations with these, so we’ll filter them out.</p>
<p>For the correlation we’ll use a nonparametric, rank-based Kendall
correlation. Given that we don’t expect NES values to be normally
distributed, and that all we’re really care to know is whether the terms
are enriched in the same direction or opposite directions between mouse
and human, this will do the job.</p>
<div class="codewrapper sourceCode" id="cb114">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">n.terms</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'hs.nes'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'mm.nes'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'kendall'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">comp</span><span class="op">)</span></span></code></pre>
</div>
<p>Ok, now let’s take a look at the correlation among enriched Biodomain
terms (without an <code>_</code> character in the <code>set</code>
name):</p>
<div class="codewrapper sourceCode" id="cb115">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">F</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>fad <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">fad</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'5xFAD, 10m'</span>,<span class="st">'5xFAD, 6m'</span>,<span class="st">'5xFAD, 4m'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set</span>, <span class="va">fad</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">amp</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>fill <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="va">jaccard</span> <span class="op">)</span>, color <span class="op">=</span> <span class="st">'black'</span>, shape <span class="op">=</span> <span class="fl">21</span>, stroke <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">p.value</span> <span class="op">&lt;=</span> <span class="fl">5e-2</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"black"</span>, shape<span class="op">=</span><span class="fl">0</span>, size<span class="op">=</span> <span class="fl">5</span>, stroke <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span> <span class="st">"Jaccard index"</span>, range <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">.5</span>, <span class="fl">4</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1e-90</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_gradient2</span><span class="op">(</span></span>
<span>    <span class="st">"Kendall's \u03C4 coefficient"</span>,</span>
<span>    low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, mid <span class="op">=</span> <span class="st">'grey95'</span></span>
<span>    , guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span>, drop <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">5</span>,<span class="fl">50</span>,<span class="fl">5</span>,<span class="fl">5</span><span class="op">)</span>,</span>
<span>    strip.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>,colour <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    legend.box <span class="op">=</span> <span class="st">'vertical'</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>The largest overlap is with the <code>Immune Response</code> domain
terms, which are positively correlated between older 5xFAD and the Mayo
and MSSM cohorts. There are also positively correlated terms for the
<code>Apoptosis</code>, <code>Autophagy</code>,
<code>Lipid Metabolism</code>, <code>Structural Stabilization</code>,
and <code>Vasculature</code> domains.</p>
<p>We can also look at the subdomains to get a more specific picture.
Let’s filter to only include significant correlations or any correlation
with a tau statistic with an absolute value &gt; 0.2:</p>
<div class="codewrapper sourceCode" id="cb116">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">l</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">T</span>, </span>
<span>     <span class="op">(</span><span class="va">p.value</span> <span class="op">&lt;</span> <span class="fl">5e-2</span> <span class="op">|</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.3</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">set</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">%in%</span> <span class="va">l</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>max.n <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">n.terms</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    fad <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">fad</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'5xFAD, 10m'</span>,<span class="st">'5xFAD, 6m'</span>,<span class="st">'5xFAD, 4m'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">rev</span><span class="op">)</span>,</span>
<span>    set1 <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">max.n</span>,<span class="st">'] '</span>,<span class="va">.</span><span class="op">)</span>,</span>
<span>    abbr <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>         <span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">fad</span>,<span class="va">set1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">abbr</span><span class="op">)</span>, cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">amp</span><span class="op">)</span>, scales<span class="op">=</span><span class="st">'free'</span>, space<span class="op">=</span><span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>fill <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="va">jaccard</span> <span class="op">)</span>, color <span class="op">=</span> <span class="st">'black'</span>, shape <span class="op">=</span> <span class="fl">21</span>, stroke <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">p.value</span> <span class="op">&lt;=</span> <span class="fl">5e-2</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"black"</span>, shape<span class="op">=</span><span class="fl">0</span>, size<span class="op">=</span> <span class="fl">5</span>, stroke <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span> <span class="st">"Jaccard index"</span>, range <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">.5</span>, <span class="fl">4</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1e-90</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_gradient2</span><span class="op">(</span></span>
<span>    <span class="st">"Kendall's \u03C4 coefficient"</span>,</span>
<span>    low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, mid <span class="op">=</span> <span class="st">'grey95'</span></span>
<span>    , guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span>, drop <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># scale_y_discrete(position = 'right', limits = rev)+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">5</span>,<span class="fl">50</span>,<span class="fl">5</span>,<span class="fl">5</span><span class="op">)</span>,</span>
<span>    strip.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>,colour <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    legend.box <span class="op">=</span> <span class="st">'vertical'</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>This tells us that the positive correlations in
<code>Immune Response</code> are primarily due to terms within “cytokine
production”, “neuroinflammatory response”, and “behavioral defense
response”, while the correlations in the <code>Apoptosis</code> domain
are primarily related to “NF-kappaB signaling”, and the late
<code>Vasculature</code> correlations have to do with
“angiogenesis”.</p>
<p>Are the phagocytosis subdomain terms really negatively
correlated?</p>
<div class="codewrapper sourceCode" id="cb117">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comp</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">==</span> <span class="st">'IR_phagocytosis'</span>, <span class="va">amp</span> <span class="op">==</span> <span class="st">'MAYO, TCX'</span>, <span class="va">fad</span> <span class="op">==</span> <span class="st">'5xFAD, 10m'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre>
</div>
<p>Not really – it is just that different terms from the subdomain are
enriched in each species. All terms from the subdomain are enriched
among up-regulated genes in each species.</p>
<div id="challenge-12" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-12" class="callout-inner">
<h3 class="callout-title">Challenge 12</h3>
<div class="callout-content">
<p>One could also perform a correlation analysis on a gene-by-gene basis
like we did with the co-expression modules, but instead using the genes
within each of the biodomains and subdomains. Compute these results and
compare the the correlations of enriched terms and modules.</p>
</div>
</div>
</div>
<div id="accordionSolution12" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution12" aria-expanded="false" aria-controls="collapseSolution12">
  <h4 class="accordion-header" id="headingSolution12"> Show me the solution </h4>
</button>
<div id="collapseSolution12" class="accordion-collapse collapse" aria-labelledby="headingSolution12" data-bs-parent="#accordionSolution12">
<div class="accordion-body">
<p>first set up the lists of genes associated with biodomain and
subdomain term lists</p>
<div class="codewrapper sourceCode" id="cb118">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bd.genes</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>set <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>genes <span class="op">=</span> <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="va">set</span>,</span>
<span>    <span class="op">~</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unlist</span></span>
<span>  <span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sd.genes</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">biodom</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">abbr</span>, <span class="va">Subdomain</span>, <span class="va">subdomain_idx</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    set <span class="op">=</span> <span class="fu">paste0</span><span class="op">(</span><span class="va">abbr</span>, <span class="st">'_'</span>, <span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    bd <span class="op">=</span> <span class="va">Biodomain</span>,</span>
<span>    sdi <span class="op">=</span> <span class="va">subdomain_idx</span>,</span>
<span>    genes <span class="op">=</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">bd</span>, <span class="va">subdomain_idx</span> <span class="op">==</span> <span class="va">sdi</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unlist</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># combine</span></span>
<span><span class="va">bd.gene.lists</span> <span class="op">=</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">bd.genes</span>, <span class="va">sd.genes</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">set</span>, <span class="va">genes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">!=</span> <span class="st">'NA_none'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">bd.genes</span>, <span class="va">sd.genes</span><span class="op">)</span></span></code></pre>
</div>
<p>next add expression data for human genes</p>
<div class="codewrapper sourceCode" id="cb119">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bd.genes</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>,</span>
<span>         <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span>,</span>
<span>         <span class="va">Tissue</span> <span class="op">==</span> <span class="st">'TCX'</span>,<span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">hgnc_symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">hgnc_symbol</span>, <span class="va">logFC</span>, <span class="va">adj.P.Val</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>,</span>
<span>            <span class="va">mouse.human.ortho</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>hgnc_symbol <span class="op">=</span> <span class="va">human_symbol</span>, symbol <span class="op">=</span> <span class="va">mouse_symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">bd.gene.lists</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>hgnc_symbol <span class="op">=</span> <span class="va">genes</span><span class="op">)</span>,</span>
<span>            <span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>now join the data and perform the correlation analysis</p>
<div class="codewrapper sourceCode" id="cb120">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_vs_ampad</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">fad.deg</span>,</span>
<span>                             <span class="va">bd.genes</span>,</span>
<span>                             by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"symbol"</span><span class="op">)</span>,</span>
<span>                             multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'5xFAD, '</span>, <span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">model</span>,</span>
<span>         <span class="va">set</span>,</span>
<span>         <span class="va">symbol</span>,</span>
<span>         <span class="va">log2FoldChange</span>,</span>
<span>         <span class="va">padj</span>,</span>
<span>         <span class="va">hgnc_symbol</span>,</span>
<span>         <span class="va">logFC</span>,</span>
<span>         <span class="va">adj.P.Val</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">set</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">padj</span>, <span class="va">hgnc_symbol</span>, <span class="va">logFC</span>, <span class="va">adj.P.Val</span><span class="op">)</span>,</span>
<span>    .by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">model</span>, <span class="va">set</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">distinct</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span>, n <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">nrow</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">n</span> <span class="op">&gt;</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">model_vs_ampad</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor_test <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">"log2FoldChange"</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">"logFC"</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"estimate"</span><span class="op">)</span>,</span>
<span>    p_value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"p.value"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">cor_test</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>significant <span class="op">=</span> <span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre>
</div>
<p>finally, plot the biodomain correlations</p>
<div class="codewrapper sourceCode" id="cb121">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>, <span class="va">set</span><span class="op">)</span> <span class="op">==</span> <span class="cn">F</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">model</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'5xFAD, 10m'</span>, <span class="st">'5xFAD, 6m'</span>, <span class="st">'5xFAD, 4m'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">rev</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set</span>, <span class="va">model</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>,</span>
<span>    stroke <span class="op">=</span> <span class="fl">1.2</span>,</span>
<span>    shape <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">6</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="st">"Correlation"</span>,</span>
<span>    low <span class="op">=</span> <span class="st">"#85070C"</span>,</span>
<span>    high <span class="op">=</span> <span class="st">"#164B6E"</span>,</span>
<span>    guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span></span>
<span>      angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>      hjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>      vjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>      size <span class="op">=</span> <span class="fl">12</span></span>
<span>    <span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span></span>
<span>      angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>      vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>,</span>
<span>      hjust <span class="op">=</span> <span class="fl">0.03</span>,</span>
<span>      size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>      face <span class="op">=</span> <span class="st">"bold"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>You can also plot the subdomain correlations, though it is best to
filter these to the strongest and most significant correlations.</p>
<div class="codewrapper sourceCode" id="cb122">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">l</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>, <span class="va">set</span><span class="op">)</span> <span class="op">==</span> <span class="cn">T</span>, <span class="op">(</span><span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">5e-2</span> <span class="op">&amp;</span></span>
<span>                                  <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.25</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">set</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">%in%</span> <span class="va">l</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>max.n <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">n</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    model <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">model</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'5xFAD, 10m'</span>, <span class="st">'5xFAD, 6m'</span>, <span class="st">'5xFAD, 4m'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">rev</span><span class="op">)</span>,</span>
<span>    set1 <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>, <span class="va">max.n</span>, <span class="st">'] '</span>, <span class="va">.</span><span class="op">)</span>,</span>
<span>    abbr <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">model</span>, <span class="va">set1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span></span>
<span>    rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">abbr</span><span class="op">)</span>,</span>
<span>    scales <span class="op">=</span> <span class="st">'free'</span>,</span>
<span>    space <span class="op">=</span> <span class="st">'free'</span>,</span>
<span>    switch <span class="op">=</span> <span class="st">'y'</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>,</span>
<span>    stroke <span class="op">=</span> <span class="fl">1.2</span>,</span>
<span>    shape <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">6</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'bottom'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="st">"Correlation"</span>,</span>
<span>    low <span class="op">=</span> <span class="st">"#85070C"</span>,</span>
<span>    high <span class="op">=</span> <span class="st">"#164B6E"</span>,</span>
<span>    guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span></span>
<span>      angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>      vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>,</span>
<span>      hjust <span class="op">=</span> <span class="fl">0.03</span>,</span>
<span>      size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>      face <span class="op">=</span> <span class="st">"bold"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"left"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Most biodomains have a significant positive correlation with AD
transcriptomes, some get stronger at older ages of 5xFAD. The most
significant correlations among subdomains are for the APP Metabolism,
Apoptosis, Autophagy, Mitochondrial Metabolism, and Myelination domains.
There is also positive correlation for “tau-protein kinase activity.</p>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="conclusion">[7] Conclusion:<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<hr class="half-width">
<p>Overall, by aligning human and mouse omics signatures through the
lens of domains affected in each context, we can get a better
understanding of the relationships between the biological processes
affected in each context.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>AMP-AD gene modules represent major transcriptomic heterogeneity in
AD</li>
<li>Correlation of logFC is a practical approach for human-mouse
alignment of AD-associated transcriptomic signatures</li>
<li>Functional gene set signatures are also a useful point of comparison
between species</li>
<li>Subsetting functions into associated Biological Domains and
Subdomains allows for more granular comparisons</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb123">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb123-2"><a href="#cb123-2" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">5.2</span> (<span class="dv">2025-10-31</span>)</span>
<span id="cb123-3"><a href="#cb123-3" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>pc<span class="sc">-</span>linux<span class="sc">-</span>gnu</span>
<span id="cb123-4"><a href="#cb123-4" tabindex="-1"></a>Running under<span class="sc">:</span> Ubuntu <span class="dv">24</span>.<span class="fl">04.3</span> LTS</span>
<span id="cb123-5"><a href="#cb123-5" tabindex="-1"></a></span>
<span id="cb123-6"><a href="#cb123-6" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb123-7"><a href="#cb123-7" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>x86_64<span class="sc">-</span>linux<span class="sc">-</span>gnu<span class="sc">/</span>openblas<span class="sc">-</span>pthread<span class="sc">/</span>libblas.so<span class="fl">.3</span> </span>
<span id="cb123-8"><a href="#cb123-8" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>x86_64<span class="sc">-</span>linux<span class="sc">-</span>gnu<span class="sc">/</span>openblas<span class="sc">-</span>pthread<span class="sc">/</span>libopenblasp<span class="sc">-</span>r0.<span class="dv">3</span>.<span class="fl">26.</span>so;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb123-9"><a href="#cb123-9" tabindex="-1"></a></span>
<span id="cb123-10"><a href="#cb123-10" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb123-11"><a href="#cb123-11" tabindex="-1"></a> [<span class="dv">1</span>] LC_CTYPE<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>       LC_NUMERIC<span class="ot">=</span>C              </span>
<span id="cb123-12"><a href="#cb123-12" tabindex="-1"></a> [<span class="dv">3</span>] LC_TIME<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>        LC_COLLATE<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>    </span>
<span id="cb123-13"><a href="#cb123-13" tabindex="-1"></a> [<span class="dv">5</span>] LC_MONETARY<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>    LC_MESSAGES<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>   </span>
<span id="cb123-14"><a href="#cb123-14" tabindex="-1"></a> [<span class="dv">7</span>] LC_PAPER<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>       LC_NAME<span class="ot">=</span>C                 </span>
<span id="cb123-15"><a href="#cb123-15" tabindex="-1"></a> [<span class="dv">9</span>] LC_ADDRESS<span class="ot">=</span>C               LC_TELEPHONE<span class="ot">=</span>C            </span>
<span id="cb123-16"><a href="#cb123-16" tabindex="-1"></a>[<span class="dv">11</span>] LC_MEASUREMENT<span class="ot">=</span>en_US.UTF<span class="dv">-8</span> LC_IDENTIFICATION<span class="ot">=</span>C       </span>
<span id="cb123-17"><a href="#cb123-17" tabindex="-1"></a></span>
<span id="cb123-18"><a href="#cb123-18" tabindex="-1"></a>time zone<span class="sc">:</span> Etc<span class="sc">/</span>UTC</span>
<span id="cb123-19"><a href="#cb123-19" tabindex="-1"></a>tzcode source<span class="sc">:</span> <span class="fu">system</span> (glibc)</span>
<span id="cb123-20"><a href="#cb123-20" tabindex="-1"></a></span>
<span id="cb123-21"><a href="#cb123-21" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb123-22"><a href="#cb123-22" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb123-23"><a href="#cb123-23" tabindex="-1"></a></span>
<span id="cb123-24"><a href="#cb123-24" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb123-25"><a href="#cb123-25" tabindex="-1"></a> [<span class="dv">1</span>] BiocManager_1.<span class="fl">30.27</span> compiler_4.<span class="fl">5.2</span>      cli_3.<span class="fl">6.5</span>          </span>
<span id="cb123-26"><a href="#cb123-26" tabindex="-1"></a> [<span class="dv">4</span>] tools_4.<span class="fl">5.2</span>         pillar_1.<span class="fl">11.1</span>       otel_0.<span class="fl">2.0</span>         </span>
<span id="cb123-27"><a href="#cb123-27" tabindex="-1"></a> [<span class="dv">7</span>] glue_1.<span class="fl">8.0</span>          yaml_2.<span class="fl">3.12</span>         vctrs_0.<span class="fl">7.1</span>        </span>
<span id="cb123-28"><a href="#cb123-28" tabindex="-1"></a>[<span class="dv">10</span>] knitr_1<span class="fl">.51</span>          xfun_0<span class="fl">.56</span>           lifecycle_1.<span class="fl">0.5</span>    </span>
<span id="cb123-29"><a href="#cb123-29" tabindex="-1"></a>[<span class="dv">13</span>] rlang_1.<span class="fl">1.7</span>         renv_1.<span class="fl">1.7</span>          evaluate_1.<span class="fl">0.5</span>     </span></code></pre>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-choosing_the_right_model"><p>Content from <a href="choosing_the_right_model.html">Choosing the right animal models</a></p>
<hr>
<p>Last updated on 2026-02-12 |

        <a href="https://github.com/asliuyar/translational-omics-AD/edit/main/episodes/choosing_the_right_model.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What are some approaches that have been used to reduce the
heterogeneity in human AD patient populations?</li>
<li>How well do the changes we observe in AD mouse models align with
human AD data?</li>
<li>How do we perform corss-species comparison?</li>
<li>Which animal models best capture features of human AD subtypes?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Approaches to align mouse data to human data</li>
<li>Review the human AD co-expression modules</li>
<li>Understand the data from AD mouse models</li>
<li>Perform correlation analysis between mouse models and human
modules</li>
<li>Perform correlation analysis using the human subtypes</li>
<li>Understand the biological domains and subdomains of AD</li>
<li>Use domain annotations to compare between species</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Authors: Ravi Pandey &amp; Greg Cary, Jackson Laboratory</p>
<div class="section level3">
<h3 id="load-required-libraries">Load required libraries<a class="anchor" aria-label="anchor" href="#load-required-libraries"></a>
</h3>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">dplyr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tibble</span><span class="op">)</span></span>
<span><span class="co">#library(limma)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">corrplot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>corrplot 0.95 loaded</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">ggplot2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">purrr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">forcats</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">gt</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
Attaching package: 'gt'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>The following object is masked from 'package:cowplot':

    as_gtable</code></pre>
</div>
<p>Mouse models of AD serve as indispensable platforms for
comprehensively characterizing AD pathology, disease progression, and
biological mechanisms. However, selection of the right model in
preclinical research and translation of findings to clinical populations
are intricate processes that require identification of
pathophysiological resemblance between model organisms and humans. Many
existing clinical trials that showed promising efficacy in one
particular mouse model later do not align with human trial results,
assuming that study had consisted of a heterogeneous group of
participants, and individual animal models may only recapitulate
features of a subgroup of human cases.</p>
<figure><img src="../figures/subtype_intro.png" width="1000px" align="center" alt="Heterogenetiy in Alzheimer's Disease" class="figure mx-auto d-block"></figure><p>To improve interspecies translation, it is necessary to
comprehensively compare molecular signatures in mouse models with
subgroup of human AD cases with distinct molecular signatures.</p>
<p>Through omics approaches, we can assess how genetic perturbations in
mice align with changes observed in human LOAD study cohorts. This
allows us to identify mouse models and genetic factors that correspond
to specific subsets of human AD subtypes. Within these subtypes,
genesets are highly co-expressed and represent distinct molecular
pathways.</p>
</div>
<section><h2 class="section-heading" id="overview-of-human-ad-data">Overview of Human AD Data<a class="anchor" aria-label="anchor" href="#overview-of-human-ad-data"></a>
</h2>
<hr class="half-width">
<p>Three independent human brain transcriptome studies ROSMAP [Religious
Orders Study and the Memory and Aging Project], MSSM [Mount Sinai School
of Medicine], and Mayo collected human postmortem brain RNA-seq data
from seven distinct regions: dorsolateral prefrontal cortex (DLPFC),
temporal cortex (TCX), inferior frontal gyrus (IFG), superior temporal
gyrus (STG), frontal pole (FP), parahippocampal gyrus (PHG), and
cerebellum (CBE).</p>
<div class="section level3">
<h3 id="amp-ad-modules">AMP-AD Modules<a class="anchor" aria-label="anchor" href="#amp-ad-modules"></a>
</h3>
<p><a href="https://doi.org/10.1016/j.celrep.2020.107908" class="external-link">Wan, et
al.</a> performed multi method co-expression network analysis followed
by differential analysis and found 30 co-expression modules related LOAD
pathology from human cohort study. Among the 30 aggregate co-expression
modules, five consensus clusters have been described by Wan, et
al. These consensus clusters consist of a subset of modules which are
associated with similar AD related changes across the multiple studies
and brain regions.</p>
<figure><img src="../figures/Picture1.png" width="1000px" align="center" alt="Transcriptomic of Alzheimer's Disease" class="figure mx-auto d-block"></figure><p>Here we are showing matrix view of gene content overlap between these
module, and you can see few strongly overlapping group of modules,
implicating similar pathology in different studies in different brain
regions.</p>
</div>
<div class="section level3">
<h3 id="ad-subtypes">AD Subtypes<a class="anchor" aria-label="anchor" href="#ad-subtypes"></a>
</h3>
<p>Post mortem transcriptomics from AMP-AD and similar studies have
enabled the partitioning of AD cases into potential disease subtypes.
These studies have often stratified AD subjects into inflammatory and
non-inflammatory subtypes.</p>
<!--html_preserve-->
<div id="meunwmbjmp" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#meunwmbjmp table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#meunwmbjmp thead, #meunwmbjmp tbody, #meunwmbjmp tfoot, #meunwmbjmp tr, #meunwmbjmp td, #meunwmbjmp th {
  border-style: none;
}

#meunwmbjmp p {
  margin: 0;
  padding: 0;
}

#meunwmbjmp .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#meunwmbjmp .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#meunwmbjmp .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#meunwmbjmp .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#meunwmbjmp .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#meunwmbjmp .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#meunwmbjmp .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#meunwmbjmp .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#meunwmbjmp .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#meunwmbjmp .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#meunwmbjmp .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#meunwmbjmp .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#meunwmbjmp .gt_spanner_row {
  border-bottom-style: hidden;
}

#meunwmbjmp .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#meunwmbjmp .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#meunwmbjmp .gt_from_md > :first-child {
  margin-top: 0;
}

#meunwmbjmp .gt_from_md > :last-child {
  margin-bottom: 0;
}

#meunwmbjmp .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#meunwmbjmp .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#meunwmbjmp .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#meunwmbjmp .gt_row_group_first td {
  border-top-width: 2px;
}

#meunwmbjmp .gt_row_group_first th {
  border-top-width: 2px;
}

#meunwmbjmp .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#meunwmbjmp .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#meunwmbjmp .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#meunwmbjmp .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#meunwmbjmp .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#meunwmbjmp .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#meunwmbjmp .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#meunwmbjmp .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#meunwmbjmp .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#meunwmbjmp .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#meunwmbjmp .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#meunwmbjmp .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#meunwmbjmp .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#meunwmbjmp .gt_left {
  text-align: left;
}

#meunwmbjmp .gt_center {
  text-align: center;
}

#meunwmbjmp .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#meunwmbjmp .gt_font_normal {
  font-weight: normal;
}

#meunwmbjmp .gt_font_bold {
  font-weight: bold;
}

#meunwmbjmp .gt_font_italic {
  font-style: italic;
}

#meunwmbjmp .gt_super {
  font-size: 65%;
}

#meunwmbjmp .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#meunwmbjmp .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#meunwmbjmp .gt_indent_1 {
  text-indent: 5px;
}

#meunwmbjmp .gt_indent_2 {
  text-indent: 10px;
}

#meunwmbjmp .gt_indent_3 {
  text-indent: 15px;
}

#meunwmbjmp .gt_indent_4 {
  text-indent: 20px;
}

#meunwmbjmp .gt_indent_5 {
  text-indent: 25px;
}

#meunwmbjmp .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#meunwmbjmp div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead><tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Study">
Study
</th>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Cohort">
Cohort
</th>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Tissue">
Tissue
</th>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="DataTypes">
DataTypes
</th>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Methods">
Methods
</th>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Subtypes">
Subtypes
</th>
</tr></thead>
<tbody class="gt_table_body">
<tr>
<td headers="Study" class="gt_row gt_left">
Milind et al.,2020
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP, MSBB, MAYO
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC, PHG, FP,TCX
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ
</td>
<td headers="Methods" class="gt_row gt_left">
Iterative WGCNA followed by NbClust R package
</td>
<td headers="Subtypes" class="gt_row gt_left">
2,2, 3
</td>
</tr>
<tr>
<td headers="Study" class="gt_row gt_left">
Neff et al., 2021
</td>
<td headers="Cohort" class="gt_row gt_left">
MSBB &amp; ROSMAP
</td>
<td headers="Tissue" class="gt_row gt_left">
PHG,DLPFC
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ
</td>
<td headers="Methods" class="gt_row gt_left">
WSCNA, Hierarchical and K-means clustering
</td>
<td headers="Subtypes" class="gt_row gt_left">
5
</td>
</tr>
<tr>
<td headers="Study" class="gt_row gt_left">
Yasser et al.,2022
</td>
<td headers="Cohort" class="gt_row gt_left">
ROSMAP
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC
</td>
<td headers="DataTypes" class="gt_row gt_left">
Multi-Omics
</td>
<td headers="Methods" class="gt_row gt_left">
Novel ML framework mcTI
</td>
<td headers="Subtypes" class="gt_row gt_left">
3
</td>
</tr>
<tr>
<td headers="Study" class="gt_row gt_left">
Mukherjee et al., 2020
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP &amp; MAYO
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC &amp; TCX
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ
</td>
<td headers="Methods" class="gt_row gt_left">
DDRTree manifold learning approach using Monocle 2 R package
</td>
<td headers="Subtypes" class="gt_row gt_left">
6
</td>
</tr>
<tr>
<td headers="Study" class="gt_row gt_left">
Laura Heath et al.(unpublished)
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ, PROTEOMICS, METABOLOMICS
</td>
<td headers="Methods" class="gt_row gt_left">
DDRTree manifold learning approach
</td>
<td headers="Subtypes" class="gt_row gt_left">
~7
</td>
</tr>
<tr>
<td headers="Study" class="gt_row gt_left">
Zheng et al. 2021
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ
</td>
<td headers="Methods" class="gt_row gt_left">
Non-negative Matrix (NMF)
</td>
<td headers="Subtypes" class="gt_row gt_left">
2
</td>
</tr>
<tr>
<td headers="Study" class="gt_row gt_left">
Yang et al., 2023
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC
</td>
<td headers="DataTypes" class="gt_row gt_left">
Multi-Omics
</td>
<td headers="Methods" class="gt_row gt_left">
Similarity Network Fusion
</td>
<td headers="Subtypes" class="gt_row gt_left">
2 and 5
</td>
</tr>
<tr>
<td headers="Study" class="gt_row gt_left">
Lee et al., 2023
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC, AC, PCC
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ
</td>
<td headers="Methods" class="gt_row gt_left">
CCA, k-means clustering, NMF
</td>
<td headers="Subtypes" class="gt_row gt_left">
2
</td>
</tr>
</tbody>
</table>
</div>
<!--/html_preserve-->
<p>For this workshop, we will work with Milind’s and Neff’s subtype.</p>
<div class="section level4">
<h4 id="milinds-ad-subtype">Milind’s AD Subtype<a class="anchor" aria-label="anchor" href="#milinds-ad-subtype"></a>
</h4>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/32492070/" class="external-link">Milind et al.</a>
integrated post mortem brain co-expression data from the frontal cortex,
temporal cortex, and hippocampus brain regions and stratified patients
into different molecular subtypes based on molecular profiles in three
independent human LOAD cohorts (ROS/MAP, Mount Sinai Brain Bank, and
Mayo Clinic).</p>
<p>Two distinct LOAD subtypes were identified in the ROSMAP cohort,
three LOAD subtypes were identified in the Mayo cohort, and two distinct
LOAD subtypes were identified in the MSBB cohort. Similar subtype
results were observed in each cohort, with LOAD subtypes found to
primarily differ in their inflammatory response based on differential
expression analysis.</p>
<figure><img src="../figures/milindsubtype.png" width="1000px" align="center" alt="Transcriptomic of Alzheimer's Disease" class="figure mx-auto d-block"></figure>
</div>
<div class="section level4">
<h4 id="neffs-ad-subtype">Neff’s AD Subtype<a class="anchor" aria-label="anchor" href="#neffs-ad-subtype"></a>
</h4>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/33523961/" class="external-link">Neff et al.</a>
investigated molecular subtypes of AD in the MSBB-AD study and
replicated findings in the ROSMAP. They have adopted weighted sample
gene network analysis (WSCNA) clustering algorithm to identify subgrups
of AD patients. WSCNA identifies five subtypes in the MSBB-AD (clusters
A, B1, B2, C1, and C2) across all 151 participants with PHG
transcriptomic data. These subtypes were classified into three larger
classes: typical AD (subtype C1 &amp; C2), intermediate (subtype B1
&amp; B2), or atypical AD (subtype A), by molecular presentation when
compared to the Blalock signatures of AD.</p>
<figure><img src="../figures/neff_fig2.png" width="1000px" align="center" alt="Transcriptomic of Alzheimer's Disease" class="figure mx-auto d-block"></figure><figure><img src="../figures/Neff6.png" width="1000px" align="center" alt="Transcriptomic of Alzheimer's Disease" class="figure mx-auto d-block"></figure>
</div>
</div>
</section><section><h2 class="section-heading" id="overview-of-mouse-model">Overview of Mouse Model<a class="anchor" aria-label="anchor" href="#overview-of-mouse-model"></a>
</h2>
<hr class="half-width">
<p>We are going to utilize transcriptomics data from mouse models
expressing human risk variants created on a more LOAD-susceptible
genetic background expressing humanized APOE with the ε4 variant and the
R47H mutation in Trem2, two of the strongest genetic risk factors for
LOAD. Chracaterization of these mouse models is publsihed here: <a href="https://alz-journals.onlinelibrary.wiley.com/doi/full/10.1002/alz.13840" class="external-link">In
vivo validation of late-onset Alzheimer’s disease genetic risk
factors</a></p>
<figure><img src="../figures/table1.png" width="1000px" align="center" alt="Transcriptomic of Alzheimer's Disease" class="figure mx-auto d-block"></figure><p>In this session, we are going to reproduce some of the results
published in the paper</p>
</section><section><h2 class="section-heading" id="mouse-data">Mouse Data<a class="anchor" aria-label="anchor" href="#mouse-data"></a>
</h2>
<hr class="half-width">
<p>The <a href="https://link.springer.com/article/10.1186/s13024-020-00412-5" class="external-link">NanoString
Mouse AD gene expression panel</a> was used for gene expression
profiling on the nCounter platform (NanoString, Seattle, WA, USA).
NanoString gene expression panels are comprised of 770 probes. Mouse
NanoString gene expression data were collected from brain hemisphere
homogenates at 4 and 12 months of age for both sexes. The nSolver
software was used for generating NanoString gene expression counts.</p>
<p>Normalization was done by dividing counts within a lane by geometric
mean of the designated housekeeping genes from the same lane. Next,
normalized count values were log-transformed and corrected for potential
batch effects using ComBat.</p>
<p>You can find data associated with this study on synapse: <a href="https://www.synapse.org/Synapse:syn21595258" class="external-link uri">https://www.synapse.org/Synapse:syn21595258</a></p>
<div class="section level3">
<h3 id="synapse-data-download">Synapse Data Download<a class="anchor" aria-label="anchor" href="#synapse-data-download"></a>
</h3>
<p>You can download the data from Synapse data repository. API clients
provide a way to use Synapse programmatically. Installation instructions
are available at <a href="https://help.synapse.org/docs/Installing-Synapse-API-Clients.1985249668.html" class="external-link">Synapse
API Documentation Site</a>.</p>
<p>The Synapse command line client is implemented in Python and comes
with the Synapse Python package. To install the Synapse command line
client, make sure that you have Python and pip installed. For more
information, see the <a href="https://www.python.org/downloads/" class="external-link">Python</a> and <a href="https://pip.pypa.io/en/stable/installing/" class="external-link">pip</a> installation
instructions.</p>
<p>First, Define the data directory:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">datadir</span> <span class="op">&lt;-</span> <span class="st">"/sbgenomics/projects/asliuyar/asliuyar/ad-omics/data/7_ADsubtypes_Pandey_Cary/"</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="count-matrix">Count matrix<a class="anchor" aria-label="anchor" href="#count-matrix"></a>
</h4>
<p>You can access the data from synapse</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NS.INTENSITY</span> <span class="op">&lt;-</span> <span class="fu">fread</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn59479928"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span>,check.names <span class="op">=</span> <span class="cn">F</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre>
</div>
<p>Above command will download the relevent file in your working
directory.</p>
<p>Now, let’s read the downloaded file</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NS.INTENSITY</span> <span class="op">&lt;-</span> <span class="fu">read.csv</span><span class="op">(</span><span class="fu">paste0</span><span class="op">(</span><span class="va">datadir</span>,                            <span class="st">"Nanostringdata_logNormalized_BatchCorrected_MergedTg_LOAD1_PrimaryScreenPaper.csv"</span><span class="op">)</span>,</span>
<span>                         check.names <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                         row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s examine the data,</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NS.INTENSITY</span> <span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span></span></code></pre>
</div>
<p>Here, row’s represent gene names and columns are all mouse
samples.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">NS.INTENSITY</span><span class="op">)</span></span></code></pre>
</div>
<p>Total 763 genes were quantified for 718 mouse samples.</p>
</div>
<div class="section level4">
<h4 id="metadata">Metadata<a class="anchor" aria-label="anchor" href="#metadata"></a>
</h4>
<p>You can access the biospecimen and individual metadata file
associated with this study from synapse.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># biospecimen metadata</span></span>
<span><span class="va">bio_meta</span> <span class="op">&lt;-</span> <span class="fu">fread</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn58614964"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span>,check.names <span class="op">=</span> <span class="cn">F</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># individual metadata</span></span>
<span><span class="va">ind_meta</span> <span class="op">&lt;-</span> <span class="fu">fread</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn59479967"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span>,check.names <span class="op">=</span> <span class="cn">F</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre>
</div>
<p>You can process and join these files to create final metadata file as
discussed earlier in this course.</p>
<p>To expedite the analysis, we’ll utilize the metadata file provided in
the session’s data folder for subsequent analyses.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">read.csv</span><span class="op">(</span><span class="fu">paste0</span><span class="op">(</span><span class="va">datadir</span>,</span>
<span>                            <span class="st">"metadata_NSdata.csv"</span><span class="op">)</span>,</span>
<span>                     check.names <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span></code></pre>
</div>
<p>You might be familiar with most of the co-variates beside Binding
Density. Briefly, In NanoString nCounter technology, binding density
refers to the number of fluorescent spots (reporter probes) per square
micron on the imaging surface of a sample in the nCounter cartridge.</p>
<p>Binding density can be influenced by several factors, including RNA
input mass, number of targets and expression of targets.Highly expressed
genes and/or higher RNA mass result in higher binding density. Low
binding density amy indicate that the assay is not sensitive enough to
detect the target molecules, while high binding density can indicate
over-saturation where multiple probes might be overlapping, potentially
leading to underestimation of target molecule counts.</p>
<p>We can modify the metadata to only include covariates we’ll need for
this analysis</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="va">metadata</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">Sample</span>, Sex<span class="op">=</span><span class="va">SEX</span>, Age<span class="op">=</span><span class="va">AGE</span>,</span>
<span>                <span class="va">Genotype</span>, BD<span class="op">=</span><span class="va">BindingDensity</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span></code></pre>
</div>
<p>Using <code>pivot_longer()</code> function to “lengthens” data
i.e. increasing the number of rows and decreasing the number of
columns.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NS.INTENSITY.long</span> <span class="op">&lt;-</span> <span class="va">NS.INTENSITY</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu">rownames_to_column</span><span class="op">(</span><span class="va">.</span>, <span class="st">"symbol"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols<span class="op">=</span><span class="op">-</span><span class="va">symbol</span>,</span>
<span>               names_to<span class="op">=</span><span class="st">"Sample"</span>,</span>
<span>               values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">NS.INTENSITY.long</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will join the countdata with metadata by <code>Sample</code>
column</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mydat_with_metadata</span> <span class="op">&lt;-</span> <span class="va">NS.INTENSITY.long</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">metadata</span>,by<span class="op">=</span><span class="st">"Sample"</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s check the joined data table</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">mydat_with_metadata</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="create-design-matrix">Create design matrix<a class="anchor" aria-label="anchor" href="#create-design-matrix"></a>
</h4>
<p>Let’s create design matrix for multiple regression analyses. In this
formulation, B6J will be used as the control for the 5xFAD and LOAD1
mouse models, whereas LOAD1 served as controls for GWAS-based models in
order to estimate the effects of individual variants.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mydat_with_design</span> <span class="op">&lt;-</span> <span class="va">mydat_with_metadata</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sex <span class="op">=</span> <span class="fu">ifelse</span><span class="op">(</span><span class="va">Sex</span> <span class="op">%in%</span> <span class="st">"M"</span>,<span class="fl">1</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>APOE4.Trem2.R47H<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"APOE4Trem2"</span>,<span class="st">"LOAD1"</span>,<span class="st">"A/T.MIXED-WT"</span>,<span class="st">"A/T.IL1RAP-KO"</span>,<span class="st">"A/T.CR1&lt;B&gt;"</span>,<span class="st">"LOAD1.Shc2"</span>,<span class="st">"LOAD1.Slc6a17"</span>,<span class="st">"LOAD1.Erc2"</span>,<span class="st">"A/T.SNX1-KI"</span>,<span class="st">"A/T.CLASP2-KI"</span>,<span class="st">"A/T.MTHFR-KI"</span>,<span class="st">"A/T.ABCA7-KI"</span>,<span class="st">"A/T.CEACAM1-KO"</span>,<span class="st">"A/T.PLCG2.M28L"</span>,<span class="st">"hATA"</span>,<span class="st">"HFD.A/T.MIXED-WT"</span>,<span class="st">"HFD.A/T.MTHFR-KI"</span>,<span class="st">"HFD.hATA"</span>,<span class="st">"HFD.A/T.PLCG2.M28L"</span>,<span class="st">"A/T.SORL1"</span>,<span class="st">"A/T.MTMR4"</span>,<span class="st">"A/T.MEOX2"</span><span class="op">)</span>,<span class="fl">1</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>FADX5<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Tg"</span>,<span class="st">"5xFAD"</span><span class="op">)</span>,<span class="fl">1</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>MTHFR.KI<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.MTHFR-KI"</span>,<span class="st">"HFD.A/T.MTHFR-KI"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>PLCG2.M28L<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.PLCG2.M28L"</span>,<span class="st">"HFD.A/T.PLCG2.M28L"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>ABCA7.KI<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.ABCA7-KI"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>CEACAM1.KO<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.CEACAM1-KO"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>SNX1<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.SNX1-KI"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>CLASP2<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.CLASP2-KI"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>SORL1<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.Sorl1"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>MTMR4<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.Mtmr4"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>MEOX2<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.MEOX2"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>SHC2<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"LOAD1.Shc2"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>SLC6A17<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"LOAD1.Slc6a17"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">Sex</span>,<span class="op">-</span><span class="va">Genotype</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">mydat_with_design</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="determine-the-effects-of-each-factor-in-mouse-models">Determine the effects of each factor in mouse models<a class="anchor" aria-label="anchor" href="#determine-the-effects-of-each-factor-in-mouse-models"></a>
</h4>
<p>Now, we will determine the effects of each factor (sex and genetic
variants) by fitting a multiple regression model using the lm function
in R.</p>
<p><strong>We are going to focus on only 12 months old mice
data.</strong></p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lm.results</span> <span class="op">&lt;-</span> <span class="va">mydat_with_design</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Age</span> <span class="op">==</span> <span class="fl">12</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">split</span><span class="op">(</span><span class="va">.</span><span class="op">$</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="op">~</span> <span class="fu">lm</span><span class="op">(</span></span>
<span>    <span class="va">value</span> <span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">BD</span> <span class="op">+</span>  <span class="va">APOE4.Trem2.R47H</span> <span class="op">+</span> <span class="va">FADX5</span> <span class="op">+</span> <span class="va">MTHFR.KI</span> <span class="op">+</span> <span class="va">ABCA7.KI</span> <span class="op">+</span> <span class="va">CEACAM1.KO</span> <span class="op">+</span> <span class="va">SNX1</span> <span class="op">+</span> <span class="va">CLASP2</span> <span class="op">+</span> <span class="va">PLCG2.M28L</span> <span class="op">+</span> <span class="va">SORL1</span> <span class="op">+</span> <span class="va">MTMR4</span> <span class="op">+</span> <span class="va">SHC2</span> <span class="op">+</span> <span class="va">MEOX2</span> <span class="op">+</span> <span class="va">SLC6A17</span>,</span>
<span>    data <span class="op">=</span> <span class="va">.x</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">map</span><span class="op">(</span><span class="va">summary</span><span class="op">)</span></span></code></pre>
</div>
<p>Results of regression analysis for each gene for each factor is
stored in <code>lm.results</code>.</p>
<p>Let’s check the result:</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lm.results</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<p>Now, we will extract effect i.e. regression coefficients for each
gene for each factor from output and will store in a data frame.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">effects</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="fu">bind_rows</span><span class="op">(</span> </span>
<span>  <span class="fu">lapply</span><span class="op">(</span><span class="va">lm.results</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span>,<span class="st">"Estimate"</span><span class="op">]</span> <span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>names <span class="op">=</span> <span class="fu">names</span><span class="op">(</span><span class="va">lm.results</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">column_to_rownames</span><span class="op">(</span><span class="va">.</span>,var<span class="op">=</span><span class="st">"names"</span><span class="op">)</span></span></code></pre>
</div>
<p>let’s see few entries of this data</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">effects</span><span class="op">)</span></span></code></pre>
</div>
<p>rename columns for clear annotation</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colnames</span><span class="op">(</span><span class="va">effects</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"(Intercept)"</span>, </span>
<span>                       <span class="st">"Sex (Male)"</span>, </span>
<span>                       <span class="st">"BD"</span>, </span>
<span>                       <span class="st">"LOAD1"</span>,</span>
<span>                       <span class="st">"5xFAD"</span>,</span>
<span>                       <span class="st">"Mthfr*677C&gt;T"</span>, </span>
<span>                       <span class="st">"Abca7*A1527G"</span>, </span>
<span>                       <span class="st">"Ceacam1 KO"</span>, </span>
<span>                       <span class="st">"Snx1*D465N"</span>, </span>
<span>                       <span class="st">"Clasp2*L163P"</span>, </span>
<span>                       <span class="st">"Plcg2*M28L"</span>,</span>
<span>                       <span class="st">"Sorl1*A528T"</span>,</span>
<span>                       <span class="st">"Mtmr4*V297G"</span>,</span>
<span>                       <span class="st">"Shc2*V433M"</span>,</span>
<span>                       <span class="st">"Meox2 KO (HET)"</span>,</span>
<span>                       <span class="st">"Slc6a17*P61P"</span></span>
<span>                       <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">effects</span><span class="op">)</span></span></code></pre>
</div>
<p>we will only keep factor of interest.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ns_effects</span> <span class="op">&lt;-</span> <span class="va">effects</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu">rownames_to_column</span><span class="op">(</span><span class="va">.</span>,<span class="st">"symbol"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="st">"(Intercept)"</span>, <span class="op">-</span><span class="st">"BD"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols<span class="op">=</span><span class="op">-</span><span class="va">symbol</span>,</span>
<span>               names_to<span class="op">=</span><span class="st">"Variant"</span>,</span>
<span>               values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ns_effects</span><span class="op">)</span></span></code></pre>
</div>
<p>You can save the results for future use.</p>
<p>We will use these values to perform correlation analyses with human
AD Data.</p>
</div>
</div>
</section><section><h2 class="section-heading" id="correlation-between-mouse-models-and-human-ad-modules">Correlation between mouse models and human AD modules<a class="anchor" aria-label="anchor" href="#correlation-between-mouse-models-and-human-ad-modules"></a>
</h2>
<hr class="half-width">
<ul>
<li>Compare Human AD to mouse genetic effects for each orthologous gene
in a given module
<ul>
<li>h = human gene expression (Log2 RNA-seq Fold Change control/AD)</li>
<li>β = mouse gene expression effect from linear regression model (Log2
RNA-seq TPM) <span class="math display">\[cor.test(LogFC(h),
β)\]</span>
</li>
</ul>
</li>
</ul>
<p>These appraoches allow us to assess directional coherence between
AMP-AD modules and the effects of genetic perturbations in mice. In this
lesson, we are going to use second approach.</p>
<p>Log2FC values for human transcripts were obtained through the AD
Knowledge Portal55 (<a href="https://www.synapse.org/#!Synapse:syn14237651" class="external-link uri">https://www.synapse.org/#!Synapse:syn14237651</a>).</p>
<p>Let’s start!</p>
<div class="section level4">
<h4 id="loading-amp-ad-module-data">Loading AMP-AD module data<a class="anchor" aria-label="anchor" href="#loading-amp-ad-module-data"></a>
</h4>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lnames</span> <span class="op">=</span> <span class="fu">load</span><span class="op">(</span><span class="fu">paste0</span><span class="op">(</span><span class="va">datadir</span>,<span class="st">"AMPADModuleData_Correlation.RData"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s check AMP-AD data</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lnames</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules_fc</span><span class="op">)</span></span></code></pre>
</div>
<p>First, we will join regression coefficients and human
ampad_modules_fc log fold change datasets for all genes.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ns_vs_ampad_fc</span> <span class="op">&lt;-</span> <span class="va">ns_effects</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">ampad_modules_fc</span>, </span>
<span>             by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"symbol"</span><span class="op">)</span></span>
<span>             <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ns_vs_ampad_fc</span><span class="op">)</span></span></code></pre>
</div>
<p>Now, we will create a list-columns of data frame using nest function
of tidyverse package. Nesting is implicitly a summarising operation: you
get one row for each group defined by the non-nested columns.</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span>  <span class="op">&lt;-</span> <span class="va">ns_vs_ampad_fc</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">module</span>,<span class="va">Variant</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">value</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">df</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we compute correlation coefficients using cor.test function
built in R as following:</p>
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor_test <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">"value"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                                    <span class="va">.x</span><span class="op">[[</span><span class="st">"ampad_fc"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                                    method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    estimate <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"estimate"</span><span class="op">)</span>,</span>
<span>    p_value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"p.value"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">cor_test</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cor.df</span><span class="op">)</span></span></code></pre>
</div>
<p>We will do some data wrangling to convert it into usable format for
plotting.</p>
<p>First, label correlations significant based on p-value, then join
module cluster information to correlation table</p>
<div class="codewrapper sourceCode" id="cb40">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">variant_module.df</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>significant <span class="op">=</span> <span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,age_group<span class="op">=</span><span class="st">"12 Months"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">module_clusters</span>, by <span class="op">=</span> <span class="st">"module"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">cluster</span>, <span class="va">cluster_label</span>, <span class="va">module</span>,<span class="va">Variant</span>,<span class="va">age_group</span>, correlation <span class="op">=</span> <span class="va">estimate</span>, <span class="va">p_value</span>, <span class="va">significant</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ordered.variant</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Sex (Male)"</span>, <span class="st">"5xFAD"</span>, <span class="st">"LOAD1"</span>,  </span>
<span>                     <span class="st">"Abca7*A1527G"</span>, <span class="st">"Ceacam1 KO"</span>, <span class="st">"Mthfr*677C&gt;T"</span>,<span class="st">"Shc2*V433M"</span>,<span class="st">"Slc6a17*P61P"</span>,<span class="st">"Clasp2*L163P"</span>,<span class="st">"Sorl1*A528T"</span>,</span>
<span>                     <span class="st">"Meox2 KO (HET)"</span>,<span class="st">"Snx1*D465N"</span> ,<span class="st">"Plcg2*M28L"</span>,<span class="st">"Mtmr4*V297G"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ordered.variant</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">ordered.variant</span><span class="op">)</span></span>
<span><span class="co"># Create a version of the data for plotting - clean up naming, order factors, etc</span></span>
<span><span class="va">correlation_for_plot</span> <span class="op">&lt;-</span> <span class="va">variant_module.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Variant <span class="op">=</span><span class="fu">factor</span><span class="op">(</span><span class="va">Variant</span>,levels<span class="op">=</span><span class="va">ordered.variant</span><span class="op">)</span>,</span>
<span>    Variant <span class="op">=</span> <span class="fu">fct_rev</span><span class="op">(</span><span class="va">Variant</span><span class="op">)</span>,</span>
<span>    module <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">module</span>,levels<span class="op">=</span><span class="va">mod</span><span class="op">)</span>,</span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot.12M</span> <span class="op">=</span> <span class="va">correlation_for_plot</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Background <span class="op">=</span> <span class="fu">ifelse</span><span class="op">(</span><span class="va">Variant</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Sex (Male)"</span><span class="op">)</span>,<span class="st">"Female"</span>,</span>
<span>  <span class="fu">ifelse</span><span class="op">(</span><span class="va">Variant</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"5xFAD"</span>, <span class="st">"LOAD1"</span><span class="op">)</span>, <span class="st">"B6"</span>, <span class="st">"LOAD1"</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plot.12M</span><span class="op">$</span><span class="va">Background</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">plot.12M</span><span class="op">$</span><span class="va">Background</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Female"</span>, <span class="st">"B6"</span>, <span class="st">"LOAD1"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">range</span><span class="op">(</span><span class="va">plot.12M</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level5">
<h5 id="visualizing-the-correlation-plot">Visualizing the Correlation plot<a class="anchor" aria-label="anchor" href="#visualizing-the-correlation-plot"></a>
</h5>
<p>Now, we will use above matrix and visualize the correlation results
using ggplot2 package.</p>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">plot.12M</span></span>
<span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_tile</span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">Variant</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span><span class="op">)</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">Variant</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span>, <span class="va">.data</span><span class="op">$</span><span class="va">significant</span><span class="op">)</span>,<span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">module</span>,y<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">Variant</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span>,color<span class="op">=</span><span class="st">"black"</span>,shape<span class="op">=</span><span class="fl">0</span>,size<span class="op">=</span><span class="fl">9</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_size</span><span class="op">(</span>guide <span class="op">=</span> <span class="st">"none"</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.6</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_color_gradient2</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">0.6</span>, <span class="fl">0.6</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">0.6</span>, <span class="fl">0</span>, <span class="fl">0.6</span><span class="op">)</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, name <span class="op">=</span> <span class="st">"Correlation"</span>, guide <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">ggtitle</span><span class="op">(</span>label<span class="op">=</span> <span class="st">"12 months"</span>,subtitle<span class="op">=</span> <span class="st">"Perturbation  | Control"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">Background</span><span class="op">)</span>,cols <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">cluster_label</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free"</span>,switch<span class="op">=</span><span class="st">"y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">theme</span><span class="op">(</span></span>
<span>      strip.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span>,</span>
<span>      strip.text.y.left <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>      strip.background.y <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_rect</span><span class="op">(</span>fill<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span>,</span>
<span>      axis.ticks <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0</span>,size<span class="op">=</span><span class="fl">16</span><span class="op">)</span>,</span>
<span>      axis.text.y <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">14</span>,face<span class="op">=</span><span class="st">"italic"</span><span class="op">)</span>,</span>
<span>      plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">56</span>, hjust <span class="op">=</span> <span class="fl">0.066</span>,size<span class="op">=</span><span class="fl">14</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      plot.subtitle <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">65</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      panel.background <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>      panel.grid <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      legend.position <span class="op">=</span> <span class="st">"right"</span></span>
<span>    <span class="op">)</span></span></code></pre>
</div>
<p>In above plot, top row represent 30 AMP-AD modules grouped into 5
consensus clusters describing the major functional groups of AD-related
alterations and left column represent mouse models. Positive
correlations are shown in blue and negative correlations in red. Color
intensity and size of the circles are proportional to the correlation
coefficient. Black square around dots represent significant correlation
at p-value=0.05 and non-significant correlations are left blank.</p>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p><strong>What can you conclude from the plot?</strong> <strong>Which
variants capture effect of neurodegeneration </strong></p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="correlation-between-mouse-models-and-milinds-ad-subtypes">Correlation between mouse models and Milind’s AD Subtypes<a class="anchor" aria-label="anchor" href="#correlation-between-mouse-models-and-milinds-ad-subtypes"></a>
</h2>
<hr class="half-width">
<div class="section level3">
<h3 id="milinds-subtype-data">Milind’s subtype data<a class="anchor" aria-label="anchor" href="#milinds-subtype-data"></a>
</h3>
<p>Briefly, Synapse repository <a href="https://www.synapse.org/#!Synapse:syn23660885" class="external-link">syn23660885</a>
contains code, data, and analyses for Milind’s LOAD subtypes. Subtype
assignment of each patient for each cohort are here on synpase: <a href="https://www.synapse.org/Synapse:syn23660969" class="external-link uri">https://www.synapse.org/Synapse:syn23660969</a>. Using
patient’s ID from these files and normalized expression from AMP-AD
cohorts (<a href="https://www.synapse.org/Synapse:syn30821562" class="external-link uri">https://www.synapse.org/Synapse:syn30821562</a>), you can
calculate change in gene expression for subtype patients vs controls.
These matrix can be used to perform correlation analysis with mouse
models.</p>
<p>Here, we have provided final processed data that is stored in
project’s data folder.</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="fu">paste0</span><span class="op">(</span><span class="va">datadir</span>,<span class="st">"Nikhil_SubtypeData.RData"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>This data object contains 3 files: <code>ampad_subtype_fc</code>,
<code>module_cohort</code>, <code>subtypes</code></p>
<p>Let’s briefly check the data objects:</p>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_subtype_fc</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">module_cohort</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">subtypes</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ns_effects</span> <span class="op">&lt;-</span> <span class="va">effects</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu">rownames_to_column</span><span class="op">(</span><span class="va">.</span>, <span class="st">"Gene"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="st">"(Intercept)"</span>, <span class="op">-</span><span class="st">"BD"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols<span class="op">=</span><span class="op">-</span><span class="va">Gene</span>,</span>
<span>               names_to<span class="op">=</span><span class="st">"Variant"</span>,</span>
<span>               values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_vs_subtype_fc</span> <span class="op">&lt;-</span> <span class="va">ns_effects</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">ampad_subtype_fc</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Gene"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">subtype</span>,<span class="va">Variant</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">Gene</span>, <span class="va">value</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor_test <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">"value"</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">"ampad_fc"</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    estimate <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"estimate"</span><span class="op">)</span>,</span>
<span>    p_value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"p.value"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">cor_test</span><span class="op">)</span></span></code></pre>
</div>
<p>Process data for plotting —- Flag for significant results, add
cluster information to modules</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="va">model_vs_subtype_fc</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>significant <span class="op">=</span> <span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,age_group<span class="op">=</span><span class="st">"12 Months"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">module_cohort</span>, by <span class="op">=</span> <span class="st">"subtype"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">cluster</span>, <span class="va">cluster_label</span>, <span class="va">subtype</span>,<span class="va">Variant</span>,<span class="va">age_group</span>, correlation <span class="op">=</span> <span class="va">estimate</span>, <span class="va">p_value</span>, <span class="va">significant</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">#ordered.variant &lt;- rev(ordered.variant)</span></span></code></pre>
</div>
<p>Create a version of the data for plotting - clean up naming, order
factors, etc</p>
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ordered.variant</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Sex (Male)"</span>, <span class="st">"5xFAD"</span>, <span class="st">"LOAD1"</span>,  </span>
<span>                     <span class="st">"Abca7*A1527G"</span>, <span class="st">"Ceacam1 KO"</span>, <span class="st">"Mthfr*677C&gt;T"</span>,<span class="st">"Shc2*V433M"</span>,<span class="st">"Slc6a17*P61P"</span>,<span class="st">"Erc2*N542S"</span>,<span class="st">"Clasp2*L163P"</span>,<span class="st">"Sorl1*A528T"</span> ,</span>
<span>                     <span class="st">"Meox2 KO (HET)"</span>,<span class="st">"Snx1*D465N"</span>, <span class="st">"Plcg2*M28L"</span>,<span class="st">"Mtmr4*V297G"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">subtype_nanostring_for_plot.12M</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Variant <span class="op">=</span><span class="fu">factor</span><span class="op">(</span><span class="va">Variant</span>,levels<span class="op">=</span><span class="va">ordered.variant</span><span class="op">)</span>,</span>
<span>    Variant <span class="op">=</span><span class="fu">fct_rev</span><span class="op">(</span><span class="va">Variant</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">range</span><span class="op">(</span><span class="va">subtype_nanostring_for_plot.12M</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">dd.subtype_PS</span> <span class="op">=</span> <span class="va">subtype_nanostring_for_plot.12M</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>Background <span class="op">=</span> <span class="fu">ifelse</span><span class="op">(</span><span class="va">Variant</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Sex (Male)"</span><span class="op">)</span>,<span class="st">"Female"</span>,<span class="fu">ifelse</span><span class="op">(</span><span class="va">Variant</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"5xFAD"</span>,<span class="st">"LOAD1"</span><span class="op">)</span>,<span class="st">"B6"</span>,<span class="st">"LOAD1"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dd.subtype_PS</span><span class="op">$</span><span class="va">Background</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">dd.subtype_PS</span><span class="op">$</span><span class="va">Background</span>,levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Female"</span>,<span class="st">"B6"</span>,<span class="st">"LOAD1"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Function for creating correlation plot</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">subtype_variant_corrplot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span>,<span class="va">ran</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_tile</span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">subtype</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">Variant</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span><span class="op">)</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">subtype</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">Variant</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span>, <span class="va">.data</span><span class="op">$</span><span class="va">significant</span><span class="op">)</span>,<span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">subtype</span>,y<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">Variant</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span>,color<span class="op">=</span><span class="st">"black"</span>,shape<span class="op">=</span><span class="fl">0</span>,size<span class="op">=</span><span class="fl">9</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_size</span><span class="op">(</span>guide <span class="op">=</span> <span class="st">"none"</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="va">ran</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_color_gradient2</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="va">ran</span>, <span class="va">ran</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="va">ran</span>, <span class="fl">0</span>, <span class="va">ran</span><span class="op">)</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, name <span class="op">=</span> <span class="st">"Correlation"</span>, guide <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">ggtitle</span><span class="op">(</span>label<span class="op">=</span> <span class="st">"12 months"</span>,subtitle<span class="op">=</span> <span class="st">"Perturbation  | Control"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">Background</span><span class="op">)</span>,cols <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">cluster_label</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free"</span>,switch<span class="op">=</span><span class="st">"y"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>strip.placement.x <span class="op">=</span> <span class="st">"outside"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">theme</span><span class="op">(</span></span>
<span>      strip.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span>,</span>
<span>      strip.text.y.left <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>      strip.background.y <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_rect</span><span class="op">(</span>fill<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span>,</span>
<span>      axis.ticks <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, hjust <span class="op">=</span> <span class="fl">0.5</span>,size<span class="op">=</span><span class="fl">11</span><span class="op">)</span>,</span>
<span>      axis.text.y <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span>,size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,</span>
<span>      plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">15</span>, hjust <span class="op">=</span> <span class="fl">0.05</span>,size<span class="op">=</span><span class="fl">13</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      plot.subtitle <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">16</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">11</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      panel.background <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>      panel.grid <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      legend.position <span class="op">=</span> <span class="st">"bottom"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<p>Let’s plot the results</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">subtype_variant_corrplot</span><span class="op">(</span><span class="va">dd.subtype_PS</span>,<span class="fl">0.4</span><span class="op">)</span></span></code></pre>
</div>
<p>We can see that mouse model of AD may match to a particular subset of
human AD subtypes but not all subtypes simultaneously, and that risk for
these subtypes may be influenced by distinct AD genetic factors.</p>
<p>We can see that mouse model of AD may match to a particular subset of
human AD subtypes but not all subtypes simultaneously, and that risk for
these subtypes may be influenced by distinct AD genetic factors.</p>
</div>
<div class="section level3">
<h3 id="neffs-subtype-data">Neff’s subtype data<a class="anchor" aria-label="anchor" href="#neffs-subtype-data"></a>
</h3>
<p>Synapse repository <a href="https://www.synapse.org/Synapse:syn25944417" class="external-link">syn25944417</a>
contains code, data, and analyses for Neff’s LOAD subtypes for MSBB
cohort. Subtype assignments from the Neff Synapse project here: <a href="https://www.synapse.org/Synapse:syn25944448" class="external-link uri">https://www.synapse.org/Synapse:syn25944448</a>. You can use
the MSBB biospecimen metadata file from <a href="https://www.synapse.org/Synapse:syn21893059" class="external-link">syn21893059</a> to
curate the subtypes with RNA-Seq IDs and individual IDs.</p>
<p>Next, you can (if have permission) extract normalized expression of
all human genes for these individuals (<a href="https://www.synapse.org/Synapse:syn16795937" class="external-link uri">https://www.synapse.org/Synapse:syn16795937</a>) and perform
differential analysis to calculate log fold change expression.</p>
<p>Here, we have provided final processed data that is stored in
project’s data folder.</p>
<div class="codewrapper sourceCode" id="cb53">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="fu">paste0</span><span class="op">(</span><span class="va">datadir</span>,<span class="st">"Neff_Subtype.RData"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">neff_subtype_fc</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p><em>You all should create correlation plot for mouse data compared
with Neff’s subtype data. Identify mouse variants significantly positive
correlated with subtype C1 and subtype A.</em></p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="milinds-vs-neffs-subtype">Milind’s vs Neff’s Subtype<a class="anchor" aria-label="anchor" href="#milinds-vs-neffs-subtype"></a>
</h4>
<p>Finally, Let’ see how does AD subtypes corresponds to each other.</p>
<div id="challenge-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-3" class="callout-inner">
<h3 class="callout-title">Challenge 3</h3>
<div class="callout-content">
<p>Using data from Milind’s and Neff’s AD subtypes perform correlation
analysis to see how AD subtypes corresponds to each other.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">

</div>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="biodomain-correlations">Biodomain Correlations<a class="anchor" aria-label="anchor" href="#biodomain-correlations"></a>
</h2>
<hr class="half-width">
<p>We can also use the gene sub-sets defined by the AD Biodomains and
Subdomains to frame these correlations. Let’s see how that works,
starting with how the LOAD1 mouse strain effects correlate with the
logFC values from AMP-AD cohorts.</p>
<p>First, we’ll download the Biodomain definitions. However, this time
there’s something different. What’s different between this table and the
one we used in the earlier session?</p>
<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># biodomain definitions</span></span>
<span><span class="va">biodom</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">synGet</span><span class="op">(</span><span class="st">'syn26592124'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biodomain labels and colors</span></span>
<span><span class="va">dom.lab</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="fu">synGet</span><span class="op">(</span><span class="st">'syn26856828'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-4" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-4" class="callout-inner">
<h3 class="callout-title">Challenge 4</h3>
<div class="callout-content">
<p>What’s different about this <code>biodom</code> table?</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">GOterm_Name</span> <span class="op">==</span> <span class="st">'amyloid-beta formation'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unlist</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>This table contains mouse gene IDs!</p>
</div>
</div>
</div>
</div>
<p>To perform the correlation on subsets of genes defined by the
domains, we need to specify the genes in each biodomain and subdomain.
For each subdomain we’ll prefix the subdomain name with the biodomain
abbreviation. This is because some subdomains are shared between two
different domains. For example, both the <code>Autophagy</code> and
<code>Immune Response</code> domains have a subdomain called
“phagocytosis”. In each case there are slightly different sets of terms
contained within the subdomain. We’ll keep track by adding a domain
prefix, e.g. “IR_phagocytosis” and “Au_phagocytosis”.</p>
<p>Let’s collect the gene lists:</p>
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="co"># first set up the list of genes associated with each biodomain</span></span>
<span><span class="va">bd.genes</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span> set <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span> genes <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">set</span>, <span class="op">~</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unlist</span><span class="op">)</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># then the list of genes associated with each subdomain </span></span>
<span><span class="va">sd.genes</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span> <span class="va">biodom</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span><span class="op">=</span><span class="st">'domain'</span><span class="op">)</span> <span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span> Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">abbr</span>, <span class="va">Subdomain</span>, <span class="va">subdomain_idx</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    set <span class="op">=</span> <span class="fu">paste0</span><span class="op">(</span><span class="va">abbr</span>,<span class="st">'_'</span>,<span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    bd <span class="op">=</span> <span class="va">Biodomain</span>, sdi <span class="op">=</span> <span class="va">subdomain_idx</span>,</span>
<span>    genes <span class="op">=</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">bd</span>, <span class="va">subdomain_idx</span> <span class="op">==</span> <span class="va">sdi</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unlist</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co"># now combine the two lists</span></span>
<span><span class="va">bd.gene.lists</span> <span class="op">=</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">bd.genes</span>, <span class="va">sd.genes</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">set</span>, <span class="va">genes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">!=</span> <span class="st">'NA_none'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">bd.genes</span>, <span class="va">sd.genes</span><span class="op">)</span></span></code></pre>
</div>
<p>We’ll also need the cohort data fold change data, and to map the
orthologous mouse genes.</p>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ortholog map between human and mouse provided by Wan et al</span></span>
<span><span class="va">mouse.human.ortho</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn17010253"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># AMP-AD cohort differential expression table</span></span>
<span><span class="va">ampad_modules_raw</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn14237651"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">mouse.human.ortho</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>hgnc_symbol <span class="op">=</span> <span class="va">human_symbol</span>, <span class="va">mouse_symbol</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="amp-ad-cohort-correlation">AMP-AD Cohort Correlation<a class="anchor" aria-label="anchor" href="#amp-ad-cohort-correlation"></a>
</h4>
<p>We’ll start by examining how the LOAD1 strain effects (β) correlate
with the human AMP-AD cohort logFC values, like we did above with the
co-expression modules. This time we’ll use the biodomain and subdomain
gene lists to sub-set the genes for correlation analysis. We’ll compute
correlations to one tissu for each cohort.</p>
<p>First we can generate a nested table with the AMP-AD data, but
filtering for the appropriate contrasts and tissues (DLPFC = ROSMAP, TCX
= MAYO, and PHG = MSSM).</p>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs.lfc</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>, </span>
<span>    <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span>,</span>
<span>    <span class="va">Tissue</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">'PHG'</span>,<span class="st">'TCX'</span>,<span class="st">'DLPFC'</span><span class="op">)</span>,</span>
<span>    <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">mouse_symbol</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">mouse_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nest</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">hgnc_symbol</span>, <span class="va">symbol</span>, <span class="va">logFC</span>, <span class="va">adj.P.Val</span><span class="op">)</span>, </span>
<span>    .by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">Study</span>, <span class="va">Tissue</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">hs.lfc</span><span class="op">)</span></span></code></pre>
</div>
<p>And look at one of those nested tables:</p>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">hs.lfc</span><span class="op">$</span><span class="va">data</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div id="a-note-about-gene-homology" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="a-note-about-gene-homology" class="callout-inner">
<h3 class="callout-title">A note about gene homology</h3>
<div class="callout-content">
<p>You’ll notice in the table above that there are two
<strong>identical</strong> logFC values for the human gene ANGPT1. This
is because it aligns with two separate mouse genes: Angpt1 and Angpt2.
Each of these mouse genes are paralogs of each other and they form an
orthogroup with the human ANGPT1 gene. There are even more mouse genes
for SLC7A2! There are other ways to map orthologous genes between
sepecies (such as the <code>gprofiler2::gorth</code> function), and I
encourage you to try them out. This is a non-trivial task and care
should be given to how to structure these relationships to best address
the questions of your research.</p>
</div>
</div>
</div>
<p>We can prepare the LOAD1 mouse effects in an analogous way:</p>
<div class="codewrapper sourceCode" id="cb61">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">load1.effects</span> <span class="op">&lt;-</span> <span class="va">ns_effects</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>mm.beta <span class="op">=</span> <span class="va">value</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">mm.beta</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Now join the two datasets based on the mouse gene symbols. If we
perform an <code>inner_join</code> we’ll get only the genes that are
present in both datasets</p>
<div class="codewrapper sourceCode" id="cb62">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">load1.vs.amp</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span>load1 <span class="op">=</span> <span class="va">load1.effects</span><span class="op">$</span><span class="va">Variant</span>, amp <span class="op">=</span> <span class="va">hs.lfc</span><span class="op">$</span><span class="va">Study</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">load1</span>, <span class="va">amp</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>    <span class="va">mm</span> <span class="op">=</span> <span class="va">load1.effects</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Variant</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">hs</span> <span class="op">=</span> <span class="va">hs.lfc</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Study</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="fu">inner_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="st">'symbol'</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">load1.vs.amp</span><span class="op">)</span></span></code></pre>
</div>
<p>One last piece of information – let’s add in the biodomain/subdomain
gene lists and subset each mouse X human data table for the genes in
each subset:</p>
<div class="codewrapper sourceCode" id="cb63">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">load1.vs.amp</span>, set <span class="op">=</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span></span>
<span>      <span class="va">data</span>, <span class="va">set</span>, </span>
<span>      <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span> <span class="va">symbol</span> <span class="op">%in%</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">genes</span><span class="op">[[</span><span class="fu">which</span><span class="op">(</span><span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">)</span> <span class="op">)</span>,</span>
<span>    n.genes <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="fu">nrow</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> </span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">cor.df</span><span class="op">)</span></span></code></pre>
</div>
<p>Finally, we can perform a correlation analysis for each comparison.
We’ll filter to remove any comparison that contains &lt; 3 genes. Also,
since we’re running so many comparisons (~150 for each mouse-human
pair), we should correct for multiple hypothesis testing. The way this
is set up is extreme (i.e. against <strong>all</strong> comparisons, not
just those within a mouse-human pairing), but it will help us to narrow
down to the most significant results.</p>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">n.genes</span> <span class="op">&gt;</span> <span class="fl">3</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'logFC'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'mm.beta'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.adj <span class="op">=</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">p.value</span>, method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span>,</span>
<span>    significant <span class="op">=</span> <span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Let’s plot the results of the correlation. We’ll start at the
biodomain level by filtering the gene set for those that don’t include
an underscore (“_“) in their name.</p>
<div class="codewrapper sourceCode" id="cb65">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">F</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set</span>, <span class="va">load1</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">amp</span><span class="op">)</span>, switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, </span>
<span>             stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>We can also take a look at the subdomains. Let’s focus on just one
cohort here - MSSM - to simplify the plot.</p>
<div class="codewrapper sourceCode" id="cb66">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the list of subdomains to plot</span></span>
<span><span class="va">l</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="va">amp</span> <span class="op">==</span> <span class="st">'ROSMAP'</span>,</span>
<span>    <span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">T</span>, </span>
<span>     <span class="op">(</span><span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">5e-2</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">set</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># prepare the cor.df for plotting</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">amp</span> <span class="op">==</span> <span class="st">'ROSMAP'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>max.n <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">n.genes</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">%in%</span> <span class="va">l</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    set1 <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>      <span class="fu">str_trunc</span><span class="op">(</span><span class="va">.</span>, <span class="fl">40</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>      <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">max.n</span>,<span class="st">'] '</span>,<span class="va">.</span><span class="op">)</span>,</span>
<span>    abbr <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># plot</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set1</span>, <span class="va">load1</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">abbr</span><span class="op">)</span>, <span class="co">#rows = vars(amp), </span></span>
<span>             scales <span class="op">=</span> <span class="st">'free'</span>, space <span class="op">=</span> <span class="st">'free'</span>, switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'left'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x.top = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x.top <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"left"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div id="challenge-5" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-5" class="callout-inner">
<h3 class="callout-title">Challenge 5</h3>
<div class="callout-content">
<p>How do these correlation results compare with those above when
considering the co-expression modules? What is similar? What is
different?</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="challenge-6" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-6" class="callout-inner">
<h3 class="callout-title">Challenge 6</h3>
<div class="callout-content">
<p>Are there different subdomains that have significant correlations for
the other AMP-AD cohorts?</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">

</div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="amp-ad-subtype-correlation">AMP-AD Subtype Correlation<a class="anchor" aria-label="anchor" href="#amp-ad-subtype-correlation"></a>
</h4>
<p>Now we’ve seen how the effects from each variant from the LOAD1 study
(mouse β) compare to the expression in the whole cohorts, let’s explore
how the mouse effects correlate with sub-type vs control LogFC. We’ll
start with Millind’s MAYO subtypes. So let’s join together the mouse
effects and human LogFCs into one table for each comparison.</p>
<div class="codewrapper sourceCode" id="cb67">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># some variable renaming</span></span>
<span><span class="va">stype.effects</span> <span class="op">&lt;-</span> <span class="va">ampad_subtype_fc</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">neff.effects</span> <span class="op">&lt;-</span> <span class="va">neff_subtype_fc</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Just the ROSMAPs</span></span>
<span><span class="va">stype.effects.RM</span> <span class="op">&lt;-</span> <span class="va">stype.effects</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'MSBB'</span>, <span class="va">subtype</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># stype.effects.RM &lt;- neff.effects</span></span>
<span></span>
<span><span class="co"># Join the tables together for each comparison</span></span>
<span><span class="va">load1.vs.stype</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span>load1 <span class="op">=</span> <span class="va">load1.effects</span><span class="op">$</span><span class="va">Variant</span>, stype <span class="op">=</span> <span class="va">stype.effects.RM</span><span class="op">$</span><span class="va">subtype</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">load1</span>, <span class="va">stype</span>,</span>
<span>                <span class="op">~</span> <span class="op">{</span> </span>
<span>                  <span class="va">mm</span> <span class="op">=</span> <span class="va">load1.effects</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Variant</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> </span>
<span>                  <span class="va">hs</span> <span class="op">=</span> <span class="va">stype.effects.RM</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">subtype</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> </span>
<span>                  <span class="fu">inner_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="st">'symbol'</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span> </span>
<span>                <span class="op">}</span></span>
<span>            <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Now let’s create a subset for each of the gene sets defined by the
biodomains and subdomains. Then we can compute a correlation for
each.</p>
<div class="codewrapper sourceCode" id="cb68">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">load1.vs.stype</span>, set <span class="op">=</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span></span>
<span>      <span class="va">data</span>, <span class="va">set</span>, </span>
<span>      <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span> <span class="va">symbol</span> <span class="op">%in%</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">genes</span><span class="op">[[</span><span class="fu">which</span><span class="op">(</span><span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">)</span> <span class="op">)</span>,</span>
<span>    n.genes <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="fu">nrow</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">n.genes</span> <span class="op">&gt;</span> <span class="fl">3</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'ampad_fc'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'mm.beta'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.adj <span class="op">=</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">p.value</span>, method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span>,</span>
<span>    significant <span class="op">=</span> <span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>And plot the correlation result for the biodomain gene sets</p>
<div class="codewrapper sourceCode" id="cb69">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">F</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set</span>, <span class="va">stype</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">load1</span><span class="op">)</span>, switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, </span>
<span>             stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>And for the subdomains, we’ll focus on only those that are
significant and have at least 10 genes</p>
<div class="codewrapper sourceCode" id="cb70">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the list of subdomains to plot</span></span>
<span><span class="va">l</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">T</span>, </span>
<span>     <span class="op">(</span><span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">5e-2</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">set</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># prepare the cor.df for plotting</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>max.n <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">n.genes</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">%in%</span> <span class="va">l</span>, <span class="va">max.n</span> <span class="op">&gt;</span> <span class="fl">10</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>any.sig <span class="op">=</span> <span class="fu">any</span><span class="op">(</span><span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">5e-2</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">load1</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span> <span class="va">any.sig</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    set1 <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>      <span class="fu">str_trunc</span><span class="op">(</span><span class="va">.</span>, <span class="fl">40</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>      <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">max.n</span>,<span class="st">'] '</span>,<span class="va">.</span><span class="op">)</span>,</span>
<span>    abbr <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="op">)</span></span>
<span>         </span>
<span><span class="co"># plot!</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set1</span>, <span class="va">stype</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">load1</span><span class="op">)</span>, cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">abbr</span><span class="op">)</span>, </span>
<span>             scales <span class="op">=</span> <span class="st">'free'</span>, space <span class="op">=</span> <span class="st">'free'</span>, switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"left"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
</div>
</section><div class="section level1">
<h1 id="wan-et-al-mouse-model-compendium">Wan et al mouse model compendium<a class="anchor" aria-label="anchor" href="#wan-et-al-mouse-model-compendium"></a>
</h1>
<p>Our ability to use the domains to assess the LOAD1 primary screen
study is constrained because the mice were only assessed for the
expression of the 770 genes that contained probes in the Nanostring
probe set. This is well suited for the module-level correlation
analysis, because the Nanostring genes were selected based on coverage
of the co-expression modules. However we can also perform these analyses
with RNA-seq data.</p>
<p>We’ve talked about the <a href="https://doi.org/10.1016/j.celrep.2020.107908" class="external-link">Wan, et al.</a>
study already this week in the context of the consensus co-expression
module generation for AMP-AD cohorts. <strong>Another</strong> useful
resource that this study generated is a compendium of mouse model
RNA-seq data from various sources. We can use these data as another
dataset to use to explore correlations between mouse and human. The data
are part of the <code>AD_CrossSpecies</code> study which can be accessed
on Synapse at <a href="https://www.synapse.org/Synapse:syn16779040" class="external-link">syn16779040</a>.</p>
<p>First, let’s list the files in the
<code>Analysis &gt; Differential Expression</code> folder for the study
(<a href="https://www.synapse.org/Synapse:syn17009951" class="external-link">syn17009951</a>):</p>
<div class="codewrapper sourceCode" id="cb71">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synLogin</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGetChildren</span><span class="op">(</span><span class="st">'syn17009951'</span><span class="op">)</span><span class="op">$</span><span class="fu">asList</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">tibble</span><span class="op">(</span>entries <span class="op">=</span> <span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unnest_wider</span><span class="op">(</span><span class="va">entries</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span> study <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">name</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">)</span> </span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span></code></pre>
</div>
<p>This shows the files in the synapse directory along with the filename
and the synapse ID for each.</p>
<p>How many are there?</p>
<div class="codewrapper sourceCode" id="cb72">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span></code></pre>
</div>
<p>There are 234 results files in this directory! Let’s not work with
all of them, but instead focus on a subset. To know which ones to use
we’ll need more information about these data. So let’s read the metadata
file for DEG analyses:</p>
<div class="codewrapper sourceCode" id="cb73">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span> <span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">'syn17023327'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">f</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Filename'</span><span class="op">=</span><span class="st">'study'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">m</span><span class="op">)</span></span></code></pre>
</div>
<p>This file gives information about each of the DEG results listed in
the directory above. This information includes the following critical
deails:</p>
<ul>
<li>
<code>Study_name</code>: GSE or Syn ID denoting the original RNAseq
or microarray data source.<br>
</li>
<li>
<code>Annotation</code>: The unique, manually-curated annotation for
the DEG set, following the convention:
<code>category_experimental condition_sex_age_brain region_cell type_ transgene</code>.
Let’s break down each of these pieces of information:
<ul>
<li>
<code>category</code> denotes the human disease being modeled or
“other” (e.g. AD, HD, SCA, ALS)</li>
<li>
<code>experimental condition</code> denotes the specific mouse
genotype or treatment condition</li>
<li>
<code>sex</code>, male or female mice</li>
<li>
<code>age</code>, in months of the animals</li>
<li>
<code>brain region</code> sampled (e.g. hippocampus, spinal
cord)</li>
<li>
<code>cell type</code> sampled (e.g. neuron, microglia)</li>
<li>
<code>transgene</code> for AD models only “APP” vs. “tau” is
noted</li>
<li>If unknown or not applicable for a given analysis, the above fields
are replaced with “na”</li>
</ul>
</li>
</ul>
<p>Great! These annotation fields will help us to filter for the data
that we want.</p>
<p>Let’s do a little processing and join these files together:</p>
<div class="codewrapper sourceCode" id="cb74">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Join the list of files from Synapse with the metadata</span></span>
<span><span class="va">wan_mice</span> <span class="op">&lt;-</span> <span class="va">f</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">id</span>, <span class="va">study</span><span class="op">)</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span></span>
<span>    <span class="va">.</span>, </span>
<span>    <span class="va">m</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Filename</span>, <span class="va">Study_name</span>, <span class="va">Annotation</span>, <span class="va">Experimental_condition</span>, <span class="va">Control_condition</span><span class="op">)</span>, </span>
<span>    by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'study'</span><span class="op">=</span><span class="st">'Filename'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># split and re-name the annotation fields</span></span>
<span><span class="va">anno.cols</span> <span class="op">&lt;-</span> <span class="va">wan_mice</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">Annotation</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>anno <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">Annotation</span>,<span class="st">'_'</span>, <span class="fl">7</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>           <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>           <span class="fu">rename_with</span><span class="op">(</span><span class="op">~</span><span class="fu">c</span><span class="op">(</span><span class="st">'category'</span>,</span>
<span>                          <span class="st">'exptCondition'</span>,</span>
<span>                          <span class="st">'sex'</span>,</span>
<span>                          <span class="st">'age'</span>,</span>
<span>                          <span class="st">'brainRegion'</span>,</span>
<span>                          <span class="st">'cellType'</span>,</span>
<span>                          <span class="st">'transgene'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">unnest_wider</span><span class="op">(</span><span class="va">anno</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="op">-</span><span class="va">Annotation</span>, <span class="op">~</span> <span class="fu">str_trim</span><span class="op">(</span><span class="va">.x</span>, <span class="st">'both'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the annotation columns to the original table</span></span>
<span><span class="va">wan_mice</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">wan_mice</span>, <span class="va">anno.cols</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">relocate</span><span class="op">(</span><span class="va">category</span><span class="op">:</span><span class="va">transgene</span>, .after <span class="op">=</span> <span class="va">Annotation</span><span class="op">)</span></span></code></pre>
</div>
<p>So how many of these studies are categorized as AD models?</p>
<div class="codewrapper sourceCode" id="cb75">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wan_mice</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">category</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarise</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">study</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Only 43 are considered AD models. Let’s take a closer look at the AD
mice category:</p>
<div class="codewrapper sourceCode" id="cb76">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wan_mice</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">category</span> <span class="op">==</span> <span class="st">'AD'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">transgene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarise</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">study</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Over half of the AD mouse model datasets are related to mice carrying
an APP transgene, there are 10 related to mice carrying a Tau transgene,
and 7 in the “other” category. What’s in this other category?</p>
<div class="codewrapper sourceCode" id="cb77">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wan_mice</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">category</span> <span class="op">==</span> <span class="st">'AD'</span>, <span class="va">transgene</span> <span class="op">==</span> <span class="st">'other'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">exptCondition</span>, <span class="va">sex</span>, <span class="va">age</span>, <span class="va">brainRegion</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s select a subset to work with. You can choose your own here, but
I’m going to pick the neurodegenerative disease models (AD, PD, HD,
FTD-ALS, SCA, and CJD). I’m going to filter to only include studies that
look at brain expression (i.e. not sampling cell types), and I’m going
to exclude a couple of other categories (e.g. “spinal cord” and “motor
neurons”). I encourage you to choose your own adventure here.</p>
<div class="codewrapper sourceCode" id="cb78">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wan.data</span> <span class="op">&lt;-</span> <span class="va">wan_mice</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="va">category</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">'AD'</span>, <span class="st">'HD'</span>, <span class="st">'FTD-ALS'</span>, <span class="st">'PD'</span>, <span class="st">'SCA'</span>, <span class="st">'CJD'</span>, <span class="st">'Rett'</span><span class="op">)</span>,</span>
<span>    <span class="va">cellType</span> <span class="op">==</span> <span class="st">'na'</span>,</span>
<span>    <span class="va">brainRegion</span> <span class="op">!=</span> <span class="st">'spinal cord'</span>,</span>
<span>    <span class="va">brainRegion</span> <span class="op">!=</span> <span class="st">'motor neurons'</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>So this should be far fewer than 234 studies</p>
<div class="codewrapper sourceCode" id="cb79">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">wan.data</span><span class="op">)</span></span></code></pre>
</div>
<p>Ok, 87 seems like a more manageable number. Now let’s read in the
data from Synapse. Each DEG results table has similarly labelled columns
– let’s grab the gene IDs, log2FoldChange values, and padj significance
values for each. This step can take a little while.</p>
<div class="codewrapper sourceCode" id="cb80">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">st</span> <span class="op">=</span> <span class="fu">Sys.time</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">wan.data</span><span class="op">$</span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span></span>
<span>  <span class="fl">1</span><span class="op">:</span><span class="fu">nrow</span><span class="op">(</span><span class="va">wan.data</span><span class="op">)</span>,</span>
<span>  <span class="op">~</span> <span class="fu">read_tsv</span><span class="op">(</span> <span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="va">wan.data</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="va">.x</span><span class="op">]</span><span class="op">)</span><span class="op">$</span><span class="va">path</span>, show_col_types <span class="op">=</span> <span class="cn">F</span> <span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Gene</span>, <span class="va">IsDEG</span>,<span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span></span>
<span>  , .progress <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> </span>
<span><span class="va">ed</span> <span class="op">=</span> <span class="fu">Sys.time</span><span class="op">(</span><span class="op">)</span><span class="op">-</span><span class="va">st</span></span>
<span><span class="fu">print</span><span class="op">(</span><span class="va">ed</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="amp-ad-module-correlation">AMP-AD Module Correlation<a class="anchor" aria-label="anchor" href="#amp-ad-module-correlation"></a>
</h4>
<p>Like with the LOAD1 study we can start with the correlation to the
co-expression module data. <a href="https://doi.org/10.1016/j.celrep.2020.107908" class="external-link">Wan, et al.</a>
included similar analyses in their results.</p>
<div class="codewrapper sourceCode" id="cb81">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># prepare the module table </span></span>
<span><span class="va">mod.lfc</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_fc</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># prepare the mouse data</span></span>
<span><span class="va">wan.lfc</span> <span class="op">&lt;-</span> <span class="va">wan.data</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">IsDEG</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join the tables together for each comparison</span></span>
<span><span class="va">wan.vs.mod</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">wan.lfc</span>, mod <span class="op">=</span> <span class="va">mod.lfc</span><span class="op">$</span><span class="va">module</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">data</span>, <span class="va">mod</span>,</span>
<span>                <span class="op">~</span> <span class="op">{</span> </span>
<span>                  <span class="va">mm</span> <span class="op">=</span> <span class="va">.x</span></span>
<span>                  <span class="va">hs</span> <span class="op">=</span> <span class="va">mod.lfc</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">module</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> </span>
<span>                  <span class="fu">inner_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="st">'symbol'</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span> </span>
<span>                <span class="op">}</span></span>
<span>            <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Compute the correlation</p>
<div class="codewrapper sourceCode" id="cb82">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">wan.vs.mod</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'ampad_fc'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'log2FoldChange'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.adj <span class="op">=</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">p.value</span>, method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span>,</span>
<span>    significant <span class="op">=</span> <span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">module_clusters</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'mod'</span><span class="op">=</span><span class="st">'module'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Plot the correlation</p>
<div class="codewrapper sourceCode" id="cb83">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model_name <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">category</span>,<span class="st">'] '</span>,<span class="va">study</span>,<span class="st">': '</span>, <span class="va">exptCondition</span>, <span class="st">' ('</span>, <span class="va">transgene</span>,<span class="st">')'</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">mod</span>, <span class="va">model_name</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">cluster_label</span><span class="op">)</span>, switch <span class="op">=</span> <span class="st">'y'</span>, scales<span class="op">=</span><span class="st">'free'</span>,space<span class="op">=</span> <span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, </span>
<span>             stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'left'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div id="challenge-7" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-7" class="callout-inner">
<h3 class="callout-title">Challenge 7</h3>
<div class="callout-content">
<p>This is a lot of information! Take a minute to explore the plot and
discuss with your neighbor what trends you notice. We’ll come back to
discuss as a group.</p>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7"> Show me the solution </h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" aria-labelledby="headingSolution7" data-bs-parent="#accordionSolution7">
<div class="accordion-body">
<p>Some things I notice: (1) The <code>AD</code> models are fairly
similar and many show strong positive correlation to clusters A and B
(2) The <code>HD</code> models show negative correlations to clusters A
and B, but have the strongest correlations with cluster E (3) The
<code>SCA</code> models tend to look a bit like the <code>AD</code>
models</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="biodomain-correlation">Biodomain Correlation<a class="anchor" aria-label="anchor" href="#biodomain-correlation"></a>
</h4>
<p>We can also perform the correlation by biodomain genes. Here we join
up the whole cohort</p>
<div class="codewrapper sourceCode" id="cb84">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs.lfc</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>, </span>
<span>    <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span>,</span>
<span>    <span class="va">Tissue</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">'PHG'</span>,<span class="st">'TCX'</span>,<span class="st">'DLPFC'</span><span class="op">)</span>,</span>
<span>    <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">mouse_symbol</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">mouse_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nest</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">hgnc_symbol</span>, <span class="va">symbol</span>, <span class="va">logFC</span>, <span class="va">adj.P.Val</span><span class="op">)</span>, </span>
<span>    .by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">Study</span>, <span class="va">Tissue</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">wan.lfc</span> <span class="op">&lt;-</span> <span class="va">wan.data</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">IsDEG</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join the tables together for each comparison</span></span>
<span><span class="va">wan.vs.amp</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">wan.lfc</span>, amp <span class="op">=</span> <span class="va">hs.lfc</span><span class="op">$</span><span class="va">Study</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">data</span>, <span class="va">amp</span>,</span>
<span>                <span class="op">~</span> <span class="op">{</span> </span>
<span>                  <span class="va">mm</span> <span class="op">=</span> <span class="va">.x</span></span>
<span>                  <span class="va">hs</span> <span class="op">=</span> <span class="va">hs.lfc</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Study</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> </span>
<span>                  <span class="fu">inner_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="st">'symbol'</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span> </span>
<span>                <span class="op">}</span></span>
<span>            <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb85">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">wan.vs.amp</span>, set <span class="op">=</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span></span>
<span>      <span class="va">data</span>, <span class="va">set</span>, </span>
<span>      <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span> <span class="va">symbol</span> <span class="op">%in%</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">genes</span><span class="op">[[</span><span class="fu">which</span><span class="op">(</span><span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">)</span> <span class="op">)</span>,</span>
<span>    n.genes <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="fu">nrow</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">n.genes</span> <span class="op">&gt;</span> <span class="fl">3</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'logFC'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'log2FoldChange'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.adj <span class="op">=</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">p.value</span>, method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span>,</span>
<span>    significant <span class="op">=</span> <span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb86">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">F</span>, <span class="va">amp</span> <span class="op">==</span> <span class="st">'MSSM'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model_name <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">category</span>,<span class="st">'] '</span>,<span class="va">study</span>,<span class="st">': '</span>, <span class="va">exptCondition</span>, <span class="st">' ('</span>, <span class="va">transgene</span>,<span class="st">')'</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">ord</span> <span class="op">&lt;-</span> <span class="va">tmp</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">==</span> <span class="st">'Lipid Metabolism'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">model_name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">tmp</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model_name <span class="op">=</span> <span class="fu">fct_relevel</span><span class="op">(</span><span class="va">model_name</span>, <span class="va">ord</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set</span>, <span class="va">model_name</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">amp</span><span class="op">)</span>, switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, </span>
<span>             stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="subtypes">subtypes<a class="anchor" aria-label="anchor" href="#subtypes"></a>
</h4>
<div class="codewrapper sourceCode" id="cb87">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># some variable renaming</span></span>
<span><span class="va">stype.effects</span> <span class="op">&lt;-</span> <span class="va">ampad_subtype_fc</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">neff.effects</span> <span class="op">&lt;-</span> <span class="va">neff_subtype_fc</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Just the ROSMAPs</span></span>
<span><span class="co"># stype.effects.RM &lt;- stype.effects %&gt;% filter(grepl('Mayo', subtype))</span></span>
<span><span class="va">stype.effects.RM</span> <span class="op">&lt;-</span> <span class="va">neff.effects</span></span>
<span></span>
<span><span class="va">wan.lfc</span> <span class="op">&lt;-</span> <span class="va">wan.data</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">IsDEG</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join the tables together for each comparison</span></span>
<span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">wan.lfc</span>, stype <span class="op">=</span> <span class="va">stype.effects.RM</span><span class="op">$</span><span class="va">subtype</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">data</span>, <span class="va">stype</span>,</span>
<span>                <span class="op">~</span> <span class="op">{</span> </span>
<span>                  <span class="va">mm</span> <span class="op">=</span> <span class="va">.x</span> </span>
<span>                  <span class="va">hs</span> <span class="op">=</span> <span class="va">stype.effects.RM</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">subtype</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> </span>
<span>                  <span class="fu">inner_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="st">'symbol'</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span></span>
<span>                <span class="op">}</span></span>
<span>            <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'ampad_fc'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'log2FoldChange'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.adj <span class="op">=</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">p.value</span>, method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span>,</span>
<span>    significant <span class="op">=</span> <span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb88">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model_name <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">category</span>,<span class="st">'] '</span>,<span class="va">study</span>,<span class="st">': '</span>, <span class="va">exptCondition</span>, <span class="st">' ('</span>, <span class="va">transgene</span>,<span class="st">')'</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">ord</span> <span class="op">&lt;-</span> <span class="va">tmp</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">stype</span> <span class="op">==</span> <span class="st">'subtype C1'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">model_name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">tmp</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model_name <span class="op">=</span> <span class="fu">fct_relevel</span><span class="op">(</span><span class="va">model_name</span>, <span class="va">ord</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">stype</span>, <span class="va">model_name</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="co"># facet_grid(rows = vars(Annotation), switch = 'y')+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, </span>
<span>             stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul>
<li>AD is complex and human patient populations are heterogenous</li>
<li>Molecular subtyping is one way to reduce that complexity</li>
<li>We can use identified human subtypes to help characterize mouse
models</li>
<li>Each existing mouse model of AD may match to a particular subset of
human AD subtypes but not all subtypes simultaneously.</li>
<li>Most of the molecular AD subtypes were identified using
transcriptomics data and there seems to be similarities between AD
subtypes across studies.</li>
</ul>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb89">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb89-2"><a href="#cb89-2" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">5.2</span> (<span class="dv">2025-10-31</span>)</span>
<span id="cb89-3"><a href="#cb89-3" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>pc<span class="sc">-</span>linux<span class="sc">-</span>gnu</span>
<span id="cb89-4"><a href="#cb89-4" tabindex="-1"></a>Running under<span class="sc">:</span> Ubuntu <span class="dv">24</span>.<span class="fl">04.3</span> LTS</span>
<span id="cb89-5"><a href="#cb89-5" tabindex="-1"></a></span>
<span id="cb89-6"><a href="#cb89-6" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb89-7"><a href="#cb89-7" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>x86_64<span class="sc">-</span>linux<span class="sc">-</span>gnu<span class="sc">/</span>openblas<span class="sc">-</span>pthread<span class="sc">/</span>libblas.so<span class="fl">.3</span> </span>
<span id="cb89-8"><a href="#cb89-8" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>x86_64<span class="sc">-</span>linux<span class="sc">-</span>gnu<span class="sc">/</span>openblas<span class="sc">-</span>pthread<span class="sc">/</span>libopenblasp<span class="sc">-</span>r0.<span class="dv">3</span>.<span class="fl">26.</span>so;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb89-9"><a href="#cb89-9" tabindex="-1"></a></span>
<span id="cb89-10"><a href="#cb89-10" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb89-11"><a href="#cb89-11" tabindex="-1"></a> [<span class="dv">1</span>] LC_CTYPE<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>       LC_NUMERIC<span class="ot">=</span>C              </span>
<span id="cb89-12"><a href="#cb89-12" tabindex="-1"></a> [<span class="dv">3</span>] LC_TIME<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>        LC_COLLATE<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>    </span>
<span id="cb89-13"><a href="#cb89-13" tabindex="-1"></a> [<span class="dv">5</span>] LC_MONETARY<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>    LC_MESSAGES<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>   </span>
<span id="cb89-14"><a href="#cb89-14" tabindex="-1"></a> [<span class="dv">7</span>] LC_PAPER<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>       LC_NAME<span class="ot">=</span>C                 </span>
<span id="cb89-15"><a href="#cb89-15" tabindex="-1"></a> [<span class="dv">9</span>] LC_ADDRESS<span class="ot">=</span>C               LC_TELEPHONE<span class="ot">=</span>C            </span>
<span id="cb89-16"><a href="#cb89-16" tabindex="-1"></a>[<span class="dv">11</span>] LC_MEASUREMENT<span class="ot">=</span>en_US.UTF<span class="dv">-8</span> LC_IDENTIFICATION<span class="ot">=</span>C       </span>
<span id="cb89-17"><a href="#cb89-17" tabindex="-1"></a></span>
<span id="cb89-18"><a href="#cb89-18" tabindex="-1"></a>time zone<span class="sc">:</span> Etc<span class="sc">/</span>UTC</span>
<span id="cb89-19"><a href="#cb89-19" tabindex="-1"></a>tzcode source<span class="sc">:</span> <span class="fu">system</span> (glibc)</span>
<span id="cb89-20"><a href="#cb89-20" tabindex="-1"></a></span>
<span id="cb89-21"><a href="#cb89-21" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb89-22"><a href="#cb89-22" tabindex="-1"></a>[<span class="dv">1</span>] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span id="cb89-23"><a href="#cb89-23" tabindex="-1"></a>[<span class="dv">8</span>] base     </span>
<span id="cb89-24"><a href="#cb89-24" tabindex="-1"></a></span>
<span id="cb89-25"><a href="#cb89-25" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb89-26"><a href="#cb89-26" tabindex="-1"></a> [<span class="dv">1</span>] gt_1.<span class="fl">3.0</span>               corrplot_0<span class="fl">.95</span>          lubridate_1.<span class="fl">9.4</span>       </span>
<span id="cb89-27"><a href="#cb89-27" tabindex="-1"></a> [<span class="dv">4</span>] forcats_1.<span class="fl">0.1</span>          stringr_1.<span class="fl">6.0</span>          purrr_1.<span class="fl">2.1</span>           </span>
<span id="cb89-28"><a href="#cb89-28" tabindex="-1"></a> [<span class="dv">7</span>] readr_2.<span class="fl">1.6</span>            tidyr_1.<span class="fl">3.2</span>            tibble_3.<span class="fl">3.1</span>          </span>
<span id="cb89-29"><a href="#cb89-29" tabindex="-1"></a>[<span class="dv">10</span>] tidyverse_2.<span class="fl">0.0</span>        data.table_1.<span class="dv">18</span>.<span class="fl">2.1</span>    clusterProfiler_4.<span class="fl">18.4</span></span>
<span id="cb89-30"><a href="#cb89-30" tabindex="-1"></a>[<span class="dv">13</span>] dplyr_1.<span class="fl">1.4</span>            cowplot_1.<span class="fl">2.0</span>          GO.db_3.<span class="fl">22.0</span>          </span>
<span id="cb89-31"><a href="#cb89-31" tabindex="-1"></a>[<span class="dv">16</span>] org.Hs.eg.db_3.<span class="fl">22.0</span>    org.Mm.eg.db_3.<span class="fl">22.0</span>    AnnotationDbi_1.<span class="fl">72.0</span>  </span>
<span id="cb89-32"><a href="#cb89-32" tabindex="-1"></a>[<span class="dv">19</span>] IRanges_2.<span class="fl">44.0</span>         S4Vectors_0.<span class="fl">48.0</span>       Biobase_2.<span class="fl">70.0</span>        </span>
<span id="cb89-33"><a href="#cb89-33" tabindex="-1"></a>[<span class="dv">22</span>] BiocGenerics_0.<span class="fl">56.0</span>    generics_0.<span class="fl">1.4</span>         ggplot2_4.<span class="fl">0.1</span>         </span>
<span id="cb89-34"><a href="#cb89-34" tabindex="-1"></a></span>
<span id="cb89-35"><a href="#cb89-35" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb89-36"><a href="#cb89-36" tabindex="-1"></a>  [<span class="dv">1</span>] DBI_1.<span class="fl">2.3</span>               gson_0.<span class="fl">1.0</span>              rlang_1.<span class="fl">1.7</span>            </span>
<span id="cb89-37"><a href="#cb89-37" tabindex="-1"></a>  [<span class="dv">4</span>] magrittr_2.<span class="fl">0.4</span>          DOSE_4.<span class="fl">4.0</span>              otel_0.<span class="fl">2.0</span>             </span>
<span id="cb89-38"><a href="#cb89-38" tabindex="-1"></a>  [<span class="dv">7</span>] compiler_4.<span class="fl">5.2</span>          RSQLite_2.<span class="fl">4.5</span>           png_0<span class="fl">.1</span><span class="dv">-8</span>              </span>
<span id="cb89-39"><a href="#cb89-39" tabindex="-1"></a> [<span class="dv">10</span>] systemfonts_1.<span class="fl">3.1</span>       vctrs_0.<span class="fl">7.1</span>             reshape2_1.<span class="fl">4.5</span>         </span>
<span id="cb89-40"><a href="#cb89-40" tabindex="-1"></a> [<span class="dv">13</span>] pkgconfig_2.<span class="fl">0.3</span>         crayon_1.<span class="fl">5.3</span>            fastmap_1.<span class="fl">2.0</span>          </span>
<span id="cb89-41"><a href="#cb89-41" tabindex="-1"></a> [<span class="dv">16</span>] XVector_0.<span class="fl">50.0</span>          tzdb_0.<span class="fl">5.0</span>              enrichplot_1.<span class="fl">30.4</span>      </span>
<span id="cb89-42"><a href="#cb89-42" tabindex="-1"></a> [<span class="dv">19</span>] bit_4.<span class="fl">6.0</span>               xfun_0<span class="fl">.56</span>               cachem_1.<span class="fl">1.0</span>           </span>
<span id="cb89-43"><a href="#cb89-43" tabindex="-1"></a> [<span class="dv">22</span>] aplot_0.<span class="fl">2.9</span>             jsonlite_2.<span class="fl">0.0</span>          blob_1.<span class="fl">3.0</span>             </span>
<span id="cb89-44"><a href="#cb89-44" tabindex="-1"></a> [<span class="dv">25</span>] tidydr_0.<span class="fl">0.6</span>            tweenr_2.<span class="fl">0.3</span>            BiocParallel_1.<span class="fl">44.0</span>    </span>
<span id="cb89-45"><a href="#cb89-45" tabindex="-1"></a> [<span class="dv">28</span>] cluster_2.<span class="dv">1</span>.<span class="fl">8.1</span>         parallel_4.<span class="fl">5.2</span>          R6_2.<span class="fl">6.1</span>               </span>
<span id="cb89-46"><a href="#cb89-46" tabindex="-1"></a> [<span class="dv">31</span>] stringi_1.<span class="fl">8.7</span>           RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>      GOSemSim_2.<span class="fl">36.0</span>        </span>
<span id="cb89-47"><a href="#cb89-47" tabindex="-1"></a> [<span class="dv">34</span>] Rcpp_1.<span class="fl">1.1</span>              Seqinfo_1.<span class="fl">0.0</span>           knitr_1<span class="fl">.51</span>             </span>
<span id="cb89-48"><a href="#cb89-48" tabindex="-1"></a> [<span class="dv">37</span>] ggtangle_0.<span class="fl">1.1</span>          R.utils_2.<span class="fl">13.0</span>          timechange_0.<span class="fl">4.0</span>       </span>
<span id="cb89-49"><a href="#cb89-49" tabindex="-1"></a> [<span class="dv">40</span>] Matrix_1<span class="fl">.7</span><span class="dv">-4</span>            splines_4.<span class="fl">5.2</span>           igraph_2.<span class="fl">2.1</span>           </span>
<span id="cb89-50"><a href="#cb89-50" tabindex="-1"></a> [<span class="dv">43</span>] tidyselect_1.<span class="fl">2.1</span>        qvalue_2.<span class="fl">42.0</span>           yaml_2.<span class="fl">3.12</span>            </span>
<span id="cb89-51"><a href="#cb89-51" tabindex="-1"></a> [<span class="dv">46</span>] codetools_0<span class="fl">.2</span><span class="dv">-20</span>        lattice_0<span class="fl">.22</span><span class="dv">-7</span>          plyr_1.<span class="fl">8.9</span>             </span>
<span id="cb89-52"><a href="#cb89-52" tabindex="-1"></a> [<span class="dv">49</span>] treeio_1.<span class="fl">34.0</span>           withr_3.<span class="fl">0.2</span>             KEGGREST_1.<span class="fl">50.0</span>        </span>
<span id="cb89-53"><a href="#cb89-53" tabindex="-1"></a> [<span class="dv">52</span>] S7_0.<span class="fl">2.1</span>                evaluate_1.<span class="fl">0.5</span>          gridGraphics_0<span class="fl">.5</span><span class="dv">-1</span>     </span>
<span id="cb89-54"><a href="#cb89-54" tabindex="-1"></a> [<span class="dv">55</span>] polyclip_1<span class="fl">.10</span><span class="dv">-7</span>         scatterpie_0.<span class="fl">2.6</span>        xml2_1.<span class="fl">5.2</span>             </span>
<span id="cb89-55"><a href="#cb89-55" tabindex="-1"></a> [<span class="dv">58</span>] Biostrings_2.<span class="fl">78.0</span>       pillar_1.<span class="fl">11.1</span>           BiocManager_1.<span class="fl">30.27</span>    </span>
<span id="cb89-56"><a href="#cb89-56" tabindex="-1"></a> [<span class="dv">61</span>] ggtree_4.<span class="fl">0.4</span>            renv_1.<span class="fl">1.7</span>              ggfun_0.<span class="fl">2.0</span>            </span>
<span id="cb89-57"><a href="#cb89-57" tabindex="-1"></a> [<span class="dv">64</span>] hms_1.<span class="fl">1.4</span>               scales_1.<span class="fl">4.0</span>            tidytree_0.<span class="fl">4.7</span>         </span>
<span id="cb89-58"><a href="#cb89-58" tabindex="-1"></a> [<span class="dv">67</span>] glue_1.<span class="fl">8.0</span>              gdtools_0.<span class="fl">4.4</span>           lazyeval_0.<span class="fl">2.2</span>         </span>
<span id="cb89-59"><a href="#cb89-59" tabindex="-1"></a> [<span class="dv">70</span>] tools_4.<span class="fl">5.2</span>             ggnewscale_0.<span class="fl">5.2</span>        fgsea_1.<span class="fl">36.2</span>           </span>
<span id="cb89-60"><a href="#cb89-60" tabindex="-1"></a> [<span class="dv">73</span>] ggiraph_0.<span class="fl">9.3</span>           fs_1.<span class="fl">6.6</span>                fastmatch_1<span class="fl">.1</span><span class="dv">-8</span>        </span>
<span id="cb89-61"><a href="#cb89-61" tabindex="-1"></a> [<span class="dv">76</span>] grid_4.<span class="fl">5.2</span>              ape_5<span class="fl">.8</span><span class="dv">-1</span>               nlme_3<span class="fl">.1</span><span class="dv">-168</span>           </span>
<span id="cb89-62"><a href="#cb89-62" tabindex="-1"></a> [<span class="dv">79</span>] patchwork_1.<span class="fl">3.2</span>         ggforce_0.<span class="fl">5.0</span>           cli_3.<span class="fl">6.5</span>              </span>
<span id="cb89-63"><a href="#cb89-63" tabindex="-1"></a> [<span class="dv">82</span>] rappdirs_0.<span class="fl">3.4</span>          fontBitstreamVera_0.<span class="fl">1.1</span> gtable_0.<span class="fl">3.6</span>           </span>
<span id="cb89-64"><a href="#cb89-64" tabindex="-1"></a> [<span class="dv">85</span>] R.methodsS3_1.<span class="fl">8.2</span>       yulab.utils_0.<span class="fl">2.4</span>       sass_0.<span class="fl">4.10</span>            </span>
<span id="cb89-65"><a href="#cb89-65" tabindex="-1"></a> [<span class="dv">88</span>] digest_0.<span class="fl">6.39</span>           fontquiver_0.<span class="fl">2.1</span>        ggrepel_0.<span class="fl">9.6</span>          </span>
<span id="cb89-66"><a href="#cb89-66" tabindex="-1"></a> [<span class="dv">91</span>] ggplotify_0.<span class="fl">1.3</span>         htmlwidgets_1.<span class="fl">6.4</span>       farver_2.<span class="fl">1.2</span>           </span>
<span id="cb89-67"><a href="#cb89-67" tabindex="-1"></a> [<span class="dv">94</span>] memoise_2.<span class="fl">0.1</span>           htmltools_0.<span class="fl">5.9</span>         R.oo_1.<span class="fl">27.1</span>            </span>
<span id="cb89-68"><a href="#cb89-68" tabindex="-1"></a> [<span class="dv">97</span>] lifecycle_1.<span class="fl">0.5</span>         httr_1.<span class="fl">4.7</span>              MASS_7<span class="fl">.3</span><span class="dv">-65</span>            </span>
<span id="cb89-69"><a href="#cb89-69" tabindex="-1"></a>[<span class="dv">100</span>] fontLiberation_0.<span class="fl">1.0</span>    bit64_4.<span class="fl">6.0</span><span class="dv">-1</span>          </span></code></pre>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</div></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/asliuyar/translational-omics-AD/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/asliuyar/translational-omics-AD/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/asliuyar/translational-omics-AD/" class="external-link">Source</a></p>
        <p>
        <a href="citation.html">Cite</a>
        | <a href="mailto:">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/" class="external-link">sandpaper (0.18.1)</a>, <a href="https://github.com/carpentries/pegboard/tree/" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/" class="external-link">varnish (1.0.9)</a></p>
		</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://asliuyar.github.io/translational-omics-AD/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "Alzheimers-disease, cross-species-translation, omics, AD-Knowledge-Portal",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://asliuyar.github.io/translational-omics-AD/aio.html",
  "identifier": "https://asliuyar.github.io/translational-omics-AD/aio.html",
  "dateCreated": "2024-10-02",
  "dateModified": "2026-02-12",
  "datePublished": "2026-02-12"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

