<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Translational Omics in Alzheimers Disease: Choosing the right animal models</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png"><link rel="manifest" href="../favicons/incubator/site.webmanifest"><link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../choosing_the_right_model.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Translational Omics in Alzheimers Disease
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Translational Omics in Alzheimers Disease
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Translational Omics in Alzheimers Disease
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 62%" class="percentage">
    62%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 62%" aria-valuenow="62" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../choosing_the_right_model.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction to Omics Translation in AD</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="exploring-data-in-synapse.html">2. Synapse and AD Knowledge Portal</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="differential-expression-analysis.html">3. Differential expression analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="mouse-human-alignment.html">4. Cross Species Functional Alignment</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        5. Choosing the right animal models
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#overview-of-human-ad-data">Overview of Human AD Data</a></li>
<li><a href="#overview-of-mouse-model">Overview of Mouse Model</a></li>
<li><a href="#mouse-data">Mouse Data</a></li>
<li><a href="#correlation-between-mouse-models-and-human-ad-modules">Correlation between mouse models and human AD modules</a></li>
<li><a href="#correlation-between-mouse-models-and-milinds-ad-subtypes">Correlation between mouse models and Milind’s AD Subtypes</a></li>
<li><a href="#biodomain-correlations">Biodomain Correlations</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/mouse-human-alignment.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/index.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/mouse-human-alignment.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Cross Species
        </a>
        <a class="chapter-link float-end" href="../instructor/index.html" rel="next">
          Home
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Choosing the right animal models</h1>
        <p>Last updated on 2026-02-12 |

        <a href="https://github.com/asliuyar/translational-omics-AD/edit/main/episodes/choosing_the_right_model.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 180 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>What are some approaches that have been used to reduce the
heterogeneity in human AD patient populations?</li>
<li>How well do the changes we observe in AD mouse models align with
human AD data?</li>
<li>How do we perform corss-species comparison?</li>
<li>Which animal models best capture features of human AD subtypes?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Approaches to align mouse data to human data</li>
<li>Review the human AD co-expression modules</li>
<li>Understand the data from AD mouse models</li>
<li>Perform correlation analysis between mouse models and human
modules</li>
<li>Perform correlation analysis using the human subtypes</li>
<li>Understand the biological domains and subdomains of AD</li>
<li>Use domain annotations to compare between species</li>
</ul></div>
</div>
</div>
</div>
</div>
<p>Authors: Ravi Pandey &amp; Greg Cary, Jackson Laboratory</p>
<div class="section level3">
<h3 id="load-required-libraries">Load required libraries<a class="anchor" aria-label="anchor" href="#load-required-libraries"></a></h3>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">dplyr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tibble</span><span class="op">)</span></span>
<span><span class="co">#library(limma)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">corrplot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>corrplot 0.95 loaded</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">ggplot2</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">tidyr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">purrr</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">forcats</span><span class="op">)</span></span>
<span><span class="kw">library</span><span class="op">(</span><span class="va">gt</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
Attaching package: 'gt'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>The following object is masked from 'package:cowplot':

    as_gtable</code></pre>
</div>
<p>Mouse models of AD serve as indispensable platforms for
comprehensively characterizing AD pathology, disease progression, and
biological mechanisms. However, selection of the right model in
preclinical research and translation of findings to clinical populations
are intricate processes that require identification of
pathophysiological resemblance between model organisms and humans. Many
existing clinical trials that showed promising efficacy in one
particular mouse model later do not align with human trial results,
assuming that study had consisted of a heterogeneous group of
participants, and individual animal models may only recapitulate
features of a subgroup of human cases.</p>
<figure><img src="../../figures/subtype_intro.png" width="1000px" align="center" alt="Heterogenetiy in Alzheimer's Disease" class="figure mx-auto d-block"></figure><p>To improve interspecies translation, it is necessary to
comprehensively compare molecular signatures in mouse models with
subgroup of human AD cases with distinct molecular signatures.</p>
<p>Through omics approaches, we can assess how genetic perturbations in
mice align with changes observed in human LOAD study cohorts. This
allows us to identify mouse models and genetic factors that correspond
to specific subsets of human AD subtypes. Within these subtypes,
genesets are highly co-expressed and represent distinct molecular
pathways.</p>
</div>
<section><h2 class="section-heading" id="overview-of-human-ad-data">Overview of Human AD Data<a class="anchor" aria-label="anchor" href="#overview-of-human-ad-data"></a></h2>
<hr class="half-width"><p>Three independent human brain transcriptome studies ROSMAP [Religious
Orders Study and the Memory and Aging Project], MSSM [Mount Sinai School
of Medicine], and Mayo collected human postmortem brain RNA-seq data
from seven distinct regions: dorsolateral prefrontal cortex (DLPFC),
temporal cortex (TCX), inferior frontal gyrus (IFG), superior temporal
gyrus (STG), frontal pole (FP), parahippocampal gyrus (PHG), and
cerebellum (CBE).</p>
<div class="section level3">
<h3 id="amp-ad-modules">AMP-AD Modules<a class="anchor" aria-label="anchor" href="#amp-ad-modules"></a></h3>
<p><a href="https://doi.org/10.1016/j.celrep.2020.107908" class="external-link">Wan, et
al.</a> performed multi method co-expression network analysis followed
by differential analysis and found 30 co-expression modules related LOAD
pathology from human cohort study. Among the 30 aggregate co-expression
modules, five consensus clusters have been described by Wan, et
al. These consensus clusters consist of a subset of modules which are
associated with similar AD related changes across the multiple studies
and brain regions.</p>
<figure><img src="../../figures/Picture1.png" width="1000px" align="center" alt="Transcriptomic of Alzheimer's Disease" class="figure mx-auto d-block"></figure><p>Here we are showing matrix view of gene content overlap between these
module, and you can see few strongly overlapping group of modules,
implicating similar pathology in different studies in different brain
regions.</p>
</div>
<div class="section level3">
<h3 id="ad-subtypes">AD Subtypes<a class="anchor" aria-label="anchor" href="#ad-subtypes"></a></h3>
<p>Post mortem transcriptomics from AMP-AD and similar studies have
enabled the partitioning of AD cases into potential disease subtypes.
These studies have often stratified AD subjects into inflammatory and
non-inflammatory subtypes.</p>
<!--html_preserve-->
<div id="meunwmbjmp" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#meunwmbjmp table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#meunwmbjmp thead, #meunwmbjmp tbody, #meunwmbjmp tfoot, #meunwmbjmp tr, #meunwmbjmp td, #meunwmbjmp th {
  border-style: none;
}

#meunwmbjmp p {
  margin: 0;
  padding: 0;
}

#meunwmbjmp .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#meunwmbjmp .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#meunwmbjmp .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#meunwmbjmp .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#meunwmbjmp .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#meunwmbjmp .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#meunwmbjmp .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#meunwmbjmp .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#meunwmbjmp .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#meunwmbjmp .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#meunwmbjmp .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#meunwmbjmp .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#meunwmbjmp .gt_spanner_row {
  border-bottom-style: hidden;
}

#meunwmbjmp .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#meunwmbjmp .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#meunwmbjmp .gt_from_md > :first-child {
  margin-top: 0;
}

#meunwmbjmp .gt_from_md > :last-child {
  margin-bottom: 0;
}

#meunwmbjmp .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#meunwmbjmp .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#meunwmbjmp .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#meunwmbjmp .gt_row_group_first td {
  border-top-width: 2px;
}

#meunwmbjmp .gt_row_group_first th {
  border-top-width: 2px;
}

#meunwmbjmp .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#meunwmbjmp .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#meunwmbjmp .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#meunwmbjmp .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#meunwmbjmp .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#meunwmbjmp .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#meunwmbjmp .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#meunwmbjmp .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#meunwmbjmp .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#meunwmbjmp .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#meunwmbjmp .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#meunwmbjmp .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#meunwmbjmp .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#meunwmbjmp .gt_left {
  text-align: left;
}

#meunwmbjmp .gt_center {
  text-align: center;
}

#meunwmbjmp .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#meunwmbjmp .gt_font_normal {
  font-weight: normal;
}

#meunwmbjmp .gt_font_bold {
  font-weight: bold;
}

#meunwmbjmp .gt_font_italic {
  font-style: italic;
}

#meunwmbjmp .gt_super {
  font-size: 65%;
}

#meunwmbjmp .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#meunwmbjmp .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#meunwmbjmp .gt_indent_1 {
  text-indent: 5px;
}

#meunwmbjmp .gt_indent_2 {
  text-indent: 10px;
}

#meunwmbjmp .gt_indent_3 {
  text-indent: 15px;
}

#meunwmbjmp .gt_indent_4 {
  text-indent: 20px;
}

#meunwmbjmp .gt_indent_5 {
  text-indent: 25px;
}

#meunwmbjmp .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#meunwmbjmp div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style><table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false"><thead><tr class="gt_col_headings"><th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Study">
Study
</th>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Cohort">
Cohort
</th>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Tissue">
Tissue
</th>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="DataTypes">
DataTypes
</th>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Methods">
Methods
</th>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Subtypes">
Subtypes
</th>
</tr></thead><tbody class="gt_table_body"><tr><td headers="Study" class="gt_row gt_left">
Milind et al.,2020
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP, MSBB, MAYO
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC, PHG, FP,TCX
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ
</td>
<td headers="Methods" class="gt_row gt_left">
Iterative WGCNA followed by NbClust R package
</td>
<td headers="Subtypes" class="gt_row gt_left">
2,2, 3
</td>
</tr><tr><td headers="Study" class="gt_row gt_left">
Neff et al., 2021
</td>
<td headers="Cohort" class="gt_row gt_left">
MSBB &amp; ROSMAP
</td>
<td headers="Tissue" class="gt_row gt_left">
PHG,DLPFC
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ
</td>
<td headers="Methods" class="gt_row gt_left">
WSCNA, Hierarchical and K-means clustering
</td>
<td headers="Subtypes" class="gt_row gt_left">
5
</td>
</tr><tr><td headers="Study" class="gt_row gt_left">
Yasser et al.,2022
</td>
<td headers="Cohort" class="gt_row gt_left">
ROSMAP
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC
</td>
<td headers="DataTypes" class="gt_row gt_left">
Multi-Omics
</td>
<td headers="Methods" class="gt_row gt_left">
Novel ML framework mcTI
</td>
<td headers="Subtypes" class="gt_row gt_left">
3
</td>
</tr><tr><td headers="Study" class="gt_row gt_left">
Mukherjee et al., 2020
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP &amp; MAYO
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC &amp; TCX
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ
</td>
<td headers="Methods" class="gt_row gt_left">
DDRTree manifold learning approach using Monocle 2 R package
</td>
<td headers="Subtypes" class="gt_row gt_left">
6
</td>
</tr><tr><td headers="Study" class="gt_row gt_left">
Laura Heath et al.(unpublished)
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ, PROTEOMICS, METABOLOMICS
</td>
<td headers="Methods" class="gt_row gt_left">
DDRTree manifold learning approach
</td>
<td headers="Subtypes" class="gt_row gt_left">
~7
</td>
</tr><tr><td headers="Study" class="gt_row gt_left">
Zheng et al. 2021
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ
</td>
<td headers="Methods" class="gt_row gt_left">
Non-negative Matrix (NMF)
</td>
<td headers="Subtypes" class="gt_row gt_left">
2
</td>
</tr><tr><td headers="Study" class="gt_row gt_left">
Yang et al., 2023
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC
</td>
<td headers="DataTypes" class="gt_row gt_left">
Multi-Omics
</td>
<td headers="Methods" class="gt_row gt_left">
Similarity Network Fusion
</td>
<td headers="Subtypes" class="gt_row gt_left">
2 and 5
</td>
</tr><tr><td headers="Study" class="gt_row gt_left">
Lee et al., 2023
</td>
<td headers="Cohort" class="gt_row gt_left">
ROS/MAP
</td>
<td headers="Tissue" class="gt_row gt_left">
DLPFC, AC, PCC
</td>
<td headers="DataTypes" class="gt_row gt_left">
RNASEQ
</td>
<td headers="Methods" class="gt_row gt_left">
CCA, k-means clustering, NMF
</td>
<td headers="Subtypes" class="gt_row gt_left">
2
</td>
</tr></tbody></table></div>
<!--/html_preserve-->
<p>For this workshop, we will work with Milind’s and Neff’s subtype.</p>
<div class="section level4">
<h4 id="milinds-ad-subtype">Milind’s AD Subtype<a class="anchor" aria-label="anchor" href="#milinds-ad-subtype"></a></h4>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/32492070/" class="external-link">Milind et al.</a>
integrated post mortem brain co-expression data from the frontal cortex,
temporal cortex, and hippocampus brain regions and stratified patients
into different molecular subtypes based on molecular profiles in three
independent human LOAD cohorts (ROS/MAP, Mount Sinai Brain Bank, and
Mayo Clinic).</p>
<p>Two distinct LOAD subtypes were identified in the ROSMAP cohort,
three LOAD subtypes were identified in the Mayo cohort, and two distinct
LOAD subtypes were identified in the MSBB cohort. Similar subtype
results were observed in each cohort, with LOAD subtypes found to
primarily differ in their inflammatory response based on differential
expression analysis.</p>
<figure><img src="../../figures/milindsubtype.png" width="1000px" align="center" alt="Transcriptomic of Alzheimer's Disease" class="figure mx-auto d-block"></figure></div>
<div class="section level4">
<h4 id="neffs-ad-subtype">Neff’s AD Subtype<a class="anchor" aria-label="anchor" href="#neffs-ad-subtype"></a></h4>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/33523961/" class="external-link">Neff et al.</a>
investigated molecular subtypes of AD in the MSBB-AD study and
replicated findings in the ROSMAP. They have adopted weighted sample
gene network analysis (WSCNA) clustering algorithm to identify subgrups
of AD patients. WSCNA identifies five subtypes in the MSBB-AD (clusters
A, B1, B2, C1, and C2) across all 151 participants with PHG
transcriptomic data. These subtypes were classified into three larger
classes: typical AD (subtype C1 &amp; C2), intermediate (subtype B1
&amp; B2), or atypical AD (subtype A), by molecular presentation when
compared to the Blalock signatures of AD.</p>
<figure><img src="../../figures/neff_fig2.png" width="1000px" align="center" alt="Transcriptomic of Alzheimer's Disease" class="figure mx-auto d-block"></figure><figure><img src="../../figures/Neff6.png" width="1000px" align="center" alt="Transcriptomic of Alzheimer's Disease" class="figure mx-auto d-block"></figure></div>
</div>
</section><section><h2 class="section-heading" id="overview-of-mouse-model">Overview of Mouse Model<a class="anchor" aria-label="anchor" href="#overview-of-mouse-model"></a></h2>
<hr class="half-width"><p>We are going to utilize transcriptomics data from mouse models
expressing human risk variants created on a more LOAD-susceptible
genetic background expressing humanized APOE with the ε4 variant and the
R47H mutation in Trem2, two of the strongest genetic risk factors for
LOAD. Chracaterization of these mouse models is publsihed here: <a href="https://alz-journals.onlinelibrary.wiley.com/doi/full/10.1002/alz.13840" class="external-link">In
vivo validation of late-onset Alzheimer’s disease genetic risk
factors</a></p>
<figure><img src="../../figures/table1.png" width="1000px" align="center" alt="Transcriptomic of Alzheimer's Disease" class="figure mx-auto d-block"></figure><p>In this session, we are going to reproduce some of the results
published in the paper</p>
</section><section><h2 class="section-heading" id="mouse-data">Mouse Data<a class="anchor" aria-label="anchor" href="#mouse-data"></a></h2>
<hr class="half-width"><p>The <a href="https://link.springer.com/article/10.1186/s13024-020-00412-5" class="external-link">NanoString
Mouse AD gene expression panel</a> was used for gene expression
profiling on the nCounter platform (NanoString, Seattle, WA, USA).
NanoString gene expression panels are comprised of 770 probes. Mouse
NanoString gene expression data were collected from brain hemisphere
homogenates at 4 and 12 months of age for both sexes. The nSolver
software was used for generating NanoString gene expression counts.</p>
<p>Normalization was done by dividing counts within a lane by geometric
mean of the designated housekeeping genes from the same lane. Next,
normalized count values were log-transformed and corrected for potential
batch effects using ComBat.</p>
<p>You can find data associated with this study on synapse: <a href="https://www.synapse.org/Synapse:syn21595258" class="external-link uri">https://www.synapse.org/Synapse:syn21595258</a></p>
<div class="section level3">
<h3 id="synapse-data-download">Synapse Data Download<a class="anchor" aria-label="anchor" href="#synapse-data-download"></a></h3>
<p>You can download the data from Synapse data repository. API clients
provide a way to use Synapse programmatically. Installation instructions
are available at <a href="https://help.synapse.org/docs/Installing-Synapse-API-Clients.1985249668.html" class="external-link">Synapse
API Documentation Site</a>.</p>
<p>The Synapse command line client is implemented in Python and comes
with the Synapse Python package. To install the Synapse command line
client, make sure that you have Python and pip installed. For more
information, see the <a href="https://www.python.org/downloads/" class="external-link">Python</a> and <a href="https://pip.pypa.io/en/stable/installing/" class="external-link">pip</a> installation
instructions.</p>
<p>First, Define the data directory:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">datadir</span> <span class="op">&lt;-</span> <span class="st">"/sbgenomics/projects/asliuyar/asliuyar/ad-omics/data/7_ADsubtypes_Pandey_Cary/"</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="count-matrix">Count matrix<a class="anchor" aria-label="anchor" href="#count-matrix"></a></h4>
<p>You can access the data from synapse</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NS.INTENSITY</span> <span class="op">&lt;-</span> <span class="fu">fread</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn59479928"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span>,check.names <span class="op">=</span> <span class="cn">F</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre>
</div>
<p>Above command will download the relevent file in your working
directory.</p>
<p>Now, let’s read the downloaded file</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NS.INTENSITY</span> <span class="op">&lt;-</span> <span class="fu">read.csv</span><span class="op">(</span><span class="fu">paste0</span><span class="op">(</span><span class="va">datadir</span>,                            <span class="st">"Nanostringdata_logNormalized_BatchCorrected_MergedTg_LOAD1_PrimaryScreenPaper.csv"</span><span class="op">)</span>,</span>
<span>                         check.names <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                         row.names<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s examine the data,</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NS.INTENSITY</span> <span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span></span></code></pre>
</div>
<p>Here, row’s represent gene names and columns are all mouse
samples.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">NS.INTENSITY</span><span class="op">)</span></span></code></pre>
</div>
<p>Total 763 genes were quantified for 718 mouse samples.</p>
</div>
<div class="section level4">
<h4 id="metadata">Metadata<a class="anchor" aria-label="anchor" href="#metadata"></a></h4>
<p>You can access the biospecimen and individual metadata file
associated with this study from synapse.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># biospecimen metadata</span></span>
<span><span class="va">bio_meta</span> <span class="op">&lt;-</span> <span class="fu">fread</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn58614964"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span>,check.names <span class="op">=</span> <span class="cn">F</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># individual metadata</span></span>
<span><span class="va">ind_meta</span> <span class="op">&lt;-</span> <span class="fu">fread</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn59479967"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span>,check.names <span class="op">=</span> <span class="cn">F</span>,header<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre>
</div>
<p>You can process and join these files to create final metadata file as
discussed earlier in this course.</p>
<p>To expedite the analysis, we’ll utilize the metadata file provided in
the session’s data folder for subsequent analyses.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">read.csv</span><span class="op">(</span><span class="fu">paste0</span><span class="op">(</span><span class="va">datadir</span>,</span>
<span>                            <span class="st">"metadata_NSdata.csv"</span><span class="op">)</span>,</span>
<span>                     check.names <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span></code></pre>
</div>
<p>You might be familiar with most of the co-variates beside Binding
Density. Briefly, In NanoString nCounter technology, binding density
refers to the number of fluorescent spots (reporter probes) per square
micron on the imaging surface of a sample in the nCounter cartridge.</p>
<p>Binding density can be influenced by several factors, including RNA
input mass, number of targets and expression of targets.Highly expressed
genes and/or higher RNA mass result in higher binding density. Low
binding density amy indicate that the assay is not sensitive enough to
detect the target molecules, while high binding density can indicate
over-saturation where multiple probes might be overlapping, potentially
leading to underestimation of target molecule counts.</p>
<p>We can modify the metadata to only include covariates we’ll need for
this analysis</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="va">metadata</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">Sample</span>, Sex<span class="op">=</span><span class="va">SEX</span>, Age<span class="op">=</span><span class="va">AGE</span>,</span>
<span>                <span class="va">Genotype</span>, BD<span class="op">=</span><span class="va">BindingDensity</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span></code></pre>
</div>
<p>Using <code>pivot_longer()</code> function to “lengthens” data
i.e. increasing the number of rows and decreasing the number of
columns.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NS.INTENSITY.long</span> <span class="op">&lt;-</span> <span class="va">NS.INTENSITY</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu">rownames_to_column</span><span class="op">(</span><span class="va">.</span>, <span class="st">"symbol"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols<span class="op">=</span><span class="op">-</span><span class="va">symbol</span>,</span>
<span>               names_to<span class="op">=</span><span class="st">"Sample"</span>,</span>
<span>               values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">NS.INTENSITY.long</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will join the countdata with metadata by <code>Sample</code>
column</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mydat_with_metadata</span> <span class="op">&lt;-</span> <span class="va">NS.INTENSITY.long</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">metadata</span>,by<span class="op">=</span><span class="st">"Sample"</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s check the joined data table</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">mydat_with_metadata</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="create-design-matrix">Create design matrix<a class="anchor" aria-label="anchor" href="#create-design-matrix"></a></h4>
<p>Let’s create design matrix for multiple regression analyses. In this
formulation, B6J will be used as the control for the 5xFAD and LOAD1
mouse models, whereas LOAD1 served as controls for GWAS-based models in
order to estimate the effects of individual variants.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mydat_with_design</span> <span class="op">&lt;-</span> <span class="va">mydat_with_metadata</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sex <span class="op">=</span> <span class="fu">ifelse</span><span class="op">(</span><span class="va">Sex</span> <span class="op">%in%</span> <span class="st">"M"</span>,<span class="fl">1</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>APOE4.Trem2.R47H<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"APOE4Trem2"</span>,<span class="st">"LOAD1"</span>,<span class="st">"A/T.MIXED-WT"</span>,<span class="st">"A/T.IL1RAP-KO"</span>,<span class="st">"A/T.CR1&lt;B&gt;"</span>,<span class="st">"LOAD1.Shc2"</span>,<span class="st">"LOAD1.Slc6a17"</span>,<span class="st">"LOAD1.Erc2"</span>,<span class="st">"A/T.SNX1-KI"</span>,<span class="st">"A/T.CLASP2-KI"</span>,<span class="st">"A/T.MTHFR-KI"</span>,<span class="st">"A/T.ABCA7-KI"</span>,<span class="st">"A/T.CEACAM1-KO"</span>,<span class="st">"A/T.PLCG2.M28L"</span>,<span class="st">"hATA"</span>,<span class="st">"HFD.A/T.MIXED-WT"</span>,<span class="st">"HFD.A/T.MTHFR-KI"</span>,<span class="st">"HFD.hATA"</span>,<span class="st">"HFD.A/T.PLCG2.M28L"</span>,<span class="st">"A/T.SORL1"</span>,<span class="st">"A/T.MTMR4"</span>,<span class="st">"A/T.MEOX2"</span><span class="op">)</span>,<span class="fl">1</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>FADX5<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Tg"</span>,<span class="st">"5xFAD"</span><span class="op">)</span>,<span class="fl">1</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>MTHFR.KI<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.MTHFR-KI"</span>,<span class="st">"HFD.A/T.MTHFR-KI"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>PLCG2.M28L<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.PLCG2.M28L"</span>,<span class="st">"HFD.A/T.PLCG2.M28L"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>ABCA7.KI<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.ABCA7-KI"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>CEACAM1.KO<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.CEACAM1-KO"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>SNX1<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.SNX1-KI"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>CLASP2<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.CLASP2-KI"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>SORL1<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.Sorl1"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>MTMR4<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.Mtmr4"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>MEOX2<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"A/T.MEOX2"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>SHC2<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"LOAD1.Shc2"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>SLC6A17<span class="op">=</span><span class="fu">ifelse</span><span class="op">(</span><span class="va">Genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"LOAD1.Slc6a17"</span><span class="op">)</span>,<span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">Sex</span>,<span class="op">-</span><span class="va">Genotype</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">mydat_with_design</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="determine-the-effects-of-each-factor-in-mouse-models">Determine the effects of each factor in mouse models<a class="anchor" aria-label="anchor" href="#determine-the-effects-of-each-factor-in-mouse-models"></a></h4>
<p>Now, we will determine the effects of each factor (sex and genetic
variants) by fitting a multiple regression model using the lm function
in R.</p>
<p><strong>We are going to focus on only 12 months old mice
data.</strong></p>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lm.results</span> <span class="op">&lt;-</span> <span class="va">mydat_with_design</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Age</span> <span class="op">==</span> <span class="fl">12</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">split</span><span class="op">(</span><span class="va">.</span><span class="op">$</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="op">~</span> <span class="fu">lm</span><span class="op">(</span></span>
<span>    <span class="va">value</span> <span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">BD</span> <span class="op">+</span>  <span class="va">APOE4.Trem2.R47H</span> <span class="op">+</span> <span class="va">FADX5</span> <span class="op">+</span> <span class="va">MTHFR.KI</span> <span class="op">+</span> <span class="va">ABCA7.KI</span> <span class="op">+</span> <span class="va">CEACAM1.KO</span> <span class="op">+</span> <span class="va">SNX1</span> <span class="op">+</span> <span class="va">CLASP2</span> <span class="op">+</span> <span class="va">PLCG2.M28L</span> <span class="op">+</span> <span class="va">SORL1</span> <span class="op">+</span> <span class="va">MTMR4</span> <span class="op">+</span> <span class="va">SHC2</span> <span class="op">+</span> <span class="va">MEOX2</span> <span class="op">+</span> <span class="va">SLC6A17</span>,</span>
<span>    data <span class="op">=</span> <span class="va">.x</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">map</span><span class="op">(</span><span class="va">summary</span><span class="op">)</span></span></code></pre>
</div>
<p>Results of regression analysis for each gene for each factor is
stored in <code>lm.results</code>.</p>
<p>Let’s check the result:</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lm.results</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<p>Now, we will extract effect i.e. regression coefficients for each
gene for each factor from output and will store in a data frame.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">effects</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="fu">bind_rows</span><span class="op">(</span> </span>
<span>  <span class="fu">lapply</span><span class="op">(</span><span class="va">lm.results</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span>,<span class="st">"Estimate"</span><span class="op">]</span> <span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>names <span class="op">=</span> <span class="fu">names</span><span class="op">(</span><span class="va">lm.results</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">column_to_rownames</span><span class="op">(</span><span class="va">.</span>,var<span class="op">=</span><span class="st">"names"</span><span class="op">)</span></span></code></pre>
</div>
<p>let’s see few entries of this data</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">effects</span><span class="op">)</span></span></code></pre>
</div>
<p>rename columns for clear annotation</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colnames</span><span class="op">(</span><span class="va">effects</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"(Intercept)"</span>, </span>
<span>                       <span class="st">"Sex (Male)"</span>, </span>
<span>                       <span class="st">"BD"</span>, </span>
<span>                       <span class="st">"LOAD1"</span>,</span>
<span>                       <span class="st">"5xFAD"</span>,</span>
<span>                       <span class="st">"Mthfr*677C&gt;T"</span>, </span>
<span>                       <span class="st">"Abca7*A1527G"</span>, </span>
<span>                       <span class="st">"Ceacam1 KO"</span>, </span>
<span>                       <span class="st">"Snx1*D465N"</span>, </span>
<span>                       <span class="st">"Clasp2*L163P"</span>, </span>
<span>                       <span class="st">"Plcg2*M28L"</span>,</span>
<span>                       <span class="st">"Sorl1*A528T"</span>,</span>
<span>                       <span class="st">"Mtmr4*V297G"</span>,</span>
<span>                       <span class="st">"Shc2*V433M"</span>,</span>
<span>                       <span class="st">"Meox2 KO (HET)"</span>,</span>
<span>                       <span class="st">"Slc6a17*P61P"</span></span>
<span>                       <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">effects</span><span class="op">)</span></span></code></pre>
</div>
<p>we will only keep factor of interest.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ns_effects</span> <span class="op">&lt;-</span> <span class="va">effects</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu">rownames_to_column</span><span class="op">(</span><span class="va">.</span>,<span class="st">"symbol"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="st">"(Intercept)"</span>, <span class="op">-</span><span class="st">"BD"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols<span class="op">=</span><span class="op">-</span><span class="va">symbol</span>,</span>
<span>               names_to<span class="op">=</span><span class="st">"Variant"</span>,</span>
<span>               values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ns_effects</span><span class="op">)</span></span></code></pre>
</div>
<p>You can save the results for future use.</p>
<p>We will use these values to perform correlation analyses with human
AD Data.</p>
</div>
</div>
</section><section><h2 class="section-heading" id="correlation-between-mouse-models-and-human-ad-modules">Correlation between mouse models and human AD modules<a class="anchor" aria-label="anchor" href="#correlation-between-mouse-models-and-human-ad-modules"></a></h2>
<hr class="half-width"><ul><li>Compare Human AD to mouse genetic effects for each orthologous gene
in a given module
<ul><li>h = human gene expression (Log2 RNA-seq Fold Change control/AD)</li>
<li>β = mouse gene expression effect from linear regression model (Log2
RNA-seq TPM) <span class="math display">\[cor.test(LogFC(h),
β)\]</span>
</li>
</ul></li>
</ul><p>These appraoches allow us to assess directional coherence between
AMP-AD modules and the effects of genetic perturbations in mice. In this
lesson, we are going to use second approach.</p>
<p>Log2FC values for human transcripts were obtained through the AD
Knowledge Portal55 (<a href="https://www.synapse.org/#!Synapse:syn14237651" class="external-link uri">https://www.synapse.org/#!Synapse:syn14237651</a>).</p>
<p>Let’s start!</p>
<div class="section level4">
<h4 id="loading-amp-ad-module-data">Loading AMP-AD module data<a class="anchor" aria-label="anchor" href="#loading-amp-ad-module-data"></a></h4>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lnames</span> <span class="op">=</span> <span class="fu">load</span><span class="op">(</span><span class="fu">paste0</span><span class="op">(</span><span class="va">datadir</span>,<span class="st">"AMPADModuleData_Correlation.RData"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s check AMP-AD data</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lnames</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules_fc</span><span class="op">)</span></span></code></pre>
</div>
<p>First, we will join regression coefficients and human
ampad_modules_fc log fold change datasets for all genes.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ns_vs_ampad_fc</span> <span class="op">&lt;-</span> <span class="va">ns_effects</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">ampad_modules_fc</span>, </span>
<span>             by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"symbol"</span><span class="op">)</span></span>
<span>             <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ns_vs_ampad_fc</span><span class="op">)</span></span></code></pre>
</div>
<p>Now, we will create a list-columns of data frame using nest function
of tidyverse package. Nesting is implicitly a summarising operation: you
get one row for each group defined by the non-nested columns.</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span>  <span class="op">&lt;-</span> <span class="va">ns_vs_ampad_fc</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">module</span>,<span class="va">Variant</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">value</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">df</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we compute correlation coefficients using cor.test function
built in R as following:</p>
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor_test <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">"value"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                                    <span class="va">.x</span><span class="op">[[</span><span class="st">"ampad_fc"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                                    method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    estimate <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"estimate"</span><span class="op">)</span>,</span>
<span>    p_value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"p.value"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">cor_test</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cor.df</span><span class="op">)</span></span></code></pre>
</div>
<p>We will do some data wrangling to convert it into usable format for
plotting.</p>
<p>First, label correlations significant based on p-value, then join
module cluster information to correlation table</p>
<div class="codewrapper sourceCode" id="cb40">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">variant_module.df</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>significant <span class="op">=</span> <span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,age_group<span class="op">=</span><span class="st">"12 Months"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">module_clusters</span>, by <span class="op">=</span> <span class="st">"module"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">cluster</span>, <span class="va">cluster_label</span>, <span class="va">module</span>,<span class="va">Variant</span>,<span class="va">age_group</span>, correlation <span class="op">=</span> <span class="va">estimate</span>, <span class="va">p_value</span>, <span class="va">significant</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ordered.variant</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Sex (Male)"</span>, <span class="st">"5xFAD"</span>, <span class="st">"LOAD1"</span>,  </span>
<span>                     <span class="st">"Abca7*A1527G"</span>, <span class="st">"Ceacam1 KO"</span>, <span class="st">"Mthfr*677C&gt;T"</span>,<span class="st">"Shc2*V433M"</span>,<span class="st">"Slc6a17*P61P"</span>,<span class="st">"Clasp2*L163P"</span>,<span class="st">"Sorl1*A528T"</span>,</span>
<span>                     <span class="st">"Meox2 KO (HET)"</span>,<span class="st">"Snx1*D465N"</span> ,<span class="st">"Plcg2*M28L"</span>,<span class="st">"Mtmr4*V297G"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ordered.variant</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">ordered.variant</span><span class="op">)</span></span>
<span><span class="co"># Create a version of the data for plotting - clean up naming, order factors, etc</span></span>
<span><span class="va">correlation_for_plot</span> <span class="op">&lt;-</span> <span class="va">variant_module.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Variant <span class="op">=</span><span class="fu">factor</span><span class="op">(</span><span class="va">Variant</span>,levels<span class="op">=</span><span class="va">ordered.variant</span><span class="op">)</span>,</span>
<span>    Variant <span class="op">=</span> <span class="fu">fct_rev</span><span class="op">(</span><span class="va">Variant</span><span class="op">)</span>,</span>
<span>    module <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">module</span>,levels<span class="op">=</span><span class="va">mod</span><span class="op">)</span>,</span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot.12M</span> <span class="op">=</span> <span class="va">correlation_for_plot</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Background <span class="op">=</span> <span class="fu">ifelse</span><span class="op">(</span><span class="va">Variant</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Sex (Male)"</span><span class="op">)</span>,<span class="st">"Female"</span>,</span>
<span>  <span class="fu">ifelse</span><span class="op">(</span><span class="va">Variant</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"5xFAD"</span>, <span class="st">"LOAD1"</span><span class="op">)</span>, <span class="st">"B6"</span>, <span class="st">"LOAD1"</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plot.12M</span><span class="op">$</span><span class="va">Background</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">plot.12M</span><span class="op">$</span><span class="va">Background</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Female"</span>, <span class="st">"B6"</span>, <span class="st">"LOAD1"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">range</span><span class="op">(</span><span class="va">plot.12M</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level5">
<h5 id="visualizing-the-correlation-plot">Visualizing the Correlation plot<a class="anchor" aria-label="anchor" href="#visualizing-the-correlation-plot"></a></h5>
<p>Now, we will use above matrix and visualize the correlation results
using ggplot2 package.</p>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">plot.12M</span></span>
<span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_tile</span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">Variant</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span><span class="op">)</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">Variant</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span>, <span class="va">.data</span><span class="op">$</span><span class="va">significant</span><span class="op">)</span>,<span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">module</span>,y<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">Variant</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span>,color<span class="op">=</span><span class="st">"black"</span>,shape<span class="op">=</span><span class="fl">0</span>,size<span class="op">=</span><span class="fl">9</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_size</span><span class="op">(</span>guide <span class="op">=</span> <span class="st">"none"</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.6</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_color_gradient2</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">0.6</span>, <span class="fl">0.6</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">0.6</span>, <span class="fl">0</span>, <span class="fl">0.6</span><span class="op">)</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, name <span class="op">=</span> <span class="st">"Correlation"</span>, guide <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">ggtitle</span><span class="op">(</span>label<span class="op">=</span> <span class="st">"12 months"</span>,subtitle<span class="op">=</span> <span class="st">"Perturbation  | Control"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">Background</span><span class="op">)</span>,cols <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">cluster_label</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free"</span>,switch<span class="op">=</span><span class="st">"y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">theme</span><span class="op">(</span></span>
<span>      strip.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span>,</span>
<span>      strip.text.y.left <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>      strip.background.y <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_rect</span><span class="op">(</span>fill<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span>,</span>
<span>      axis.ticks <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0</span>,size<span class="op">=</span><span class="fl">16</span><span class="op">)</span>,</span>
<span>      axis.text.y <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">14</span>,face<span class="op">=</span><span class="st">"italic"</span><span class="op">)</span>,</span>
<span>      plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">56</span>, hjust <span class="op">=</span> <span class="fl">0.066</span>,size<span class="op">=</span><span class="fl">14</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      plot.subtitle <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">65</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      panel.background <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>      panel.grid <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      legend.position <span class="op">=</span> <span class="st">"right"</span></span>
<span>    <span class="op">)</span></span></code></pre>
</div>
<p>In above plot, top row represent 30 AMP-AD modules grouped into 5
consensus clusters describing the major functional groups of AD-related
alterations and left column represent mouse models. Positive
correlations are shown in blue and negative correlations in red. Color
intensity and size of the circles are proportional to the correlation
coefficient. Black square around dots represent significant correlation
at p-value=0.05 and non-significant correlations are left blank.</p>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p><strong>What can you conclude from the plot?</strong> <strong>Which
variants capture effect of neurodegeneration </strong></p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="correlation-between-mouse-models-and-milinds-ad-subtypes">Correlation between mouse models and Milind’s AD Subtypes<a class="anchor" aria-label="anchor" href="#correlation-between-mouse-models-and-milinds-ad-subtypes"></a></h2>
<hr class="half-width"><div class="section level3">
<h3 id="milinds-subtype-data">Milind’s subtype data<a class="anchor" aria-label="anchor" href="#milinds-subtype-data"></a></h3>
<p>Briefly, Synapse repository <a href="https://www.synapse.org/#!Synapse:syn23660885" class="external-link">syn23660885</a>
contains code, data, and analyses for Milind’s LOAD subtypes. Subtype
assignment of each patient for each cohort are here on synpase: <a href="https://www.synapse.org/Synapse:syn23660969" class="external-link uri">https://www.synapse.org/Synapse:syn23660969</a>. Using
patient’s ID from these files and normalized expression from AMP-AD
cohorts (<a href="https://www.synapse.org/Synapse:syn30821562" class="external-link uri">https://www.synapse.org/Synapse:syn30821562</a>), you can
calculate change in gene expression for subtype patients vs controls.
These matrix can be used to perform correlation analysis with mouse
models.</p>
<p>Here, we have provided final processed data that is stored in
project’s data folder.</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="fu">paste0</span><span class="op">(</span><span class="va">datadir</span>,<span class="st">"Nikhil_SubtypeData.RData"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>This data object contains 3 files: <code>ampad_subtype_fc</code>,
<code>module_cohort</code>, <code>subtypes</code></p>
<p>Let’s briefly check the data objects:</p>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_subtype_fc</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">module_cohort</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">subtypes</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ns_effects</span> <span class="op">&lt;-</span> <span class="va">effects</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu">rownames_to_column</span><span class="op">(</span><span class="va">.</span>, <span class="st">"Gene"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="st">"(Intercept)"</span>, <span class="op">-</span><span class="st">"BD"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols<span class="op">=</span><span class="op">-</span><span class="va">Gene</span>,</span>
<span>               names_to<span class="op">=</span><span class="st">"Variant"</span>,</span>
<span>               values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_vs_subtype_fc</span> <span class="op">&lt;-</span> <span class="va">ns_effects</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">ampad_subtype_fc</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Gene"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">subtype</span>,<span class="va">Variant</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">Gene</span>, <span class="va">value</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor_test <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">"value"</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">"ampad_fc"</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    estimate <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"estimate"</span><span class="op">)</span>,</span>
<span>    p_value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"p.value"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">cor_test</span><span class="op">)</span></span></code></pre>
</div>
<p>Process data for plotting —- Flag for significant results, add
cluster information to modules</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="va">model_vs_subtype_fc</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>significant <span class="op">=</span> <span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span>,age_group<span class="op">=</span><span class="st">"12 Months"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">module_cohort</span>, by <span class="op">=</span> <span class="st">"subtype"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">cluster</span>, <span class="va">cluster_label</span>, <span class="va">subtype</span>,<span class="va">Variant</span>,<span class="va">age_group</span>, correlation <span class="op">=</span> <span class="va">estimate</span>, <span class="va">p_value</span>, <span class="va">significant</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">#ordered.variant &lt;- rev(ordered.variant)</span></span></code></pre>
</div>
<p>Create a version of the data for plotting - clean up naming, order
factors, etc</p>
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ordered.variant</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Sex (Male)"</span>, <span class="st">"5xFAD"</span>, <span class="st">"LOAD1"</span>,  </span>
<span>                     <span class="st">"Abca7*A1527G"</span>, <span class="st">"Ceacam1 KO"</span>, <span class="st">"Mthfr*677C&gt;T"</span>,<span class="st">"Shc2*V433M"</span>,<span class="st">"Slc6a17*P61P"</span>,<span class="st">"Erc2*N542S"</span>,<span class="st">"Clasp2*L163P"</span>,<span class="st">"Sorl1*A528T"</span> ,</span>
<span>                     <span class="st">"Meox2 KO (HET)"</span>,<span class="st">"Snx1*D465N"</span>, <span class="st">"Plcg2*M28L"</span>,<span class="st">"Mtmr4*V297G"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">subtype_nanostring_for_plot.12M</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Variant <span class="op">=</span><span class="fu">factor</span><span class="op">(</span><span class="va">Variant</span>,levels<span class="op">=</span><span class="va">ordered.variant</span><span class="op">)</span>,</span>
<span>    Variant <span class="op">=</span><span class="fu">fct_rev</span><span class="op">(</span><span class="va">Variant</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">range</span><span class="op">(</span><span class="va">subtype_nanostring_for_plot.12M</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">dd.subtype_PS</span> <span class="op">=</span> <span class="va">subtype_nanostring_for_plot.12M</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>Background <span class="op">=</span> <span class="fu">ifelse</span><span class="op">(</span><span class="va">Variant</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Sex (Male)"</span><span class="op">)</span>,<span class="st">"Female"</span>,<span class="fu">ifelse</span><span class="op">(</span><span class="va">Variant</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"5xFAD"</span>,<span class="st">"LOAD1"</span><span class="op">)</span>,<span class="st">"B6"</span>,<span class="st">"LOAD1"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dd.subtype_PS</span><span class="op">$</span><span class="va">Background</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">dd.subtype_PS</span><span class="op">$</span><span class="va">Background</span>,levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"Female"</span>,<span class="st">"B6"</span>,<span class="st">"LOAD1"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Function for creating correlation plot</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">subtype_variant_corrplot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span>,<span class="va">ran</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_tile</span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">subtype</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">Variant</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span><span class="op">)</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">subtype</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">Variant</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span>, <span class="va">.data</span><span class="op">$</span><span class="va">significant</span><span class="op">)</span>,<span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">subtype</span>,y<span class="op">=</span><span class="va">.data</span><span class="op">$</span><span class="va">Variant</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span>,color<span class="op">=</span><span class="st">"black"</span>,shape<span class="op">=</span><span class="fl">0</span>,size<span class="op">=</span><span class="fl">9</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_size</span><span class="op">(</span>guide <span class="op">=</span> <span class="st">"none"</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="va">ran</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">scale_color_gradient2</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="va">ran</span>, <span class="va">ran</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="va">ran</span>, <span class="fl">0</span>, <span class="va">ran</span><span class="op">)</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, name <span class="op">=</span> <span class="st">"Correlation"</span>, guide <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">ggtitle</span><span class="op">(</span>label<span class="op">=</span> <span class="st">"12 months"</span>,subtitle<span class="op">=</span> <span class="st">"Perturbation  | Control"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">Background</span><span class="op">)</span>,cols <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">cluster_label</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free"</span>,switch<span class="op">=</span><span class="st">"y"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>strip.placement.x <span class="op">=</span> <span class="st">"outside"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">theme</span><span class="op">(</span></span>
<span>      strip.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span>,</span>
<span>      strip.text.y.left <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>      strip.background.y <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_rect</span><span class="op">(</span>fill<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span>,</span>
<span>      axis.ticks <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, hjust <span class="op">=</span> <span class="fl">0.5</span>,size<span class="op">=</span><span class="fl">11</span><span class="op">)</span>,</span>
<span>      axis.text.y <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span>,size<span class="op">=</span><span class="fl">12</span><span class="op">)</span>,</span>
<span>      plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">15</span>, hjust <span class="op">=</span> <span class="fl">0.05</span>,size<span class="op">=</span><span class="fl">13</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      plot.subtitle <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">16</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">11</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      panel.background <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>      panel.grid <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      legend.position <span class="op">=</span> <span class="st">"bottom"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<p>Let’s plot the results</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">subtype_variant_corrplot</span><span class="op">(</span><span class="va">dd.subtype_PS</span>,<span class="fl">0.4</span><span class="op">)</span></span></code></pre>
</div>
<p>We can see that mouse model of AD may match to a particular subset of
human AD subtypes but not all subtypes simultaneously, and that risk for
these subtypes may be influenced by distinct AD genetic factors.</p>
<p>We can see that mouse model of AD may match to a particular subset of
human AD subtypes but not all subtypes simultaneously, and that risk for
these subtypes may be influenced by distinct AD genetic factors.</p>
</div>
<div class="section level3">
<h3 id="neffs-subtype-data">Neff’s subtype data<a class="anchor" aria-label="anchor" href="#neffs-subtype-data"></a></h3>
<p>Synapse repository <a href="https://www.synapse.org/Synapse:syn25944417" class="external-link">syn25944417</a>
contains code, data, and analyses for Neff’s LOAD subtypes for MSBB
cohort. Subtype assignments from the Neff Synapse project here: <a href="https://www.synapse.org/Synapse:syn25944448" class="external-link uri">https://www.synapse.org/Synapse:syn25944448</a>. You can use
the MSBB biospecimen metadata file from <a href="https://www.synapse.org/Synapse:syn21893059" class="external-link">syn21893059</a> to
curate the subtypes with RNA-Seq IDs and individual IDs.</p>
<p>Next, you can (if have permission) extract normalized expression of
all human genes for these individuals (<a href="https://www.synapse.org/Synapse:syn16795937" class="external-link uri">https://www.synapse.org/Synapse:syn16795937</a>) and perform
differential analysis to calculate log fold change expression.</p>
<p>Here, we have provided final processed data that is stored in
project’s data folder.</p>
<div class="codewrapper sourceCode" id="cb53">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="fu">paste0</span><span class="op">(</span><span class="va">datadir</span>,<span class="st">"Neff_Subtype.RData"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">neff_subtype_fc</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p><em>You all should create correlation plot for mouse data compared
with Neff’s subtype data. Identify mouse variants significantly positive
correlated with subtype C1 and subtype A.</em></p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="milinds-vs-neffs-subtype">Milind’s vs Neff’s Subtype<a class="anchor" aria-label="anchor" href="#milinds-vs-neffs-subtype"></a></h4>
<p>Finally, Let’ see how does AD subtypes corresponds to each other.</p>
<div id="challenge-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-3" class="callout-inner">
<h3 class="callout-title">Challenge 3</h3>
<div class="callout-content">
<p>Using data from Milind’s and Neff’s AD subtypes perform correlation
analysis to see how AD subtypes corresponds to each other.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">

</div>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="biodomain-correlations">Biodomain Correlations<a class="anchor" aria-label="anchor" href="#biodomain-correlations"></a></h2>
<hr class="half-width"><p>We can also use the gene sub-sets defined by the AD Biodomains and
Subdomains to frame these correlations. Let’s see how that works,
starting with how the LOAD1 mouse strain effects correlate with the
logFC values from AMP-AD cohorts.</p>
<p>First, we’ll download the Biodomain definitions. However, this time
there’s something different. What’s different between this table and the
one we used in the earlier session?</p>
<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># biodomain definitions</span></span>
<span><span class="va">biodom</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="fu">synGet</span><span class="op">(</span><span class="st">'syn26592124'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biodomain labels and colors</span></span>
<span><span class="va">dom.lab</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="fu">synGet</span><span class="op">(</span><span class="st">'syn26856828'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-4" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-4" class="callout-inner">
<h3 class="callout-title">Challenge 4</h3>
<div class="callout-content">
<p>What’s different about this <code>biodom</code> table?</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">GOterm_Name</span> <span class="op">==</span> <span class="st">'amyloid-beta formation'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unlist</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>This table contains mouse gene IDs!</p>
</div>
</div>
</div>
</div>
<p>To perform the correlation on subsets of genes defined by the
domains, we need to specify the genes in each biodomain and subdomain.
For each subdomain we’ll prefix the subdomain name with the biodomain
abbreviation. This is because some subdomains are shared between two
different domains. For example, both the <code>Autophagy</code> and
<code>Immune Response</code> domains have a subdomain called
“phagocytosis”. In each case there are slightly different sets of terms
contained within the subdomain. We’ll keep track by adding a domain
prefix, e.g. “IR_phagocytosis” and “Au_phagocytosis”.</p>
<p>Let’s collect the gene lists:</p>
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="co"># first set up the list of genes associated with each biodomain</span></span>
<span><span class="va">bd.genes</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span> set <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span> genes <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">set</span>, <span class="op">~</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unlist</span><span class="op">)</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># then the list of genes associated with each subdomain </span></span>
<span><span class="va">sd.genes</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span> <span class="va">biodom</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span><span class="op">=</span><span class="st">'domain'</span><span class="op">)</span> <span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span> Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">abbr</span>, <span class="va">Subdomain</span>, <span class="va">subdomain_idx</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    set <span class="op">=</span> <span class="fu">paste0</span><span class="op">(</span><span class="va">abbr</span>,<span class="st">'_'</span>,<span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    bd <span class="op">=</span> <span class="va">Biodomain</span>, sdi <span class="op">=</span> <span class="va">subdomain_idx</span>,</span>
<span>    genes <span class="op">=</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">bd</span>, <span class="va">subdomain_idx</span> <span class="op">==</span> <span class="va">sdi</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unlist</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co"># now combine the two lists</span></span>
<span><span class="va">bd.gene.lists</span> <span class="op">=</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">bd.genes</span>, <span class="va">sd.genes</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">set</span>, <span class="va">genes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">!=</span> <span class="st">'NA_none'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">bd.genes</span>, <span class="va">sd.genes</span><span class="op">)</span></span></code></pre>
</div>
<p>We’ll also need the cohort data fold change data, and to map the
orthologous mouse genes.</p>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ortholog map between human and mouse provided by Wan et al</span></span>
<span><span class="va">mouse.human.ortho</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn17010253"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># AMP-AD cohort differential expression table</span></span>
<span><span class="va">ampad_modules_raw</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">"syn14237651"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">mouse.human.ortho</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>hgnc_symbol <span class="op">=</span> <span class="va">human_symbol</span>, <span class="va">mouse_symbol</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="amp-ad-cohort-correlation">AMP-AD Cohort Correlation<a class="anchor" aria-label="anchor" href="#amp-ad-cohort-correlation"></a></h4>
<p>We’ll start by examining how the LOAD1 strain effects (β) correlate
with the human AMP-AD cohort logFC values, like we did above with the
co-expression modules. This time we’ll use the biodomain and subdomain
gene lists to sub-set the genes for correlation analysis. We’ll compute
correlations to one tissu for each cohort.</p>
<p>First we can generate a nested table with the AMP-AD data, but
filtering for the appropriate contrasts and tissues (DLPFC = ROSMAP, TCX
= MAYO, and PHG = MSSM).</p>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs.lfc</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>, </span>
<span>    <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span>,</span>
<span>    <span class="va">Tissue</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">'PHG'</span>,<span class="st">'TCX'</span>,<span class="st">'DLPFC'</span><span class="op">)</span>,</span>
<span>    <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">mouse_symbol</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">mouse_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nest</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">hgnc_symbol</span>, <span class="va">symbol</span>, <span class="va">logFC</span>, <span class="va">adj.P.Val</span><span class="op">)</span>, </span>
<span>    .by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">Study</span>, <span class="va">Tissue</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">hs.lfc</span><span class="op">)</span></span></code></pre>
</div>
<p>And look at one of those nested tables:</p>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">hs.lfc</span><span class="op">$</span><span class="va">data</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div id="a-note-about-gene-homology" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<span class="callout-header">Callout</span>
<div id="a-note-about-gene-homology" class="callout-inner">
<h3 class="callout-title">A note about gene homology</h3>
<div class="callout-content">
<p>You’ll notice in the table above that there are two
<strong>identical</strong> logFC values for the human gene ANGPT1. This
is because it aligns with two separate mouse genes: Angpt1 and Angpt2.
Each of these mouse genes are paralogs of each other and they form an
orthogroup with the human ANGPT1 gene. There are even more mouse genes
for SLC7A2! There are other ways to map orthologous genes between
sepecies (such as the <code>gprofiler2::gorth</code> function), and I
encourage you to try them out. This is a non-trivial task and care
should be given to how to structure these relationships to best address
the questions of your research.</p>
</div>
</div>
</div>
<p>We can prepare the LOAD1 mouse effects in an analogous way:</p>
<div class="codewrapper sourceCode" id="cb61">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">load1.effects</span> <span class="op">&lt;-</span> <span class="va">ns_effects</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>mm.beta <span class="op">=</span> <span class="va">value</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">mm.beta</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Now join the two datasets based on the mouse gene symbols. If we
perform an <code>inner_join</code> we’ll get only the genes that are
present in both datasets</p>
<div class="codewrapper sourceCode" id="cb62">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">load1.vs.amp</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span>load1 <span class="op">=</span> <span class="va">load1.effects</span><span class="op">$</span><span class="va">Variant</span>, amp <span class="op">=</span> <span class="va">hs.lfc</span><span class="op">$</span><span class="va">Study</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">load1</span>, <span class="va">amp</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>    <span class="va">mm</span> <span class="op">=</span> <span class="va">load1.effects</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Variant</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">hs</span> <span class="op">=</span> <span class="va">hs.lfc</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Study</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="fu">inner_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="st">'symbol'</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">load1.vs.amp</span><span class="op">)</span></span></code></pre>
</div>
<p>One last piece of information – let’s add in the biodomain/subdomain
gene lists and subset each mouse X human data table for the genes in
each subset:</p>
<div class="codewrapper sourceCode" id="cb63">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">load1.vs.amp</span>, set <span class="op">=</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span></span>
<span>      <span class="va">data</span>, <span class="va">set</span>, </span>
<span>      <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span> <span class="va">symbol</span> <span class="op">%in%</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">genes</span><span class="op">[[</span><span class="fu">which</span><span class="op">(</span><span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">)</span> <span class="op">)</span>,</span>
<span>    n.genes <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="fu">nrow</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> </span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">cor.df</span><span class="op">)</span></span></code></pre>
</div>
<p>Finally, we can perform a correlation analysis for each comparison.
We’ll filter to remove any comparison that contains &lt; 3 genes. Also,
since we’re running so many comparisons (~150 for each mouse-human
pair), we should correct for multiple hypothesis testing. The way this
is set up is extreme (i.e. against <strong>all</strong> comparisons, not
just those within a mouse-human pairing), but it will help us to narrow
down to the most significant results.</p>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">n.genes</span> <span class="op">&gt;</span> <span class="fl">3</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'logFC'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'mm.beta'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.adj <span class="op">=</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">p.value</span>, method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span>,</span>
<span>    significant <span class="op">=</span> <span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Let’s plot the results of the correlation. We’ll start at the
biodomain level by filtering the gene set for those that don’t include
an underscore (“_“) in their name.</p>
<div class="codewrapper sourceCode" id="cb65">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">F</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set</span>, <span class="va">load1</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">amp</span><span class="op">)</span>, switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, </span>
<span>             stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>We can also take a look at the subdomains. Let’s focus on just one
cohort here - MSSM - to simplify the plot.</p>
<div class="codewrapper sourceCode" id="cb66">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the list of subdomains to plot</span></span>
<span><span class="va">l</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="va">amp</span> <span class="op">==</span> <span class="st">'ROSMAP'</span>,</span>
<span>    <span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">T</span>, </span>
<span>     <span class="op">(</span><span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">5e-2</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">set</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># prepare the cor.df for plotting</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">amp</span> <span class="op">==</span> <span class="st">'ROSMAP'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>max.n <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">n.genes</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">%in%</span> <span class="va">l</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    set1 <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>      <span class="fu">str_trunc</span><span class="op">(</span><span class="va">.</span>, <span class="fl">40</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>      <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">max.n</span>,<span class="st">'] '</span>,<span class="va">.</span><span class="op">)</span>,</span>
<span>    abbr <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># plot</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set1</span>, <span class="va">load1</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">abbr</span><span class="op">)</span>, <span class="co">#rows = vars(amp), </span></span>
<span>             scales <span class="op">=</span> <span class="st">'free'</span>, space <span class="op">=</span> <span class="st">'free'</span>, switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'left'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x.top = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x.top <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"left"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div id="challenge-5" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-5" class="callout-inner">
<h3 class="callout-title">Challenge 5</h3>
<div class="callout-content">
<p>How do these correlation results compare with those above when
considering the co-expression modules? What is similar? What is
different?</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="challenge-6" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-6" class="callout-inner">
<h3 class="callout-title">Challenge 6</h3>
<div class="callout-content">
<p>Are there different subdomains that have significant correlations for
the other AMP-AD cohorts?</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">

</div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="amp-ad-subtype-correlation">AMP-AD Subtype Correlation<a class="anchor" aria-label="anchor" href="#amp-ad-subtype-correlation"></a></h4>
<p>Now we’ve seen how the effects from each variant from the LOAD1 study
(mouse β) compare to the expression in the whole cohorts, let’s explore
how the mouse effects correlate with sub-type vs control LogFC. We’ll
start with Millind’s MAYO subtypes. So let’s join together the mouse
effects and human LogFCs into one table for each comparison.</p>
<div class="codewrapper sourceCode" id="cb67">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># some variable renaming</span></span>
<span><span class="va">stype.effects</span> <span class="op">&lt;-</span> <span class="va">ampad_subtype_fc</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">neff.effects</span> <span class="op">&lt;-</span> <span class="va">neff_subtype_fc</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Just the ROSMAPs</span></span>
<span><span class="va">stype.effects.RM</span> <span class="op">&lt;-</span> <span class="va">stype.effects</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'MSBB'</span>, <span class="va">subtype</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># stype.effects.RM &lt;- neff.effects</span></span>
<span></span>
<span><span class="co"># Join the tables together for each comparison</span></span>
<span><span class="va">load1.vs.stype</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span>load1 <span class="op">=</span> <span class="va">load1.effects</span><span class="op">$</span><span class="va">Variant</span>, stype <span class="op">=</span> <span class="va">stype.effects.RM</span><span class="op">$</span><span class="va">subtype</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">load1</span>, <span class="va">stype</span>,</span>
<span>                <span class="op">~</span> <span class="op">{</span> </span>
<span>                  <span class="va">mm</span> <span class="op">=</span> <span class="va">load1.effects</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Variant</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> </span>
<span>                  <span class="va">hs</span> <span class="op">=</span> <span class="va">stype.effects.RM</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">subtype</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> </span>
<span>                  <span class="fu">inner_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="st">'symbol'</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span> </span>
<span>                <span class="op">}</span></span>
<span>            <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Now let’s create a subset for each of the gene sets defined by the
biodomains and subdomains. Then we can compute a correlation for
each.</p>
<div class="codewrapper sourceCode" id="cb68">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">load1.vs.stype</span>, set <span class="op">=</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span></span>
<span>      <span class="va">data</span>, <span class="va">set</span>, </span>
<span>      <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span> <span class="va">symbol</span> <span class="op">%in%</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">genes</span><span class="op">[[</span><span class="fu">which</span><span class="op">(</span><span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">)</span> <span class="op">)</span>,</span>
<span>    n.genes <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="fu">nrow</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">n.genes</span> <span class="op">&gt;</span> <span class="fl">3</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'ampad_fc'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'mm.beta'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.adj <span class="op">=</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">p.value</span>, method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span>,</span>
<span>    significant <span class="op">=</span> <span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>And plot the correlation result for the biodomain gene sets</p>
<div class="codewrapper sourceCode" id="cb69">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">F</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set</span>, <span class="va">stype</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">load1</span><span class="op">)</span>, switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, </span>
<span>             stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>And for the subdomains, we’ll focus on only those that are
significant and have at least 10 genes</p>
<div class="codewrapper sourceCode" id="cb70">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the list of subdomains to plot</span></span>
<span><span class="va">l</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">T</span>, </span>
<span>     <span class="op">(</span><span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">5e-2</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">set</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># prepare the cor.df for plotting</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>max.n <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">n.genes</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">%in%</span> <span class="va">l</span>, <span class="va">max.n</span> <span class="op">&gt;</span> <span class="fl">10</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>any.sig <span class="op">=</span> <span class="fu">any</span><span class="op">(</span><span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">5e-2</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">load1</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span> <span class="va">any.sig</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    set1 <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>      <span class="fu">str_trunc</span><span class="op">(</span><span class="va">.</span>, <span class="fl">40</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>      <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">max.n</span>,<span class="st">'] '</span>,<span class="va">.</span><span class="op">)</span>,</span>
<span>    abbr <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="op">)</span></span>
<span>         </span>
<span><span class="co"># plot!</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set1</span>, <span class="va">stype</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">load1</span><span class="op">)</span>, cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">abbr</span><span class="op">)</span>, </span>
<span>             scales <span class="op">=</span> <span class="st">'free'</span>, space <span class="op">=</span> <span class="st">'free'</span>, switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"left"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
</div>
</section><div class="section level1">
<h1 id="wan-et-al-mouse-model-compendium">Wan et al mouse model compendium<a class="anchor" aria-label="anchor" href="#wan-et-al-mouse-model-compendium"></a></h1>
<p>Our ability to use the domains to assess the LOAD1 primary screen
study is constrained because the mice were only assessed for the
expression of the 770 genes that contained probes in the Nanostring
probe set. This is well suited for the module-level correlation
analysis, because the Nanostring genes were selected based on coverage
of the co-expression modules. However we can also perform these analyses
with RNA-seq data.</p>
<p>We’ve talked about the <a href="https://doi.org/10.1016/j.celrep.2020.107908" class="external-link">Wan, et al.</a>
study already this week in the context of the consensus co-expression
module generation for AMP-AD cohorts. <strong>Another</strong> useful
resource that this study generated is a compendium of mouse model
RNA-seq data from various sources. We can use these data as another
dataset to use to explore correlations between mouse and human. The data
are part of the <code>AD_CrossSpecies</code> study which can be accessed
on Synapse at <a href="https://www.synapse.org/Synapse:syn16779040" class="external-link">syn16779040</a>.</p>
<p>First, let’s list the files in the
<code>Analysis &gt; Differential Expression</code> folder for the study
(<a href="https://www.synapse.org/Synapse:syn17009951" class="external-link">syn17009951</a>):</p>
<div class="codewrapper sourceCode" id="cb71">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">synapser</span><span class="fu">::</span><span class="fu">synLogin</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGetChildren</span><span class="op">(</span><span class="st">'syn17009951'</span><span class="op">)</span><span class="op">$</span><span class="fu">asList</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">tibble</span><span class="op">(</span>entries <span class="op">=</span> <span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unnest_wider</span><span class="op">(</span><span class="va">entries</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span> study <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">name</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">)</span> </span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span></code></pre>
</div>
<p>This shows the files in the synapse directory along with the filename
and the synapse ID for each.</p>
<p>How many are there?</p>
<div class="codewrapper sourceCode" id="cb72">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span></code></pre>
</div>
<p>There are 234 results files in this directory! Let’s not work with
all of them, but instead focus on a subset. To know which ones to use
we’ll need more information about these data. So let’s read the metadata
file for DEG analyses:</p>
<div class="codewrapper sourceCode" id="cb73">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span> <span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="st">'syn17023327'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span> <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">f</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Filename'</span><span class="op">=</span><span class="st">'study'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">m</span><span class="op">)</span></span></code></pre>
</div>
<p>This file gives information about each of the DEG results listed in
the directory above. This information includes the following critical
deails:</p>
<ul><li>
<code>Study_name</code>: GSE or Syn ID denoting the original RNAseq
or microarray data source.<br></li>
<li>
<code>Annotation</code>: The unique, manually-curated annotation for
the DEG set, following the convention:
<code>category_experimental condition_sex_age_brain region_cell type_ transgene</code>.
Let’s break down each of these pieces of information:
<ul><li>
<code>category</code> denotes the human disease being modeled or
“other” (e.g. AD, HD, SCA, ALS)</li>
<li>
<code>experimental condition</code> denotes the specific mouse
genotype or treatment condition</li>
<li>
<code>sex</code>, male or female mice</li>
<li>
<code>age</code>, in months of the animals</li>
<li>
<code>brain region</code> sampled (e.g. hippocampus, spinal
cord)</li>
<li>
<code>cell type</code> sampled (e.g. neuron, microglia)</li>
<li>
<code>transgene</code> for AD models only “APP” vs. “tau” is
noted</li>
<li>If unknown or not applicable for a given analysis, the above fields
are replaced with “na”</li>
</ul></li>
</ul><p>Great! These annotation fields will help us to filter for the data
that we want.</p>
<p>Let’s do a little processing and join these files together:</p>
<div class="codewrapper sourceCode" id="cb74">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Join the list of files from Synapse with the metadata</span></span>
<span><span class="va">wan_mice</span> <span class="op">&lt;-</span> <span class="va">f</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">id</span>, <span class="va">study</span><span class="op">)</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span></span>
<span>    <span class="va">.</span>, </span>
<span>    <span class="va">m</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Filename</span>, <span class="va">Study_name</span>, <span class="va">Annotation</span>, <span class="va">Experimental_condition</span>, <span class="va">Control_condition</span><span class="op">)</span>, </span>
<span>    by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'study'</span><span class="op">=</span><span class="st">'Filename'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># split and re-name the annotation fields</span></span>
<span><span class="va">anno.cols</span> <span class="op">&lt;-</span> <span class="va">wan_mice</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">Annotation</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>anno <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">Annotation</span>,<span class="st">'_'</span>, <span class="fl">7</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>           <span class="fu">as.data.frame</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>           <span class="fu">rename_with</span><span class="op">(</span><span class="op">~</span><span class="fu">c</span><span class="op">(</span><span class="st">'category'</span>,</span>
<span>                          <span class="st">'exptCondition'</span>,</span>
<span>                          <span class="st">'sex'</span>,</span>
<span>                          <span class="st">'age'</span>,</span>
<span>                          <span class="st">'brainRegion'</span>,</span>
<span>                          <span class="st">'cellType'</span>,</span>
<span>                          <span class="st">'transgene'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">unnest_wider</span><span class="op">(</span><span class="va">anno</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="op">-</span><span class="va">Annotation</span>, <span class="op">~</span> <span class="fu">str_trim</span><span class="op">(</span><span class="va">.x</span>, <span class="st">'both'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the annotation columns to the original table</span></span>
<span><span class="va">wan_mice</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">wan_mice</span>, <span class="va">anno.cols</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">relocate</span><span class="op">(</span><span class="va">category</span><span class="op">:</span><span class="va">transgene</span>, .after <span class="op">=</span> <span class="va">Annotation</span><span class="op">)</span></span></code></pre>
</div>
<p>So how many of these studies are categorized as AD models?</p>
<div class="codewrapper sourceCode" id="cb75">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wan_mice</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">category</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarise</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">study</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Only 43 are considered AD models. Let’s take a closer look at the AD
mice category:</p>
<div class="codewrapper sourceCode" id="cb76">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wan_mice</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">category</span> <span class="op">==</span> <span class="st">'AD'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">transgene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarise</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">study</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Over half of the AD mouse model datasets are related to mice carrying
an APP transgene, there are 10 related to mice carrying a Tau transgene,
and 7 in the “other” category. What’s in this other category?</p>
<div class="codewrapper sourceCode" id="cb77">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wan_mice</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">category</span> <span class="op">==</span> <span class="st">'AD'</span>, <span class="va">transgene</span> <span class="op">==</span> <span class="st">'other'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">exptCondition</span>, <span class="va">sex</span>, <span class="va">age</span>, <span class="va">brainRegion</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s select a subset to work with. You can choose your own here, but
I’m going to pick the neurodegenerative disease models (AD, PD, HD,
FTD-ALS, SCA, and CJD). I’m going to filter to only include studies that
look at brain expression (i.e. not sampling cell types), and I’m going
to exclude a couple of other categories (e.g. “spinal cord” and “motor
neurons”). I encourage you to choose your own adventure here.</p>
<div class="codewrapper sourceCode" id="cb78">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wan.data</span> <span class="op">&lt;-</span> <span class="va">wan_mice</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="va">category</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">'AD'</span>, <span class="st">'HD'</span>, <span class="st">'FTD-ALS'</span>, <span class="st">'PD'</span>, <span class="st">'SCA'</span>, <span class="st">'CJD'</span>, <span class="st">'Rett'</span><span class="op">)</span>,</span>
<span>    <span class="va">cellType</span> <span class="op">==</span> <span class="st">'na'</span>,</span>
<span>    <span class="va">brainRegion</span> <span class="op">!=</span> <span class="st">'spinal cord'</span>,</span>
<span>    <span class="va">brainRegion</span> <span class="op">!=</span> <span class="st">'motor neurons'</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>So this should be far fewer than 234 studies</p>
<div class="codewrapper sourceCode" id="cb79">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">wan.data</span><span class="op">)</span></span></code></pre>
</div>
<p>Ok, 87 seems like a more manageable number. Now let’s read in the
data from Synapse. Each DEG results table has similarly labelled columns
– let’s grab the gene IDs, log2FoldChange values, and padj significance
values for each. This step can take a little while.</p>
<div class="codewrapper sourceCode" id="cb80">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">st</span> <span class="op">=</span> <span class="fu">Sys.time</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">wan.data</span><span class="op">$</span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span></span>
<span>  <span class="fl">1</span><span class="op">:</span><span class="fu">nrow</span><span class="op">(</span><span class="va">wan.data</span><span class="op">)</span>,</span>
<span>  <span class="op">~</span> <span class="fu">read_tsv</span><span class="op">(</span> <span class="fu">synapser</span><span class="fu">::</span><span class="fu">synGet</span><span class="op">(</span><span class="va">wan.data</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="va">.x</span><span class="op">]</span><span class="op">)</span><span class="op">$</span><span class="va">path</span>, show_col_types <span class="op">=</span> <span class="cn">F</span> <span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Gene</span>, <span class="va">IsDEG</span>,<span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span></span>
<span>  , .progress <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> </span>
<span><span class="va">ed</span> <span class="op">=</span> <span class="fu">Sys.time</span><span class="op">(</span><span class="op">)</span><span class="op">-</span><span class="va">st</span></span>
<span><span class="fu">print</span><span class="op">(</span><span class="va">ed</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="amp-ad-module-correlation">AMP-AD Module Correlation<a class="anchor" aria-label="anchor" href="#amp-ad-module-correlation"></a></h4>
<p>Like with the LOAD1 study we can start with the correlation to the
co-expression module data. <a href="https://doi.org/10.1016/j.celrep.2020.107908" class="external-link">Wan, et al.</a>
included similar analyses in their results.</p>
<div class="codewrapper sourceCode" id="cb81">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># prepare the module table </span></span>
<span><span class="va">mod.lfc</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_fc</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># prepare the mouse data</span></span>
<span><span class="va">wan.lfc</span> <span class="op">&lt;-</span> <span class="va">wan.data</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">IsDEG</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join the tables together for each comparison</span></span>
<span><span class="va">wan.vs.mod</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">wan.lfc</span>, mod <span class="op">=</span> <span class="va">mod.lfc</span><span class="op">$</span><span class="va">module</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">data</span>, <span class="va">mod</span>,</span>
<span>                <span class="op">~</span> <span class="op">{</span> </span>
<span>                  <span class="va">mm</span> <span class="op">=</span> <span class="va">.x</span></span>
<span>                  <span class="va">hs</span> <span class="op">=</span> <span class="va">mod.lfc</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">module</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> </span>
<span>                  <span class="fu">inner_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="st">'symbol'</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span> </span>
<span>                <span class="op">}</span></span>
<span>            <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Compute the correlation</p>
<div class="codewrapper sourceCode" id="cb82">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">wan.vs.mod</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'ampad_fc'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'log2FoldChange'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.adj <span class="op">=</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">p.value</span>, method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span>,</span>
<span>    significant <span class="op">=</span> <span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">module_clusters</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'mod'</span><span class="op">=</span><span class="st">'module'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Plot the correlation</p>
<div class="codewrapper sourceCode" id="cb83">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model_name <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">category</span>,<span class="st">'] '</span>,<span class="va">study</span>,<span class="st">': '</span>, <span class="va">exptCondition</span>, <span class="st">' ('</span>, <span class="va">transgene</span>,<span class="st">')'</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">mod</span>, <span class="va">model_name</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">cluster_label</span><span class="op">)</span>, switch <span class="op">=</span> <span class="st">'y'</span>, scales<span class="op">=</span><span class="st">'free'</span>,space<span class="op">=</span> <span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, </span>
<span>             stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'left'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div id="challenge-7" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-7" class="callout-inner">
<h3 class="callout-title">Challenge 7</h3>
<div class="callout-content">
<p>This is a lot of information! Take a minute to explore the plot and
discuss with your neighbor what trends you notice. We’ll come back to
discuss as a group.</p>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7"> Show me the solution </h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" aria-labelledby="headingSolution7" data-bs-parent="#accordionSolution7">
<div class="accordion-body">
<p>Some things I notice: (1) The <code>AD</code> models are fairly
similar and many show strong positive correlation to clusters A and B
(2) The <code>HD</code> models show negative correlations to clusters A
and B, but have the strongest correlations with cluster E (3) The
<code>SCA</code> models tend to look a bit like the <code>AD</code>
models</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="biodomain-correlation">Biodomain Correlation<a class="anchor" aria-label="anchor" href="#biodomain-correlation"></a></h4>
<p>We can also perform the correlation by biodomain genes. Here we join
up the whole cohort</p>
<div class="codewrapper sourceCode" id="cb84">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs.lfc</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>, </span>
<span>    <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span>,</span>
<span>    <span class="va">Tissue</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">'PHG'</span>,<span class="st">'TCX'</span>,<span class="st">'DLPFC'</span><span class="op">)</span>,</span>
<span>    <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">mouse_symbol</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">mouse_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nest</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">hgnc_symbol</span>, <span class="va">symbol</span>, <span class="va">logFC</span>, <span class="va">adj.P.Val</span><span class="op">)</span>, </span>
<span>    .by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">Study</span>, <span class="va">Tissue</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">wan.lfc</span> <span class="op">&lt;-</span> <span class="va">wan.data</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">IsDEG</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join the tables together for each comparison</span></span>
<span><span class="va">wan.vs.amp</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">wan.lfc</span>, amp <span class="op">=</span> <span class="va">hs.lfc</span><span class="op">$</span><span class="va">Study</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">data</span>, <span class="va">amp</span>,</span>
<span>                <span class="op">~</span> <span class="op">{</span> </span>
<span>                  <span class="va">mm</span> <span class="op">=</span> <span class="va">.x</span></span>
<span>                  <span class="va">hs</span> <span class="op">=</span> <span class="va">hs.lfc</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Study</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> </span>
<span>                  <span class="fu">inner_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="st">'symbol'</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span> </span>
<span>                <span class="op">}</span></span>
<span>            <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb85">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">wan.vs.amp</span>, set <span class="op">=</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span></span>
<span>      <span class="va">data</span>, <span class="va">set</span>, </span>
<span>      <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span> <span class="va">symbol</span> <span class="op">%in%</span> <span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">genes</span><span class="op">[[</span><span class="fu">which</span><span class="op">(</span><span class="va">bd.gene.lists</span><span class="op">$</span><span class="va">set</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op">)</span> <span class="op">)</span>,</span>
<span>    n.genes <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="fu">nrow</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">n.genes</span> <span class="op">&gt;</span> <span class="fl">3</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'logFC'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'log2FoldChange'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.adj <span class="op">=</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">p.value</span>, method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span>,</span>
<span>    significant <span class="op">=</span> <span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb86">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">F</span>, <span class="va">amp</span> <span class="op">==</span> <span class="st">'MSSM'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model_name <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">category</span>,<span class="st">'] '</span>,<span class="va">study</span>,<span class="st">': '</span>, <span class="va">exptCondition</span>, <span class="st">' ('</span>, <span class="va">transgene</span>,<span class="st">')'</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">ord</span> <span class="op">&lt;-</span> <span class="va">tmp</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">==</span> <span class="st">'Lipid Metabolism'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">model_name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">tmp</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model_name <span class="op">=</span> <span class="fu">fct_relevel</span><span class="op">(</span><span class="va">model_name</span>, <span class="va">ord</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set</span>, <span class="va">model_name</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">amp</span><span class="op">)</span>, switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, </span>
<span>             stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="subtypes">subtypes<a class="anchor" aria-label="anchor" href="#subtypes"></a></h4>
<div class="codewrapper sourceCode" id="cb87">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># some variable renaming</span></span>
<span><span class="va">stype.effects</span> <span class="op">&lt;-</span> <span class="va">ampad_subtype_fc</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">neff.effects</span> <span class="op">&lt;-</span> <span class="va">neff_subtype_fc</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Just the ROSMAPs</span></span>
<span><span class="co"># stype.effects.RM &lt;- stype.effects %&gt;% filter(grepl('Mayo', subtype))</span></span>
<span><span class="va">stype.effects.RM</span> <span class="op">&lt;-</span> <span class="va">neff.effects</span></span>
<span></span>
<span><span class="va">wan.lfc</span> <span class="op">&lt;-</span> <span class="va">wan.data</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">IsDEG</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join the tables together for each comparison</span></span>
<span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">wan.lfc</span>, stype <span class="op">=</span> <span class="va">stype.effects.RM</span><span class="op">$</span><span class="va">subtype</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">data</span>, <span class="va">stype</span>,</span>
<span>                <span class="op">~</span> <span class="op">{</span> </span>
<span>                  <span class="va">mm</span> <span class="op">=</span> <span class="va">.x</span> </span>
<span>                  <span class="va">hs</span> <span class="op">=</span> <span class="va">stype.effects.RM</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">subtype</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> </span>
<span>                  <span class="fu">inner_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="st">'symbol'</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span></span>
<span>                <span class="op">}</span></span>
<span>            <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'ampad_fc'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'log2FoldChange'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.adj <span class="op">=</span> <span class="fu">p.adjust</span><span class="op">(</span><span class="va">p.value</span>, method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span>,</span>
<span>    significant <span class="op">=</span> <span class="va">p.adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb88">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model_name <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">category</span>,<span class="st">'] '</span>,<span class="va">study</span>,<span class="st">': '</span>, <span class="va">exptCondition</span>, <span class="st">' ('</span>, <span class="va">transgene</span>,<span class="st">')'</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">ord</span> <span class="op">&lt;-</span> <span class="va">tmp</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">stype</span> <span class="op">==</span> <span class="st">'subtype C1'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">model_name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">tmp</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model_name <span class="op">=</span> <span class="fu">fct_relevel</span><span class="op">(</span><span class="va">model_name</span>, <span class="va">ord</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">stype</span>, <span class="va">model_name</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="co"># facet_grid(rows = vars(Annotation), switch = 'y')+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>, </span>
<span>             stroke <span class="op">=</span> <span class="fl">1.2</span>,shape <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Correlation"</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                         guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a></h2>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul><li>AD is complex and human patient populations are heterogenous</li>
<li>Molecular subtyping is one way to reduce that complexity</li>
<li>We can use identified human subtypes to help characterize mouse
models</li>
<li>Each existing mouse model of AD may match to a particular subset of
human AD subtypes but not all subtypes simultaneously.</li>
<li>Most of the molecular AD subtypes were identified using
transcriptomics data and there seems to be similarities between AD
subtypes across studies.</li>
</ul></div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb89">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb89-2"><a href="#cb89-2" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">5.2</span> (<span class="dv">2025-10-31</span>)</span>
<span id="cb89-3"><a href="#cb89-3" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>pc<span class="sc">-</span>linux<span class="sc">-</span>gnu</span>
<span id="cb89-4"><a href="#cb89-4" tabindex="-1"></a>Running under<span class="sc">:</span> Ubuntu <span class="dv">24</span>.<span class="fl">04.3</span> LTS</span>
<span id="cb89-5"><a href="#cb89-5" tabindex="-1"></a></span>
<span id="cb89-6"><a href="#cb89-6" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb89-7"><a href="#cb89-7" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>x86_64<span class="sc">-</span>linux<span class="sc">-</span>gnu<span class="sc">/</span>openblas<span class="sc">-</span>pthread<span class="sc">/</span>libblas.so<span class="fl">.3</span> </span>
<span id="cb89-8"><a href="#cb89-8" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>x86_64<span class="sc">-</span>linux<span class="sc">-</span>gnu<span class="sc">/</span>openblas<span class="sc">-</span>pthread<span class="sc">/</span>libopenblasp<span class="sc">-</span>r0.<span class="dv">3</span>.<span class="fl">26.</span>so;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb89-9"><a href="#cb89-9" tabindex="-1"></a></span>
<span id="cb89-10"><a href="#cb89-10" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb89-11"><a href="#cb89-11" tabindex="-1"></a> [<span class="dv">1</span>] LC_CTYPE<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>       LC_NUMERIC<span class="ot">=</span>C              </span>
<span id="cb89-12"><a href="#cb89-12" tabindex="-1"></a> [<span class="dv">3</span>] LC_TIME<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>        LC_COLLATE<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>    </span>
<span id="cb89-13"><a href="#cb89-13" tabindex="-1"></a> [<span class="dv">5</span>] LC_MONETARY<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>    LC_MESSAGES<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>   </span>
<span id="cb89-14"><a href="#cb89-14" tabindex="-1"></a> [<span class="dv">7</span>] LC_PAPER<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>       LC_NAME<span class="ot">=</span>C                 </span>
<span id="cb89-15"><a href="#cb89-15" tabindex="-1"></a> [<span class="dv">9</span>] LC_ADDRESS<span class="ot">=</span>C               LC_TELEPHONE<span class="ot">=</span>C            </span>
<span id="cb89-16"><a href="#cb89-16" tabindex="-1"></a>[<span class="dv">11</span>] LC_MEASUREMENT<span class="ot">=</span>en_US.UTF<span class="dv">-8</span> LC_IDENTIFICATION<span class="ot">=</span>C       </span>
<span id="cb89-17"><a href="#cb89-17" tabindex="-1"></a></span>
<span id="cb89-18"><a href="#cb89-18" tabindex="-1"></a>time zone<span class="sc">:</span> Etc<span class="sc">/</span>UTC</span>
<span id="cb89-19"><a href="#cb89-19" tabindex="-1"></a>tzcode source<span class="sc">:</span> <span class="fu">system</span> (glibc)</span>
<span id="cb89-20"><a href="#cb89-20" tabindex="-1"></a></span>
<span id="cb89-21"><a href="#cb89-21" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb89-22"><a href="#cb89-22" tabindex="-1"></a>[<span class="dv">1</span>] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span id="cb89-23"><a href="#cb89-23" tabindex="-1"></a>[<span class="dv">8</span>] base     </span>
<span id="cb89-24"><a href="#cb89-24" tabindex="-1"></a></span>
<span id="cb89-25"><a href="#cb89-25" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb89-26"><a href="#cb89-26" tabindex="-1"></a> [<span class="dv">1</span>] gt_1.<span class="fl">3.0</span>               corrplot_0<span class="fl">.95</span>          lubridate_1.<span class="fl">9.4</span>       </span>
<span id="cb89-27"><a href="#cb89-27" tabindex="-1"></a> [<span class="dv">4</span>] forcats_1.<span class="fl">0.1</span>          stringr_1.<span class="fl">6.0</span>          purrr_1.<span class="fl">2.1</span>           </span>
<span id="cb89-28"><a href="#cb89-28" tabindex="-1"></a> [<span class="dv">7</span>] readr_2.<span class="fl">1.6</span>            tidyr_1.<span class="fl">3.2</span>            tibble_3.<span class="fl">3.1</span>          </span>
<span id="cb89-29"><a href="#cb89-29" tabindex="-1"></a>[<span class="dv">10</span>] tidyverse_2.<span class="fl">0.0</span>        data.table_1.<span class="dv">18</span>.<span class="fl">2.1</span>    clusterProfiler_4.<span class="fl">18.4</span></span>
<span id="cb89-30"><a href="#cb89-30" tabindex="-1"></a>[<span class="dv">13</span>] dplyr_1.<span class="fl">1.4</span>            cowplot_1.<span class="fl">2.0</span>          GO.db_3.<span class="fl">22.0</span>          </span>
<span id="cb89-31"><a href="#cb89-31" tabindex="-1"></a>[<span class="dv">16</span>] org.Hs.eg.db_3.<span class="fl">22.0</span>    org.Mm.eg.db_3.<span class="fl">22.0</span>    AnnotationDbi_1.<span class="fl">72.0</span>  </span>
<span id="cb89-32"><a href="#cb89-32" tabindex="-1"></a>[<span class="dv">19</span>] IRanges_2.<span class="fl">44.0</span>         S4Vectors_0.<span class="fl">48.0</span>       Biobase_2.<span class="fl">70.0</span>        </span>
<span id="cb89-33"><a href="#cb89-33" tabindex="-1"></a>[<span class="dv">22</span>] BiocGenerics_0.<span class="fl">56.0</span>    generics_0.<span class="fl">1.4</span>         ggplot2_4.<span class="fl">0.1</span>         </span>
<span id="cb89-34"><a href="#cb89-34" tabindex="-1"></a></span>
<span id="cb89-35"><a href="#cb89-35" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb89-36"><a href="#cb89-36" tabindex="-1"></a>  [<span class="dv">1</span>] DBI_1.<span class="fl">2.3</span>               gson_0.<span class="fl">1.0</span>              rlang_1.<span class="fl">1.7</span>            </span>
<span id="cb89-37"><a href="#cb89-37" tabindex="-1"></a>  [<span class="dv">4</span>] magrittr_2.<span class="fl">0.4</span>          DOSE_4.<span class="fl">4.0</span>              otel_0.<span class="fl">2.0</span>             </span>
<span id="cb89-38"><a href="#cb89-38" tabindex="-1"></a>  [<span class="dv">7</span>] compiler_4.<span class="fl">5.2</span>          RSQLite_2.<span class="fl">4.5</span>           png_0<span class="fl">.1</span><span class="dv">-8</span>              </span>
<span id="cb89-39"><a href="#cb89-39" tabindex="-1"></a> [<span class="dv">10</span>] systemfonts_1.<span class="fl">3.1</span>       vctrs_0.<span class="fl">7.1</span>             reshape2_1.<span class="fl">4.5</span>         </span>
<span id="cb89-40"><a href="#cb89-40" tabindex="-1"></a> [<span class="dv">13</span>] pkgconfig_2.<span class="fl">0.3</span>         crayon_1.<span class="fl">5.3</span>            fastmap_1.<span class="fl">2.0</span>          </span>
<span id="cb89-41"><a href="#cb89-41" tabindex="-1"></a> [<span class="dv">16</span>] XVector_0.<span class="fl">50.0</span>          tzdb_0.<span class="fl">5.0</span>              enrichplot_1.<span class="fl">30.4</span>      </span>
<span id="cb89-42"><a href="#cb89-42" tabindex="-1"></a> [<span class="dv">19</span>] bit_4.<span class="fl">6.0</span>               xfun_0<span class="fl">.56</span>               cachem_1.<span class="fl">1.0</span>           </span>
<span id="cb89-43"><a href="#cb89-43" tabindex="-1"></a> [<span class="dv">22</span>] aplot_0.<span class="fl">2.9</span>             jsonlite_2.<span class="fl">0.0</span>          blob_1.<span class="fl">3.0</span>             </span>
<span id="cb89-44"><a href="#cb89-44" tabindex="-1"></a> [<span class="dv">25</span>] tidydr_0.<span class="fl">0.6</span>            tweenr_2.<span class="fl">0.3</span>            BiocParallel_1.<span class="fl">44.0</span>    </span>
<span id="cb89-45"><a href="#cb89-45" tabindex="-1"></a> [<span class="dv">28</span>] cluster_2.<span class="dv">1</span>.<span class="fl">8.1</span>         parallel_4.<span class="fl">5.2</span>          R6_2.<span class="fl">6.1</span>               </span>
<span id="cb89-46"><a href="#cb89-46" tabindex="-1"></a> [<span class="dv">31</span>] stringi_1.<span class="fl">8.7</span>           RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>      GOSemSim_2.<span class="fl">36.0</span>        </span>
<span id="cb89-47"><a href="#cb89-47" tabindex="-1"></a> [<span class="dv">34</span>] Rcpp_1.<span class="fl">1.1</span>              Seqinfo_1.<span class="fl">0.0</span>           knitr_1<span class="fl">.51</span>             </span>
<span id="cb89-48"><a href="#cb89-48" tabindex="-1"></a> [<span class="dv">37</span>] ggtangle_0.<span class="fl">1.1</span>          R.utils_2.<span class="fl">13.0</span>          timechange_0.<span class="fl">4.0</span>       </span>
<span id="cb89-49"><a href="#cb89-49" tabindex="-1"></a> [<span class="dv">40</span>] Matrix_1<span class="fl">.7</span><span class="dv">-4</span>            splines_4.<span class="fl">5.2</span>           igraph_2.<span class="fl">2.1</span>           </span>
<span id="cb89-50"><a href="#cb89-50" tabindex="-1"></a> [<span class="dv">43</span>] tidyselect_1.<span class="fl">2.1</span>        qvalue_2.<span class="fl">42.0</span>           yaml_2.<span class="fl">3.12</span>            </span>
<span id="cb89-51"><a href="#cb89-51" tabindex="-1"></a> [<span class="dv">46</span>] codetools_0<span class="fl">.2</span><span class="dv">-20</span>        lattice_0<span class="fl">.22</span><span class="dv">-7</span>          plyr_1.<span class="fl">8.9</span>             </span>
<span id="cb89-52"><a href="#cb89-52" tabindex="-1"></a> [<span class="dv">49</span>] treeio_1.<span class="fl">34.0</span>           withr_3.<span class="fl">0.2</span>             KEGGREST_1.<span class="fl">50.0</span>        </span>
<span id="cb89-53"><a href="#cb89-53" tabindex="-1"></a> [<span class="dv">52</span>] S7_0.<span class="fl">2.1</span>                evaluate_1.<span class="fl">0.5</span>          gridGraphics_0<span class="fl">.5</span><span class="dv">-1</span>     </span>
<span id="cb89-54"><a href="#cb89-54" tabindex="-1"></a> [<span class="dv">55</span>] polyclip_1<span class="fl">.10</span><span class="dv">-7</span>         scatterpie_0.<span class="fl">2.6</span>        xml2_1.<span class="fl">5.2</span>             </span>
<span id="cb89-55"><a href="#cb89-55" tabindex="-1"></a> [<span class="dv">58</span>] Biostrings_2.<span class="fl">78.0</span>       pillar_1.<span class="fl">11.1</span>           BiocManager_1.<span class="fl">30.27</span>    </span>
<span id="cb89-56"><a href="#cb89-56" tabindex="-1"></a> [<span class="dv">61</span>] ggtree_4.<span class="fl">0.4</span>            renv_1.<span class="fl">1.7</span>              ggfun_0.<span class="fl">2.0</span>            </span>
<span id="cb89-57"><a href="#cb89-57" tabindex="-1"></a> [<span class="dv">64</span>] hms_1.<span class="fl">1.4</span>               scales_1.<span class="fl">4.0</span>            tidytree_0.<span class="fl">4.7</span>         </span>
<span id="cb89-58"><a href="#cb89-58" tabindex="-1"></a> [<span class="dv">67</span>] glue_1.<span class="fl">8.0</span>              gdtools_0.<span class="fl">4.4</span>           lazyeval_0.<span class="fl">2.2</span>         </span>
<span id="cb89-59"><a href="#cb89-59" tabindex="-1"></a> [<span class="dv">70</span>] tools_4.<span class="fl">5.2</span>             ggnewscale_0.<span class="fl">5.2</span>        fgsea_1.<span class="fl">36.2</span>           </span>
<span id="cb89-60"><a href="#cb89-60" tabindex="-1"></a> [<span class="dv">73</span>] ggiraph_0.<span class="fl">9.3</span>           fs_1.<span class="fl">6.6</span>                fastmatch_1<span class="fl">.1</span><span class="dv">-8</span>        </span>
<span id="cb89-61"><a href="#cb89-61" tabindex="-1"></a> [<span class="dv">76</span>] grid_4.<span class="fl">5.2</span>              ape_5<span class="fl">.8</span><span class="dv">-1</span>               nlme_3<span class="fl">.1</span><span class="dv">-168</span>           </span>
<span id="cb89-62"><a href="#cb89-62" tabindex="-1"></a> [<span class="dv">79</span>] patchwork_1.<span class="fl">3.2</span>         ggforce_0.<span class="fl">5.0</span>           cli_3.<span class="fl">6.5</span>              </span>
<span id="cb89-63"><a href="#cb89-63" tabindex="-1"></a> [<span class="dv">82</span>] rappdirs_0.<span class="fl">3.4</span>          fontBitstreamVera_0.<span class="fl">1.1</span> gtable_0.<span class="fl">3.6</span>           </span>
<span id="cb89-64"><a href="#cb89-64" tabindex="-1"></a> [<span class="dv">85</span>] R.methodsS3_1.<span class="fl">8.2</span>       yulab.utils_0.<span class="fl">2.4</span>       sass_0.<span class="fl">4.10</span>            </span>
<span id="cb89-65"><a href="#cb89-65" tabindex="-1"></a> [<span class="dv">88</span>] digest_0.<span class="fl">6.39</span>           fontquiver_0.<span class="fl">2.1</span>        ggrepel_0.<span class="fl">9.6</span>          </span>
<span id="cb89-66"><a href="#cb89-66" tabindex="-1"></a> [<span class="dv">91</span>] ggplotify_0.<span class="fl">1.3</span>         htmlwidgets_1.<span class="fl">6.4</span>       farver_2.<span class="fl">1.2</span>           </span>
<span id="cb89-67"><a href="#cb89-67" tabindex="-1"></a> [<span class="dv">94</span>] memoise_2.<span class="fl">0.1</span>           htmltools_0.<span class="fl">5.9</span>         R.oo_1.<span class="fl">27.1</span>            </span>
<span id="cb89-68"><a href="#cb89-68" tabindex="-1"></a> [<span class="dv">97</span>] lifecycle_1.<span class="fl">0.5</span>         httr_1.<span class="fl">4.7</span>              MASS_7<span class="fl">.3</span><span class="dv">-65</span>            </span>
<span id="cb89-69"><a href="#cb89-69" tabindex="-1"></a>[<span class="dv">100</span>] fontLiberation_0.<span class="fl">1.0</span>    bit64_4.<span class="fl">6.0</span><span class="dv">-1</span>          </span></code></pre>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</div>



      </div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/mouse-human-alignment.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/index.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/mouse-human-alignment.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Cross Species
        </a>
        <a class="chapter-link float-end" href="../instructor/index.html" rel="next">
          Home
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/asliuyar/translational-omics-AD/edit/main/episodes/choosing_the_right_model.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/asliuyar/translational-omics-AD/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/asliuyar/translational-omics-AD/" class="external-link">Source</a></p>
        <p>
        <a href="../citation.html">Cite</a>
        | <a href="mailto:">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/" class="external-link">sandpaper (0.18.1)</a>, <a href="https://github.com/carpentries/pegboard/tree/" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/" class="external-link">varnish (1.0.9)</a></p>
		</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://asliuyar.github.io/translational-omics-AD/instructor/choosing_the_right_model.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "Alzheimers-disease, cross-species-translation, omics, AD-Knowledge-Portal",
  "name": "Choosing the right animal models",
  "creativeWorkStatus": "active",
  "url": "https://asliuyar.github.io/translational-omics-AD/instructor/choosing_the_right_model.html",
  "identifier": "https://asliuyar.github.io/translational-omics-AD/instructor/choosing_the_right_model.html",
  "dateCreated": "2024-10-02",
  "dateModified": "2026-02-12",
  "datePublished": "2026-02-12"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

