<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Translational Omics in Alzheimers Disease: Cross Species Functional Alignment</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png"><link rel="manifest" href="../favicons/incubator/site.webmanifest"><link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>


      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../mouse-human-alignment.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Translational Omics in Alzheimers Disease
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Translational Omics in Alzheimers Disease
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Translational Omics in Alzheimers Disease
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 23%" class="percentage">
    23%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 23%" aria-valuenow="23" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../mouse-human-alignment.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Introduction to Omics Translation in AD</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="exploring-data-in-synapse.html">2. Synapse and AD Knowledge Portal</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="differential-expression-analysis.html">3. Differential expression analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        4. Cross Species Functional Alignment
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#setup">Setup</a></li>
<li><a href="#aligning-human-and-mouse-phenotypes">[1] Aligning Human and Mouse Phenotypes</a></li>
<li><a href="#differential-expression-analysis-in-ad-mouse-models">[2] Differential Expression Analysis in AD Mouse Models</a></li>
<li><a href="#overview-of-human-cohort-data">[3] Overview of Human cohort data</a></li>
<li><a href="#correlation-between-mouse-models-and-human-ad-modules">[4] Correlation between mouse models and human AD modules</a></li>
<li><a href="#detecting-functional-coherence-of-gene-sets-from-omics-data">[5] Detecting functional coherence of gene sets from omics data</a></li>
<li><a href="#alzheimerss-disease-biological-domains">[6] Alzheimers’s Disease Biological Domains</a></li>
<li><a href="#conclusion">[7] Conclusion:</a></li>
<li><a href="#session-info">Session Info</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="choosing_the_right_model.html">5. Choosing the right animal models</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/differential-expression-analysis.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/choosing_the_right_model.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/differential-expression-analysis.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Differential
        </a>
        <a class="chapter-link float-end" href="../instructor/choosing_the_right_model.html" rel="next">
          Next: Choosing the right...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Cross Species Functional Alignment</h1>
        <p>Last updated on 2026-02-12 |

        <a href="https://github.com/asliuyar/translational-omics-AD/edit/main/episodes/mouse-human-alignment.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 180 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How do we perfrom a cross-species comparison?</li>
<li>What transcriptomic changes do we observe in mouse models?</li>
<li>Which aspects of disease does a model capture?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Approaches to align mouse data to human data</li>
<li>Understand the human AD co-expression modules</li>
<li>Understand the data from AD mouse models</li>
<li>Perform differential analysis using DESeq2</li>
<li>Perform correlation analysis between mouse models and human
modules</li>
<li>Understand the biological domains and subdomains of AD</li>
<li>Use domain annotations to compare between species</li>
</ul></div>
</div>
</div>
</div>
</div>
<p>Author: Gregory Cary, Jackson Laboratory</p>
<section><h2 class="section-heading" id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a></h2>
<hr class="half-width"><p>First, let’s install the necessary packages into the workspace,
starting with the annotation packages:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Mouse and Human annotation databases</span></span>
<span><span class="co"># BiocManager::install(c('org.Mm.eg.db','org.Hs.eg.db', 'GO.db'))</span></span></code></pre>
</div>
<p>Next we’ll install the <code>fgsea</code> and
<code>clusterProfiler</code> packages, which will be used for GO term
enrichment.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># fgsea package</span></span>
<span><span class="co"># BiocManager::install(c('fgsea', 'clusterProfiler'))</span></span></code></pre>
</div>
<p>Finally we can instally the <code>synapseclient</code> python package
using the <code>reticulate</code> R package, this will enable us
comand-line access to objects in the Synapse data repository.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># synapseclient python package</span></span>
<span><span class="co"># reticulate::py_install('synapseclient')</span></span></code></pre>
</div>
<p>Finally we’ll need to generate a personal access token on Synapse.
Login to Synapse, go to
<code>Your Account &gt; Account Settings &gt; Personal Access Tokens &gt; Manage Personal Access Tokens</code>.
Click on <code>Create New Token</code> and make sure to enable both View
and Download permissions. Paste the resulting PAT in the quotes below
and save it to your workspace.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># synToken &lt;- </span></span></code></pre>
</div>
<p><strong>NOTE</strong>: This is for learning purposes only. Storing
your PAT in an environmental variable in your workspace is not secure
and you could easily loose access to the PAT. The preferred method is to
store it in a separate file called <code>.synapseConfig</code> under
your home directory.</p>
<p>Now we should be able to login to Synapse through our R session using
the following commands:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># import the synapseclient python package</span></span>
<span><span class="va">syn.client</span> <span class="op">&lt;-</span> <span class="fu">reticulate</span><span class="fu">::</span><span class="fu">import</span><span class="op">(</span><span class="st">'synapseclient'</span><span class="op">)</span></span>
<span><span class="va">syn</span>        <span class="op">&lt;-</span> <span class="va">syn.client</span><span class="op">$</span><span class="fu">Synapse</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># log in to Synapse</span></span>
<span><span class="co"># syn$login('', authToken = synToken)</span></span></code></pre>
</div>
<p>Finally, let’s load the R packages we’ll need for today’s lesson:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load necessary libraries for the analysis</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">DESeq2</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">clusterProfiler</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">fgsea</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span> <span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># set ggplot plotting theme (personal preference)</span></span>
<span><span class="fu">theme_set</span><span class="op">(</span> <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="aligning-human-and-mouse-phenotypes">[1] Aligning Human and Mouse Phenotypes<a class="anchor" aria-label="anchor" href="#aligning-human-and-mouse-phenotypes"></a></h2>
<hr class="half-width"><p>Alzheimer’s Disease (AD) is complex, and we can not expect any single
animal model to fully recapitulate all aspects of late onset AD (LOAD)
pathology. To study AD with animal models we must find dimensions
through which we can align phenotypes between the models and human
cohorts. In MODEL-AD we use the following data modalities to identify
commonalities between mouse models and human cohorts:</p>
<ol style="list-style-type: decimal"><li>Imaging (i.e. MRI and PET) to correspond with human imaging studies
(e.g. <a href="https://adni.loni.usc.edu/" class="external-link">ADNI</a>)<br></li>
<li>Neuropathological and biomarker phenotypes<br></li>
<li>
<em>Lots</em> of ’omics — genomics, proteomics, and
metabolomics</li>
</ol><p>The ’omics comparisons allow for very rich comparisons because a
significant proportion of genes are shared between these two species.
Furthermore, homology at the anatomical and neuropathological levels is
less clear.</p>
<figure><img src="../fig/dimensions_of_model_alignment.png" alt="image showing humans and mice along with neuroimaging, neuropathology, and heatmap images representing the different modalities to align." class="figure mx-auto d-block"><div class="figcaption">Aligning Human and Mouse Phenotypes</div>
</figure><p>In this session we will explore several ways to compare ’omics
signatures between human AD patients and mouse models. We’ll focus on
transcriptomic alignment for this session, but we’ll consider other
modalities in later sessions. We’ll consider several different
approaches to compare gene expression between human cohorts and model
systems:</p>
<ol style="list-style-type: decimal"><li>Correlation of genes within human co-expression modules<br></li>
</ol><ul><li>correlations will be generally weak for all expression, but animal
models may recapitulate specific aspects of the disease<br></li>
<li>we can use subsets of genes from co-expression modules, which
represent genes expressed in similar patterns in AD, and look for
correlations within these subsets<br></li>
</ul><ol start="2" style="list-style-type: decimal"><li>Correlation of functional enrichment results<br></li>
</ol><ul><li>another approach is to consider the functional annotation enriched
among differentially expressed genes in human and mouse.</li>
<li>we can similarly sub-divide these groups of co-functional genes into
biological domains to aid our interpretation</li>
</ul><p>Let’s start by briefly reviewing how to assess differential
expression in our mouse RNA-seq datasets using the <code>DESeq2</code>
package. Then we’ll move on to discussing the background of the human
cohorts and co-expression modules.</p>
</section><section><h2 class="section-heading" id="differential-expression-analysis-in-ad-mouse-models">[2] Differential Expression Analysis in AD Mouse Models<a class="anchor" aria-label="anchor" href="#differential-expression-analysis-in-ad-mouse-models"></a></h2>
<hr class="half-width"><p>Let’s analyze the 5xFAD RNA-seq expression data we explored
yesterday. Specifically, we want to know which genes are differentially
expressed at each age as a result of the transgenes that constitute the
5xFAD model.</p>
<p>The 5xFAD mouse model is a 5x transgenic model consisting of mutatnt
human transgenes of the amyloid precursor protein (APP) and presenilin 1
(PSEN1) genes. The specific variants are all causal variants for
Familial Alzheimer’s Disease (FAD) and include three variants in the APP
gene - Swedish (K670N, M671L), Florida (I716V), and London (V717I) - and
two in the PSEN1 gene - M146L and L286V. The expression of both
transgenes is under control of the neural-specific elements of the mouse
Thy1 promoter, which drives overexpression of the transgenes in the
brain. More information about this generation and maintenance of this
strain can be obtained from the <a href="https://www.jax.org/strain/008730" class="external-link">JAX strain catalog</a>.</p>
<p>This model has been extensively characterized by the MODEL-AD
consortium, and others, including the study that we explored yesterday
(i.e. <a href="https://www.synapse.org/Synapse:syn21983020" class="external-link">syn21983020</a>). For
more information about this specific MODEL-AD study, see the publication
by <a href="https://pubmed.ncbi.nlm.nih.gov/34366832/" class="external-link">Oblak et al
(2021)</a>. The primary patho-physiological phenotypes are summarized in
the figure below from <a href="https://www.alzforum.org/research-models/5xfad-b6sjl" class="external-link">AlzForum</a>,
and include (1) early deposition of amyloid plaques, (2) gliosis and
neuroinflammation, (3) synaptic changes and cognitive impairment, and
(4) neuronal loss, specifically in cortical layer V and the subiculum.
Importantly, Tau tangles are absent from this model.</p>
<figure><img src="../fig/5xFAD_phenotypes.png" alt="list of AD phenotypes associated with 5xFAD mouse model" class="figure mx-auto d-block"><div class="figcaption">5xFAD Mouse Phenotypes</div>
</figure><p>But what <strong>else</strong> can the RNA-seq data tell us about the
transcriptomic response to the 5xFAD model in the brain? To know more,
we need to assess the differentially expressed transcripts. We’ll use
the <code>DESeq2</code> package to perform differential expression
analysis of the 5xFAD RNA-seq data.</p>
<div class="section level3">
<h3 id="read-5xfad-rna-seq-count-data">read 5xFAD RNA-seq count data<a class="anchor" aria-label="anchor" href="#read-5xfad-rna-seq-count-data"></a></h3>
<p>First, let’s re-access the RNA-seq data and metadata from Synapse</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># RNA-seq counts</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">'syn22108847'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">$</span><span class="va">path</span> <span class="op">%&gt;%</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biospecimen  metadata</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">'syn22103213'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">$</span><span class="va">path</span> <span class="op">%&gt;%</span> <span class="fu">read_csv</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">individualID</span>, <span class="va">specimenID</span><span class="op">)</span> </span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="st">"data/htseqcounts_5XFAD.txt"</span>, </span>
<span>                   show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># individual metadata</span></span>
<span><span class="va">ind_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_individual_metadata.csv"</span>, </span>
<span>                     show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biospecimen metadata</span></span>
<span><span class="va">bio_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_biospecimen_metadata.csv"</span>, </span>
<span>                     show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># assay metadata</span></span>
<span><span class="va">rna_meta</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/Jax.IU.Pitt_5XFAD_assay_RNAseq_metadata.csv"</span>, </span>
<span>                     show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># individual metadata, joined to the above</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">'syn22103212'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">$</span><span class="va">path</span> <span class="op">%&gt;%</span> <span class="fu">read_csv</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">meta</span>, by <span class="op">=</span> <span class="st">'individualID'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">specimenID</span><span class="op">)</span><span class="op">)</span> </span></code></pre>
</div>
<p>We can modify the metadata to only include covariates we’ll need for
this analysis</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># order rows that have corresponding IDs in the counts table</span></span>
<span><span class="va">covars</span> <span class="op">&lt;-</span> <span class="va">meta</span> <span class="op">%&gt;%</span> <span class="fu">slice</span><span class="op">(</span> <span class="fu">match</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="va">specimenID</span><span class="op">)</span> <span class="op">)</span> </span>
<span></span>
<span><span class="co"># compute the age of animals in months</span></span>
<span><span class="va">covars</span> <span class="op">&lt;-</span> <span class="va">covars</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    dateBirth <span class="op">=</span> <span class="fu">mdy</span><span class="op">(</span><span class="va">dateBirth</span><span class="op">)</span>,</span>
<span>    dateDeath <span class="op">=</span> <span class="fu">mdy</span><span class="op">(</span><span class="va">dateDeath</span><span class="op">)</span>,</span>
<span>    age <span class="op">=</span> <span class="fu">interval</span><span class="op">(</span><span class="va">dateBirth</span>,<span class="va">dateDeath</span><span class="op">)</span> <span class="op">%/%</span> <span class="fu">months</span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># change the group variable based on the animal genotype</span></span>
<span><span class="va">covars</span> <span class="op">&lt;-</span> <span class="va">covars</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span> group <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">genotype</span> <span class="op">==</span> <span class="st">'5XFAD_carrier'</span>, <span class="st">'5xFAD'</span>, <span class="st">'WT'</span><span class="op">)</span> <span class="op">)</span></span>
<span>    </span>
<span><span class="co"># finally, only keep the columns we'll need</span></span>
<span><span class="va">covars</span> <span class="op">&lt;-</span> <span class="va">covars</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">specimenID</span>, <span class="va">group</span>, <span class="va">sex</span>, <span class="va">age</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">covars</span><span class="op">)</span></span></code></pre>
</div>
<p>First, let’s make sure we have all relevant metadata</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">all</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">==</span><span class="va">covars</span><span class="op">$</span><span class="va">specimenID</span><span class="op">)</span></span></code></pre>
</div>
<p>How many animals do we have in each group?</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covars</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">group</span>, <span class="va">age</span>, <span class="va">sex</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarise</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Looks like we have 6 samples each from two genotypes (5xFAD or WT),
three ages (4 months, 6 months, and 10 months), and both sexes, for a
total of 72 samples.</p>
</div>
<div class="section level3">
<h3 id="accounting-for-transgenes">accounting for transgenes<a class="anchor" aria-label="anchor" href="#accounting-for-transgenes"></a></h3>
<p>The 5xFAD model has two copies each of the APP and PSEN1 genes - one
endogenous mouse gene, and the orthologous human transgene. The RNA-seq
data was assessed using a custom transcriptome definition that included
the sequences of both the mouse and human versions of each gene.</p>
<p>Ultimately we are going to sum the counts from both ortholgous genes
(human APP and mouse App; human PSEN1 and mouse Psen1). But first, let’s
look at the expression of each of these genes in the different groups.
To start we’ll filter the counts down to just those four relevant gene
IDs and join the counts up with the covariates to explore the expression
of these genes.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tg.counts</span> <span class="op">&lt;-</span> <span class="va">counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">gene_id</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000080815"</span>,<span class="st">"ENSMUSG00000019969"</span>,</span>
<span>                        <span class="st">"ENSG00000142192"</span>,<span class="st">"ENSMUSG00000022892"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="va">.</span>,cols <span class="op">=</span> <span class="op">-</span><span class="st">"gene_id"</span>,names_to <span class="op">=</span> <span class="st">"specimenID"</span>,values_to<span class="op">=</span><span class="st">"counts"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">covars</span> ,by<span class="op">=</span><span class="st">"specimenID"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">tg.counts</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s do a little data housekeeping:</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># make an age column that is a factor and re-order the levels</span></span>
<span><span class="va">tg.counts</span> <span class="op">&lt;-</span> <span class="va">tg.counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    age.m <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">age</span>, <span class="st">'m'</span><span class="op">)</span>,</span>
<span>    age.m <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">age.m</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4m'</span>,<span class="st">'6m'</span>,<span class="st">'10m'</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># add gene symbols</span></span>
<span><span class="va">tg.counts</span> <span class="op">&lt;-</span> <span class="va">tg.counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    symbol <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span></span>
<span>      <span class="va">gene_id</span> <span class="op">==</span> <span class="st">"ENSG00000142192"</span> <span class="op">~</span> <span class="st">"Human APP"</span>,</span>
<span>      <span class="va">gene_id</span> <span class="op">==</span> <span class="st">"ENSG00000080815"</span> <span class="op">~</span> <span class="st">"Human PSEN1"</span>,</span>
<span>      <span class="va">gene_id</span> <span class="op">==</span> <span class="st">"ENSMUSG00000022892"</span> <span class="op">~</span> <span class="st">"Mouse App"</span>,</span>
<span>      <span class="va">gene_id</span> <span class="op">==</span> <span class="st">"ENSMUSG00000019969"</span> <span class="op">~</span> <span class="st">"Mouse Psen1"</span></span>
<span>      <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Okay, now let’s plot the counts for each gene across the samples:</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">tg.counts</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">age.m</span>, y<span class="op">=</span><span class="va">counts</span>, color<span class="op">=</span><span class="va">group</span>, shape <span class="op">=</span> <span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>position<span class="op">=</span><span class="fu">position_jitterdodge</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">symbol</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>The human transgenes all have a counts of zero in the WT animals
(where the transgenes are absent), while the endogenous mouse genes are
expressed relatively consistently across both groups.</p>
<p>Let’s combine the expression of corresponding human and mouse genes
by summing the expression and saving the summed expression as expression
of mouse genes, respectively to match with gene names in control
mice.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># move the gene_id column to rownames, to enable summing across rows</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span> <span class="op">%&gt;%</span> <span class="fu">column_to_rownames</span><span class="op">(</span><span class="st">"gene_id"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co">#merge mouse and human APP gene raw count</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000022892"</span>,<span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000022892"</span>,<span class="op">]</span> <span class="op">+</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSG00000142192"</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000142192"</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">#merge mouse and human PS1 gene raw count</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000019969"</span>,<span class="op">]</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSMUSG00000019969"</span>,<span class="op">]</span> <span class="op">+</span> </span>
<span>  <span class="va">counts</span><span class="op">[</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"ENSG00000080815"</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"ENSG00000080815"</span><span class="op">)</span>,<span class="op">]</span></span></code></pre>
</div>
<p>We can confirm that the human genes are now absent from the counts
table:</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">str_detect</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, <span class="st">"MUS"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="prepare-data-and-run-deseq-analysis">prepare data and run DESeq analysis<a class="anchor" aria-label="anchor" href="#prepare-data-and-run-deseq-analysis"></a></h3>
<p>Next we’ll prepare the data for differential expression analysis.
We’ll use DESeq2 today, though there are other approaches. Another
disclaimer: there are multiple steps to a DESeq2 analysis and we’re not
going to get into nitty-gritty details here. We’ll briefly cover some of
the basics, but for more information, please refer to the <a href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html" class="external-link">DESeq2
vignette</a>.</p>
<p>Let’s perform this analysis stratified by age group while controlling
for the sex of the animals as a covariate. We can start with the
youngest animals (4 months old). Let’s sub-set the data and covariates
to these data:</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covars.4m</span> <span class="op">&lt;-</span> <span class="va">covars</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">age</span> <span class="op">==</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">counts.4m</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span>,<span class="fu">colnames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">covars.4m</span><span class="op">$</span><span class="va">specimenID</span><span class="op">]</span></span></code></pre>
</div>
<p>Next we’ll build the DESeq object</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData<span class="op">=</span><span class="va">counts.4m</span>, </span>
<span>                                   colData<span class="op">=</span><span class="va">covars.4m</span>, </span>
<span>                                   design <span class="op">=</span> <span class="op">~</span><span class="va">group</span><span class="op">+</span><span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span></span></code></pre>
</div>
<p>Now we have a <code>DESeqDataSet</code> object covering counts data
for 55k genes across 24 mice.</p>
<p>Let’s take a closer look at the counts that go into this object</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts.4m</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre>
</div>
<p>You can see that ENSMUSG00000000003 has <code>0</code> reads across
the samples listed here. Let’s find out how many genes are
<code>0 counts</code> across all samples.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_sums</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>gene_id <span class="op">=</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>,</span>
<span>                        sums    <span class="op">=</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu">rowSums</span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">sum</span><span class="op">(</span><span class="va">gene_sums</span><span class="op">$</span><span class="va">sums</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span></code></pre>
</div>
<p>We can see that 9691 (17.5%) genes have no reads at all. Let’s filter
these out. While it is not necessary to pre-filter low count genes
before running the DESeq2 functions, there are two reasons which make
pre-filtering useful: by removing rows in which there are very few
reads, we reduce the memory size of the dds data object, and we increase
the speed of the transformation and testing functions within DESeq2. It
can also improve visualizations, as features with no information for
differential expression are not plotted.</p>
<p>Here we perform a minimal pre-filtering to keep only rows that have
at least 10 reads in at least 6 separate samples.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span> <span class="op">&lt;-</span> <span class="va">ddsHTSeq</span><span class="op">[</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">6</span>,<span class="op">]</span></span></code></pre>
</div>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>What proportion of the 55k genes we started with remain after this
filter?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span></span></code></pre>
</div>
<p>There are 24765 genes, or 44.6% (24765/55487).</p>
</div>
</div>
</div>
</div>
<p>Let’s also make sure DESeq knows which group is our control or
“reference” group. By default this is arbitrarily assigned to the first
group in the factor. We can use the <code>relevel</code> function to set
the reference group to “WT” (wild type).</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu">relevel</span><span class="op">(</span><span class="va">ddsHTSeq</span><span class="op">$</span><span class="va">group</span>,ref<span class="op">=</span><span class="st">"WT"</span><span class="op">)</span>  </span></code></pre>
</div>
<p>Now we’re ready to run the differential expression analysis; it’ll
take a few seconds to process this step:</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq</span><span class="op">(</span><span class="va">ddsHTSeq</span>, parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>What are the results that have been computed:</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">resultsNames</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></code></pre>
</div>
<p>Because we specified <code>design = ~ group + sex</code> when setting
up the DESeq object, we now have the results from these two contrasts.
We can pull the results specifically for the 5xFAD vs WT comparison
using the <code>results</code> function. The results table contains the
log2 fold change, p-value, and adjusted p-value for each gene in the
analysis.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">results</span><span class="op">(</span><span class="va">dds</span>, contrast <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'group'</span>,<span class="st">'5xFAD'</span>,<span class="st">'WT'</span><span class="op">)</span>, alpha<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span>  </span>
<span><span class="va">fad.res.4m</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre>
</div>
<p>We can get a summary of the DE results using the <code>summary</code>
function:</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre>
</div>
<p>Ok, there are a total of 425 significantly differentially expressed
genes at 4 months of age when comparing 5xFAD to WT brain tissue. The
vast majority of these (392) are expressed at higher levels in 5xFAD
mice relative to WT. Let’s take a look at the most significantly DE
genes by ordering the results table by the adjusted p-value:</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fad.res.4m</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">head</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>This is a little difficult to interpret given all genes are
identified by their ENSEMBL IDs. Let’s map in the gene symbols using the
<code>org.Mm.eg.db</code> package. This package contains a mapping of
ENSEMBL IDs to gene symbols, and we can use the
<code>AnnotationDbi::mapIds</code> function to get the gene symbols for
our results table.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fad.res.4m</span><span class="op">$</span><span class="va">symbol</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">mapIds</span><span class="op">(</span></span>
<span>  <span class="fu">org.Mm.eg.db</span><span class="fu">::</span><span class="va">org.Mm.eg.db</span>,</span>
<span>  keys <span class="op">=</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">fad.res.4m</span><span class="op">)</span>,</span>
<span>  column <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span>  keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  multiVals <span class="op">=</span> <span class="st">"first"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">fad.res.4m</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">head</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Ok! Now we can see that among the most significantly up-regulated
genes in 5xFAD mice brains at 4 months of age are Psen1 and App (which
we saw previously), along with Thy1, Cst7, Ccl6, and Clec7a. Let’s plot
all DE genes:</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">fad.res.4m</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">log2FoldChange</span>, <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.3</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="op">(</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="fu">abs</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fu">log2</span><span class="op">(</span><span class="fl">1.2</span><span class="op">)</span><span class="op">)</span> <span class="op">)</span>, show.legend <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'grey'</span>,<span class="st">'red'</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu">geom_text_repel</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">fad.res.4m</span>, <span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="fu">abs</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fu">log2</span><span class="op">(</span><span class="fl">1.2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span>, min.segment.length <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s put it all together and compute results for each age group.
This code does all of the steps outlined above for each age cohort. The
input data, <code>DESeq</code> objects, and result tables are stored in
columns of a data frame. Feel free to copy and paste this code, but
investigate it and be sure you understand what each part is doing. It
may take about a minute or two to complete.</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">st</span> <span class="op">=</span> <span class="fu">Sys.time</span><span class="op">(</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">fad.deg</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  age <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">4</span>,<span class="fl">6</span>,<span class="fl">10</span><span class="op">)</span>,</span>
<span>  meta <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">age</span>, <span class="op">~</span> <span class="va">covars</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">age</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  counts <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">meta</span>, <span class="op">~</span> <span class="va">counts</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">specimenID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    dds <span class="op">=</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span></span>
<span>      countData <span class="op">=</span> <span class="va">counts</span>,</span>
<span>      colData <span class="op">=</span> <span class="va">meta</span>,</span>
<span>      design <span class="op">=</span> <span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>    </span>
<span>    dds <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">dds</span>, <span class="op">~</span> <span class="va">.x</span><span class="op">[</span><span class="fu">rowSums</span><span class="op">(</span><span class="fu">counts</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">6</span>,<span class="op">]</span><span class="op">)</span>,</span>
<span>    dds <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">dds</span>, <span class="op">~</span> <span class="op">{</span><span class="va">.x</span><span class="op">$</span><span class="va">group</span> <span class="op">=</span> <span class="fu">relevel</span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">group</span>, ref <span class="op">=</span> <span class="st">'WT'</span><span class="op">)</span>; <span class="va">.x</span><span class="op">}</span><span class="op">)</span>,</span>
<span>    dds <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">dds</span>, <span class="op">~</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu">DESeq</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    res <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">dds</span>, <span class="op">~</span> <span class="fu">results</span><span class="op">(</span><span class="va">.x</span>, contrast <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'group'</span>,<span class="st">'5xFAD'</span>,<span class="st">'WT'</span><span class="op">)</span>, alpha<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    res.t <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">res</span>, <span class="op">~</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                  <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">'Ensembl_gene_id'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                  <span class="fu">mutate</span><span class="op">(</span>symbol <span class="op">=</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">mapIds</span><span class="op">(</span></span>
<span>                    <span class="fu">org.Mm.eg.db</span><span class="fu">::</span><span class="va">org.Mm.eg.db</span>,</span>
<span>                    keys <span class="op">=</span> <span class="va">Ensembl_gene_id</span>,</span>
<span>                    column <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span>                    keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>                    multiVals <span class="op">=</span> <span class="st">"first"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span></span>
<span><span class="va">ed</span> <span class="op">=</span> <span class="fu">Sys.time</span><span class="op">(</span><span class="op">)</span> <span class="op">-</span> <span class="va">st</span></span>
<span></span>
<span><span class="fu">print</span><span class="op">(</span><span class="va">ed</span><span class="op">)</span></span></code></pre>
</div>
<p>What information does the <code>fad.r</code> tibble contain? Let’s
take a look:</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">glimpse</span><span class="op">(</span><span class="va">fad.deg</span><span class="op">)</span></span></code></pre>
</div>
<p>The first column is the age brackets in integers. The second and
third columns are the metadata and count data for each age cohort (24
samples per age bracket). The fourth column is the DESeq object and the
fifth and sixth columns are the results from the DE analysis.</p>
<p>So by running:</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">fad.deg</span><span class="op">$</span><span class="va">res</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<p>We can see that there are more significantly DE genes at 6 months
than we saw at four months.</p>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p>How many up- and down-regulated genes are found for each age?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">map_dbl</span><span class="op">(</span><span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span>, </span>
<span>  <span class="op">~</span><span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">length</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">map_dbl</span><span class="op">(</span><span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span>, </span>
<span>  <span class="op">~</span><span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">length</span><span class="op">)</span></span></code></pre>
</div>
<p>There are between 392 and 1855 up-regulated genes, and between 33 and
1827 down-regulated genes.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-3" class="callout-inner">
<h3 class="callout-title">Challenge 3</h3>
<div class="callout-content">
<p>How many down-regulated genes overlap between each timepoint?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>The comparisons to check are 4m+6m, 4m+10m, and 6m+10m; these are in
rows 1+2, 1+3, and 2+3, respectively.</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">length</span><span class="op">(</span><span class="fu">intersect</span><span class="op">(</span></span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span>,</span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">length</span><span class="op">(</span><span class="fu">intersect</span><span class="op">(</span></span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span>,</span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">length</span><span class="op">(</span><span class="fu">intersect</span><span class="op">(</span></span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span>,</span>
<span>  <span class="va">fad.deg</span><span class="op">$</span><span class="va">res.t</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">Ensembl_gene_id</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>It looks like there are 22, 22, and 424 overlapping down-regulated
genes, respectively.</p>
</div>
</div>
</div>
</div>
<p>This is a good point to save these data before we move on.</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">fad.deg</span>, <span class="fu">here</span><span class="fu">::</span><span class="fu">here</span><span class="op">(</span><span class="st">"5xFAD_DESeq_analysis.rds"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
</section><section><h2 class="section-heading" id="overview-of-human-cohort-data">[3] Overview of Human cohort data<a class="anchor" aria-label="anchor" href="#overview-of-human-cohort-data"></a></h2>
<hr class="half-width"><p>Now that we have our mouse transcriptomes analyzed, let’s switch
gears and think about the human datasets. The Accelerating Medicines
Partnership-Alzheimer’s Disease Consortium (<a href="https://adknowledgeportal.synapse.org/Explore/Programs/DetailsPage?Program=AMP-AD" class="external-link">AMP-AD</a>)
has generated extensive sets of ’omics data from a variety of human
Alzheimer’s Disease cohorts. AMP-AD researchers are applying systems
biology approaches toward the goal of elucidating AD mechanisms and
highlighting potential therapeutic targets.</p>
<p>There are three large, independent human cohorts that are part of
AMP-AD:</p>
<ol style="list-style-type: decimal"><li>The Religious Orders Study and the Memory and Aging Project (<a href="https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn3219045" class="external-link">ROSMAP,
syn3219045</a>)</li>
<li>Mount Sinai Brain Bank (<a href="https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn3159438" class="external-link">MSBB,
syn3159438</a>)<br></li>
<li>Mayo Clinic (<a href="https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn5550404" class="external-link">Mayo,
syn5550404</a>)</li>
</ol><p>These studies have collected postmortem RNA-seq profiles from over
1,200 individuals spanning seven distinct brain regions:<br>
- dorsolateral prefrontal cortex (DLPFC)<br>
- temporal cortex (TCX)<br>
- inferior frontal gyrus (IFG)<br>
- superior temporal gyrus (STG)<br>
- frontal pole (FP)<br>
- parahippocampal gyrus (PHG)<br>
- cerebellum (CBE)</p>
<p>These samples are generally balanced for AD, MCI, and non-affected
controls. The data provide a broad assessment on how AD affects multiple
brain regions in 3 different populations in the US.</p>
<figure><img src="../fig/ampad_studies_regions.png" alt="table showing AMP-AD studies, the number of samples, and brain regions sampled" class="figure mx-auto d-block"><div class="figcaption">Transcriptomic studies of Alzheimer’s
Disease</div>
</figure><div class="section level3">
<h3 id="overview-of-human-consensus-rna-seq-coexpression-modules">Overview of Human Consensus RNA-Seq Coexpression Modules<a class="anchor" aria-label="anchor" href="#overview-of-human-consensus-rna-seq-coexpression-modules"></a></h3>
<p><a href="https://doi.org/10.1016/j.celrep.2020.107908" class="external-link">Wan, et
al. (2020)</a> performed meta analysis including all available AMP-AD
RNA-seq datasets and systematically defined correspondences between gene
expression changes associated with AD in human brains. Briefly, the
RNA-seq read libraries were normalized and covariates were adjusted for
each human study separately before testing for differential expression
using fixed/mixed effects modeling to account for batch effects. The
expression data from each brain region was used to perform co-expression
analysis using a variety of different algorithms, generating in total
2,978 co-expression modules across all tissues. Of these, 660 modules
showed enrichment for at least one AD-specific differentially expressed
gene from the meta-analysis of all cases compared to controls.</p>
<p>Wan <em>et al</em> clustered these modules together using network
analyses and found 30 co-expression modules related to LOAD pathology.
Among the 30 aggregate co-expression modules, five consensus clusters
were described that span brain region and study cohorts. These consensus
clusters consist of subsets of modules which are associated with similar
AD related changes across brain regions and cohorts. Wan <em>et al</em>
looked for enrichment of cell -type signatures within the modules using
expression-weighted cell type enrichment analysis (<a href="https://doi.org/10.3389/fnins.2016.00016" class="external-link">Skene and Grant
(2016)</a>) and examined enrichment of functional annotations within the
modules.</p>
<figure><img src="../fig/ampad_consensus_clusters.png" alt="Heatmap showing the overlap between co-expression clusters and the consensus clusters." class="figure mx-auto d-block"><div class="figcaption">AMP-AD GENE Modules</div>
</figure><p>This figure shows a matrix view of gene content overlap between the
30 co-expression modules. You’ll note a few strongly overlapping group
of modules, implicating similar expression profiles in different studies
and brain regions, which are the consensus clusters (A-E).</p>
<p>The first module block (consensus cluster A) is enriched for
signatures of astrocytes, while the next block (consensus cluster B) is
enriched for signatures of other cell types, including endothelial and
microglial expressed genes, suggesting an inflammation component. The
third module block (consensus cluster C) is strongly enriched for
signatures of neuronal gene expression, linking the modules within this
cluster to neurodegenerative processes. The fourth module block
(consensus cluster D) is enriched for oligodendrocyte and other glial
genes, indicating myelination and other neuronal support functions
associated with these modules. Finally, consensus cluster E contains
mixed modules that don’t have clear cell type enrichments, but do have
enrichments for processes involved in response to stress or unfolded
proteins. Stress response is not cell specific, so the expression
represented by these modules may be throughout many cells in the
brain.</p>
</div>
<div class="section level3">
<h3 id="accessing-amp-ad-module-data">Accessing AMP-AD module data<a class="anchor" aria-label="anchor" href="#accessing-amp-ad-module-data"></a></h3>
<p>These AMP-AD co-expression modules are very useful for making
comparisons between animal models and the human cohorts. In order to use
these modules to make the comparisons, we’ll need to download data
pertaining to the 30 co-expression modules. These data are available
from the Synapse data repository (<a href="https://www.synapse.org/Synapse:syn11932957" class="external-link">syn11932957</a>);
let’s download and take a closer look at the data.</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="va">syn</span><span class="op">$</span><span class="fu">tableQuery</span><span class="op">(</span><span class="st">"SELECT * FROM syn11932957"</span><span class="op">)</span></span>
<span><span class="va">module_table</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">query</span><span class="op">$</span><span class="va">filepath</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">module_table</span><span class="op">)</span></span></code></pre>
</div>
<p>Here you see 9 columns in this table. Columns we’re interested in
are:</p>
<ul><li>column 3: <code>GeneID</code> contains Ensembl gene IDs<br></li>
<li>column 4: <code>Module</code> is the module name in which gene is
clustered<br></li>
<li>column 7: <code>brainRegion</code> is the tissue of the
corresponding module<br></li>
<li>column 9: <code>external_gene_name</code> are gene symbols</li>
</ul><p>How many unique modules are in the table?</p>
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">Module</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>What are the names of the modules?</p>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">unique</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">Module</span><span class="op">)</span></span></code></pre>
</div>
<p>How many genes are in each module?</p>
<div class="codewrapper sourceCode" id="cb40">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">table</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">Module</span><span class="op">)</span></span></code></pre>
</div>
<p>We can visualize this as bar plot using ggplot2 package.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">module_table</span>,<span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">Module</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_bar</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> </span></code></pre>
</div>
<div id="challenge-4" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-4" class="callout-inner">
<h3 class="callout-title">Challenge 4</h3>
<div class="callout-content">
<p>What are other ways to count genes in each module?</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>You could also try:</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">count</span><span class="op">(</span><span class="va">module_table</span> ,<span class="va">Module</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>We can also check the total number of unique genes in the table</p>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">GeneID</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="mouse-orthologs-of-human-module-genes">Mouse orthologs of Human module genes<a class="anchor" aria-label="anchor" href="#mouse-orthologs-of-human-module-genes"></a></h4>
<p>In the module table we’ve downloaded we have human ENSEMBL ids and
human gene symbols. In order to compare between human and mouse models,
we will need to identify the corresponding (i.e. orthologous) mouse gene
IDs. We are going to add the gene IDs of orthologous genes in mouse to
the corresponding human genes in the module table.</p>
<p>Orthology mapping can be tricky, but thankfully Wan <em>et al</em>
have already identified mouse orthologs for each of the human genes
using the HGNC Comparison of Orthology Predictions (<a href="https://www.genenames.org/tools/hcop/" class="external-link">HCOP</a>) tool. While there
are a variety of different ways to get data about gene orthology, for
the sake of simplicity we are going to read that table from Synapse (<a href="https://doi.org/10.7303/syn17010253.1" class="external-link">syn17010253</a>).</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouse.human.ortho</span> <span class="op">&lt;-</span> <span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">"syn17010253"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span> <span class="op">%&gt;%</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">mouse.human.ortho</span><span class="op">)</span></span></code></pre>
</div>
<p>There are 15 columns with various gene identifiers for each species.
We’ll add mouse gene symbols from the ortholog table to the module table
by matching the human ENSEMBL IDs from both tables
(i.e. <code>GeneID</code> from the module table and
<code>human_ensembl_gene</code> from the orthology table).</p>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">module_table</span><span class="op">$</span><span class="va">Mouse_gene_symbol</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">mouse.human.ortho</span><span class="op">$</span><span class="va">mouse_symbol</span><span class="op">[</span></span>
<span>    <span class="fu">match</span><span class="op">(</span><span class="va">module_table</span><span class="op">$</span><span class="va">GeneID</span>,<span class="va">mouse.human.ortho</span><span class="op">$</span><span class="va">human_ensembl_gene</span><span class="op">)</span></span>
<span>    <span class="op">]</span></span></code></pre>
</div>
<p>Taking a look at the module table, we can see the new column of mouse
orthologs we just added.</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">module_table</span><span class="op">)</span></span></code></pre>
</div>
<p>Some genes don’t have identified orthologs. Also there’s some
redundant information. Let’s only keep the columns of interest and rows
that contain a mouse ortholog mapping:</p>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules</span> <span class="op">&lt;-</span> <span class="va">module_table</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">distinct</span><span class="op">(</span>tissue <span class="op">=</span> <span class="va">brainRegion</span>, module <span class="op">=</span> <span class="va">Module</span>, gene <span class="op">=</span> <span class="va">GeneID</span>, <span class="va">Mouse_gene_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Mouse_gene_symbol</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span></span></code></pre>
</div>
<p>Take a look at this new data table:</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-5" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-5" class="callout-inner">
<h3 class="callout-title">Challenge 5</h3>
<div class="callout-content">
<p>How many human genes are we removing that don’t have identified
orthologs?</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">module_table</span>, <span class="fu">is.na</span><span class="op">(</span><span class="va">Mouse_gene_symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">external_gene_name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">nrow</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>2998 genes</p>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="reading-differential-expression-result-of-human-data-from-meta-analysis">Reading differential expression result of human data from
meta-analysis<a class="anchor" aria-label="anchor" href="#reading-differential-expression-result-of-human-data-from-meta-analysis"></a></h3>
<p>Now we know the genes that are in each AMP-AD co-expression cluster,
along with the ID of the corresponding orthologous genes in mouse. We’ll
also need to know how the expression of these genes change in AD.</p>
<p>We’ll download the results from differential expression analysis of
reprocessed AMP-AD RNA-seq data (all 7 brain regions). Log fold change
values for human transcripts can be obtained from Synapse (<a href="https://www.synapse.org/#!Synapse:syn14237651" class="external-link">syn14237651</a>).</p>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules_raw</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">"syn14237651"</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules_raw</span><span class="op">)</span></span></code></pre>
</div>
<p>Data from which tissues are in this table?</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">unique</span><span class="op">(</span><span class="va">ampad_modules_raw</span><span class="op">$</span><span class="va">Tissue</span><span class="op">)</span></span></code></pre>
</div>
<p>All 7 brain regions are represented.</p>
<p>The AMP-AD data has been processed many ways and using different
models and comparisons. Let’s take a look at how many ways the data have
been analyzed:</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Model</span>, <span class="va">Comparison</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>For our analyses we’ll use data from the “Diagnosis” model and
comparisons between AD cases vs controls. We’ll filter the table for
these conditions and only keep the three columns we’ll need:
<code>Tissue</code>, <code>Gene</code> and <code>logFC</code>.</p>
<div class="codewrapper sourceCode" id="cb53">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_fc</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">as_tibble</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>, <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span>tissue <span class="op">=</span> <span class="va">Tissue</span>, gene <span class="op">=</span> <span class="va">ensembl_gene_id</span>, ampad_fc <span class="op">=</span> <span class="va">logFC</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="combine-with-modules-so-correlation-can-be-done-per-module">Combine with modules so correlation can be done per module<a class="anchor" aria-label="anchor" href="#combine-with-modules-so-correlation-can-be-done-per-module"></a></h4>
<p>Next, we will combine the fold change table we just downloaded
(<code>ampad_fc</code>) and module table from before
(<code>ampad_modules</code>). First, let’s look at both tables to check
how can we merge them together?</p>
<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_fc</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules</span><span class="op">)</span></span></code></pre>
</div>
<p>The columns common to both tables are <code>gene</code> (the human
Ensembl gene IDs) and <code>tissue</code> (the brain region
corresponding to the module/measurement). So we will merge the data sets
using these two columns.</p>
<p><strong>Reminder</strong>: Each gene can be present in multiple brain
regions, but should only be in one module per brain region. Let’s double
check using the first gene in the table:</p>
<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules</span><span class="op">[</span><span class="va">ampad_modules</span><span class="op">$</span><span class="va">gene</span> <span class="op">%in%</span> <span class="st">"ENSG00000168439"</span>,<span class="op">]</span></span></code></pre>
</div>
<p>This gene is present in six different co-expression modules all from
different brain regions. You can do this for any other gene as well.</p>
<p>We’ll merge the two tables using the <code>dplyr::inner_join</code>
function:</p>
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ampad_modules_fc</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">ampad_modules</span>, <span class="va">ampad_fc</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"tissue"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span>symbol <span class="op">=</span> <span class="va">Mouse_gene_symbol</span>, <span class="va">module</span>, <span class="va">ampad_fc</span><span class="op">)</span> </span></code></pre>
</div>
<p>Take a look at the new table we just made:</p>
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">ampad_modules_fc</span><span class="op">)</span></span></code></pre>
</div>
<p>We will use the data in <code>ampad_modules_fc</code> to compare with
log fold change data measured in mouse models.</p>
</div>
<div class="section level4">
<h4 id="preparing-module-information-for-correlation-plots">Preparing module information for correlation plots<a class="anchor" aria-label="anchor" href="#preparing-module-information-for-correlation-plots"></a></h4>
<p>Let’s package up these data and save this progress so far. This is
some manual book-keeping to arrange the modules into consensus clusters
for plotting later. You can just copy-paste this code into your
<code>Rstudio</code> session.</p>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cluster_a</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"TCXblue"</span>, <span class="st">"PHGyellow"</span>, <span class="st">"IFGyellow"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster A (ECM organization)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster A\n(ECM organization)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_b</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"DLPFCblue"</span>, <span class="st">"CBEturquoise"</span>, <span class="st">"STGblue"</span>, <span class="st">"PHGturquoise"</span>, <span class="st">"IFGturquoise"</span>, <span class="st">"TCXturquoise"</span>, <span class="st">"FPturquoise"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster B (Immune system)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster B\n(Immune system)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_c</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"IFGbrown"</span>, <span class="st">"STGbrown"</span>, <span class="st">"DLPFCyellow"</span>, <span class="st">"TCXgreen"</span>, <span class="st">"FPyellow"</span>, <span class="st">"CBEyellow"</span>, <span class="st">"PHGbrown"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster C (Neuronal system)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster C\n(Neuronal system)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_d</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"DLPFCbrown"</span>, <span class="st">"STGyellow"</span>, <span class="st">"PHGgreen"</span>, <span class="st">"CBEbrown"</span>, <span class="st">"TCXyellow"</span>, <span class="st">"IFGblue"</span>, <span class="st">"FPblue"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster D (Cell Cycle, NMD)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster D\n(Cell Cycle, NMD)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cluster_e</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span></span>
<span>  module <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"FPbrown"</span>, <span class="st">"CBEblue"</span>, <span class="st">"DLPFCturquoise"</span>, <span class="st">"TCXbrown"</span>, <span class="st">"STGturquoise"</span>, <span class="st">"PHGblue"</span><span class="op">)</span>,</span>
<span>  cluster <span class="op">=</span> <span class="st">"Consensus Cluster E (Organelle Biogensis, Cellular stress response)"</span>,</span>
<span>  cluster_label <span class="op">=</span> <span class="st">"Consensus Cluster E\n(Organelle Biogenesis,\nCellular stress response)"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">module_clusters</span> <span class="op">&lt;-</span> <span class="va">cluster_a</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_b</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_c</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_d</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cluster_e</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cluster_label <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">cluster_label</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">module_clusters</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="va">module_clusters</span><span class="op">$</span><span class="va">module</span></span>
<span></span>
<span><span class="fu">save</span><span class="op">(</span><span class="va">ampad_modules_fc</span>,<span class="va">module_clusters</span>,<span class="va">mod</span>, file<span class="op">=</span> <span class="fu">here</span><span class="fu">::</span><span class="fu">here</span><span class="op">(</span><span class="st">"AMPAD_Module_Data.RData"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="correlation-between-mouse-models-and-human-ad-modules">[4] Correlation between mouse models and human AD modules<a class="anchor" aria-label="anchor" href="#correlation-between-mouse-models-and-human-ad-modules"></a></h2>
<hr class="half-width"><p>Now we’ll get to the cross-species alignment. Our goal, as
demonstrated in the plots below, is to identify modules where the gene
expression is correlated between human and mouse orthologs (left) as
well as modules where there is no correlation (right).</p>
<figure><img src="../fig/hs_mm_logfc_correlation.png" alt="Example correlation plots for AMP-AD co-expression module genes and mouse model expression." class="figure mx-auto d-block"><div class="figcaption">Human-Mouse logFC correlation</div>
</figure><p>There are two approaches that we commonly use to compute correlation
between mouse data and human AD data:</p>
<ol style="list-style-type: decimal"><li>Compare change in expression in Human AD cases vs controls with
change in expression in mouse models for each gene in a given
module:<br></li>
</ol><ul><li>LogFC(h) = log fold change in expression of human AD patients
compared to control patients.</li>
<li>LogFC(m) = log fold change in expression of mouse AD models compared
to control mouse models.</li>
</ul><p><span class="math display">\[cor.test(LogFC(h),
LogFC(m))\]</span></p>
<ol start="2" style="list-style-type: decimal"><li>Compare Human AD expression changes to mouse genetic effects for
each gene in a given module:<br></li>
</ol><ul><li>h = human gene expression (Log2 RNA-seq Fold Change AD/control)</li>
<li>β = mouse gene expression effect from linear regression model (Log2
RNA-seq TPM)</li>
</ul><p><span class="math display">\[cor.test(LogFC(h), β)\]</span></p>
<p>Both approaches allow us to assess directional coherence between gene
expression for genes in AMP-AD modules and the effects of genetic
manipulations in mice. For this session we are going to use the first
approach; we’ll return to approach #2 later in the week.</p>
<div id="challenge-6" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-6" class="callout-inner">
<h3 class="callout-title">Challenge 6</h3>
<div class="callout-content">
<p>What are the relative advantages of each approach?</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<p><span class="math display">\[cor.test(LogFC(h),
LogFC(m))\]</span></p>
<p>• direct comparison of effect sizes and direction • intuitive
interpretation</p>
<p><span class="math display">\[cor.test(LogFC(h), β)\]</span></p>
<p>• identify the genetic contributions • human AD data often has age,
sex, and other covariates regressed out, to derive the AD specific
effect • controlling for analogous variables by computing the genetic
effect (β) is often advantageous</p>
<p>Others?</p>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="step-0-reading-gene-expression-count-matrix-from-previous-lesson">Step 0: Reading Gene Expression Count matrix from Previous
Lesson<a class="anchor" aria-label="anchor" href="#step-0-reading-gene-expression-count-matrix-from-previous-lesson"></a></h4>
<p>First we’ll read the results saved after differential expression
analysis (above). We’ll only keep the information about which age cohort
and the differential expression results.</p>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fad.deg</span> <span class="op">&lt;-</span> <span class="fu">load</span><span class="op">(</span><span class="st">"results/DEAnalysis_5XFAD.Rdata"</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">WARNING<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="warning" tabindex="0"><code>Warning in readChar(con, 5L, useBytes = TRUE): cannot open compressed file
'results/DEAnalysis_5XFAD.Rdata', probable reason 'No such file or directory'</code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in `readChar()`:
! cannot open the connection</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb62">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">fad.deg</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error:
! object 'fad.deg' not found</code></pre>
</div>
<p>Let’s also load the AMP-AD module data.</p>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"data/AMPAD_Module_Data.RData"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="step-1-measure-the-correlation-between-mouse-models-for-each-sex-at-each-age-and-amp-ad-modules-using-common-genes-from-both-datasets">Step 1: Measure the correlation between mouse models for each sex at
each age and AMP-AD modules using common genes from both datasets<a class="anchor" aria-label="anchor" href="#step-1-measure-the-correlation-between-mouse-models-for-each-sex-at-each-age-and-amp-ad-modules-using-common-genes-from-both-datasets"></a></h4>
<p>We compute a Pearson correlation between changes in expression for
each gene within a given module (log fold change for cases vs controls)
with each mouse model (log fold change of the 5xFAD mice vs sex- and
age-matched B6 mice).</p>
<p>First, we’ll combine both mouse <code>fad.deg</code> and human
<code>ampad_modules_fc</code> log fold change data sets for all
genes.</p>
<div class="codewrapper sourceCode" id="cb65">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_vs_ampad</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">fad.deg</span>, </span>
<span>                             <span class="va">ampad_modules_fc</span>, </span>
<span>                             by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"symbol"</span><span class="op">)</span>, </span>
<span>                             multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> </span></code></pre>
</div>
<p><strong>Note:</strong> for this join we specify
<code>multiple = "all"</code> so that the same gene can be matched
across multiple human tissues and multiple mouse ages.</p>
<div class="codewrapper sourceCode" id="cb66">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">model_vs_ampad</span><span class="op">)</span></span></code></pre>
</div>
<p>Now we’ll create a list column containing data frames using the <a href="https://tidyr.tidyverse.org/reference/nest.html" class="external-link">tidyr::nest</a>
function. Nesting is implicitly a summarizing operation: you get one row
for each group defined by the non-nested columns.</p>
<div class="codewrapper sourceCode" id="cb67">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="va">model_vs_ampad</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">module</span>, <span class="va">age</span>, <span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">ampad_fc</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">module</span>, <span class="va">age</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">ampad_fc</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<p>And we can also look at some of the nested data:</p>
<div class="codewrapper sourceCode" id="cb68">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">data</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<p>These are the mouse and human log fold-change values for all genes in
the TCXblue module; the mouse data corresponds to 4 month old 5xFAD
mice.</p>
<p>The total number of groups in the data table:</p>
<div class="codewrapper sourceCode" id="cb69">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we’ll compute correlation coefficients using the
<code>cor.test</code> function:</p>
<div class="codewrapper sourceCode" id="cb70">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">df</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>      cor_test <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">"log2FoldChange"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                                      <span class="va">.x</span><span class="op">[[</span><span class="st">"ampad_fc"</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>      estimate <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"estimate"</span><span class="op">)</span>,</span>
<span>      p_value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"p.value"</span><span class="op">)</span></span>
<span>      <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">cor_test</span><span class="op">)</span></span></code></pre>
</div>
<p>Here we’re using <code>purrr::map</code> based functions to apply the
correlation test to every entry in the <code>data</code> column. We can
pull out specific features from the <code>cor_test</code> list column,
including the computed correlation coefficient (<code>estimate</code>)
and the significance (<code>p.value</code>).</p>
<p>In the end we should have correlation coefficients and significance
values for every comparison in our data table:</p>
<div class="codewrapper sourceCode" id="cb71">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cor.df</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="step-2-annotate-correlation-table-to-prepare-for-visualization">Step 2: Annotate correlation table to prepare for visualization<a class="anchor" aria-label="anchor" href="#step-2-annotate-correlation-table-to-prepare-for-visualization"></a></h4>
<p>These steps will make it easier to make a nice looking plot during
the next step. We’ll add a column that flags whether the correlation is
significant or not, and we’ll add in the information about which
consensus cluster each module belongs to:</p>
<div class="codewrapper sourceCode" id="cb72">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_module</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>significant <span class="op">=</span> <span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">module_clusters</span>, by <span class="op">=</span> <span class="st">"module"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="va">cluster</span>, <span class="va">cluster_label</span>, <span class="va">module</span>, <span class="va">age</span>,</span>
<span>                correlation <span class="op">=</span> <span class="va">estimate</span>, <span class="va">p_value</span>, <span class="va">significant</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">model_module</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level4">
<h4 id="step-3-create-a-dataframe-to-use-as-input-for-plotting-the-results">Step 3: Create a dataframe to use as input for plotting the
results<a class="anchor" aria-label="anchor" href="#step-3-create-a-dataframe-to-use-as-input-for-plotting-the-results"></a></h4>
<p>More preparations for plotting, here we’ll get all of the values in
the right order so that they are grouped together nicely on the
plot.</p>
<div class="codewrapper sourceCode" id="cb73">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">correlation_for_plot</span> <span class="op">&lt;-</span> <span class="va">model_module</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">arrange</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span> </span>
<span>      module <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">module</span>,levels<span class="op">=</span><span class="va">mod</span><span class="op">)</span>,</span>
<span>      age <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">age</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'10m'</span>,<span class="st">'6m'</span>,<span class="st">'4m'</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="op">)</span> </span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">correlation_for_plot</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="visualizing-the-correlation-plot">Visualizing the Correlation plot<a class="anchor" aria-label="anchor" href="#visualizing-the-correlation-plot"></a></h3>
<p>Now, we will use the above matrix and visualize the correlation
results using <code>ggplot2</code> package functions.</p>
<div class="codewrapper sourceCode" id="cb74">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">correlation_for_plot</span></span>
<span><span class="fu">range</span><span class="op">(</span><span class="va">correlation_for_plot</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># the AMP-AD modules will be along the x-axis, the mouse models will be along the y-axis</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">age</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># each tile of the grid will be filled with a circle where the fill and size correspond to the correlation coefficient</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">age</span>, </span>
<span>                              colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># we'll draw a box arround significant correlations</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">filter</span><span class="op">(</span><span class="va">data</span>, <span class="va">.data</span><span class="op">$</span><span class="va">significant</span><span class="op">)</span>, </span>
<span>             <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">module</span>, y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">age</span>, colour <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">correlation</span><span class="op">)</span>,</span>
<span>             color<span class="op">=</span><span class="st">"black"</span>,shape<span class="op">=</span><span class="fl">0</span>,size<span class="op">=</span><span class="fl">9</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># plot the x-axis on the top of the plot, set the parameters of the scales</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span>guide <span class="op">=</span> <span class="st">"none"</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">0.5</span>, <span class="fl">0.5</span><span class="op">)</span>, low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, </span>
<span>                                 name <span class="op">=</span> <span class="st">"Correlation"</span>, guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># remove axis labels</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># facet the plot based on age range for the mice (rows) and consensus cluster (columns)</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span> rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="st">'5xFAD'</span><span class="op">)</span>, cols <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">vars</span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">cluster_label</span><span class="op">)</span>, </span>
<span>                       scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free"</span>,switch <span class="op">=</span> <span class="st">'y'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  </span>
<span>  <span class="co"># specify how different aspects of the plot will look</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    strip.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>,colour <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, hjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"right"</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>In the above plot, categories along the x-axis are the 30 AMP-AD
co-expression modules grouped into 5 consensus clusters, while the
categories along the y-axis show the different groupings of mouse models
tested (split by age). Positive correlations are shown in blue and
negative correlations in red. Color intensity and size of the circles
are proportional to the correlation coefficient. Black squares around
dots represent significant correlation at p-value=0.05 and
non-significant correlations are left blank.</p>
<p>5xFAD mice display gene expression alterations that are correlated
with human disease across all five consensus clusters, with the
strongest correlations observed among modules/genes in Consensus Cluster
B, which generally consists of immune system pathways and functions.</p>
</div>
<div class="section level3">
<h3 id="examining-individual-correlation-results">Examining individual correlation results<a class="anchor" aria-label="anchor" href="#examining-individual-correlation-results"></a></h3>
<p>Let’s say we want to know more about a single comparison in the plot
above and which genes are contributing to the correlation result. Maybe
we’re really interested in the correlations to the FPbrown module
between 5xFAD mice at 4 months and 6 months. We can plot the individual
correlations for each comparison in the plot above with the data we
have. We’ll label genes with large fold change in the mouse using the
<code>ggrepel::geom_label_repel</code> function.</p>
<div class="codewrapper sourceCode" id="cb75">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># specify which comparisons to consider</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="st">'FPbrown'</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4m'</span>,<span class="st">'6m'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter the correlation data frame to these comparisons</span></span>
<span><span class="va">indiv.corr</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">module</span> <span class="op">==</span> <span class="va">m</span>, <span class="va">age</span> <span class="op">%in%</span> <span class="va">a</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span> facet <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">age</span>, <span class="st">'\n'</span>, <span class="st">'r = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">estimate</span>,<span class="fl">3</span><span class="op">)</span>,<span class="st">' ; p = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">p_value</span>,<span class="fl">3</span><span class="op">)</span> <span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">aes</span><span class="op">(</span> <span class="va">log2FoldChange</span> , <span class="va">ampad_fc</span> <span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span> size <span class="op">=</span> <span class="fl">.5</span>, color <span class="op">=</span> <span class="st">'darkred'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span>method <span class="op">=</span> <span class="st">'lm'</span>, lwd <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu">geom_text_repel</span><span class="op">(</span> data <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">desc</span><span class="op">(</span><span class="fu">abs</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                             <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">3</span>, min.segment.length <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">'5xFAD logFC'</span>, y <span class="op">=</span> <span class="st">'AMP-AD logFC'</span>,</span>
<span>       title <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">indiv.corr</span><span class="op">$</span><span class="va">module</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">facet</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Here we can see that the 5xFAD logFC are pretty small in general. The
correlations are relatively weak and driven by individual genes that
have relatively large changes (e.g. Heatr4, Nirp1a, Rpl39l). If we
compare a different module, say STGblue, we can see a stronger
relationship between mouse and human expression changes.</p>
<div class="codewrapper sourceCode" id="cb76">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">&lt;-</span> <span class="st">'STGblue'</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4m'</span>,<span class="st">'6m'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter the correlation data frame to these comparisons</span></span>
<span><span class="va">indiv.corr</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">module</span> <span class="op">==</span> <span class="va">m</span>, <span class="va">age</span> <span class="op">%in%</span> <span class="va">a</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span> facet <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">age</span>, <span class="st">'\n'</span>, <span class="st">'r = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">estimate</span>,<span class="fl">3</span><span class="op">)</span>,<span class="st">' ; p = '</span>,<span class="fu">signif</span><span class="op">(</span><span class="va">p_value</span>,<span class="fl">3</span><span class="op">)</span> <span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">aes</span><span class="op">(</span> <span class="va">log2FoldChange</span> , <span class="va">ampad_fc</span> <span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span> size <span class="op">=</span> <span class="fl">.5</span>, color <span class="op">=</span> <span class="st">'darkred'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span>method <span class="op">=</span> <span class="st">'lm'</span>, lwd <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu">geom_text_repel</span><span class="op">(</span> data <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">indiv.corr</span>, <span class="fu">desc</span><span class="op">(</span><span class="fu">abs</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                             <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">3</span>, min.segment.length <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">'5xFAD logFC'</span>, y <span class="op">=</span> <span class="st">'AMP-AD logFC'</span>,</span>
<span>       title <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">indiv.corr</span><span class="op">$</span><span class="va">module</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">facet</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>These correlations are much stronger (r is approximately 0.3 vs 0.1
for the previous module), and there is a consistent pattern between
young mice and old mice, with similar genes being expressed in similar
ways (e.g. Itgax and Clec7a are up-regulated at both ages). These
significant positive correlations suggest that the 5xFAD model captures
inflammatory changes observed in human AD patients.</p>
</div>
</section><section><h2 class="section-heading" id="detecting-functional-coherence-of-gene-sets-from-omics-data">[5] Detecting functional coherence of gene sets from omics data<a class="anchor" aria-label="anchor" href="#detecting-functional-coherence-of-gene-sets-from-omics-data"></a></h2>
<hr class="half-width"><p>Most omics analyses produce data on many thousands of genomic
features (i.e. transcripts, proteins, etc.) for each condition tested.
Simply looking at these lists of genes and associated statistics can be
daunting and uninformative. We need approaches to identify which
biological functions are being impacted by a given experiment from these
systems-level measurements.</p>
<p>Gene functional enrichment analysis describes a variety of
statistical methods that identify groups of genes that share a
particular biological function or process and show differential
association with experimental conditions. Most approaches compare some
statistically valid set of differentially expressed features to sets of
functional annotations for those features. There are many different
functional annotation sets available, some of the more commonly used
include:</p>
<ul><li>gene function resources, such as the <a href="../amigo.geneontology.org/amigo">Gene Ontology</a> (i.e. GO)<br></li>
<li>pathway databases, such as <a href="../reactome.org">Reactome</a> or <a href="../www.genome.jp/kegg">KEGG</a><br></li>
<li>disease and phenotype ontologies, such as the <a href="../hpo.jax.org/app">Human Phenotype Ontology</a>, the <a href="../www.informatics.jax.org/vocab/mp_ontology">Mammalian Phenotype
Ontology</a>, and the <a href="https://disease-ontology.org/" class="external-link">Disease
Ontology</a>
</li>
</ul><p>These are the resources that are the foundation for many genomics
knowledge bases, such as <a href="../www.informatics.jax.org">MGI</a>, <a href="../monarchinitiative.org">monarch initiative</a>, and the <a href="../alliancegenome.org">Alliance of Genome Resources</a>. The precise
nature of each of these resources varies, but the core information
contained within each is the relationship of sets of genes to
biologically meaningful annotations. These annotations are typically
expertly curated from the published literature.</p>
<p>There are a variety of statistical approaches that can be employed to
test for functional enrichment among genes identified from an omics
dataset. Two of the most common broad classes of tests are
<em>over-representation analysis (ORA)</em> and <em>gene set enrichment
analysis (GSEA)</em>. For example, consider the figure below from <a href="https://doi.org/10.1016/j.tig.2023.01.003" class="external-link">Zhao &amp; Rhee, Trends
in Genetics (2023)</a>. Let’s consider each in a bit more detail.</p>
<figure><img src="../fig/pathway_enrichment_Zhao_Rhee_TrendsGenetics.jpg" alt="Figure showing different methods to test functional enrichment." class="figure mx-auto d-block"><div class="figcaption">Functional enrichment approaches</div>
</figure><div class="section level3">
<h3 id="over-representation-analysis">Over-representation analysis<a class="anchor" aria-label="anchor" href="#over-representation-analysis"></a></h3>
<p><code>ORA</code> involves statistical tests of overlap between two
lists of genes: one derived from the experiment and one from the
functional annotations. For example, one might ask what is the overlap
between the genes in an annotation class, such as “Lysosome”, and the
genes that are significantly up-regulated in a given experimental
condition. These tests usually rely on some form of Fisher’s exact test
(e.g. <code>fisher.test()</code>) or hypergeometric test
(e.g. <code>phyper()</code>). If the gene lists consist of a larger
number of overlapping genes than would be expected at random - given the
sample sizes involved - then there is said to be a statistical
over-representation of the annotation class within the experimental
condition.</p>
<p>Of course, these overlap tests are generally considered for all
annotation classes, which can number in the hundreds to thousands.
Performing this many statistical tests ensures that many will be
identified as significant by chance. Therefore, there is typically a
requirement to correct for multiple testing errors
(e.g. <code>p.adjust()</code>).</p>
<p>There are many R packages available to handle the statistical tests
and corrections involved in ORA. Today we’ll use
<code>clusterProfiler::enrichGO()</code>, which wraps statistical
testing for overlap with GO terms and multiple test correction in one
function.</p>
<p>Let’s start by considering the enrichments from the mouse data
analyzed previously. We’ll start by considering the genes that are
significantly DE in 6 month old animals</p>
<div class="codewrapper sourceCode" id="cb77">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene.list.up</span> <span class="op">&lt;-</span> <span class="va">fad.deg</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">age</span> <span class="op">==</span> <span class="st">'6m'</span>,</span>
<span>         <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, </span>
<span>         <span class="va">log2FoldChange</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">duplicated</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unique</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gene.list.dn</span> <span class="op">&lt;-</span> <span class="va">fad.deg</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">age</span> <span class="op">==</span> <span class="st">'6m'</span>,</span>
<span>         <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, </span>
<span>         <span class="va">log2FoldChange</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">duplicated</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>, <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unique</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">length</span><span class="op">(</span><span class="va">gene.list.up</span><span class="op">)</span></span>
<span><span class="fu">length</span><span class="op">(</span><span class="va">gene.list.dn</span><span class="op">)</span></span></code></pre>
</div>
<p>There are a total of # r length(gene.list.up) significantly
up-regulated genes and r length(gene.list.dn) significantly
down-regulated genes in this cohort. Now test for over representation of
GO terms among the DEGs. First, we need to identify the universe of all
possible genes which includes the genes that were both measured by the
experiment and contained within the annotation set.</p>
<div class="codewrapper sourceCode" id="cb78">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">univ</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame</span><span class="op">(</span><span class="va">org.Mm.egGO</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">bitr</span><span class="op">(</span><span class="va">.</span>, fromType <span class="op">=</span> <span class="st">"ENTREZID"</span>, toType <span class="op">=</span> <span class="st">'SYMBOL'</span>, OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, drop <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="st">'SYMBOL'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">intersect</span><span class="op">(</span><span class="va">.</span>, <span class="va">fad.deg</span><span class="op">$</span><span class="va">symbol</span><span class="op">)</span></span></code></pre>
</div>
<p>Now let’s test for enriched GO terms (this can take 3-4 minutes)</p>
<div class="codewrapper sourceCode" id="cb79">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.up</span> <span class="op">&lt;-</span> <span class="fu">enrichGO</span><span class="op">(</span><span class="va">gene.list.up</span>, </span>
<span>                   ont <span class="op">=</span> <span class="st">'all'</span>, </span>
<span>                   OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, </span>
<span>                   keyType <span class="op">=</span> <span class="st">'SYMBOL'</span>,</span>
<span>                   universe <span class="op">=</span> <span class="va">univ</span></span>
<span>                   <span class="op">)</span></span>
<span><span class="va">enr.dn</span> <span class="op">&lt;-</span> <span class="fu">enrichGO</span><span class="op">(</span><span class="va">gene.list.dn</span>, </span>
<span>                   ont <span class="op">=</span> <span class="st">'all'</span>, </span>
<span>                   OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, </span>
<span>                   keyType <span class="op">=</span> <span class="st">'SYMBOL'</span>,</span>
<span>                   universe <span class="op">=</span> <span class="va">univ</span></span>
<span>                   <span class="op">)</span></span></code></pre>
</div>
<p>How many significant terms are identified:</p>
<div class="codewrapper sourceCode" id="cb80">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.up</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">enr.dn</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-7" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-7" class="callout-inner">
<h3 class="callout-title">Challenge 7</h3>
<div class="callout-content">
<p>How many significant terms are identified from the up-regulated gene
list if you do not specify the “universe”?</p>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7"> Show me the solution </h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" aria-labelledby="headingSolution7" data-bs-parent="#accordionSolution7">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb81">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">enrichGO</span><span class="op">(</span><span class="va">gene.list.up</span>, ont <span class="op">=</span> <span class="st">'all'</span>, OrgDb <span class="op">=</span> <span class="va">org.Mm.eg.db</span>, keyType <span class="op">=</span> <span class="st">'SYMBOL'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="va">.</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span> </span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Plot the top 10 significant terms:</p>
<div class="codewrapper sourceCode" id="cb82">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu">plot_grid</span><span class="op">(</span></span>
<span>  <span class="fu">dotplot</span><span class="op">(</span><span class="va">enr.dn</span>, showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'down'</span><span class="op">)</span>,</span>
<span>  <span class="fu">dotplot</span><span class="op">(</span><span class="va">enr.up</span>, showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'up'</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre>
</div>
<p>From this we can see that nervous system related terms
(e.g. “dendrite development” and “protein localization to synapse”) are
down in 5xFAD mice at 6 months, while immune functions (e.g. “regulation
of innate immune response” and “leukocyte mediated immunity”) are up in
5xFAD mice at 6 months.</p>
</div>
<div class="section level3">
<h3 id="gene-set-enrichment-analysis">Gene set enrichment analysis<a class="anchor" aria-label="anchor" href="#gene-set-enrichment-analysis"></a></h3>
<p><code>GSEA</code> is an alternative approach that uses a statistical
measure from the omics data (e.g. the log fold change or significance)
to rank the genes. An “enrichment score” is calculated for each
annotation set based on where the genes annotated to the term sit in the
overall distribution.</p>
<p>Let’s analyze those same data with the
<code>fgsea::fgseaMultilevel()</code> function. First we’ll specify the
gene list and use the <code>log2FoldChange</code> value to rank the
genes in the list.</p>
<div class="codewrapper sourceCode" id="cb83">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene.list</span> <span class="op">&lt;-</span> <span class="va">fad.deg</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">age</span> <span class="op">==</span> <span class="st">'6m'</span>, <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>,<span class="op">!</span><span class="fu">duplicated</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">log2FoldChange</span>, name <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span></span></code></pre>
</div>
<p>We’ll also need a list of GO terms and the genes annotated. We can
get this from the <code>org.Mm.eg.db</code> annotation package using the
<code>AnnotationDbi::select</code> function</p>
<div class="codewrapper sourceCode" id="cb84">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get GO annotations</span></span>
<span><span class="va">go.terms</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span></span>
<span>    <span class="va">org.Mm.eg.db</span>, </span>
<span>    keys <span class="op">=</span> <span class="fu">keys</span><span class="op">(</span> <span class="va">org.Mm.eg.db</span> , keytype <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span>,</span>
<span>    columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"GO"</span>, <span class="st">"SYMBOL"</span><span class="op">)</span>,</span>
<span>    keytype <span class="op">=</span> <span class="st">"SYMBOL"</span> </span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the list of gene sets</span></span>
<span><span class="va">go.gene.sets</span> <span class="op">&lt;-</span> <span class="fu">split</span><span class="op">(</span><span class="va">go.terms</span><span class="op">$</span><span class="va">SYMBOL</span>, <span class="va">go.terms</span><span class="op">$</span><span class="va">GO</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter gene sets by size (minSize and maxSize are arguments in fgsea)</span></span>
<span><span class="va">min_size</span> <span class="op">&lt;-</span> <span class="fl">15</span></span>
<span><span class="va">max_size</span> <span class="op">&lt;-</span> <span class="fl">500</span></span>
<span><span class="va">go.sets.filtered</span> <span class="op">&lt;-</span> <span class="va">go.gene.sets</span><span class="op">[</span><span class="fu">sapply</span><span class="op">(</span><span class="va">go.gene.sets</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">length</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">min_size</span> <span class="op">&amp;&amp;</span> <span class="fu">length</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="va">max_size</span><span class="op">)</span><span class="op">]</span></span></code></pre>
</div>
<p>Now we’ll test for enrichment:</p>
<div class="codewrapper sourceCode" id="cb85">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr</span> <span class="op">&lt;-</span> <span class="fu">fgsea</span><span class="fu">::</span><span class="fu">fgseaMultilevel</span><span class="op">(</span> </span>
<span>    pathways <span class="op">=</span> <span class="va">go.sets.filtered</span>,</span>
<span>    stats <span class="op">=</span> <span class="va">gene.list</span>, </span>
<span>    minSize <span class="op">=</span> <span class="va">min_size</span>,</span>
<span>    maxSize <span class="op">=</span> <span class="va">max_size</span>,</span>
<span>    nproc <span class="op">=</span> <span class="fl">1</span> <span class="op">)</span></span></code></pre>
</div>
<p>Let’s take a look</p>
<div class="codewrapper sourceCode" id="cb86">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">head</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<p>This isn’t especially informative without the GO term names. Let’s
pull those in</p>
<div class="codewrapper sourceCode" id="cb87">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get term names from the GO.db package</span></span>
<span><span class="va">go_term_map</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span></span>
<span>    <span class="fu">GO.db</span><span class="fu">::</span><span class="va">GO.db</span>,</span>
<span>    keys <span class="op">=</span> <span class="va">gse.enr</span><span class="op">$</span><span class="va">pathway</span>,</span>
<span>    columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"GOID"</span>, <span class="st">"TERM"</span><span class="op">)</span>,</span>
<span>    keytype <span class="op">=</span> <span class="st">"GOID"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># join with results table</span></span>
<span><span class="va">gse.enr</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">gse.enr</span>, <span class="va">go_term_map</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>pathway <span class="op">=</span> <span class="va">GOID</span>, <span class="va">TERM</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">relocate</span><span class="op">(</span><span class="va">TERM</span>, .after <span class="op">=</span> <span class="va">pathway</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># now take another look</span></span>
<span><span class="va">gse.enr</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">head</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<p>This is more helpful. We can see that the GO terms associated with
up-regulated genes include “inflammatory response” and “chemotaxis”,
while there’s one term among the top 10 that is associated with
down-regulated genes, “chemical synaptic transmission”.</p>
<p>How many significant terms are identified:</p>
<div class="codewrapper sourceCode" id="cb88">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-8" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-8" class="callout-inner">
<h3 class="callout-title">Challenge 8</h3>
<div class="callout-content">
<p>The tally above represents all genes, both up- and down-regulatd. To
compare between GSEA and ORA, can you identify how many GSEA enriched
terms are associated with up-regulated genes and how many are associated
with down-regulated genes? (Hint: the <code>NES</code> value within the
results relates to the directionality of enrichment).</p>
</div>
</div>
</div>
<div id="accordionSolution8" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution8" aria-expanded="false" aria-controls="collapseSolution8">
  <h4 class="accordion-header" id="headingSolution8"> Show me the solution </h4>
</button>
<div id="collapseSolution8" class="accordion-collapse collapse" aria-labelledby="headingSolution8" data-bs-parent="#accordionSolution8">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb89">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span>  </span>
<span><span class="va">gse.enr</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>40 terms are up-regulated, while only 2 are associated with
down-regulated genes</p>
</div>
</div>
</div>
</div>
<div id="challenge-9" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-9" class="callout-inner">
<h3 class="callout-title">Challenge 9</h3>
<div class="callout-content">
<p>You may notice that your numbers of significantly enriched terms are
slightly different. Why would this be the case?</p>
</div>
</div>
</div>
<div id="accordionSolution9" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution9" aria-expanded="false" aria-controls="collapseSolution9">
  <h4 class="accordion-header" id="headingSolution9"> Show me the solution </h4>
</button>
<div id="collapseSolution9" class="accordion-collapse collapse" aria-labelledby="headingSolution9" data-bs-parent="#accordionSolution9">
<div class="accordion-body">
<p><code>fgsea</code> uses a permutation-based approach to estimate the
significance of the gene set enrichments. Because this involves random
sampling, running the analysis multiple times with the same settings can
result in subtle, minor differences in the output, particularly in the
p-values.</p>
<p>Increasing the <code>nPermSimple</code> parameter increases the
number of permutations performed. This leads to a more thorough sampling
of the null distribution, thereby improving the precision of the
estimated p-values, especially for highly significant pathways. However,
it’s a trade-off, as a higher <code>nPermSimple</code> value will also
increase the computational time required to run the analysis.</p>
</div>
</div>
</div>
</div>
<p>Let’s compute the enriched terms for all age groups:</p>
<div class="codewrapper sourceCode" id="cb90">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fad.enr</span> <span class="op">&lt;-</span> <span class="va">fad.deg</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span></span>
<span>          , <span class="va">padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span></span>
<span>          <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">age</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>gl <span class="op">=</span> <span class="va">log2FoldChange</span> <span class="op">%&gt;%</span> <span class="fu">setNames</span><span class="op">(</span><span class="va">.</span>, <span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">sort</span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>  gse <span class="op">=</span> <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="va">gl</span>,</span>
<span>    <span class="op">~</span> <span class="fu">fgsea</span><span class="fu">::</span><span class="fu">fgseaMultilevel</span><span class="op">(</span></span>
<span>      pathways <span class="op">=</span>  <span class="va">go.sets.filtered</span>,</span>
<span>      stats <span class="op">=</span> <span class="va">.x</span>,</span>
<span>      minSize <span class="op">=</span> <span class="va">min_size</span>,</span>
<span>      maxSize <span class="op">=</span> <span class="va">max_size</span>,</span>
<span>      nproc <span class="op">=</span> <span class="fl">1</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span>,</span>
<span>  res <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">gse</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>    <span class="va">go_term_map</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span></span>
<span>      <span class="fu">GO.db</span><span class="fu">::</span><span class="va">GO.db</span>,</span>
<span>      keys <span class="op">=</span> <span class="va">.x</span><span class="op">$</span><span class="va">pathway</span>,</span>
<span>      columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"GOID"</span>, <span class="st">"TERM"</span><span class="op">)</span>,</span>
<span>      keytype <span class="op">=</span> <span class="st">"GOID"</span></span>
<span>    <span class="op">)</span></span>
<span>    </span>
<span>    <span class="fu">inner_join</span><span class="op">(</span><span class="va">.x</span>, <span class="va">go_term_map</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>pathway <span class="op">=</span> <span class="va">GOID</span>, <span class="va">TERM</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">relocate</span><span class="op">(</span><span class="va">TERM</span>, .after <span class="op">=</span> <span class="va">pathway</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">saveRDS</span><span class="op">(</span><span class="va">fad.enr</span>, <span class="fu">here</span><span class="fu">::</span><span class="fu">here</span><span class="op">(</span><span class="st">'results'</span>, <span class="st">'5xFAD_fgsea_results.rds'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="common-pitfalls-best-practices">Common pitfalls &amp; best practices<a class="anchor" aria-label="anchor" href="#common-pitfalls-best-practices"></a></h3>
<p>These kinds of functional enrichment analyses are very common, but
not all results reported are equal! A recent paper describes an <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009935" class="external-link">“Urgent
need for consistent standards in functional enrichment analysis”</a>.
They examine how the results of functional enrichment analyses are
reported in the literature, and identify several common shortcomings.
Watch out for these common mistakes when performing and reporting your
own analyses! We’ll have more opportunities to discuss issues with
reproducibility in computational biology research in future
sessions.</p>
<figure><img src="../fig/common_pitfalls_Wijesooriya_PLOS_CompBio.png" alt="Figure from the paper showing the number of papers examined that failed to meet certain standards for reporting the results of functional enrichment testing." class="figure mx-auto d-block"><div class="figcaption">Common pitfalls</div>
</figure></div>
</section><section><h2 class="section-heading" id="alzheimerss-disease-biological-domains">[6] Alzheimers’s Disease Biological Domains<a class="anchor" aria-label="anchor" href="#alzheimerss-disease-biological-domains"></a></h2>
<hr class="half-width"><p>While these results are informative and help to provide essential
biological context to the results of the omics experiment, it is
difficult to understand all of the enriched terms and what that means
for the biology of disease. It would be useful to group each of the
enriched terms within broader areas of disease biology. There are
several common molecular endophenotypes that are detected within human
Alzheimer’s cohorts across omics data types (transcriptomics,
proteomics, and genetics). Several of these endophenotypic areas are
shown in the figure below (source: Jesse Wiley).</p>
<figure><img src="../fig/AD_endophenotypes.png" alt="An image of a brain surrounded by different endophenotype categories that are associated with AD." class="figure mx-auto d-block"><div class="figcaption">Endophenotypes of AD</div>
</figure><p>These endophenotypes describe molecular processes that are
dysregulated by or during the disease process. Very similar sets of
endophenotypes have been identified among the targets of therapeutics in
clinical trials for AD (see below, <a href="https://doi.org/10.1002/trc2.12465" class="external-link">Cummings et al, Alzheimer’s
disease drug development pipeline: 2024, Alz Dem TRCI</a>.</p>
<figure><img src="../fig/CADRO_pipeline_2024.png" alt="Circle plot showing therapeutics in various stages of clinical trials coded by the CADRO target class." class="figure mx-auto d-block"><div class="figcaption">Common AD Research Ontology</div>
</figure><figure><img src="../fig/CADRO_agent_distribution_2024.png" alt="Bar plot showing the numbers of agents in clinical trials across each class." class="figure mx-auto d-block"><div class="figcaption">Drugs in Pipeline by CADRO</div>
</figure><p>In order to formalize and operationalize these endophenotypic areas,
the <a href="../treatad.org/emory-sage-sgc">Emory-Sage-SGC-JAX (ESSJ)
TREAT-AD center</a> has developed the <a href="https://alz-journals.onlinelibrary.wiley.com/doi/pdfdirect/10.1002/trc2.12461" class="external-link">Biological
Domains of AD</a>. In the enumeration of these domains, several criteria
were established; the defined biological domains should be:</p>
<ul><li>
<em>objective:</em> leverage existing well-annotated resources<br></li>
<li>
<em>automatable:</em> in CADRO, therapeutics and targets are
manually assigned<br></li>
<li>
<em>intelligible:</em> groupings should be easy to understand<br></li>
<li>
<em>modifiable:</em> definitions should be (and are!) continuously
be updated.</li>
</ul><p>In all, 19 distinct biological domains (aka biodomains) have been
identified. These biological domains are defined using sets of GO terms
that align with the endophenotype. For example, terms like “cytokine
receptor binding” and “leukocyte migration” are annotated to the
<code>Immune Response</code> biodomain, while terms like “postsynaptic
density” and “synaptic vesicle cycle” are annotated to the
<code>Synapse</code> biodomain. Of all terms in the GO, 6,837 terms
(15.7%) are annotated to one of the biological domains. Because the GO
terms have genes annotated, genes can be associated with specific
endophenotypes via the biological domain groupings of terms. In all,
16,275 genes are annotated to at least 1 biodomain term. While the
biodomains exhibit very few overlapping GO terms (panel A), due to gene
pleiotropy (etc) the number of overlapping genes between biological
domains is quite a bit higher (panel B). The development of the
biological domains is described in more detail in the published paper <a href="../doi.org/10.1002/trc2.12461">Genetic and Multi-omic Risk Assessment
of Alzheimer’s Disease Implicates Core Associated Biological
Domains</a>.</p>
<figure><img src="../fig/AD_biodomains.png" alt="Heatmaps showing the fraction of overlapping GO terms and genes between the various biological domains." class="figure mx-auto d-block"><div class="figcaption">AD Biodomain Demographics</div>
</figure><p>The ESSJ TREAT-AD center has developed approaches to group terms
within each biodomain into functionally coherent sub-domains. The
sub-domains are driven by gene co-annotation and disease risk score
enrichment.</p>
<figure><img src="../fig/AD_subdomains.png" alt="Plot showing the number of subdomains associated with each domain and how many GO terms are assigned to each domain." class="figure mx-auto d-block"><div class="figcaption">AD Subdomains</div>
</figure><div class="section level3">
<h3 id="download-and-explore-the-biological-domain-annotations">Download and explore the biological domain annotations<a class="anchor" aria-label="anchor" href="#download-and-explore-the-biological-domain-annotations"></a></h3>
<p>First let’s download the biodomain definition file from synapse.</p>
<div class="codewrapper sourceCode" id="cb91">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># biodomain definitions</span></span>
<span><span class="va">biodom</span> <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">'syn25428992'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># biodomain labels and colors</span></span>
<span><span class="va">dom.lab</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">syn</span><span class="op">$</span><span class="fu">get</span><span class="op">(</span><span class="st">'syn26856828'</span><span class="op">)</span><span class="op">$</span><span class="va">path</span><span class="op">)</span></span></code></pre>
</div>
<p>What is in the <code>dom.lab</code> file?</p>
<div class="codewrapper sourceCode" id="cb92">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">glimpse</span><span class="op">(</span><span class="va">dom.lab</span><span class="op">)</span></span></code></pre>
</div>
<p>This file contains some useful standardized abbreviations and colors
that will be useful as we work with and plot domain information
later.</p>
<p>What is in the <code>biodom</code> file?</p>
<div class="codewrapper sourceCode" id="cb93">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">glimpse</span><span class="op">(</span><span class="va">biodom</span><span class="op">)</span></span></code></pre>
</div>
<p>You can see the file is a list of GO term accessions
(<code>GO_ID</code>) and names (<code>GOterm_Name</code>) as well as the
corresponding endophenotype areas (e.g. <code>Biodomain</code> and
<code>Subdomain</code>) to which the term is annotated. There are also
gene annotations for each term. These annotation sets come from two
sources: (1) the <code>symbol</code> and <code>uniprot</code>
annotations are drawn directly from the Gene Ontology via the provided
API, (2) <code>ensembl_id</code>, <code>entrez_id</code>, and
<code>hgnc_symbol</code> are from BioMart annotations
(e.g. <code>biomaRt::getLDS()</code>).</p>
<p>We can see how many GO terms are annotated to each biodomain:</p>
<div class="codewrapper sourceCode" id="cb94">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biodom</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarise</span><span class="op">(</span>n_term <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">GO_ID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">n_term</span><span class="op">)</span> </span></code></pre>
</div>
<p>The biodomains range in size from <code>Tau Homeostasis</code> (10
terms) up to <code>Synapse</code> (1,379 terms).</p>
<p>What about the size of the subdomains? For simplicity let’s focus on
the Subdomains within the <code>Synapse</code> domain.</p>
<div class="codewrapper sourceCode" id="cb95">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biodom</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="st">'Synapse'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarise</span><span class="op">(</span>n_term <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">GO_ID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">n_term</span><span class="op">)</span> </span></code></pre>
</div>
<p>There are 8 subdomains within the <code>Synapse</code> biodomain,
plus a set of terms that aren’t assigned to any subdomain (i.e. where
<code>Subdomain</code> is <code>NA</code>). The subdomains range in size
from <code>axon regeneration</code> (20 terms) up to
<code>trans-synaptic signaling</code> (162 terms). There are 822 terms
within the <code>Synapse</code> domain that aren’t assigned to any
sub-domains.</p>
<p>We can also investigate the individual genes annotated to each
biodomain GO term.</p>
<div class="codewrapper sourceCode" id="cb96">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">GOterm_Name</span> <span class="op">==</span> <span class="st">'amyloid-beta formation'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unlist</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>So the genes associated with <code>amyloid-beta formation</code>
within the <code>APP Metabolism</code> biodomain include PSEN1, PSEN2,
BACE1, ABCA7, NCSTN, and others.</p>
</div>
<div class="section level3">
<h3 id="annotate-enriched-terms-with-biological-domain">Annotate enriched terms with biological domain<a class="anchor" aria-label="anchor" href="#annotate-enriched-terms-with-biological-domain"></a></h3>
<p>Let’s re-characterize the 5xFAD functional enrichments we computed
previously with the biological domain annotations and see if we can get
more context about what is changing in that model. We’ll focus on the
GSEA results and start by annotating the results with biodomain
groupings.</p>
<div class="codewrapper sourceCode" id="cb97">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr.bd</span> <span class="op">=</span> <span class="va">gse.enr</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, pathway <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'pathway'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">relocate</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">gse.enr.bd</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">pathway</span>, <span class="va">TERM</span>, <span class="va">Biodomain</span>, <span class="va">padj</span>, <span class="va">NES</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<p>Not all of the enriched terms are annotated to a biological domain.
Some terms are too broad and not specific (e.g. ‘cell mophogenesis’ or
‘MAPK cascade’), while others may not have been captured by a biological
domain annotation yet (e.g. ‘transcription cis-regulatory region
binding’). <em>Remember</em> that the conception of the biodomains
involved a requirement that they be modifiable, and these terms may be
added to the biodomain definintions in the future. Let’s modify the
<code>Biodomain</code> value for terms that aren’t annotated to a domain
to ‘none’.</p>
<div class="codewrapper sourceCode" id="cb98">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse.enr.bd</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">[</span><span class="fu">is.na</span><span class="op">(</span><span class="va">gse.enr.bd</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'no domain'</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">gse.enr.bd</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">pathway</span>, <span class="va">TERM</span>, <span class="va">Biodomain</span>, <span class="va">padj</span>, <span class="va">NES</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<p>How many terms are enriched from each domain?</p>
<div class="codewrapper sourceCode" id="cb99">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bd.tally</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span>domain <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'no domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    n_term <span class="op">=</span> <span class="va">biodom</span><span class="op">$</span><span class="va">GO_ID</span><span class="op">[</span> <span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span> <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    n_sig_term <span class="op">=</span> <span class="va">gse.enr.bd</span><span class="op">$</span><span class="va">pathway</span><span class="op">[</span> <span class="va">gse.enr.bd</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span> <span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu">arrange</span><span class="op">(</span><span class="va">bd.tally</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">n_sig_term</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Many enriched terms don’t map to a domain r bd.tally %&gt;%
filter(domain == ‘no domain’) %&gt;% pull(n_sig_term) %&gt;% sum()), but
nearly half do (r bd.tally %&gt;% filter(domain != ‘no domain’) %&gt;%
pull(n_sig_term) %&gt;% sum()). Of those that do, the vast majority map
into the <code>Immune Response</code> biodomain.</p>
<p>We can plot the enrichment results, stratified by biodomain:</p>
<div class="codewrapper sourceCode" id="cb100">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">gse.enr.bd</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">Biodomain</span>, levels <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">bd.tally</span>, <span class="va">n_sig_term</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">domain</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">padj</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, </span>
<span>              color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>This makes it really clear that in the 6 month old 5xFAD mice, the
enriched <code>Immune Response</code> domain terms are all associated
with up-regulated genes, while the enriched <code>Synapse</code> domain
terms are associated with a mix of up- and down-regulated genes. It also
highlights the several other domains with significantly enriched terms
(e.g. <code>Proteostasis</code>, <code>Structural Stabilization</code>,
<code>Lipid Metabolism</code>, <code>Endolysosome</code>, etc; even
<code>APP Metabolism</code>).</p>
<p>Now let’s look across age groups to see how domain enrichments
change:</p>
<div class="codewrapper sourceCode" id="cb101">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span> <span class="va">fad.enr</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">age</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>pathway <span class="op">=</span> <span class="va">GO_ID</span>, <span class="va">Biodomain</span>, <span class="va">Subdomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Biodomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Biodomain</span><span class="op">)</span>,</span>
<span>    Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    age <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">age</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4m'</span>,<span class="st">'6m'</span>,<span class="st">'10m'</span><span class="op">)</span><span class="op">)</span></span>
<span>         <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_sig <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">fct_reorder</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">n_sig</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">padj</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">age</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, </span>
<span>              color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>The earliest transcriptomic changes in the 5xFAD animals are
associated with <code>Immune Response</code> and
<code>Lipid Metabolism</code> domain terms. By 6 months, there are
enrichments for several other domains (as above), and by 10 months the
term enrichments are even more exaggerated (i.e. <em>more</em>
significance, larger NES values, etc).</p>
<p>We can also break down these results by sub-domain to get a more
clear idea of the processes affected in each case</p>
<div class="codewrapper sourceCode" id="cb102">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span> <span class="va">fad.enr</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">age</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>pathway <span class="op">=</span> <span class="va">GO_ID</span>, <span class="va">Biodomain</span>, <span class="va">Subdomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Biodomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Biodomain</span><span class="op">)</span>,</span>
<span>    Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    sd_lab <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">abbr</span>,<span class="st">'_'</span>,<span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    age <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">age</span>, levels <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'4m'</span>,<span class="st">'6m'</span>,<span class="st">'10m'</span><span class="op">)</span><span class="op">)</span></span>
<span>         <span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">sd_lab</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">age</span><span class="op">)</span>, rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">abbr</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">'free'</span>, space <span class="op">=</span> <span class="st">'free_y'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>label <span class="op">=</span> <span class="op">~</span> <span class="fu">str_remove_all</span><span class="op">(</span><span class="va">.x</span>, <span class="st">'^[A-Za-z]{2}_'</span><span class="op">)</span>, drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>What trends do you notice from these results? Which sub-domain
processes are affected at the earliest ages? Are there any sub-domains
that change across the age groups?</p>
<div id="challenge-10" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-10" class="callout-inner">
<h3 class="callout-title">Challenge 10</h3>
<div class="callout-content">
<p>How could you plot the results from the ORA to show biodomain
enrichements?</p>
</div>
</div>
</div>
<div id="accordionSolution10" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution10" aria-expanded="false" aria-controls="collapseSolution10">
  <h4 class="accordion-header" id="headingSolution10"> Show me the solution </h4>
</button>
<div id="collapseSolution10" class="accordion-collapse collapse" aria-labelledby="headingSolution10" data-bs-parent="#accordionSolution10">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb103">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.ora</span> <span class="op">=</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">enr.up</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>dir <span class="op">=</span> <span class="st">'up'</span><span class="op">)</span>,</span>
<span>                    <span class="va">enr.dn</span><span class="op">@</span><span class="va">result</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>dir <span class="op">=</span> <span class="st">'dn'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, ID <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'ID'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">relocate</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span></span>
<span></span>
<span><span class="va">enr.ora</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">[</span><span class="fu">is.na</span><span class="op">(</span><span class="va">enr.ora</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'none'</span></span>
<span></span>
<span><span class="va">bd.tally</span> <span class="op">=</span> <span class="fu">tibble</span><span class="op">(</span>domain <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    n_term <span class="op">=</span> <span class="va">biodom</span><span class="op">$</span><span class="va">GO_ID</span><span class="op">[</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    n_sig_term <span class="op">=</span> <span class="va">enr.ora</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="va">enr.ora</span><span class="op">$</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">domain</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">unique</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">length</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">enr.ora</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">enr.ora</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">Biodomain</span>, levels <span class="op">=</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">bd.tally</span>, <span class="va">n_sig_term</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">domain</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">p.adjust</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    signed_logP <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">p.adjust</span><span class="op">)</span>,</span>
<span>    signed_logP <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">dir</span> <span class="op">==</span> <span class="st">'dn'</span>, <span class="op">-</span><span class="fl">1</span> <span class="op">*</span> <span class="va">signed_logP</span>, <span class="va">signed_logP</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.ora</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">signed_logP</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.ora</span>, <span class="va">dir</span> <span class="op">==</span> <span class="st">'up'</span><span class="op">)</span>,</span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>,</span>
<span>              scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.ora</span>, <span class="va">dir</span> <span class="op">==</span> <span class="st">'dn'</span><span class="op">)</span>,</span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>,</span>
<span>              scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span></span>
<span>    <span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="va">Count</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>,</span>
<span>    color <span class="op">=</span> <span class="st">'grey20'</span>,</span>
<span>    shape <span class="op">=</span> <span class="fl">21</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">.3</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Based on the gene list (up or down) we can add a sign to the
significance. When we plot this we can see there are many more
significantly enriched terms from the ORA. The dominant signal is still
the up-regulation of terms from the <code>Immune Response</code>
biodomain. There is also nearly exclusive up-regulation of terms from
the <code>Autophagy</code>, <code>Oxidative Stress</code>, and
<code>APP Metabolism</code> domains. The most down-regulated terms are
from the <code>Synapse</code> biodomain.</p>
</div>
</div>
</div>
</div>
<div id="challenge-11" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-11" class="callout-inner">
<h3 class="callout-title">Challenge 11</h3>
<div class="callout-content">
<p>Which biodomain terms are over-represented among the gene
co-expression modules? Choose one module and test for over-represented
terms at a <code>p.adjust</code> value less than <code>0.05</code>.</p>
<p>What if you look at the <em>most</em> significantly enriched terms,
with <code>p.adjust</code> values less than <code>1e-5</code>?</p>
</div>
</div>
</div>
<div id="accordionSolution11" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution11" aria-expanded="false" aria-controls="collapseSolution11">
  <h4 class="accordion-header" id="headingSolution11"> Show me the solution </h4>
</button>
<div id="collapseSolution11" class="accordion-collapse collapse" aria-labelledby="headingSolution11" data-bs-parent="#accordionSolution11">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb104">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stg_blue.ora</span> <span class="op">&lt;-</span> <span class="va">ampad_modules</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">module</span> <span class="op">==</span> <span class="st">'STGblue'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">Mouse_gene_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">enrichGO</span><span class="op">(</span><span class="va">.</span>, <span class="va">org.Mm.eg.db</span>, keyType <span class="op">=</span> <span class="st">'SYMBOL'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">inner_join</span><span class="op">(</span><span class="va">stg_blue.ora</span><span class="op">@</span><span class="va">result</span>,</span>
<span>           <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>ID <span class="op">=</span> <span class="va">GO_ID</span>, <span class="va">Biodomain</span><span class="op">:</span><span class="va">Subdomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">inner_join</span><span class="op">(</span><span class="va">stg_blue.ora</span><span class="op">@</span><span class="va">result</span>,</span>
<span>           <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>ID <span class="op">=</span> <span class="va">GO_ID</span>, <span class="va">Biodomain</span><span class="op">:</span><span class="va">Subdomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;=</span> <span class="fl">1e-5</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>n <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">ID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>At the default p-value the domains with the most terms enriched
include <code>Synapse</code>, <code>Lipid Metabolism</code>, and
<code>Immune Response</code>. If we filter to the <em>most</em>
significantly enriched terms, terms from the
<code>Immune Response</code> and <code>Structural Stabilization</code>
domains are represented.</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="comparing-enrichments">Comparing enrichments<a class="anchor" aria-label="anchor" href="#comparing-enrichments"></a></h3>
<p>Now we know which domain processes are affected in the mouse model,
let’s consider how we can compare the functional enrichments in these
models with functional enrichments from the AMP-AD cohorts. We’ll need
to compute and examine the functional enrichments from the human cohort
data.</p>
<p>First let’s get the human GO set gene annotations</p>
<div class="codewrapper sourceCode" id="cb105">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get GO annotations</span></span>
<span><span class="va">hs.go.terms</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span></span>
<span>    <span class="va">org.Hs.eg.db</span>, </span>
<span>    keys <span class="op">=</span> <span class="fu">keys</span><span class="op">(</span> <span class="va">org.Hs.eg.db</span> , keytype <span class="op">=</span> <span class="st">"SYMBOL"</span><span class="op">)</span>,</span>
<span>    columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"GO"</span>, <span class="st">"SYMBOL"</span><span class="op">)</span>,</span>
<span>    keytype <span class="op">=</span> <span class="st">"SYMBOL"</span> </span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the list of gene sets</span></span>
<span><span class="va">hs.go.gene.sets</span> <span class="op">&lt;-</span> <span class="fu">split</span><span class="op">(</span><span class="va">hs.go.terms</span><span class="op">$</span><span class="va">SYMBOL</span>, <span class="va">hs.go.terms</span><span class="op">$</span><span class="va">GO</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter gene sets by size (minSize and maxSize are arguments in fgsea)</span></span>
<span><span class="va">min_size</span> <span class="op">&lt;-</span> <span class="fl">15</span></span>
<span><span class="va">max_size</span> <span class="op">&lt;-</span> <span class="fl">500</span></span>
<span><span class="va">hs.go.sets.filtered</span> <span class="op">&lt;-</span> <span class="va">hs.go.gene.sets</span><span class="op">[</span><span class="fu">sapply</span><span class="op">(</span><span class="va">hs.go.gene.sets</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">length</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">min_size</span> <span class="op">&amp;&amp;</span> <span class="fu">length</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="va">max_size</span><span class="op">)</span><span class="op">]</span></span></code></pre>
</div>
<p>Now we can run the enrichments</p>
<div class="codewrapper sourceCode" id="cb106">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs.gsea</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>, </span>
<span>    <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span>,</span>
<span>    <span class="va">Tissue</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">'DLPFC'</span>, <span class="st">'PHG'</span>,<span class="st">'TCX'</span><span class="op">)</span>,</span>
<span>    <span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">hgnc_symbol</span><span class="op">)</span></span>
<span>    , <span class="va">adj.P.Val</span> <span class="op">&lt;=</span> <span class="fl">0.05</span></span>
<span>    <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Study</span>, <span class="va">Tissue</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarise</span><span class="op">(</span>gl <span class="op">=</span> <span class="va">logFC</span> <span class="op">%&gt;%</span> <span class="fu">setNames</span><span class="op">(</span><span class="va">.</span>, <span class="va">hgnc_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">sort</span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    gse <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">gl</span>, </span>
<span>        <span class="op">~</span><span class="fu">fgsea</span><span class="fu">::</span><span class="fu">fgseaMultilevel</span><span class="op">(</span></span>
<span>              pathways <span class="op">=</span>  <span class="va">hs.go.sets.filtered</span>,</span>
<span>              stats <span class="op">=</span> <span class="va">.x</span>,</span>
<span>              minSize <span class="op">=</span> <span class="va">min_size</span>,</span>
<span>              maxSize <span class="op">=</span> <span class="va">max_size</span>,</span>
<span>              nproc <span class="op">=</span> <span class="fl">1</span></span>
<span>            <span class="op">)</span><span class="op">)</span>,</span>
<span>    res <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">gse</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>            <span class="va">go_term_map</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span></span>
<span>              <span class="fu">GO.db</span><span class="fu">::</span><span class="va">GO.db</span>,</span>
<span>              keys <span class="op">=</span> <span class="va">.x</span><span class="op">$</span><span class="va">pathway</span>,</span>
<span>              columns <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"GOID"</span>, <span class="st">"TERM"</span><span class="op">)</span>,</span>
<span>              keytype <span class="op">=</span> <span class="st">"GOID"</span></span>
<span>            <span class="op">)</span></span>
<span>            </span>
<span>            <span class="fu">inner_join</span><span class="op">(</span><span class="va">.x</span>, <span class="va">go_term_map</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>pathway <span class="op">=</span> <span class="va">GOID</span>, <span class="va">TERM</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu">relocate</span><span class="op">(</span><span class="va">TERM</span>, .after <span class="op">=</span> <span class="va">pathway</span><span class="op">)</span></span>
<span>          <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Let’s map the biodomains onto the enriched GO terms, and plot the
results</p>
<div class="codewrapper sourceCode" id="cb107">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span> <span class="va">hs.gsea</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Study</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">Subdomain</span>, pathway <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'pathway'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Biodomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Biodomain</span><span class="op">)</span>,</span>
<span>    Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span></span>
<span>         <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_sig <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">fct_reorder</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">n_sig</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">padj</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">Study</span><span class="op">)</span>, space <span class="op">=</span> <span class="st">'free'</span>, scales <span class="op">=</span> <span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, </span>
<span>              color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>We can see fairly similar domain term enrichments from the
transcriptomic data across AMP-AD cohorts. There is strong evidence for
up-regulation among transcripts annotated to GO terms in the
<code>Immune Response</code>, <code>Structural Stabilization</code>, and
<code>Vasculature</code> domains, along with similarly strong evidence
of down-regulation among transcripts annotated to GO terms in the
<code>Synapse</code> domain.</p>
<p>Now lets compare enrichments between species. First we’ll combine the
enrichment results for both species and plot them side-by-side:</p>
<div class="codewrapper sourceCode" id="cb108">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enr.bd_plot</span> <span class="op">&lt;-</span>  </span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span></span>
<span>    <span class="va">fad.enr</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'5xFAD, '</span>,<span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">model</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span>,</span>
<span>    <span class="va">hs.gsea</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">Study</span>,<span class="st">', '</span>, <span class="va">Tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">model</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>  </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">Subdomain</span>, pathway <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'pathway'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">model</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    Biodomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Biodomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Biodomain</span><span class="op">)</span>,</span>
<span>    Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span></span>
<span>         <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_sig <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="fu">unique</span><span class="op">(</span><span class="va">pathway</span><span class="op">)</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">Biodomain</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Biodomain <span class="op">=</span> <span class="fu">fct_reorder</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">n_sig</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">padj</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">model</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">enr.bd_plot</span>, <span class="va">NES</span> <span class="op">&lt;</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>              <span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, scale <span class="op">=</span> <span class="st">'width'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_jitter</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>size <span class="op">=</span> <span class="op">-</span><span class="fu">log10</span><span class="op">(</span><span class="va">padj</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span>, </span>
<span>              color <span class="op">=</span> <span class="st">'grey20'</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>drop <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">scale_fill_identity</span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu">scale_color_identity</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Ok, there’s definitely similarity in terms of the direction of
enrichment for domain terms between the human and mouse data. But how
can we assess if the same processes are affected in both mouse and
human? Well, let’s take a look at <code>Immune Response</code> terms
more specifically.</p>
<div class="codewrapper sourceCode" id="cb109">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comb.gsea</span> <span class="op">&lt;-</span> <span class="va">fad.enr</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'5xFAD, '</span>,<span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">model</span>, <span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">hs.gsea</span><span class="op">$</span><span class="va">res</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">pathway</span>, hs.NES <span class="op">=</span> <span class="va">NES</span>, hs.padj <span class="op">=</span> <span class="va">padj</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>, <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">Subdomain</span>, pathway <span class="op">=</span> <span class="va">GO_ID</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'pathway'</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">contains</span><span class="op">(</span><span class="st">'NES'</span><span class="op">)</span>, <span class="op">~</span><span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>, <span class="fl">0</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">comb.gsea</span>, <span class="va">Biodomain</span> <span class="op">==</span> <span class="st">"Immune Response"</span><span class="op">)</span>, </span>
<span>  <span class="fu">aes</span><span class="op">(</span><span class="va">NES</span>, <span class="va">hs.NES</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">model</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>In this case, there are several <code>Immune Response</code> terms
that are significantly enriched in both datasets and all overlapping
terms are enriched with a positive NES value, indicating they are
up-regulated processes in both AD vs Control and the 5xFAD vs WT. We can
more systematically compare enriched terms within each Biodomain. First,
lets generate a list of GO terms that is specific to each Biodomain and
each Subdomain.</p>
<div class="codewrapper sourceCode" id="cb110">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># curate biodomain and sub-domain gene lists</span></span>
<span><span class="va">bd.terms</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span> set <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span> terms <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">set</span>, <span class="op">~</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">GO_ID</span><span class="op">)</span><span class="op">)</span> <span class="op">)</span></span>
<span></span>
<span><span class="va">sd.terms</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span> <span class="va">biodom</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span><span class="op">=</span><span class="st">'domain'</span><span class="op">)</span> <span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span> Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">abbr</span>, <span class="va">Subdomain</span>, <span class="va">subdomain_idx</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    set <span class="op">=</span> <span class="fu">paste0</span><span class="op">(</span><span class="va">abbr</span>,<span class="st">'_'</span>,<span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    bd <span class="op">=</span> <span class="va">Biodomain</span>, sdi <span class="op">=</span> <span class="va">subdomain_idx</span>,</span>
<span>    terms <span class="op">=</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">bd</span>, <span class="va">subdomain_idx</span> <span class="op">==</span> <span class="va">sdi</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">GO_ID</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co"># combine</span></span>
<span><span class="va">bd.term.list</span> <span class="op">=</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">bd.terms</span>, <span class="va">sd.terms</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">set</span>, <span class="va">terms</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">!=</span> <span class="st">'NA_none'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">bd.terms</span>, <span class="va">sd.terms</span><span class="op">)</span></span></code></pre>
</div>
<p>Next let’s combine all of the human GSEA results with each of the
5xFAD age-stratified GSEA results. Any term that is enriched with an
adjusted p-value &gt; 0 we will set the NES value to 0 and the p-value
to 1.</p>
<div class="codewrapper sourceCode" id="cb111">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fad.vs.amp</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span>fad <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'5xFAD, '</span>, <span class="va">fad.enr</span><span class="op">$</span><span class="va">age</span><span class="op">)</span>, </span>
<span>                       amp <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">hs.gsea</span><span class="op">$</span><span class="va">Study</span>,<span class="st">', '</span>, <span class="va">hs.gsea</span><span class="op">$</span><span class="va">Tissue</span><span class="op">)</span> <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">fad</span>, <span class="va">amp</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>    <span class="va">mm</span> <span class="op">=</span> <span class="va">fad.enr</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'5xFAD, '</span>, <span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">model</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>          <span class="fu">select</span><span class="op">(</span>GOID <span class="op">=</span> <span class="va">pathway</span>, term <span class="op">=</span> <span class="va">TERM</span>, mm.nes <span class="op">=</span> <span class="va">NES</span>, mm.padj <span class="op">=</span> <span class="va">padj</span><span class="op">)</span></span>
<span>    <span class="va">hs</span> <span class="op">=</span> <span class="va">hs.gsea</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="va">Study</span>,<span class="st">', '</span>, <span class="va">Tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">model</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>          <span class="fu">select</span><span class="op">(</span>GOID <span class="op">=</span> <span class="va">pathway</span>, term <span class="op">=</span> <span class="va">TERM</span>, hs.nes <span class="op">=</span> <span class="va">NES</span>, hs.padj <span class="op">=</span> <span class="va">padj</span><span class="op">)</span></span>
<span>    <span class="fu">full_join</span><span class="op">(</span><span class="va">mm</span>, <span class="va">hs</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'GOID'</span>,<span class="st">'term'</span><span class="op">)</span>, na_matches <span class="op">=</span> <span class="st">'never'</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span></span>
<span>      <span class="fu">across</span><span class="op">(</span><span class="fu">contains</span><span class="op">(</span><span class="st">'padj'</span><span class="op">)</span>, <span class="op">~</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>      <span class="fu">across</span><span class="op">(</span><span class="fu">contains</span><span class="op">(</span><span class="st">'nes'</span><span class="op">)</span>, <span class="op">~</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>, <span class="fl">0</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>      hs.nes <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">hs.padj</span> <span class="op">&gt;</span> <span class="fl">0.05</span>, <span class="fl">0</span>, <span class="va">hs.nes</span><span class="op">)</span>,</span>
<span>      mm.nes <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">mm.padj</span> <span class="op">&gt;</span> <span class="fl">0.05</span>, <span class="fl">0</span>, <span class="va">mm.nes</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span>    <span class="op">}</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Then we can go through each of the GO term lists and perform the
correlation analysis. First we’ll make a list combining all mouse
comparisons and each GO term list to test using the
<code>crossing</code> function:</p>
<div class="codewrapper sourceCode" id="cb112">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="fu">crossing</span><span class="op">(</span><span class="va">fad.vs.amp</span>, set <span class="op">=</span> <span class="va">bd.term.list</span><span class="op">$</span><span class="va">set</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">comp</span><span class="op">)</span></span></code></pre>
</div>
<p>Now we can add a column for the term sub-set to consider for each
correlation:</p>
<div class="codewrapper sourceCode" id="cb113">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span> data <span class="op">=</span> <span class="fu">map2</span><span class="op">(</span><span class="va">data</span>, <span class="va">set</span>, <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">GOID</span> <span class="op">%in%</span> <span class="va">bd.term.list</span><span class="op">$</span><span class="va">terms</span><span class="op">[[</span><span class="fu">which</span><span class="op">(</span><span class="va">bd.term.list</span><span class="op">$</span><span class="va">set</span> <span class="op">==</span> <span class="va">.y</span><span class="op">)</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            n.terms <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="fu">nrow</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            n.sig.both <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="va">.x</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">mm.padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">hs.padj</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">nrow</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            jaccard <span class="op">=</span> <span class="va">n.sig.both</span><span class="op">/</span><span class="va">n.terms</span> <span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">comp</span><span class="op">)</span></span></code></pre>
</div>
<p>For several term sets there aren’t many terms enriched. We won’t be
able to compute correlations with these, so we’ll filter them out.</p>
<p>For the correlation we’ll use a nonparametric, rank-based Kendall
correlation. Given that we don’t expect NES values to be normally
distributed, and that all we’re really care to know is whether the terms
are enriched in the same direction or opposite directions between mouse
and human, this will do the job.</p>
<div class="codewrapper sourceCode" id="cb114">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">n.terms</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">'hs.nes'</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">'mm.nes'</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">'kendall'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">estimate</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor</span>, <span class="op">~</span><span class="fu">broom</span><span class="fu">::</span><span class="fu">tidy</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">comp</span><span class="op">)</span></span></code></pre>
</div>
<p>Ok, now let’s take a look at the correlation among enriched Biodomain
terms (without an <code>_</code> character in the <code>set</code>
name):</p>
<div class="codewrapper sourceCode" id="cb115">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">F</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>fad <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">fad</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'5xFAD, 10m'</span>,<span class="st">'5xFAD, 6m'</span>,<span class="st">'5xFAD, 4m'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set</span>, <span class="va">fad</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">amp</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>fill <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="va">jaccard</span> <span class="op">)</span>, color <span class="op">=</span> <span class="st">'black'</span>, shape <span class="op">=</span> <span class="fl">21</span>, stroke <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">p.value</span> <span class="op">&lt;=</span> <span class="fl">5e-2</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"black"</span>, shape<span class="op">=</span><span class="fl">0</span>, size<span class="op">=</span> <span class="fl">5</span>, stroke <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span> <span class="st">"Jaccard index"</span>, range <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">.5</span>, <span class="fl">4</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1e-90</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_gradient2</span><span class="op">(</span></span>
<span>    <span class="st">"Kendall's \u03C4 coefficient"</span>,</span>
<span>    low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, mid <span class="op">=</span> <span class="st">'grey95'</span></span>
<span>    , guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span>, drop <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">5</span>,<span class="fl">50</span>,<span class="fl">5</span>,<span class="fl">5</span><span class="op">)</span>,</span>
<span>    strip.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>,colour <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    legend.box <span class="op">=</span> <span class="st">'vertical'</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>The largest overlap is with the <code>Immune Response</code> domain
terms, which are positively correlated between older 5xFAD and the Mayo
and MSSM cohorts. There are also positively correlated terms for the
<code>Apoptosis</code>, <code>Autophagy</code>,
<code>Lipid Metabolism</code>, <code>Structural Stabilization</code>,
and <code>Vasculature</code> domains.</p>
<p>We can also look at the subdomains to get a more specific picture.
Let’s filter to only include significant correlations or any correlation
with a tau statistic with an absolute value &gt; 0.2:</p>
<div class="codewrapper sourceCode" id="cb116">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">l</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span></span>
<span>    <span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>,<span class="va">set</span><span class="op">)</span><span class="op">==</span><span class="cn">T</span>, </span>
<span>     <span class="op">(</span><span class="va">p.value</span> <span class="op">&lt;</span> <span class="fl">5e-2</span> <span class="op">|</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.3</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">set</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">%in%</span> <span class="va">l</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>max.n <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">n.terms</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    fad <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">fad</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'5xFAD, 10m'</span>,<span class="st">'5xFAD, 6m'</span>,<span class="st">'5xFAD, 4m'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">rev</span><span class="op">)</span>,</span>
<span>    set1 <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>,<span class="va">max.n</span>,<span class="st">'] '</span>,<span class="va">.</span><span class="op">)</span>,</span>
<span>    abbr <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>         <span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">fad</span>,<span class="va">set1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span>rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">abbr</span><span class="op">)</span>, cols <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">amp</span><span class="op">)</span>, scales<span class="op">=</span><span class="st">'free'</span>, space<span class="op">=</span><span class="st">'free'</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>fill <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="va">jaccard</span> <span class="op">)</span>, color <span class="op">=</span> <span class="st">'black'</span>, shape <span class="op">=</span> <span class="fl">21</span>, stroke <span class="op">=</span> <span class="fl">.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">p.value</span> <span class="op">&lt;=</span> <span class="fl">5e-2</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"black"</span>, shape<span class="op">=</span><span class="fl">0</span>, size<span class="op">=</span> <span class="fl">5</span>, stroke <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span> <span class="st">"Jaccard index"</span>, range <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">.5</span>, <span class="fl">4</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1e-90</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_gradient2</span><span class="op">(</span></span>
<span>    <span class="st">"Kendall's \u03C4 coefficient"</span>,</span>
<span>    low <span class="op">=</span> <span class="st">"#85070C"</span>, high <span class="op">=</span> <span class="st">"#164B6E"</span>, mid <span class="op">=</span> <span class="st">'grey95'</span></span>
<span>    , guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"top"</span>, drop <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># scale_y_discrete(position = 'right', limits = rev)+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">5</span>,<span class="fl">50</span>,<span class="fl">5</span>,<span class="fl">5</span><span class="op">)</span>,</span>
<span>    strip.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>,colour <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>,size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>, hjust <span class="op">=</span> <span class="fl">0.03</span>,size<span class="op">=</span><span class="fl">12</span>,face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    legend.box <span class="op">=</span> <span class="st">'vertical'</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>This tells us that the positive correlations in
<code>Immune Response</code> are primarily due to terms within “cytokine
production”, “neuroinflammatory response”, and “behavioral defense
response”, while the correlations in the <code>Apoptosis</code> domain
are primarily related to “NF-kappaB signaling”, and the late
<code>Vasculature</code> correlations have to do with
“angiogenesis”.</p>
<p>Are the phagocytosis subdomain terms really negatively
correlated?</p>
<div class="codewrapper sourceCode" id="cb117">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comp</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">==</span> <span class="st">'IR_phagocytosis'</span>, <span class="va">amp</span> <span class="op">==</span> <span class="st">'MAYO, TCX'</span>, <span class="va">fad</span> <span class="op">==</span> <span class="st">'5xFAD, 10m'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre>
</div>
<p>Not really – it is just that different terms from the subdomain are
enriched in each species. All terms from the subdomain are enriched
among up-regulated genes in each species.</p>
<div id="challenge-12" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<span class="callout-header">Challenge</span>
<div id="challenge-12" class="callout-inner">
<h3 class="callout-title">Challenge 12</h3>
<div class="callout-content">
<p>One could also perform a correlation analysis on a gene-by-gene basis
like we did with the co-expression modules, but instead using the genes
within each of the biodomains and subdomains. Compute these results and
compare the the correlations of enriched terms and modules.</p>
</div>
</div>
</div>
<div id="accordionSolution12" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution12" aria-expanded="false" aria-controls="collapseSolution12">
  <h4 class="accordion-header" id="headingSolution12"> Show me the solution </h4>
</button>
<div id="collapseSolution12" class="accordion-collapse collapse" aria-labelledby="headingSolution12" data-bs-parent="#accordionSolution12">
<div class="accordion-body">
<p>first set up the lists of genes associated with biodomain and
subdomain term lists</p>
<div class="codewrapper sourceCode" id="cb118">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bd.genes</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>set <span class="op">=</span> <span class="fu">unique</span><span class="op">(</span><span class="va">biodom</span><span class="op">$</span><span class="va">Biodomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>genes <span class="op">=</span> <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="va">set</span>,</span>
<span>    <span class="op">~</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unlist</span></span>
<span>  <span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sd.genes</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">biodom</span>, <span class="va">dom.lab</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Biodomain'</span> <span class="op">=</span> <span class="st">'domain'</span><span class="op">)</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Subdomain <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="fu">is.na</span><span class="op">(</span><span class="va">Subdomain</span><span class="op">)</span>, <span class="st">'none'</span>, <span class="va">Subdomain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">Biodomain</span>, <span class="va">abbr</span>, <span class="va">Subdomain</span>, <span class="va">subdomain_idx</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    set <span class="op">=</span> <span class="fu">paste0</span><span class="op">(</span><span class="va">abbr</span>, <span class="st">'_'</span>, <span class="va">Subdomain</span><span class="op">)</span>,</span>
<span>    bd <span class="op">=</span> <span class="va">Biodomain</span>,</span>
<span>    sdi <span class="op">=</span> <span class="va">subdomain_idx</span>,</span>
<span>    genes <span class="op">=</span> <span class="va">biodom</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="va">Biodomain</span> <span class="op">==</span> <span class="va">bd</span>, <span class="va">subdomain_idx</span> <span class="op">==</span> <span class="va">sdi</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unlist</span> <span class="op">%&gt;%</span> <span class="fu">list</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># combine</span></span>
<span><span class="va">bd.gene.lists</span> <span class="op">=</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">bd.genes</span>, <span class="va">sd.genes</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">set</span>, <span class="va">genes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">!=</span> <span class="st">'NA_none'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">rm</span><span class="op">(</span><span class="va">bd.genes</span>, <span class="va">sd.genes</span><span class="op">)</span></span></code></pre>
</div>
<p>next add expression data for human genes</p>
<div class="codewrapper sourceCode" id="cb119">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bd.genes</span> <span class="op">&lt;-</span> <span class="va">ampad_modules_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">Model</span> <span class="op">==</span> <span class="st">"Diagnosis"</span>,</span>
<span>         <span class="va">Comparison</span> <span class="op">==</span> <span class="st">"AD-CONTROL"</span>,</span>
<span>         <span class="va">Tissue</span> <span class="op">==</span> <span class="st">'TCX'</span>,<span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">hgnc_symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">hgnc_symbol</span>, <span class="va">logFC</span>, <span class="va">adj.P.Val</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">.</span>,</span>
<span>            <span class="va">mouse.human.ortho</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span>hgnc_symbol <span class="op">=</span> <span class="va">human_symbol</span>, symbol <span class="op">=</span> <span class="va">mouse_symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">bd.gene.lists</span> <span class="op">%&gt;%</span> <span class="fu">unnest</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>hgnc_symbol <span class="op">=</span> <span class="va">genes</span><span class="op">)</span>,</span>
<span>            <span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>now join the data and perform the correlation analysis</p>
<div class="codewrapper sourceCode" id="cb120">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model_vs_ampad</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">fad.deg</span>,</span>
<span>                             <span class="va">bd.genes</span>,</span>
<span>                             by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"symbol"</span><span class="op">)</span>,</span>
<span>                             multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'5xFAD, '</span>, <span class="va">age</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">model</span>,</span>
<span>         <span class="va">set</span>,</span>
<span>         <span class="va">symbol</span>,</span>
<span>         <span class="va">log2FoldChange</span>,</span>
<span>         <span class="va">padj</span>,</span>
<span>         <span class="va">hgnc_symbol</span>,</span>
<span>         <span class="va">logFC</span>,</span>
<span>         <span class="va">adj.P.Val</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="op">!</span><span class="fu">is.na</span><span class="op">(</span><span class="va">set</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">nest</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">symbol</span>, <span class="va">log2FoldChange</span>, <span class="va">padj</span>, <span class="va">hgnc_symbol</span>, <span class="va">logFC</span>, <span class="va">adj.P.Val</span><span class="op">)</span>,</span>
<span>    .by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="va">model</span>, <span class="va">set</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>data <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">distinct</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span>, n <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">nrow</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">n</span> <span class="op">&gt;</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cor.df</span> <span class="op">&lt;-</span> <span class="va">model_vs_ampad</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    cor_test <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span> <span class="fu">cor.test</span><span class="op">(</span><span class="va">.x</span><span class="op">[[</span><span class="st">"log2FoldChange"</span><span class="op">]</span><span class="op">]</span>, <span class="va">.x</span><span class="op">[[</span><span class="st">"logFC"</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    correlation <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"estimate"</span><span class="op">)</span>,</span>
<span>    p_value <span class="op">=</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">cor_test</span>, <span class="st">"p.value"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">cor_test</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>significant <span class="op">=</span> <span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre>
</div>
<p>finally, plot the biodomain correlations</p>
<div class="codewrapper sourceCode" id="cb121">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span> <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>, <span class="va">set</span><span class="op">)</span> <span class="op">==</span> <span class="cn">F</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>model <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">model</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'5xFAD, 10m'</span>, <span class="st">'5xFAD, 6m'</span>, <span class="st">'5xFAD, 4m'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">rev</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">set</span>, <span class="va">model</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>,</span>
<span>    stroke <span class="op">=</span> <span class="fl">1.2</span>,</span>
<span>    shape <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">6</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'top'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="st">"Correlation"</span>,</span>
<span>    low <span class="op">=</span> <span class="st">"#85070C"</span>,</span>
<span>    high <span class="op">=</span> <span class="st">"#164B6E"</span>,</span>
<span>    guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span></span>
<span>      angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>      hjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>      vjust <span class="op">=</span> <span class="fl">0</span>,</span>
<span>      size <span class="op">=</span> <span class="fl">12</span></span>
<span>    <span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span></span>
<span>      angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>      vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>,</span>
<span>      hjust <span class="op">=</span> <span class="fl">0.03</span>,</span>
<span>      size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>      face <span class="op">=</span> <span class="st">"bold"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>You can also plot the subdomain correlations, though it is best to
filter these to the strongest and most significant correlations.</p>
<div class="codewrapper sourceCode" id="cb122">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">l</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="fu">grepl</span><span class="op">(</span><span class="st">'_'</span>, <span class="va">set</span><span class="op">)</span> <span class="op">==</span> <span class="cn">T</span>, <span class="op">(</span><span class="va">p_value</span> <span class="op">&lt;</span> <span class="fl">5e-2</span> <span class="op">&amp;</span></span>
<span>                                  <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.25</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">set</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">cor.df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">set</span> <span class="op">%in%</span> <span class="va">l</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>max.n <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">n</span><span class="op">)</span>, .by <span class="op">=</span> <span class="va">set</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    model <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">model</span>, <span class="fu">c</span><span class="op">(</span><span class="st">'5xFAD, 10m'</span>, <span class="st">'5xFAD, 6m'</span>, <span class="st">'5xFAD, 4m'</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">rev</span><span class="op">)</span>,</span>
<span>    set1 <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">str_c</span><span class="op">(</span><span class="st">'['</span>, <span class="va">max.n</span>, <span class="st">'] '</span>, <span class="va">.</span><span class="op">)</span>,</span>
<span>    abbr <span class="op">=</span> <span class="fu">str_split_fixed</span><span class="op">(</span><span class="va">set</span>, <span class="st">'_'</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">tmp</span>, <span class="fu">aes</span><span class="op">(</span><span class="va">model</span>, <span class="va">set1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span></span>
<span>    rows <span class="op">=</span> <span class="fu">vars</span><span class="op">(</span><span class="va">abbr</span><span class="op">)</span>,</span>
<span>    scales <span class="op">=</span> <span class="st">'free'</span>,</span>
<span>    space <span class="op">=</span> <span class="st">'free'</span>,</span>
<span>    switch <span class="op">=</span> <span class="st">'y'</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_tile</span><span class="op">(</span>color <span class="op">=</span> <span class="st">'black'</span>, fill <span class="op">=</span> <span class="st">'white'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">correlation</span>, size <span class="op">=</span> <span class="fu">abs</span><span class="op">(</span><span class="va">correlation</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">tmp</span>, <span class="va">significant</span><span class="op">)</span>,</span>
<span>    stroke <span class="op">=</span> <span class="fl">1.2</span>,</span>
<span>    shape <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">6</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>position <span class="op">=</span> <span class="st">'bottom'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_discrete</span><span class="op">(</span>limits <span class="op">=</span> <span class="va">rev</span>, position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_size</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="co">#guide = 'none'</span></span>
<span>  <span class="fu">scale_color_gradient2</span><span class="op">(</span></span>
<span>    name <span class="op">=</span> <span class="st">"Correlation"</span>,</span>
<span>    low <span class="op">=</span> <span class="st">"#85070C"</span>,</span>
<span>    high <span class="op">=</span> <span class="st">"#164B6E"</span>,</span>
<span>    guide <span class="op">=</span> <span class="fu">guide_colorbar</span><span class="op">(</span>ticks <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    <span class="co"># strip.text.x = element_text(angle = 90, size = 10,colour = c("black")),</span></span>
<span>    strip.text.y.left <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span></span>
<span>      angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>      vjust <span class="op">=</span> <span class="op">-</span><span class="fl">54</span>,</span>
<span>      hjust <span class="op">=</span> <span class="fl">0.03</span>,</span>
<span>      size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>      face <span class="op">=</span> <span class="st">"bold"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    plot.title.position <span class="op">=</span> <span class="st">"plot"</span>,</span>
<span>    panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"left"</span>,</span>
<span>    plot.margin <span class="op">=</span> <span class="fu">margin</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">20</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre>
</div>
<p>Most biodomains have a significant positive correlation with AD
transcriptomes, some get stronger at older ages of 5xFAD. The most
significant correlations among subdomains are for the APP Metabolism,
Apoptosis, Autophagy, Mitochondrial Metabolism, and Myelination domains.
There is also positive correlation for “tau-protein kinase activity.</p>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="conclusion">[7] Conclusion:<a class="anchor" aria-label="anchor" href="#conclusion"></a></h2>
<hr class="half-width"><p>Overall, by aligning human and mouse omics signatures through the
lens of domains affected in each context, we can get a better
understanding of the relationships between the biological processes
affected in each context.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<span class="callout-header">Key Points</span>
<div class="callout-inner">
<div class="callout-content">
<ul><li>AMP-AD gene modules represent major transcriptomic heterogeneity in
AD</li>
<li>Correlation of logFC is a practical approach for human-mouse
alignment of AD-associated transcriptomic signatures</li>
<li>Functional gene set signatures are also a useful point of comparison
between species</li>
<li>Subsetting functions into associated Biological Domains and
Subdomains allows for more granular comparisons</li>
</ul></div>
</div>
</div>
</section><section><h2 class="section-heading" id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a></h2>
<hr class="half-width"><div class="codewrapper sourceCode" id="cb123">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb123-2"><a href="#cb123-2" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">5.2</span> (<span class="dv">2025-10-31</span>)</span>
<span id="cb123-3"><a href="#cb123-3" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>pc<span class="sc">-</span>linux<span class="sc">-</span>gnu</span>
<span id="cb123-4"><a href="#cb123-4" tabindex="-1"></a>Running under<span class="sc">:</span> Ubuntu <span class="dv">24</span>.<span class="fl">04.3</span> LTS</span>
<span id="cb123-5"><a href="#cb123-5" tabindex="-1"></a></span>
<span id="cb123-6"><a href="#cb123-6" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb123-7"><a href="#cb123-7" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>x86_64<span class="sc">-</span>linux<span class="sc">-</span>gnu<span class="sc">/</span>openblas<span class="sc">-</span>pthread<span class="sc">/</span>libblas.so<span class="fl">.3</span> </span>
<span id="cb123-8"><a href="#cb123-8" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>usr<span class="sc">/</span>lib<span class="sc">/</span>x86_64<span class="sc">-</span>linux<span class="sc">-</span>gnu<span class="sc">/</span>openblas<span class="sc">-</span>pthread<span class="sc">/</span>libopenblasp<span class="sc">-</span>r0.<span class="dv">3</span>.<span class="fl">26.</span>so;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb123-9"><a href="#cb123-9" tabindex="-1"></a></span>
<span id="cb123-10"><a href="#cb123-10" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb123-11"><a href="#cb123-11" tabindex="-1"></a> [<span class="dv">1</span>] LC_CTYPE<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>       LC_NUMERIC<span class="ot">=</span>C              </span>
<span id="cb123-12"><a href="#cb123-12" tabindex="-1"></a> [<span class="dv">3</span>] LC_TIME<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>        LC_COLLATE<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>    </span>
<span id="cb123-13"><a href="#cb123-13" tabindex="-1"></a> [<span class="dv">5</span>] LC_MONETARY<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>    LC_MESSAGES<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>   </span>
<span id="cb123-14"><a href="#cb123-14" tabindex="-1"></a> [<span class="dv">7</span>] LC_PAPER<span class="ot">=</span>en_US.UTF<span class="dv">-8</span>       LC_NAME<span class="ot">=</span>C                 </span>
<span id="cb123-15"><a href="#cb123-15" tabindex="-1"></a> [<span class="dv">9</span>] LC_ADDRESS<span class="ot">=</span>C               LC_TELEPHONE<span class="ot">=</span>C            </span>
<span id="cb123-16"><a href="#cb123-16" tabindex="-1"></a>[<span class="dv">11</span>] LC_MEASUREMENT<span class="ot">=</span>en_US.UTF<span class="dv">-8</span> LC_IDENTIFICATION<span class="ot">=</span>C       </span>
<span id="cb123-17"><a href="#cb123-17" tabindex="-1"></a></span>
<span id="cb123-18"><a href="#cb123-18" tabindex="-1"></a>time zone<span class="sc">:</span> Etc<span class="sc">/</span>UTC</span>
<span id="cb123-19"><a href="#cb123-19" tabindex="-1"></a>tzcode source<span class="sc">:</span> <span class="fu">system</span> (glibc)</span>
<span id="cb123-20"><a href="#cb123-20" tabindex="-1"></a></span>
<span id="cb123-21"><a href="#cb123-21" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb123-22"><a href="#cb123-22" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb123-23"><a href="#cb123-23" tabindex="-1"></a></span>
<span id="cb123-24"><a href="#cb123-24" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb123-25"><a href="#cb123-25" tabindex="-1"></a> [<span class="dv">1</span>] BiocManager_1.<span class="fl">30.27</span> compiler_4.<span class="fl">5.2</span>      cli_3.<span class="fl">6.5</span>          </span>
<span id="cb123-26"><a href="#cb123-26" tabindex="-1"></a> [<span class="dv">4</span>] tools_4.<span class="fl">5.2</span>         pillar_1.<span class="fl">11.1</span>       otel_0.<span class="fl">2.0</span>         </span>
<span id="cb123-27"><a href="#cb123-27" tabindex="-1"></a> [<span class="dv">7</span>] glue_1.<span class="fl">8.0</span>          yaml_2.<span class="fl">3.12</span>         vctrs_0.<span class="fl">7.1</span>        </span>
<span id="cb123-28"><a href="#cb123-28" tabindex="-1"></a>[<span class="dv">10</span>] knitr_1<span class="fl">.51</span>          xfun_0<span class="fl">.56</span>           lifecycle_1.<span class="fl">0.5</span>    </span>
<span id="cb123-29"><a href="#cb123-29" tabindex="-1"></a>[<span class="dv">13</span>] rlang_1.<span class="fl">1.7</span>         renv_1.<span class="fl">1.7</span>          evaluate_1.<span class="fl">0.5</span>     </span></code></pre>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/differential-expression-analysis.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/choosing_the_right_model.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/differential-expression-analysis.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Differential
        </a>
        <a class="chapter-link float-end" href="../instructor/choosing_the_right_model.html" rel="next">
          Next: Choosing the right...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/asliuyar/translational-omics-AD/edit/main/episodes/mouse-human-alignment.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/asliuyar/translational-omics-AD/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/asliuyar/translational-omics-AD/" class="external-link">Source</a></p>
        <p>
        <a href="../citation.html">Cite</a>
        | <a href="mailto:">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
		</div>
		<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/" class="external-link">sandpaper (0.18.1)</a>, <a href="https://github.com/carpentries/pegboard/tree/" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/" class="external-link">varnish (1.0.9)</a></p>
		</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://asliuyar.github.io/translational-omics-AD/instructor/mouse-human-alignment.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "Alzheimers-disease, cross-species-translation, omics, AD-Knowledge-Portal",
  "name": "Cross Species Functional Alignment",
  "creativeWorkStatus": "active",
  "url": "https://asliuyar.github.io/translational-omics-AD/instructor/mouse-human-alignment.html",
  "identifier": "https://asliuyar.github.io/translational-omics-AD/instructor/mouse-human-alignment.html",
  "dateCreated": "2024-10-02",
  "dateModified": "2026-02-12",
  "datePublished": "2026-02-12"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

